// @forge/cli-shared/out/graphql/gateway.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isFedrampEnv = exports.getEnvironmentConfig = exports.CDNEnvironments = exports.Environments = exports.getGraphqlGateway = exports.DEFAULT_GRAPHQL_GATEWAY = void 0;
exports.DEFAULT_GRAPHQL_GATEWAY = 'https://api.atlassian.com/graphql';
const getGraphqlGateway = () => {
    return process.env.FORGE_GRAPHQL_GATEWAY || exports.DEFAULT_GRAPHQL_GATEWAY;
};
exports.getGraphqlGateway = getGraphqlGateway;
var Environments;
(function (Environments) {
    Environments["DEV"] = "dev";
    Environments["STG"] = "staging";
    Environments["PROD"] = "prod";
    Environments["FEDRAMP_MODERATE_STAGING"] = "fedm_staging";
    Environments["FEDRAMP_MODERATE_PROD"] = "fedm_prod";
})(Environments = exports.Environments || (exports.Environments = {}));
var CDNEnvironments;
(function (CDNEnvironments) {
    CDNEnvironments["DEV"] = "dev";
    CDNEnvironments["STG"] = "stg";
    CDNEnvironments["PROD"] = "prod";
    CDNEnvironments["FEDRAMP_MODERATE_STAGING"] = "fedm_staging";
    CDNEnvironments["FEDRAMP_MODERATE_PROD"] = "fedm_prod";
})(CDNEnvironments = exports.CDNEnvironments || (exports.CDNEnvironments = {}));
const getEnvironmentConfig = (envOptions = Environments) => {
    const graphqlGateway = (0, exports.getGraphqlGateway)();
    if (graphqlGateway.includes('.dev.')) {
        return envOptions.DEV;
    }
    if (graphqlGateway.includes('.stg.atlassian-us-gov-mod.com')) {
        return envOptions.FEDRAMP_MODERATE_STAGING;
    }
    if (graphqlGateway.includes('.stg.')) {
        return envOptions.STG;
    }
    if (graphqlGateway.includes('atlassian-us-gov-mod.com')) {
        return envOptions.FEDRAMP_MODERATE_PROD;
    }
    return envOptions.PROD;
};
exports.getEnvironmentConfig = getEnvironmentConfig;
const isFedrampEnv = () => {
    const currentEnvironment = (0, exports.getEnvironmentConfig)();
    return (currentEnvironment === Environments.FEDRAMP_MODERATE_STAGING ||
        currentEnvironment === Environments.FEDRAMP_MODERATE_PROD);
};
exports.isFedrampEnv = isFedrampEnv;
