// @forge/bundler/out/config/nativeui.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getNativeUiBuildConfig = void 0;
const tslib_1 = require("tslib");
const path_1 = tslib_1.__importDefault(require("path"));
const html_webpack_plugin_1 = tslib_1.__importDefault(require("html-webpack-plugin"));
const webpack_bundle_analyzer_1 = require("webpack-bundle-analyzer");
const cli_shared_1 = require("@forge/cli-shared");
const common_1 = require("./common");
const getNativeUiBuildConfig = (entrypoints, i18nConfig) => {
    const resolvedEntryPoints = {};
    for (const entrypoint of entrypoints) {
        resolvedEntryPoints[entrypoint.name] = path_1.default.resolve(entrypoint.path);
    }
    return {
        entry: {
            ...resolvedEntryPoints
        },
        name: 'native-ui',
        mode: 'production',
        devtool: 'source-map',
        output: {
            filename: '[name].js',
            path: (0, cli_shared_1.tmpDir)('native-ui'),
            publicPath: 'auto'
        },
        node: {
            __dirname: true
        },
        performance: {
            hints: false
        },
        plugins: [
            new html_webpack_plugin_1.default({
                filename: 'index.html'
            }),
            new webpack_bundle_analyzer_1.BundleAnalyzerPlugin({ analyzerMode: 'disabled', generateStatsFile: true, logLevel: 'silent' })
        ],
        resolve: {
            extensions: ['.ts', '.tsx', '.js', '.jsx', '.jpg', '.jpeg', '.png', '.gif', '.svg']
        },
        resolveLoader: {
            modules: require.resolve.paths('babel-loader') || undefined
        },
        target: 'web',
        module: {
            rules: [
                ...(i18nConfig ? [(0, common_1.geti18nRule)(i18nConfig)] : []),
                {
                    test: /\.(jpe?g|png|gif|svg)$/i,
                    type: 'asset/resource'
                },
                { test: /\.tsx?$/, loader: 'ts-loader' },
                {
                    test: /\.jsx?$/,
                    exclude: /node_modules/,
                    use: [
                        {
                            loader: 'babel-loader',
                            options: {
                                plugins: [
                                    [
                                        require.resolve('@babel/plugin-transform-react-jsx'),
                                        {
                                            pragma: 'React.createElement'
                                        }
                                    ],
                                    require.resolve('@babel/plugin-transform-numeric-separator'),
                                    require.resolve('@babel/plugin-transform-class-static-block'),
                                    require.resolve('@babel/plugin-transform-class-properties'),
                                    require.resolve('@babel/plugin-transform-optional-chaining')
                                ],
                                cacheDirectory: true
                            }
                        }
                    ]
                }
            ]
        }
    };
};
exports.getNativeUiBuildConfig = getNativeUiBuildConfig;
