// @forge/util/packages/ari/chunk-MFK2774G.mjs
import {
  JiraIssueSearchViewTypeAriResourceOwner,
  JiraIssueSearchViewTypeAriResourceType
} from "./chunk-XE6YVVYA.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira/issue-search-view-type/manifest.ts
var jiraIssueSearchViewTypeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraIssueSearchViewTypeAriResourceOwner,
  resourceType: JiraIssueSearchViewTypeAriResourceType,
  resourceIdSlug: "activation/{activationId}/{namespaceId}/{viewId}/{viewType}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    namespaceId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    viewId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    viewType: /(list_view|grouped_list_view|detailed_view)/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/issue-search-view-type/index.ts
var JiraIssueSearchViewTypeAri = class _JiraIssueSearchViewTypeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._namespaceId = opts.resourceIdSegmentValues.namespaceId, this._viewId = opts.resourceIdSegmentValues.viewId, this._viewType = opts.resourceIdSegmentValues.viewType;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get namespaceId() {
    return this._namespaceId;
  }
  get viewId() {
    return this._viewId;
  }
  get viewType() {
    return this._viewType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraIssueSearchViewTypeAriStaticOpts.qualifier,
      platformQualifier: jiraIssueSearchViewTypeAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraIssueSearchViewTypeAriStaticOpts.resourceOwner,
      resourceType: jiraIssueSearchViewTypeAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.namespaceId}/${opts.viewId}/${opts.viewType}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        namespaceId: opts.namespaceId,
        viewId: opts.viewId,
        viewType: opts.viewType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraIssueSearchViewTypeAriStaticOpts);
    return new _JiraIssueSearchViewTypeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraIssueSearchViewTypeAriStaticOpts);
    return new _JiraIssueSearchViewTypeAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      namespaceId: this.namespaceId,
      viewId: this.viewId,
      viewType: this.viewType
    };
  }
};

export {
  JiraIssueSearchViewTypeAri
};
