// @forge/util/packages/ari/chunk-F5CSQIKL.mjs
import {
  MicrosoftTeamsConversationAriResourceOwner,
  MicrosoftTeamsConversationAriResourceType
} from "./chunk-WU5T7JXE.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft.teams/conversation/manifest.ts
var microsoftTeamsConversationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftTeamsConversationAriResourceOwner,
  resourceType: MicrosoftTeamsConversationAriResourceType,
  resourceIdSlug: "{compositeConversation}",
  resourceIdSegmentFormats: {
    compositeConversation: /(teamId\/[a-f0-9]{8}-[a-f0-9]{4}-[1-5][a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}\/channelId\/19:[a-zA-Z0-9_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9_-]+|19:[a-zA-Z0-9_]+(?:-[a-zA-Z0-9_]+)*@[a-zA-Z0-9._-]+\.[a-zA-Z0-9_-]+)/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft.teams/conversation/index.ts
var MicrosoftTeamsConversationAri = class _MicrosoftTeamsConversationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._compositeConversation = opts.resourceIdSegmentValues.compositeConversation;
  }
  get compositeConversation() {
    return this._compositeConversation;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftTeamsConversationAriStaticOpts.qualifier,
      platformQualifier: microsoftTeamsConversationAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftTeamsConversationAriStaticOpts.resourceOwner,
      resourceType: microsoftTeamsConversationAriStaticOpts.resourceType,
      resourceId: `${opts.compositeConversation}`,
      resourceIdSegmentValues: {
        compositeConversation: opts.compositeConversation
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftTeamsConversationAriStaticOpts);
    return new _MicrosoftTeamsConversationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftTeamsConversationAriStaticOpts);
    return new _MicrosoftTeamsConversationAri(opts);
  }
  getVariables() {
    return {
      compositeConversation: this.compositeConversation
    };
  }
};

export {
  MicrosoftTeamsConversationAri
};
