// @forge/util/packages/ari/chunk-MTF7V6HO.mjs
import {
  AhaAhaWorkItemAriResourceOwner,
  AhaAhaWorkItemAriResourceType
} from "./chunk-AESEW5OU.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/aha.aha/work-item/manifest.ts
var ahaAhaWorkItemAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AhaAhaWorkItemAriResourceOwner,
  resourceType: AhaAhaWorkItemAriResourceType,
  resourceIdSlug: "domain/{domain}/feature/{id}",
  resourceIdSegmentFormats: {
    domain: /[a-zA-Z0-9\-_.]*/,
    // eslint-disable-line no-useless-escape
    id: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/aha.aha/work-item/index.ts
var AhaAhaWorkItemAri = class _AhaAhaWorkItemAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._domain = opts.resourceIdSegmentValues.domain, this._id = opts.resourceIdSegmentValues.id;
  }
  get domain() {
    return this._domain;
  }
  get id() {
    return this._id;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: ahaAhaWorkItemAriStaticOpts.qualifier,
      platformQualifier: ahaAhaWorkItemAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: ahaAhaWorkItemAriStaticOpts.resourceOwner,
      resourceType: ahaAhaWorkItemAriStaticOpts.resourceType,
      resourceId: `domain/${opts.domain}/feature/${opts.id}`,
      resourceIdSegmentValues: {
        domain: opts.domain || "",
        id: opts.id
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, ahaAhaWorkItemAriStaticOpts);
    return new _AhaAhaWorkItemAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, ahaAhaWorkItemAriStaticOpts);
    return new _AhaAhaWorkItemAri(opts);
  }
  getVariables() {
    return {
      domain: this.domain,
      id: this.id
    };
  }
};

export {
  AhaAhaWorkItemAri
};
