// @forge/util/packages/ari/chunk-SLSKSO7Q.mjs
import {
  TownsquareGoalMetricUpdateAriResourceOwner,
  TownsquareGoalMetricUpdateAriResourceType
} from "./chunk-7GGI72TL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/goal-metric-update/manifest.ts
var townsquareGoalMetricUpdateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareGoalMetricUpdateAriResourceOwner,
  resourceType: TownsquareGoalMetricUpdateAriResourceType,
  resourceIdSlug: "{goalMetricUpdateUuid}",
  resourceIdSegmentFormats: {
    goalMetricUpdateUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/goal-metric-update/index.ts
var TownsquareGoalMetricUpdateAri = class _TownsquareGoalMetricUpdateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._goalMetricUpdateUuid = opts.resourceIdSegmentValues.goalMetricUpdateUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get goalMetricUpdateUuid() {
    return this._goalMetricUpdateUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareGoalMetricUpdateAriStaticOpts.qualifier,
      platformQualifier: townsquareGoalMetricUpdateAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareGoalMetricUpdateAriStaticOpts.resourceOwner,
      resourceType: townsquareGoalMetricUpdateAriStaticOpts.resourceType,
      resourceId: `${opts.goalMetricUpdateUuid}`,
      resourceIdSegmentValues: {
        goalMetricUpdateUuid: opts.goalMetricUpdateUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareGoalMetricUpdateAriStaticOpts);
    return new _TownsquareGoalMetricUpdateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareGoalMetricUpdateAriStaticOpts);
    return new _TownsquareGoalMetricUpdateAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      goalMetricUpdateUuid: this.goalMetricUpdateUuid
    };
  }
};

export {
  TownsquareGoalMetricUpdateAri
};
