// @forge/util/packages/ari/chunk-MLCBFNO5.mjs
import {
  ProjectSiteAriResourceOwner,
  ProjectSiteAriResourceType
} from "./chunk-FAZ4I6YC.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/project/site/manifest.ts
var projectSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ProjectSiteAriResourceOwner,
  resourceType: ProjectSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/project/site/index.ts
var ProjectSiteAri = class _ProjectSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: projectSiteAriStaticOpts.qualifier,
      platformQualifier: projectSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: projectSiteAriStaticOpts.resourceOwner,
      resourceType: projectSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, projectSiteAriStaticOpts);
    return new _ProjectSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, projectSiteAriStaticOpts);
    return new _ProjectSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  ProjectSiteAri
};
