// @forge/util/packages/ari/chunk-RLVKCH5J.mjs
import {
  RovoWorkspaceAriResourceOwner,
  RovoWorkspaceAriResourceType
} from "./chunk-WPBXANQL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/rovo/workspace/manifest.ts
var rovoWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: RovoWorkspaceAriResourceOwner,
  resourceType: RovoWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/rovo/workspace/index.ts
var RovoWorkspaceAri = class _RovoWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: rovoWorkspaceAriStaticOpts.qualifier,
      platformQualifier: rovoWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: rovoWorkspaceAriStaticOpts.resourceOwner,
      resourceType: rovoWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, rovoWorkspaceAriStaticOpts);
    return new _RovoWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, rovoWorkspaceAriStaticOpts);
    return new _RovoWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId
    };
  }
};

export {
  RovoWorkspaceAri
};
