// @forge/util/packages/ari/chunk-Q5L67A6J.mjs
import {
  HubspotHubspotCustomerOrgAriResourceOwner,
  HubspotHubspotCustomerOrgAriResourceType
} from "./chunk-SGZ5NPU7.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/hubspot.hubspot/customer-org/manifest.ts
var hubspotHubspotCustomerOrgAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: HubspotHubspotCustomerOrgAriResourceOwner,
  resourceType: HubspotHubspotCustomerOrgAriResourceType,
  resourceIdSlug: "portal/{portalId}/customer-org/{customerOrgId}",
  resourceIdSegmentFormats: {
    portalId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    customerOrgId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/hubspot.hubspot/customer-org/index.ts
var HubspotHubspotCustomerOrgAri = class _HubspotHubspotCustomerOrgAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._portalId = opts.resourceIdSegmentValues.portalId, this._customerOrgId = opts.resourceIdSegmentValues.customerOrgId;
  }
  get portalId() {
    return this._portalId;
  }
  get customerOrgId() {
    return this._customerOrgId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: hubspotHubspotCustomerOrgAriStaticOpts.qualifier,
      platformQualifier: hubspotHubspotCustomerOrgAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: hubspotHubspotCustomerOrgAriStaticOpts.resourceOwner,
      resourceType: hubspotHubspotCustomerOrgAriStaticOpts.resourceType,
      resourceId: `portal/${opts.portalId}/customer-org/${opts.customerOrgId}`,
      resourceIdSegmentValues: {
        portalId: opts.portalId,
        customerOrgId: opts.customerOrgId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, hubspotHubspotCustomerOrgAriStaticOpts);
    return new _HubspotHubspotCustomerOrgAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, hubspotHubspotCustomerOrgAriStaticOpts);
    return new _HubspotHubspotCustomerOrgAri(opts);
  }
  getVariables() {
    return {
      portalId: this.portalId,
      customerOrgId: this.customerOrgId
    };
  }
};

export {
  HubspotHubspotCustomerOrgAri
};
