// @forge/util/packages/ari/chunk-OMSDF73F.mjs
import {
  KnowledgeDiscoveryBookmarkAriResourceOwner,
  KnowledgeDiscoveryBookmarkAriResourceType
} from "./chunk-CSESCAWL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/knowledge-discovery/bookmark/manifest.ts
var knowledgeDiscoveryBookmarkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: KnowledgeDiscoveryBookmarkAriResourceOwner,
  resourceType: KnowledgeDiscoveryBookmarkAriResourceType,
  resourceIdSlug: "activation/{activationId}/{bookmarkId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    bookmarkId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/knowledge-discovery/bookmark/index.ts
var KnowledgeDiscoveryBookmarkAri = class _KnowledgeDiscoveryBookmarkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._bookmarkId = opts.resourceIdSegmentValues.bookmarkId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get bookmarkId() {
    return this._bookmarkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: knowledgeDiscoveryBookmarkAriStaticOpts.qualifier,
      platformQualifier: knowledgeDiscoveryBookmarkAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: knowledgeDiscoveryBookmarkAriStaticOpts.resourceOwner,
      resourceType: knowledgeDiscoveryBookmarkAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.bookmarkId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        bookmarkId: opts.bookmarkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, knowledgeDiscoveryBookmarkAriStaticOpts);
    return new _KnowledgeDiscoveryBookmarkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, knowledgeDiscoveryBookmarkAriStaticOpts);
    return new _KnowledgeDiscoveryBookmarkAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      bookmarkId: this.bookmarkId
    };
  }
};

export {
  KnowledgeDiscoveryBookmarkAri
};
