// @forge/util/packages/ari/chunk-5Q6HR2L4.mjs
import {
  OpsgenieTeamMemberAriResourceOwner,
  OpsgenieTeamMemberAriResourceType
} from "./chunk-X2LQHTML.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/opsgenie/team-member/manifest.ts
var opsgenieTeamMemberAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: OpsgenieTeamMemberAriResourceOwner,
  resourceType: OpsgenieTeamMemberAriResourceType,
  resourceIdSlug: "{workspaceId}/{teamId}/{accountId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    teamId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    accountId: /[a-zA-Z0-9_\-\:]{1,128}/
    // eslint-disable-line no-useless-escape
  }
};

// src/opsgenie/team-member/index.ts
var OpsgenieTeamMemberAri = class _OpsgenieTeamMemberAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._teamId = opts.resourceIdSegmentValues.teamId, this._accountId = opts.resourceIdSegmentValues.accountId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get teamId() {
    return this._teamId;
  }
  get accountId() {
    return this._accountId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: opsgenieTeamMemberAriStaticOpts.qualifier,
      platformQualifier: opsgenieTeamMemberAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: opsgenieTeamMemberAriStaticOpts.resourceOwner,
      resourceType: opsgenieTeamMemberAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.teamId}/${opts.accountId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        teamId: opts.teamId,
        accountId: opts.accountId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, opsgenieTeamMemberAriStaticOpts);
    return new _OpsgenieTeamMemberAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, opsgenieTeamMemberAriStaticOpts);
    return new _OpsgenieTeamMemberAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      teamId: this.teamId,
      accountId: this.accountId
    };
  }
};

export {
  OpsgenieTeamMemberAri
};
