// @forge/util/packages/ari/chunk-P3YAJMFO.mjs
import {
  AvpSubscriptionAriResourceOwner,
  AvpSubscriptionAriResourceType
} from "./chunk-GEK45VES.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/avp/subscription/manifest.ts
var avpSubscriptionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AvpSubscriptionAriResourceOwner,
  resourceType: AvpSubscriptionAriResourceType,
  resourceIdSlug: "activation/{activationId}/{subscriptionId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    subscriptionId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/subscription/index.ts
var AvpSubscriptionAri = class _AvpSubscriptionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._subscriptionId = opts.resourceIdSegmentValues.subscriptionId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get subscriptionId() {
    return this._subscriptionId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpSubscriptionAriStaticOpts.qualifier,
      platformQualifier: avpSubscriptionAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: avpSubscriptionAriStaticOpts.resourceOwner,
      resourceType: avpSubscriptionAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.subscriptionId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        subscriptionId: opts.subscriptionId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpSubscriptionAriStaticOpts);
    return new _AvpSubscriptionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpSubscriptionAriStaticOpts);
    return new _AvpSubscriptionAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      subscriptionId: this.subscriptionId
    };
  }
};

export {
  AvpSubscriptionAri
};
