// @forge/util/packages/ari/chunk-TKKIRZDB.mjs
import {
  TownsquareMetricTargetAriResourceOwner,
  TownsquareMetricTargetAriResourceType
} from "./chunk-TQEX6RU4.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/metric-target/manifest.ts
var townsquareMetricTargetAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareMetricTargetAriResourceOwner,
  resourceType: TownsquareMetricTargetAriResourceType,
  resourceIdSlug: "{metricTargetUuid}",
  resourceIdSegmentFormats: {
    metricTargetUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/metric-target/index.ts
var TownsquareMetricTargetAri = class _TownsquareMetricTargetAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._metricTargetUuid = opts.resourceIdSegmentValues.metricTargetUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get metricTargetUuid() {
    return this._metricTargetUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareMetricTargetAriStaticOpts.qualifier,
      platformQualifier: townsquareMetricTargetAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareMetricTargetAriStaticOpts.resourceOwner,
      resourceType: townsquareMetricTargetAriStaticOpts.resourceType,
      resourceId: `${opts.metricTargetUuid}`,
      resourceIdSegmentValues: {
        metricTargetUuid: opts.metricTargetUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareMetricTargetAriStaticOpts);
    return new _TownsquareMetricTargetAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareMetricTargetAriStaticOpts);
    return new _TownsquareMetricTargetAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      metricTargetUuid: this.metricTargetUuid
    };
  }
};

export {
  TownsquareMetricTargetAri
};
