// @forge/util/packages/ari/chunk-7C3P4PYG.mjs
import {
  PlatformCollaborationContextAriResourceOwner,
  PlatformCollaborationContextAriResourceType
} from "./chunk-AX7CUYP3.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/platform/collaboration-context/manifest.ts
var platformCollaborationContextAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PlatformCollaborationContextAriResourceOwner,
  resourceType: PlatformCollaborationContextAriResourceType,
  resourceIdSlug: "{collaborationContextId}",
  resourceIdSegmentFormats: {
    collaborationContextId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/platform/collaboration-context/index.ts
var PlatformCollaborationContextAri = class _PlatformCollaborationContextAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._collaborationContextId = opts.resourceIdSegmentValues.collaborationContextId;
  }
  get collaborationContextId() {
    return this._collaborationContextId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: platformCollaborationContextAriStaticOpts.qualifier,
      platformQualifier: platformCollaborationContextAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: platformCollaborationContextAriStaticOpts.resourceOwner,
      resourceType: platformCollaborationContextAriStaticOpts.resourceType,
      resourceId: `${opts.collaborationContextId}`,
      resourceIdSegmentValues: {
        collaborationContextId: opts.collaborationContextId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, platformCollaborationContextAriStaticOpts);
    return new _PlatformCollaborationContextAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, platformCollaborationContextAriStaticOpts);
    return new _PlatformCollaborationContextAri(opts);
  }
  getVariables() {
    return {
      collaborationContextId: this.collaborationContextId
    };
  }
};

export {
  PlatformCollaborationContextAri
};
