// @forge/util/packages/ari/chunk-QY7LJYM7.mjs
import {
  DropboxDropboxDocumentAriResourceOwner,
  DropboxDropboxDocumentAriResourceType
} from "./chunk-BXFVNICZ.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/dropbox.dropbox/document/manifest.ts
var dropboxDropboxDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: DropboxDropboxDocumentAriResourceOwner,
  resourceType: DropboxDropboxDocumentAriResourceType,
  resourceIdSlug: "{documentId}",
  resourceIdSegmentFormats: {
    documentId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/dropbox.dropbox/document/index.ts
var DropboxDropboxDocumentAri = class _DropboxDropboxDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._documentId = opts.resourceIdSegmentValues.documentId;
  }
  get documentId() {
    return this._documentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: dropboxDropboxDocumentAriStaticOpts.qualifier,
      platformQualifier: dropboxDropboxDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: dropboxDropboxDocumentAriStaticOpts.resourceOwner,
      resourceType: dropboxDropboxDocumentAriStaticOpts.resourceType,
      resourceId: `${opts.documentId}`,
      resourceIdSegmentValues: {
        documentId: opts.documentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, dropboxDropboxDocumentAriStaticOpts);
    return new _DropboxDropboxDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, dropboxDropboxDocumentAriStaticOpts);
    return new _DropboxDropboxDocumentAri(opts);
  }
  getVariables() {
    return {
      documentId: this.documentId
    };
  }
};

export {
  DropboxDropboxDocumentAri
};
