// @forge/util/packages/ari/chunk-INEXAAI2.mjs
import {
  AvpControlAriResourceOwner,
  AvpControlAriResourceType
} from "./chunk-SQJOVVPI.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/avp/control/manifest.ts
var avpControlAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AvpControlAriResourceOwner,
  resourceType: AvpControlAriResourceType,
  resourceIdSlug: "activation/{activationId}/{controlId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    controlId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/control/index.ts
var AvpControlAri = class _AvpControlAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._controlId = opts.resourceIdSegmentValues.controlId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get controlId() {
    return this._controlId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpControlAriStaticOpts.qualifier,
      platformQualifier: avpControlAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: avpControlAriStaticOpts.resourceOwner,
      resourceType: avpControlAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.controlId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        controlId: opts.controlId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpControlAriStaticOpts);
    return new _AvpControlAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpControlAriStaticOpts);
    return new _AvpControlAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      controlId: this.controlId
    };
  }
};

export {
  AvpControlAri
};
