// @forge/util/packages/ari/chunk-XCNIKIGY.mjs
import {
  StripeStripeRemoteLinkAriResourceOwner,
  StripeStripeRemoteLinkAriResourceType
} from "./chunk-NK2XSAGB.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/stripe.stripe/remote-link/manifest.ts
var stripeStripeRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: StripeStripeRemoteLinkAriResourceOwner,
  resourceType: StripeStripeRemoteLinkAriResourceType,
  resourceIdSlug: "invoice/{invoiceId}",
  resourceIdSegmentFormats: {
    invoiceId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/stripe.stripe/remote-link/index.ts
var StripeStripeRemoteLinkAri = class _StripeStripeRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._invoiceId = opts.resourceIdSegmentValues.invoiceId;
  }
  get invoiceId() {
    return this._invoiceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: stripeStripeRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: stripeStripeRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: stripeStripeRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: stripeStripeRemoteLinkAriStaticOpts.resourceType,
      resourceId: `invoice/${opts.invoiceId}`,
      resourceIdSegmentValues: {
        invoiceId: opts.invoiceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, stripeStripeRemoteLinkAriStaticOpts);
    return new _StripeStripeRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, stripeStripeRemoteLinkAriStaticOpts);
    return new _StripeStripeRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      invoiceId: this.invoiceId
    };
  }
};

export {
  StripeStripeRemoteLinkAri
};
