// @forge/util/packages/ari/chunk-AB5I33K3.mjs
import {
  PassionfruitAskUpdateAriResourceOwner,
  PassionfruitAskUpdateAriResourceType
} from "./chunk-WRANQECF.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/ask-update/manifest.ts
var passionfruitAskUpdateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitAskUpdateAriResourceOwner,
  resourceType: PassionfruitAskUpdateAriResourceType,
  resourceIdSlug: "workspace/{activationId}/{askId}/{askUpdateId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    askId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    askUpdateId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/ask-update/index.ts
var PassionfruitAskUpdateAri = class _PassionfruitAskUpdateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._askId = opts.resourceIdSegmentValues.askId, this._askUpdateId = opts.resourceIdSegmentValues.askUpdateId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get askId() {
    return this._askId;
  }
  get askUpdateId() {
    return this._askUpdateId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitAskUpdateAriStaticOpts.qualifier,
      platformQualifier: passionfruitAskUpdateAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: passionfruitAskUpdateAriStaticOpts.resourceOwner,
      resourceType: passionfruitAskUpdateAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.activationId}/${opts.askId}/${opts.askUpdateId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        askId: opts.askId,
        askUpdateId: opts.askUpdateId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitAskUpdateAriStaticOpts);
    return new _PassionfruitAskUpdateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitAskUpdateAriStaticOpts);
    return new _PassionfruitAskUpdateAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      askId: this.askId,
      askUpdateId: this.askUpdateId
    };
  }
};

export {
  PassionfruitAskUpdateAri
};
