// @forge/util/packages/ari/chunk-P5BYJVW7.mjs
import {
  GraphIntegrationContextAriResourceOwner,
  GraphIntegrationContextAriResourceType
} from "./chunk-CAJWUMVE.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/integration-context/manifest.ts
var graphIntegrationContextAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphIntegrationContextAriResourceOwner,
  resourceType: GraphIntegrationContextAriResourceType,
  resourceIdSlug: "{integrationContextId}",
  resourceIdSegmentFormats: {
    integrationContextId: /(ecda99d9-9b42-4bf7-8b4f-ecb5fcf5868c|3d1b1176-a2e3-4233-82bd-29a9cfc80cab|161675af-ebe2-4e6b-b5fd-e3fdf9f5ad84)/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/integration-context/index.ts
var GraphIntegrationContextAri = class _GraphIntegrationContextAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._integrationContextId = opts.resourceIdSegmentValues.integrationContextId;
  }
  get integrationContextId() {
    return this._integrationContextId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphIntegrationContextAriStaticOpts.qualifier,
      platformQualifier: graphIntegrationContextAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphIntegrationContextAriStaticOpts.resourceOwner,
      resourceType: graphIntegrationContextAriStaticOpts.resourceType,
      resourceId: `${opts.integrationContextId}`,
      resourceIdSegmentValues: {
        integrationContextId: opts.integrationContextId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphIntegrationContextAriStaticOpts);
    return new _GraphIntegrationContextAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphIntegrationContextAriStaticOpts);
    return new _GraphIntegrationContextAri(opts);
  }
  getVariables() {
    return {
      integrationContextId: this.integrationContextId
    };
  }
};

export {
  GraphIntegrationContextAri
};
