// @forge/util/packages/ari/chunk-IRX4UDFX.mjs
import {
  ConfluenceDcConfluenceDcCommentAriResourceOwner,
  ConfluenceDcConfluenceDcCommentAriResourceType
} from "./chunk-WWSSF667.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/confluence-dc.confluence-dc/comment/manifest.ts
var confluenceDcConfluenceDcCommentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ConfluenceDcConfluenceDcCommentAriResourceOwner,
  resourceType: ConfluenceDcConfluenceDcCommentAriResourceType,
  resourceIdSlug: "{siteId}/{commentId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    commentId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/confluence-dc.confluence-dc/comment/index.ts
var ConfluenceDcConfluenceDcCommentAri = class _ConfluenceDcConfluenceDcCommentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._commentId = opts.resourceIdSegmentValues.commentId;
  }
  get siteId() {
    return this._siteId;
  }
  get commentId() {
    return this._commentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: confluenceDcConfluenceDcCommentAriStaticOpts.qualifier,
      platformQualifier: confluenceDcConfluenceDcCommentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: confluenceDcConfluenceDcCommentAriStaticOpts.resourceOwner,
      resourceType: confluenceDcConfluenceDcCommentAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.commentId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        commentId: opts.commentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, confluenceDcConfluenceDcCommentAriStaticOpts);
    return new _ConfluenceDcConfluenceDcCommentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, confluenceDcConfluenceDcCommentAriStaticOpts);
    return new _ConfluenceDcConfluenceDcCommentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      commentId: this.commentId
    };
  }
};

export {
  ConfluenceDcConfluenceDcCommentAri
};
