// @forge/util/packages/ari/chunk-3IOWV2Z2.mjs
import {
  SmartsheetSmartsheetDocumentAriResourceOwner,
  SmartsheetSmartsheetDocumentAriResourceType
} from "./chunk-EOWM7GT2.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/smartsheet.smartsheet/document/manifest.ts
var smartsheetSmartsheetDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: SmartsheetSmartsheetDocumentAriResourceOwner,
  resourceType: SmartsheetSmartsheetDocumentAriResourceType,
  resourceIdSlug: "sheet/{sheetId}",
  resourceIdSegmentFormats: {
    sheetId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/smartsheet.smartsheet/document/index.ts
var SmartsheetSmartsheetDocumentAri = class _SmartsheetSmartsheetDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._sheetId = opts.resourceIdSegmentValues.sheetId;
  }
  get sheetId() {
    return this._sheetId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: smartsheetSmartsheetDocumentAriStaticOpts.qualifier,
      platformQualifier: smartsheetSmartsheetDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: smartsheetSmartsheetDocumentAriStaticOpts.resourceOwner,
      resourceType: smartsheetSmartsheetDocumentAriStaticOpts.resourceType,
      resourceId: `sheet/${opts.sheetId}`,
      resourceIdSegmentValues: {
        sheetId: opts.sheetId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, smartsheetSmartsheetDocumentAriStaticOpts);
    return new _SmartsheetSmartsheetDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, smartsheetSmartsheetDocumentAriStaticOpts);
    return new _SmartsheetSmartsheetDocumentAri(opts);
  }
  getVariables() {
    return {
      sheetId: this.sheetId
    };
  }
};

export {
  SmartsheetSmartsheetDocumentAri
};
