// @forge/util/packages/ari/chunk-MHTDAQEL.mjs
import {
  AdobeAdobexdRemoteLinkAriResourceOwner,
  AdobeAdobexdRemoteLinkAriResourceType
} from "./chunk-H2R5HU73.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/adobe.adobexd/remote-link/manifest.ts
var adobeAdobexdRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AdobeAdobexdRemoteLinkAriResourceOwner,
  resourceType: AdobeAdobexdRemoteLinkAriResourceType,
  resourceIdSlug: "file/{fileId}",
  resourceIdSegmentFormats: {
    fileId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/adobe.adobexd/remote-link/index.ts
var AdobeAdobexdRemoteLinkAri = class _AdobeAdobexdRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._fileId = opts.resourceIdSegmentValues.fileId;
  }
  get fileId() {
    return this._fileId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: adobeAdobexdRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: adobeAdobexdRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: adobeAdobexdRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: adobeAdobexdRemoteLinkAriStaticOpts.resourceType,
      resourceId: `file/${opts.fileId}`,
      resourceIdSegmentValues: {
        fileId: opts.fileId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, adobeAdobexdRemoteLinkAriStaticOpts);
    return new _AdobeAdobexdRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, adobeAdobexdRemoteLinkAriStaticOpts);
    return new _AdobeAdobexdRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      fileId: this.fileId
    };
  }
};

export {
  AdobeAdobexdRemoteLinkAri
};
