// @forge/util/packages/ari/chunk-QKXY4FYK.mjs
import {
  MicrosoftChatAriResourceOwner,
  MicrosoftChatAriResourceType
} from "./chunk-TEIQ5OS2.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft/chat/manifest.ts
var microsoftChatAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftChatAriResourceOwner,
  resourceType: MicrosoftChatAriResourceType,
  resourceIdSlug: "chatId/{chatId}/messageId/{messageId}",
  resourceIdSegmentFormats: {
    chatId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    messageId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft/chat/index.ts
var MicrosoftChatAri = class _MicrosoftChatAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._chatId = opts.resourceIdSegmentValues.chatId, this._messageId = opts.resourceIdSegmentValues.messageId;
  }
  get chatId() {
    return this._chatId;
  }
  get messageId() {
    return this._messageId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftChatAriStaticOpts.qualifier,
      platformQualifier: microsoftChatAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftChatAriStaticOpts.resourceOwner,
      resourceType: microsoftChatAriStaticOpts.resourceType,
      resourceId: `chatId/${opts.chatId}/messageId/${opts.messageId}`,
      resourceIdSegmentValues: {
        chatId: opts.chatId,
        messageId: opts.messageId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftChatAriStaticOpts);
    return new _MicrosoftChatAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftChatAriStaticOpts);
    return new _MicrosoftChatAri(opts);
  }
  getVariables() {
    return {
      chatId: this.chatId,
      messageId: this.messageId
    };
  }
};

export {
  MicrosoftChatAri
};
