// @forge/util/packages/ari/chunk-IXQQVAFS.mjs
import {
  AvpDataShareAriResourceOwner,
  AvpDataShareAriResourceType
} from "./chunk-FFZGZTIN.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/avp/data-share/manifest.ts
var avpDataShareAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AvpDataShareAriResourceOwner,
  resourceType: AvpDataShareAriResourceType,
  resourceIdSlug: "activation/{activationId}/{datashareId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    datashareId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/data-share/index.ts
var AvpDataShareAri = class _AvpDataShareAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._datashareId = opts.resourceIdSegmentValues.datashareId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get datashareId() {
    return this._datashareId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpDataShareAriStaticOpts.qualifier,
      platformQualifier: avpDataShareAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: avpDataShareAriStaticOpts.resourceOwner,
      resourceType: avpDataShareAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.datashareId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        datashareId: opts.datashareId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpDataShareAriStaticOpts);
    return new _AvpDataShareAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpDataShareAriStaticOpts);
    return new _AvpDataShareAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      datashareId: this.datashareId
    };
  }
};

export {
  AvpDataShareAri
};
