// @forge/util/packages/ari/chunk-553W4TP2.mjs
import {
  RovoMessageAriResourceOwner,
  RovoMessageAriResourceType
} from "./chunk-7DUOYN26.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/rovo/message/manifest.ts
var rovoMessageAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: RovoMessageAriResourceOwner,
  resourceType: RovoMessageAriResourceType,
  resourceIdSlug: "activation/{activationId}/{uuid}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    uuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/rovo/message/index.ts
var RovoMessageAri = class _RovoMessageAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._uuid = opts.resourceIdSegmentValues.uuid;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get uuid() {
    return this._uuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: rovoMessageAriStaticOpts.qualifier,
      platformQualifier: rovoMessageAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: rovoMessageAriStaticOpts.resourceOwner,
      resourceType: rovoMessageAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.uuid}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        uuid: opts.uuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, rovoMessageAriStaticOpts);
    return new _RovoMessageAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, rovoMessageAriStaticOpts);
    return new _RovoMessageAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      uuid: this.uuid
    };
  }
};

export {
  RovoMessageAri
};
