// @forge/util/packages/ari/chunk-YE4P4KFX.mjs
import {
  TrelloCustomFieldAriResourceOwner,
  TrelloCustomFieldAriResourceType
} from "./chunk-KGWMZBU5.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/trello/custom-field/manifest.ts
var trelloCustomFieldAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TrelloCustomFieldAriResourceOwner,
  resourceType: TrelloCustomFieldAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{customFieldId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-_.]*/,
    // eslint-disable-line no-useless-escape
    customFieldId: /[a-zA-Z0-9\-_.]*/
    // eslint-disable-line no-useless-escape
  }
};

// src/trello/custom-field/index.ts
var TrelloCustomFieldAri = class _TrelloCustomFieldAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._customFieldId = opts.resourceIdSegmentValues.customFieldId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get customFieldId() {
    return this._customFieldId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: trelloCustomFieldAriStaticOpts.qualifier,
      platformQualifier: trelloCustomFieldAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: trelloCustomFieldAriStaticOpts.resourceOwner,
      resourceType: trelloCustomFieldAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.customFieldId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId || "",
        customFieldId: opts.customFieldId || ""
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, trelloCustomFieldAriStaticOpts);
    return new _TrelloCustomFieldAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, trelloCustomFieldAriStaticOpts);
    return new _TrelloCustomFieldAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      customFieldId: this.customFieldId
    };
  }
};

export {
  TrelloCustomFieldAri
};
