// @forge/util/packages/ari/chunk-YDMQDUYV.mjs
import {
  DocusignDocusignDocumentAriResourceOwner,
  DocusignDocusignDocumentAriResourceType
} from "./chunk-T7GLL24I.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/docusign.docusign/document/manifest.ts
var docusignDocusignDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: DocusignDocusignDocumentAriResourceOwner,
  resourceType: DocusignDocusignDocumentAriResourceType,
  resourceIdSlug: "shard/{shardRegion}/account/{accountId}/envelope/{envelopeId}",
  resourceIdSegmentFormats: {
    shardRegion: /[a-zA-Z0-9_-]+/,
    // eslint-disable-line no-useless-escape
    accountId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    envelopeId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/docusign.docusign/document/index.ts
var DocusignDocusignDocumentAri = class _DocusignDocusignDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._shardRegion = opts.resourceIdSegmentValues.shardRegion, this._accountId = opts.resourceIdSegmentValues.accountId, this._envelopeId = opts.resourceIdSegmentValues.envelopeId;
  }
  get shardRegion() {
    return this._shardRegion;
  }
  get accountId() {
    return this._accountId;
  }
  get envelopeId() {
    return this._envelopeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: docusignDocusignDocumentAriStaticOpts.qualifier,
      platformQualifier: docusignDocusignDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: docusignDocusignDocumentAriStaticOpts.resourceOwner,
      resourceType: docusignDocusignDocumentAriStaticOpts.resourceType,
      resourceId: `shard/${opts.shardRegion}/account/${opts.accountId}/envelope/${opts.envelopeId}`,
      resourceIdSegmentValues: {
        shardRegion: opts.shardRegion,
        accountId: opts.accountId,
        envelopeId: opts.envelopeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, docusignDocusignDocumentAriStaticOpts);
    return new _DocusignDocusignDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, docusignDocusignDocumentAriStaticOpts);
    return new _DocusignDocusignDocumentAri(opts);
  }
  getVariables() {
    return {
      shardRegion: this.shardRegion,
      accountId: this.accountId,
      envelopeId: this.envelopeId
    };
  }
};

export {
  DocusignDocusignDocumentAri
};
