// @forge/util/packages/ari/chunk-GSQYTWNQ.mjs
import {
  JiraAiCrocsWorkspaceAriResourceOwner,
  JiraAiCrocsWorkspaceAriResourceType
} from "./chunk-KYL4TWI6.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-ai-crocs/workspace/manifest.ts
var jiraAiCrocsWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraAiCrocsWorkspaceAriResourceOwner,
  resourceType: JiraAiCrocsWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-ai-crocs/workspace/index.ts
var JiraAiCrocsWorkspaceAri = class _JiraAiCrocsWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraAiCrocsWorkspaceAriStaticOpts.qualifier,
      platformQualifier: jiraAiCrocsWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraAiCrocsWorkspaceAriStaticOpts.resourceOwner,
      resourceType: jiraAiCrocsWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraAiCrocsWorkspaceAriStaticOpts);
    return new _JiraAiCrocsWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraAiCrocsWorkspaceAriStaticOpts);
    return new _JiraAiCrocsWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId
    };
  }
};

export {
  JiraAiCrocsWorkspaceAri
};
