// @forge/util/packages/ari/chunk-BSKQUBQK.mjs
import {
  BoxBoxDocumentAriResourceOwner,
  BoxBoxDocumentAriResourceType
} from "./chunk-4QS2SFIR.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/box.box/document/manifest.ts
var boxBoxDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: BoxBoxDocumentAriResourceOwner,
  resourceType: BoxBoxDocumentAriResourceType,
  resourceIdSlug: "{item}/{itemId}",
  resourceIdSegmentFormats: {
    item: /(?:fileId|folderId)/,
    // eslint-disable-line no-useless-escape
    itemId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/box.box/document/index.ts
var BoxBoxDocumentAri = class _BoxBoxDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._item = opts.resourceIdSegmentValues.item, this._itemId = opts.resourceIdSegmentValues.itemId;
  }
  get item() {
    return this._item;
  }
  get itemId() {
    return this._itemId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: boxBoxDocumentAriStaticOpts.qualifier,
      platformQualifier: boxBoxDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: boxBoxDocumentAriStaticOpts.resourceOwner,
      resourceType: boxBoxDocumentAriStaticOpts.resourceType,
      resourceId: `${opts.item}/${opts.itemId}`,
      resourceIdSegmentValues: {
        item: opts.item,
        itemId: opts.itemId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, boxBoxDocumentAriStaticOpts);
    return new _BoxBoxDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, boxBoxDocumentAriStaticOpts);
    return new _BoxBoxDocumentAri(opts);
  }
  getVariables() {
    return {
      item: this.item,
      itemId: this.itemId
    };
  }
};

export {
  BoxBoxDocumentAri
};
