// @forge/util/packages/ari/chunk-IFNDEODP.mjs
import {
  JiraAlignWorkspaceAriResourceOwner,
  JiraAlignWorkspaceAriResourceType
} from "./chunk-KSSMXCNA.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-align/workspace/manifest.ts
var jiraAlignWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraAlignWorkspaceAriResourceOwner,
  resourceType: JiraAlignWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-align/workspace/index.ts
var JiraAlignWorkspaceAri = class _JiraAlignWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraAlignWorkspaceAriStaticOpts.qualifier,
      platformQualifier: jiraAlignWorkspaceAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraAlignWorkspaceAriStaticOpts.resourceOwner,
      resourceType: jiraAlignWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraAlignWorkspaceAriStaticOpts);
    return new _JiraAlignWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraAlignWorkspaceAriStaticOpts);
    return new _JiraAlignWorkspaceAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId
    };
  }
};

export {
  JiraAlignWorkspaceAri
};
