// @forge/util/packages/ari/chunk-635QCDHJ.mjs
import {
  JiraFormAriResourceOwner,
  JiraFormAriResourceType
} from "./chunk-UWOGO6UN.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira/form/manifest.ts
var jiraFormAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraFormAriResourceOwner,
  resourceType: JiraFormAriResourceType,
  resourceIdSlug: "activation/{activationId}/{formId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    formId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/form/index.ts
var JiraFormAri = class _JiraFormAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._formId = opts.resourceIdSegmentValues.formId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get formId() {
    return this._formId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraFormAriStaticOpts.qualifier,
      platformQualifier: jiraFormAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraFormAriStaticOpts.resourceOwner,
      resourceType: jiraFormAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.formId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        formId: opts.formId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraFormAriStaticOpts);
    return new _JiraFormAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraFormAriStaticOpts);
    return new _JiraFormAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      formId: this.formId
    };
  }
};

export {
  JiraFormAri
};
