// @forge/util/packages/ari/chunk-EXUTBA3J.mjs
import {
  PassionfruitDependencyUpdateAriResourceOwner,
  PassionfruitDependencyUpdateAriResourceType
} from "./chunk-HI6PTWWS.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/dependency-update/manifest.ts
var passionfruitDependencyUpdateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitDependencyUpdateAriResourceOwner,
  resourceType: PassionfruitDependencyUpdateAriResourceType,
  resourceIdSlug: "workspace/{activationId}/{dependencyId}/{dependencyUpdateId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    dependencyId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    dependencyUpdateId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/dependency-update/index.ts
var PassionfruitDependencyUpdateAri = class _PassionfruitDependencyUpdateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._dependencyId = opts.resourceIdSegmentValues.dependencyId, this._dependencyUpdateId = opts.resourceIdSegmentValues.dependencyUpdateId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get dependencyId() {
    return this._dependencyId;
  }
  get dependencyUpdateId() {
    return this._dependencyUpdateId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitDependencyUpdateAriStaticOpts.qualifier,
      platformQualifier: passionfruitDependencyUpdateAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: passionfruitDependencyUpdateAriStaticOpts.resourceOwner,
      resourceType: passionfruitDependencyUpdateAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.activationId}/${opts.dependencyId}/${opts.dependencyUpdateId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        dependencyId: opts.dependencyId,
        dependencyUpdateId: opts.dependencyUpdateId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitDependencyUpdateAriStaticOpts);
    return new _PassionfruitDependencyUpdateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitDependencyUpdateAriStaticOpts);
    return new _PassionfruitDependencyUpdateAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      dependencyId: this.dependencyId,
      dependencyUpdateId: this.dependencyUpdateId
    };
  }
};

export {
  PassionfruitDependencyUpdateAri
};
