// @forge/util/packages/ari/chunk-QOHF4ZZX.mjs
import {
  GoogleCalendarEventAriResourceOwner,
  GoogleCalendarEventAriResourceType
} from "./chunk-QEA64F52.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/google/calendar-event/manifest.ts
var googleCalendarEventAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GoogleCalendarEventAriResourceOwner,
  resourceType: GoogleCalendarEventAriResourceType,
  resourceIdSlug: "calendarId/{calendarId}/eventId/{eventId}",
  resourceIdSegmentFormats: {
    calendarId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    eventId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/google/calendar-event/index.ts
var GoogleCalendarEventAri = class _GoogleCalendarEventAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._calendarId = opts.resourceIdSegmentValues.calendarId, this._eventId = opts.resourceIdSegmentValues.eventId;
  }
  get calendarId() {
    return this._calendarId;
  }
  get eventId() {
    return this._eventId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googleCalendarEventAriStaticOpts.qualifier,
      platformQualifier: googleCalendarEventAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googleCalendarEventAriStaticOpts.resourceOwner,
      resourceType: googleCalendarEventAriStaticOpts.resourceType,
      resourceId: `calendarId/${opts.calendarId}/eventId/${opts.eventId}`,
      resourceIdSegmentValues: {
        calendarId: opts.calendarId,
        eventId: opts.eventId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googleCalendarEventAriStaticOpts);
    return new _GoogleCalendarEventAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googleCalendarEventAriStaticOpts);
    return new _GoogleCalendarEventAri(opts);
  }
  getVariables() {
    return {
      calendarId: this.calendarId,
      eventId: this.eventId
    };
  }
};

export {
  GoogleCalendarEventAri
};
