// @forge/util/packages/ari/chunk-VTHMKZL4.mjs
import {
  PassionfruitRoleAriResourceOwner,
  PassionfruitRoleAriResourceType
} from "./chunk-YYESCI2M.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/role/manifest.ts
var passionfruitRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitRoleAriResourceOwner,
  resourceType: PassionfruitRoleAriResourceType,
  resourceIdSlug: "product/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:admin|member|user-admin)/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/role/index.ts
var PassionfruitRoleAri = class _PassionfruitRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitRoleAriStaticOpts.qualifier,
      platformQualifier: passionfruitRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: passionfruitRoleAriStaticOpts.resourceOwner,
      resourceType: passionfruitRoleAriStaticOpts.resourceType,
      resourceId: `product/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitRoleAriStaticOpts);
    return new _PassionfruitRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitRoleAriStaticOpts);
    return new _PassionfruitRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

export {
  PassionfruitRoleAri
};
