// @forge/util/packages/ari/chunk-GQU6V3UK.mjs
import {
  MuralMuralRemoteLinkAriResourceOwner,
  MuralMuralRemoteLinkAriResourceType
} from "./chunk-JPXNYGEW.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/mural.mural/remote-link/manifest.ts
var muralMuralRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MuralMuralRemoteLinkAriResourceOwner,
  resourceType: MuralMuralRemoteLinkAriResourceType,
  resourceIdSlug: "design/{designId}",
  resourceIdSegmentFormats: {
    designId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/mural.mural/remote-link/index.ts
var MuralMuralRemoteLinkAri = class _MuralMuralRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._designId = opts.resourceIdSegmentValues.designId;
  }
  get designId() {
    return this._designId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: muralMuralRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: muralMuralRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: muralMuralRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: muralMuralRemoteLinkAriStaticOpts.resourceType,
      resourceId: `design/${opts.designId}`,
      resourceIdSegmentValues: {
        designId: opts.designId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, muralMuralRemoteLinkAriStaticOpts);
    return new _MuralMuralRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, muralMuralRemoteLinkAriStaticOpts);
    return new _MuralMuralRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      designId: this.designId
    };
  }
};

export {
  MuralMuralRemoteLinkAri
};
