// @forge/util/packages/ari/chunk-OKIMO5EO.mjs
import {
  ProformaRoleAriResourceOwner,
  ProformaRoleAriResourceType
} from "./chunk-ZNKCCQXV.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/proforma/role/manifest.ts
var proformaRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ProformaRoleAriResourceOwner,
  resourceType: ProformaRoleAriResourceType,
  resourceIdSlug: "form/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:responder|viewer)/
    // eslint-disable-line no-useless-escape
  }
};

// src/proforma/role/index.ts
var ProformaRoleAri = class _ProformaRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: proformaRoleAriStaticOpts.qualifier,
      platformQualifier: proformaRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: proformaRoleAriStaticOpts.resourceOwner,
      resourceType: proformaRoleAriStaticOpts.resourceType,
      resourceId: `form/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, proformaRoleAriStaticOpts);
    return new _ProformaRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, proformaRoleAriStaticOpts);
    return new _ProformaRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

export {
  ProformaRoleAri
};
