// @forge/util/packages/ari/chunk-QVEWE3BL.mjs
import {
  JiraDcSecondaryJiraDcSpaceAriResourceOwner,
  JiraDcSecondaryJiraDcSpaceAriResourceType
} from "./chunk-5Q5UL6WL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc-secondary.jira-dc/space/manifest.ts
var jiraDcSecondaryJiraDcSpaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcSecondaryJiraDcSpaceAriResourceOwner,
  resourceType: JiraDcSecondaryJiraDcSpaceAriResourceType,
  resourceIdSlug: "{siteId}/{spaceId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    spaceId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc-secondary.jira-dc/space/index.ts
var JiraDcSecondaryJiraDcSpaceAri = class _JiraDcSecondaryJiraDcSpaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._spaceId = opts.resourceIdSegmentValues.spaceId;
  }
  get siteId() {
    return this._siteId;
  }
  get spaceId() {
    return this._spaceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcSecondaryJiraDcSpaceAriStaticOpts.qualifier,
      platformQualifier: jiraDcSecondaryJiraDcSpaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcSecondaryJiraDcSpaceAriStaticOpts.resourceOwner,
      resourceType: jiraDcSecondaryJiraDcSpaceAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.spaceId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        spaceId: opts.spaceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcSecondaryJiraDcSpaceAriStaticOpts);
    return new _JiraDcSecondaryJiraDcSpaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcSecondaryJiraDcSpaceAriStaticOpts);
    return new _JiraDcSecondaryJiraDcSpaceAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      spaceId: this.spaceId
    };
  }
};

export {
  JiraDcSecondaryJiraDcSpaceAri
};
