// @forge/util/packages/ari/chunk-K424B5PK.mjs
import {
  JiraScreenAriResourceOwner,
  JiraScreenAriResourceType
} from "./chunk-WX6GALMK.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira/screen/manifest.ts
var jiraScreenAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraScreenAriResourceOwner,
  resourceType: JiraScreenAriResourceType,
  resourceIdSlug: "activation/{activationId}/{screenId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    screenId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/screen/index.ts
var JiraScreenAri = class _JiraScreenAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._screenId = opts.resourceIdSegmentValues.screenId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get screenId() {
    return this._screenId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraScreenAriStaticOpts.qualifier,
      platformQualifier: jiraScreenAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraScreenAriStaticOpts.resourceOwner,
      resourceType: jiraScreenAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.screenId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        screenId: opts.screenId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraScreenAriStaticOpts);
    return new _JiraScreenAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraScreenAriStaticOpts);
    return new _JiraScreenAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      screenId: this.screenId
    };
  }
};

export {
  JiraScreenAri
};
