// @forge/util/packages/ari/chunk-HGTZQJ3H.mjs
import {
  PassionfruitSiteAriResourceOwner,
  PassionfruitSiteAriResourceType
} from "./chunk-QLHTJETH.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/site/manifest.ts
var passionfruitSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitSiteAriResourceOwner,
  resourceType: PassionfruitSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/site/index.ts
var PassionfruitSiteAri = class _PassionfruitSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitSiteAriStaticOpts.qualifier,
      platformQualifier: passionfruitSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: passionfruitSiteAriStaticOpts.resourceOwner,
      resourceType: passionfruitSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitSiteAriStaticOpts);
    return new _PassionfruitSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitSiteAriStaticOpts);
    return new _PassionfruitSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  PassionfruitSiteAri
};
