// @forge/util/packages/ari/chunk-NT7VKYPC.mjs
import {
  PostOfficeTriggerAriResourceOwner,
  PostOfficeTriggerAriResourceType
} from "./chunk-VNEE4Y6C.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/post-office/trigger/manifest.ts
var postOfficeTriggerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PostOfficeTriggerAriResourceOwner,
  resourceType: PostOfficeTriggerAriResourceType,
  resourceIdSlug: "{messageTemplateId}/{messageIdempotencyKey}",
  resourceIdSegmentFormats: {
    messageTemplateId: /[a-z0-9-]{2,61}/,
    // eslint-disable-line no-useless-escape
    messageIdempotencyKey: /[a-zA-Z0-9_-]{8,64}/
    // eslint-disable-line no-useless-escape
  }
};

// src/post-office/trigger/index.ts
var PostOfficeTriggerAri = class _PostOfficeTriggerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._messageTemplateId = opts.resourceIdSegmentValues.messageTemplateId, this._messageIdempotencyKey = opts.resourceIdSegmentValues.messageIdempotencyKey;
  }
  get messageTemplateId() {
    return this._messageTemplateId;
  }
  get messageIdempotencyKey() {
    return this._messageIdempotencyKey;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: postOfficeTriggerAriStaticOpts.qualifier,
      platformQualifier: postOfficeTriggerAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: postOfficeTriggerAriStaticOpts.resourceOwner,
      resourceType: postOfficeTriggerAriStaticOpts.resourceType,
      resourceId: `${opts.messageTemplateId}/${opts.messageIdempotencyKey}`,
      resourceIdSegmentValues: {
        messageTemplateId: opts.messageTemplateId,
        messageIdempotencyKey: opts.messageIdempotencyKey
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, postOfficeTriggerAriStaticOpts);
    return new _PostOfficeTriggerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, postOfficeTriggerAriStaticOpts);
    return new _PostOfficeTriggerAri(opts);
  }
  getVariables() {
    return {
      messageTemplateId: this.messageTemplateId,
      messageIdempotencyKey: this.messageIdempotencyKey
    };
  }
};

export {
  PostOfficeTriggerAri
};
