// @forge/util/packages/ari/chunk-GNIYE5T5.mjs
import {
  TodoistTodoistRemoteLinkAriResourceOwner,
  TodoistTodoistRemoteLinkAriResourceType
} from "./chunk-JKOOQ576.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/todoist.todoist/remote-link/manifest.ts
var todoistTodoistRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TodoistTodoistRemoteLinkAriResourceOwner,
  resourceType: TodoistTodoistRemoteLinkAriResourceType,
  resourceIdSlug: "task/{taskId}",
  resourceIdSegmentFormats: {
    taskId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/todoist.todoist/remote-link/index.ts
var TodoistTodoistRemoteLinkAri = class _TodoistTodoistRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._taskId = opts.resourceIdSegmentValues.taskId;
  }
  get taskId() {
    return this._taskId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: todoistTodoistRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: todoistTodoistRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: todoistTodoistRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: todoistTodoistRemoteLinkAriStaticOpts.resourceType,
      resourceId: `task/${opts.taskId}`,
      resourceIdSegmentValues: {
        taskId: opts.taskId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, todoistTodoistRemoteLinkAriStaticOpts);
    return new _TodoistTodoistRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, todoistTodoistRemoteLinkAriStaticOpts);
    return new _TodoistTodoistRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      taskId: this.taskId
    };
  }
};

export {
  TodoistTodoistRemoteLinkAri
};
