// @forge/util/packages/ari/chunk-6ZB6ECVZ.mjs
import {
  UnifiedHelpSiteAriResourceOwner,
  UnifiedHelpSiteAriResourceType
} from "./chunk-OKKAA5RV.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/unified-help/site/manifest.ts
var unifiedHelpSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: UnifiedHelpSiteAriResourceOwner,
  resourceType: UnifiedHelpSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/unified-help/site/index.ts
var UnifiedHelpSiteAri = class _UnifiedHelpSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: unifiedHelpSiteAriStaticOpts.qualifier,
      platformQualifier: unifiedHelpSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: unifiedHelpSiteAriStaticOpts.resourceOwner,
      resourceType: unifiedHelpSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, unifiedHelpSiteAriStaticOpts);
    return new _UnifiedHelpSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, unifiedHelpSiteAriStaticOpts);
    return new _UnifiedHelpSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  UnifiedHelpSiteAri
};
