// @forge/util/packages/ari/chunk-ESXRJUNW.mjs
import {
  EcosystemConnectModuleAriResourceOwner,
  EcosystemConnectModuleAriResourceType
} from "./chunk-4UCHZIA4.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/ecosystem/connect-module/manifest.ts
var ecosystemConnectModuleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: EcosystemConnectModuleAriResourceOwner,
  resourceType: EcosystemConnectModuleAriResourceType,
  resourceIdSlug: "{appKey}/{moduleKey}",
  resourceIdSegmentFormats: {
    appKey: /[a-zA-Z0-9-._]+/,
    // eslint-disable-line no-useless-escape
    moduleKey: /[a-zA-Z0-9-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/ecosystem/connect-module/index.ts
var EcosystemConnectModuleAri = class _EcosystemConnectModuleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._appKey = opts.resourceIdSegmentValues.appKey, this._moduleKey = opts.resourceIdSegmentValues.moduleKey;
  }
  get appKey() {
    return this._appKey;
  }
  get moduleKey() {
    return this._moduleKey;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: ecosystemConnectModuleAriStaticOpts.qualifier,
      platformQualifier: ecosystemConnectModuleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: ecosystemConnectModuleAriStaticOpts.resourceOwner,
      resourceType: ecosystemConnectModuleAriStaticOpts.resourceType,
      resourceId: `${opts.appKey}/${opts.moduleKey}`,
      resourceIdSegmentValues: {
        appKey: opts.appKey,
        moduleKey: opts.moduleKey
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, ecosystemConnectModuleAriStaticOpts);
    return new _EcosystemConnectModuleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, ecosystemConnectModuleAriStaticOpts);
    return new _EcosystemConnectModuleAri(opts);
  }
  getVariables() {
    return {
      appKey: this.appKey,
      moduleKey: this.moduleKey
    };
  }
};

export {
  EcosystemConnectModuleAri
};
