// @forge/util/packages/ari/chunk-GGIQW4Z4.mjs
import {
  JiraDcTertiaryJiraDcProjectAriResourceOwner,
  JiraDcTertiaryJiraDcProjectAriResourceType
} from "./chunk-EBM6ALFK.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc-tertiary.jira-dc/project/manifest.ts
var jiraDcTertiaryJiraDcProjectAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcTertiaryJiraDcProjectAriResourceOwner,
  resourceType: JiraDcTertiaryJiraDcProjectAriResourceType,
  resourceIdSlug: "{siteId}/{projectId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc-tertiary.jira-dc/project/index.ts
var JiraDcTertiaryJiraDcProjectAri = class _JiraDcTertiaryJiraDcProjectAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._projectId = opts.resourceIdSegmentValues.projectId;
  }
  get siteId() {
    return this._siteId;
  }
  get projectId() {
    return this._projectId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcTertiaryJiraDcProjectAriStaticOpts.qualifier,
      platformQualifier: jiraDcTertiaryJiraDcProjectAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcTertiaryJiraDcProjectAriStaticOpts.resourceOwner,
      resourceType: jiraDcTertiaryJiraDcProjectAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.projectId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        projectId: opts.projectId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcTertiaryJiraDcProjectAriStaticOpts);
    return new _JiraDcTertiaryJiraDcProjectAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcTertiaryJiraDcProjectAriStaticOpts);
    return new _JiraDcTertiaryJiraDcProjectAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      projectId: this.projectId
    };
  }
};

export {
  JiraDcTertiaryJiraDcProjectAri
};
