// @forge/util/packages/ari/chunk-U7RQ3KPV.mjs
import {
  DevaiAutodevJobAriResourceOwner,
  DevaiAutodevJobAriResourceType
} from "./chunk-3DEYGGN2.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/devai/autodev-job/manifest.ts
var devaiAutodevJobAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: DevaiAutodevJobAriResourceOwner,
  resourceType: DevaiAutodevJobAriResourceType,
  resourceIdSlug: "{workspaceId}/{jobId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    jobId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/devai/autodev-job/index.ts
var DevaiAutodevJobAri = class _DevaiAutodevJobAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._jobId = opts.resourceIdSegmentValues.jobId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get jobId() {
    return this._jobId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: devaiAutodevJobAriStaticOpts.qualifier,
      platformQualifier: devaiAutodevJobAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: devaiAutodevJobAriStaticOpts.resourceOwner,
      resourceType: devaiAutodevJobAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.jobId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        jobId: opts.jobId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, devaiAutodevJobAriStaticOpts);
    return new _DevaiAutodevJobAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, devaiAutodevJobAriStaticOpts);
    return new _DevaiAutodevJobAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      jobId: this.jobId
    };
  }
};

export {
  DevaiAutodevJobAri
};
