// @forge/util/packages/ari/chunk-RBBGLGZT.mjs
import {
  DovetailDovetailRemoteLinkAriResourceOwner,
  DovetailDovetailRemoteLinkAriResourceType
} from "./chunk-IQXXUNUN.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/dovetail.dovetail/remote-link/manifest.ts
var dovetailDovetailRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: DovetailDovetailRemoteLinkAriResourceOwner,
  resourceType: DovetailDovetailRemoteLinkAriResourceType,
  resourceIdSlug: "org/{orgId}/note/{noteId}",
  resourceIdSegmentFormats: {
    orgId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    noteId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/dovetail.dovetail/remote-link/index.ts
var DovetailDovetailRemoteLinkAri = class _DovetailDovetailRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._orgId = opts.resourceIdSegmentValues.orgId, this._noteId = opts.resourceIdSegmentValues.noteId;
  }
  get orgId() {
    return this._orgId;
  }
  get noteId() {
    return this._noteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: dovetailDovetailRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: dovetailDovetailRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: dovetailDovetailRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: dovetailDovetailRemoteLinkAriStaticOpts.resourceType,
      resourceId: `org/${opts.orgId}/note/${opts.noteId}`,
      resourceIdSegmentValues: {
        orgId: opts.orgId,
        noteId: opts.noteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, dovetailDovetailRemoteLinkAriStaticOpts);
    return new _DovetailDovetailRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, dovetailDovetailRemoteLinkAriStaticOpts);
    return new _DovetailDovetailRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      orgId: this.orgId,
      noteId: this.noteId
    };
  }
};

export {
  DovetailDovetailRemoteLinkAri
};
