// @forge/util/packages/ari/chunk-HDJ6C7QB.mjs
import {
  CanvaCanvaRemoteLinkAriResourceOwner,
  CanvaCanvaRemoteLinkAriResourceType
} from "./chunk-4JERSUI5.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/canva.canva/remote-link/manifest.ts
var canvaCanvaRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: CanvaCanvaRemoteLinkAriResourceOwner,
  resourceType: CanvaCanvaRemoteLinkAriResourceType,
  resourceIdSlug: "designId/{designId}",
  resourceIdSegmentFormats: {
    designId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/canva.canva/remote-link/index.ts
var CanvaCanvaRemoteLinkAri = class _CanvaCanvaRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._designId = opts.resourceIdSegmentValues.designId;
  }
  get designId() {
    return this._designId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: canvaCanvaRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: canvaCanvaRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: canvaCanvaRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: canvaCanvaRemoteLinkAriStaticOpts.resourceType,
      resourceId: `designId/${opts.designId}`,
      resourceIdSegmentValues: {
        designId: opts.designId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, canvaCanvaRemoteLinkAriStaticOpts);
    return new _CanvaCanvaRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, canvaCanvaRemoteLinkAriStaticOpts);
    return new _CanvaCanvaRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      designId: this.designId
    };
  }
};

export {
  CanvaCanvaRemoteLinkAri
};
