// @forge/util/packages/ari/chunk-Y7ZOW42H.mjs
import {
  PureRoleAriResourceOwner,
  PureRoleAriResourceType
} from "./chunk-4QXPSVRT.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/pure/role/manifest.ts
var pureRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PureRoleAriResourceOwner,
  resourceType: PureRoleAriResourceType,
  resourceIdSlug: "{roleId}",
  resourceIdSegmentFormats: {
    roleId: /(?:partner-admin|partner-sales|partner-sales-rep|marketing|marketing-admin)/
    // eslint-disable-line no-useless-escape
  }
};

// src/pure/role/index.ts
var PureRoleAri = class _PureRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleId = opts.resourceIdSegmentValues.roleId;
  }
  get roleId() {
    return this._roleId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: pureRoleAriStaticOpts.qualifier,
      platformQualifier: pureRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: pureRoleAriStaticOpts.resourceOwner,
      resourceType: pureRoleAriStaticOpts.resourceType,
      resourceId: `${opts.roleId}`,
      resourceIdSegmentValues: {
        roleId: opts.roleId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, pureRoleAriStaticOpts);
    return new _PureRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, pureRoleAriStaticOpts);
    return new _PureRoleAri(opts);
  }
  getVariables() {
    return {
      roleId: this.roleId
    };
  }
};

export {
  PureRoleAri
};
