// @forge/util/packages/ari/chunk-N73WVBAG.mjs
import {
  DocusignDocusignRemoteLinkAriResourceOwner,
  DocusignDocusignRemoteLinkAriResourceType
} from "./chunk-ELGAGBXE.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/docusign.docusign/remote-link/manifest.ts
var docusignDocusignRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: DocusignDocusignRemoteLinkAriResourceOwner,
  resourceType: DocusignDocusignRemoteLinkAriResourceType,
  resourceIdSlug: "shard/{shardRegion}/account/{accountId}/envelope/{envelopeId}",
  resourceIdSegmentFormats: {
    shardRegion: /[a-zA-Z0-9_-]+/,
    // eslint-disable-line no-useless-escape
    accountId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    envelopeId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/docusign.docusign/remote-link/index.ts
var DocusignDocusignRemoteLinkAri = class _DocusignDocusignRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._shardRegion = opts.resourceIdSegmentValues.shardRegion, this._accountId = opts.resourceIdSegmentValues.accountId, this._envelopeId = opts.resourceIdSegmentValues.envelopeId;
  }
  get shardRegion() {
    return this._shardRegion;
  }
  get accountId() {
    return this._accountId;
  }
  get envelopeId() {
    return this._envelopeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: docusignDocusignRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: docusignDocusignRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: docusignDocusignRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: docusignDocusignRemoteLinkAriStaticOpts.resourceType,
      resourceId: `shard/${opts.shardRegion}/account/${opts.accountId}/envelope/${opts.envelopeId}`,
      resourceIdSegmentValues: {
        shardRegion: opts.shardRegion,
        accountId: opts.accountId,
        envelopeId: opts.envelopeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, docusignDocusignRemoteLinkAriStaticOpts);
    return new _DocusignDocusignRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, docusignDocusignRemoteLinkAriStaticOpts);
    return new _DocusignDocusignRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      shardRegion: this.shardRegion,
      accountId: this.accountId,
      envelopeId: this.envelopeId
    };
  }
};

export {
  DocusignDocusignRemoteLinkAri
};
