// @forge/util/packages/ari/chunk-RBPTEEQE.mjs
import {
  ConfluenceDcConfluenceDcSpaceAriResourceOwner,
  ConfluenceDcConfluenceDcSpaceAriResourceType
} from "./chunk-F4ZZPTKG.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/confluence-dc.confluence-dc/space/manifest.ts
var confluenceDcConfluenceDcSpaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ConfluenceDcConfluenceDcSpaceAriResourceOwner,
  resourceType: ConfluenceDcConfluenceDcSpaceAriResourceType,
  resourceIdSlug: "{siteId}/{spaceId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    spaceId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/confluence-dc.confluence-dc/space/index.ts
var ConfluenceDcConfluenceDcSpaceAri = class _ConfluenceDcConfluenceDcSpaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._spaceId = opts.resourceIdSegmentValues.spaceId;
  }
  get siteId() {
    return this._siteId;
  }
  get spaceId() {
    return this._spaceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: confluenceDcConfluenceDcSpaceAriStaticOpts.qualifier,
      platformQualifier: confluenceDcConfluenceDcSpaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: confluenceDcConfluenceDcSpaceAriStaticOpts.resourceOwner,
      resourceType: confluenceDcConfluenceDcSpaceAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.spaceId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        spaceId: opts.spaceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, confluenceDcConfluenceDcSpaceAriStaticOpts);
    return new _ConfluenceDcConfluenceDcSpaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, confluenceDcConfluenceDcSpaceAriStaticOpts);
    return new _ConfluenceDcConfluenceDcSpaceAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      spaceId: this.spaceId
    };
  }
};

export {
  ConfluenceDcConfluenceDcSpaceAri
};
