// @forge/util/packages/ari/chunk-6KII5J3E.mjs
import {
  RovoScenarioAriResourceOwner,
  RovoScenarioAriResourceType
} from "./chunk-XBHB5O4E.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/rovo/scenario/manifest.ts
var rovoScenarioAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: RovoScenarioAriResourceOwner,
  resourceType: RovoScenarioAriResourceType,
  resourceIdSlug: "activation/{activationId}/{scenarioId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    scenarioId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/rovo/scenario/index.ts
var RovoScenarioAri = class _RovoScenarioAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._scenarioId = opts.resourceIdSegmentValues.scenarioId;
  }
  get activationId() {
    return this._activationId;
  }
  get scenarioId() {
    return this._scenarioId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: rovoScenarioAriStaticOpts.qualifier,
      platformQualifier: rovoScenarioAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: rovoScenarioAriStaticOpts.resourceOwner,
      resourceType: rovoScenarioAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.scenarioId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        scenarioId: opts.scenarioId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, rovoScenarioAriStaticOpts);
    return new _RovoScenarioAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, rovoScenarioAriStaticOpts);
    return new _RovoScenarioAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      scenarioId: this.scenarioId
    };
  }
};

export {
  RovoScenarioAri
};
