// @forge/util/packages/ari/chunk-V5HUGOZP.mjs
import {
  MicrosoftOutlookMessageAriResourceOwner,
  MicrosoftOutlookMessageAriResourceType
} from "./chunk-KOCC56FS.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft.outlook/message/manifest.ts
var microsoftOutlookMessageAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftOutlookMessageAriResourceOwner,
  resourceType: MicrosoftOutlookMessageAriResourceType,
  resourceIdSlug: "{messageId}",
  resourceIdSegmentFormats: {
    messageId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft.outlook/message/index.ts
var MicrosoftOutlookMessageAri = class _MicrosoftOutlookMessageAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._messageId = opts.resourceIdSegmentValues.messageId;
  }
  get messageId() {
    return this._messageId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftOutlookMessageAriStaticOpts.qualifier,
      platformQualifier: microsoftOutlookMessageAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftOutlookMessageAriStaticOpts.resourceOwner,
      resourceType: microsoftOutlookMessageAriStaticOpts.resourceType,
      resourceId: `${opts.messageId}`,
      resourceIdSegmentValues: {
        messageId: opts.messageId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftOutlookMessageAriStaticOpts);
    return new _MicrosoftOutlookMessageAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftOutlookMessageAriStaticOpts);
    return new _MicrosoftOutlookMessageAri(opts);
  }
  getVariables() {
    return {
      messageId: this.messageId
    };
  }
};

export {
  MicrosoftOutlookMessageAri
};
