// @forge/util/packages/ari/chunk-RZCM2CZO.mjs
import {
  TrelloPlannerEventCardAriResourceOwner,
  TrelloPlannerEventCardAriResourceType
} from "./chunk-I4CIEUHI.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/trello/planner-event-card/manifest.ts
var trelloPlannerEventCardAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TrelloPlannerEventCardAriResourceOwner,
  resourceType: TrelloPlannerEventCardAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{plannerEventCardId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-_.]*/,
    // eslint-disable-line no-useless-escape
    plannerEventCardId: /[a-zA-Z0-9\-_.]*/
    // eslint-disable-line no-useless-escape
  }
};

// src/trello/planner-event-card/index.ts
var TrelloPlannerEventCardAri = class _TrelloPlannerEventCardAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._plannerEventCardId = opts.resourceIdSegmentValues.plannerEventCardId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get plannerEventCardId() {
    return this._plannerEventCardId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: trelloPlannerEventCardAriStaticOpts.qualifier,
      platformQualifier: trelloPlannerEventCardAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: trelloPlannerEventCardAriStaticOpts.resourceOwner,
      resourceType: trelloPlannerEventCardAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.plannerEventCardId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId || "",
        plannerEventCardId: opts.plannerEventCardId || ""
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, trelloPlannerEventCardAriStaticOpts);
    return new _TrelloPlannerEventCardAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, trelloPlannerEventCardAriStaticOpts);
    return new _TrelloPlannerEventCardAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      plannerEventCardId: this.plannerEventCardId
    };
  }
};

export {
  TrelloPlannerEventCardAri
};
