// @forge/util/packages/ari/chunk-7YLRCETH.mjs
import {
  GraphProjectHistoryAriResourceOwner,
  GraphProjectHistoryAriResourceType
} from "./chunk-FKSVEAW3.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/project-history/manifest.ts
var graphProjectHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphProjectHistoryAriResourceOwner,
  resourceType: GraphProjectHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{projectId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/project-history/index.ts
var GraphProjectHistoryAri = class _GraphProjectHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._projectId = opts.resourceIdSegmentValues.projectId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get projectId() {
    return this._projectId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphProjectHistoryAriStaticOpts.qualifier,
      platformQualifier: graphProjectHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphProjectHistoryAriStaticOpts.resourceOwner,
      resourceType: graphProjectHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.projectId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        projectId: opts.projectId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphProjectHistoryAriStaticOpts);
    return new _GraphProjectHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphProjectHistoryAriStaticOpts);
    return new _GraphProjectHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      projectId: this.projectId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphProjectHistoryAri
};
