// @forge/util/packages/ari/chunk-RNK5J6WA.mjs
import {
  TownsquareMetricAriResourceOwner,
  TownsquareMetricAriResourceType
} from "./chunk-XKTWZDLI.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/metric/manifest.ts
var townsquareMetricAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareMetricAriResourceOwner,
  resourceType: TownsquareMetricAriResourceType,
  resourceIdSlug: "{metricUuid}",
  resourceIdSegmentFormats: {
    metricUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/metric/index.ts
var TownsquareMetricAri = class _TownsquareMetricAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._metricUuid = opts.resourceIdSegmentValues.metricUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get metricUuid() {
    return this._metricUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareMetricAriStaticOpts.qualifier,
      platformQualifier: townsquareMetricAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareMetricAriStaticOpts.resourceOwner,
      resourceType: townsquareMetricAriStaticOpts.resourceType,
      resourceId: `${opts.metricUuid}`,
      resourceIdSegmentValues: {
        metricUuid: opts.metricUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareMetricAriStaticOpts);
    return new _TownsquareMetricAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareMetricAriStaticOpts);
    return new _TownsquareMetricAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      metricUuid: this.metricUuid
    };
  }
};

export {
  TownsquareMetricAri
};
