// @forge/util/packages/ari/chunk-E5ZA5FLJ.mjs
import {
  WorkdayWorkdayWorkerAriResourceOwner,
  WorkdayWorkdayWorkerAriResourceType
} from "./chunk-AT3AV22J.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/workday.workday/worker/manifest.ts
var workdayWorkdayWorkerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: WorkdayWorkdayWorkerAriResourceOwner,
  resourceType: WorkdayWorkdayWorkerAriResourceType,
  resourceIdSlug: "{workdayId}",
  resourceIdSegmentFormats: {
    workdayId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday.workday/worker/index.ts
var WorkdayWorkdayWorkerAri = class _WorkdayWorkdayWorkerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayId = opts.resourceIdSegmentValues.workdayId;
  }
  get workdayId() {
    return this._workdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayWorkdayWorkerAriStaticOpts.qualifier,
      platformQualifier: workdayWorkdayWorkerAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayWorkdayWorkerAriStaticOpts.resourceOwner,
      resourceType: workdayWorkdayWorkerAriStaticOpts.resourceType,
      resourceId: `${opts.workdayId}`,
      resourceIdSegmentValues: {
        workdayId: opts.workdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayWorkdayWorkerAriStaticOpts);
    return new _WorkdayWorkdayWorkerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayWorkdayWorkerAriStaticOpts);
    return new _WorkdayWorkdayWorkerAri(opts);
  }
  getVariables() {
    return {
      workdayId: this.workdayId
    };
  }
};

export {
  WorkdayWorkdayWorkerAri
};
