// @forge/util/packages/ari/chunk-L4LDQPBH.mjs
import {
  SalesforceSalesforceCustomerOrgAriResourceOwner,
  SalesforceSalesforceCustomerOrgAriResourceType
} from "./chunk-AVGRLOXF.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/salesforce.salesforce/customer-org/manifest.ts
var salesforceSalesforceCustomerOrgAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: SalesforceSalesforceCustomerOrgAriResourceOwner,
  resourceType: SalesforceSalesforceCustomerOrgAriResourceType,
  resourceIdSlug: "org/{instanceDomain}/account/{id}",
  resourceIdSegmentFormats: {
    instanceDomain: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    id: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/salesforce.salesforce/customer-org/index.ts
var SalesforceSalesforceCustomerOrgAri = class _SalesforceSalesforceCustomerOrgAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._instanceDomain = opts.resourceIdSegmentValues.instanceDomain, this._id = opts.resourceIdSegmentValues.id;
  }
  get instanceDomain() {
    return this._instanceDomain;
  }
  get id() {
    return this._id;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: salesforceSalesforceCustomerOrgAriStaticOpts.qualifier,
      platformQualifier: salesforceSalesforceCustomerOrgAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: salesforceSalesforceCustomerOrgAriStaticOpts.resourceOwner,
      resourceType: salesforceSalesforceCustomerOrgAriStaticOpts.resourceType,
      resourceId: `org/${opts.instanceDomain}/account/${opts.id}`,
      resourceIdSegmentValues: {
        instanceDomain: opts.instanceDomain,
        id: opts.id
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, salesforceSalesforceCustomerOrgAriStaticOpts);
    return new _SalesforceSalesforceCustomerOrgAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, salesforceSalesforceCustomerOrgAriStaticOpts);
    return new _SalesforceSalesforceCustomerOrgAri(opts);
  }
  getVariables() {
    return {
      instanceDomain: this.instanceDomain,
      id: this.id
    };
  }
};

export {
  SalesforceSalesforceCustomerOrgAri
};
