// @forge/util/packages/ari/chunk-ATB4PB3W.mjs
import {
  ZeplinZeplinRemoteLinkAriResourceOwner,
  ZeplinZeplinRemoteLinkAriResourceType
} from "./chunk-ONT6HQKB.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/zeplin.zeplin/remote-link/manifest.ts
var zeplinZeplinRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ZeplinZeplinRemoteLinkAriResourceOwner,
  resourceType: ZeplinZeplinRemoteLinkAriResourceType,
  resourceIdSlug: "{remoteLinkId}",
  resourceIdSegmentFormats: {
    remoteLinkId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/zeplin.zeplin/remote-link/index.ts
var ZeplinZeplinRemoteLinkAri = class _ZeplinZeplinRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._remoteLinkId = opts.resourceIdSegmentValues.remoteLinkId;
  }
  get remoteLinkId() {
    return this._remoteLinkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: zeplinZeplinRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: zeplinZeplinRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: zeplinZeplinRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: zeplinZeplinRemoteLinkAriStaticOpts.resourceType,
      resourceId: `${opts.remoteLinkId}`,
      resourceIdSegmentValues: {
        remoteLinkId: opts.remoteLinkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, zeplinZeplinRemoteLinkAriStaticOpts);
    return new _ZeplinZeplinRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, zeplinZeplinRemoteLinkAriStaticOpts);
    return new _ZeplinZeplinRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      remoteLinkId: this.remoteLinkId
    };
  }
};

export {
  ZeplinZeplinRemoteLinkAri
};
