// @forge/util/packages/ari/chunk-EMSLGZKT.mjs
import {
  ConvoAiWorkspaceAriResourceOwner,
  ConvoAiWorkspaceAriResourceType
} from "./chunk-E3D6AV43.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/convo-ai/workspace/manifest.ts
var convoAiWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ConvoAiWorkspaceAriResourceOwner,
  resourceType: ConvoAiWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/convo-ai/workspace/index.ts
var ConvoAiWorkspaceAri = class _ConvoAiWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: convoAiWorkspaceAriStaticOpts.qualifier,
      platformQualifier: convoAiWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: convoAiWorkspaceAriStaticOpts.resourceOwner,
      resourceType: convoAiWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, convoAiWorkspaceAriStaticOpts);
    return new _ConvoAiWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, convoAiWorkspaceAriStaticOpts);
    return new _ConvoAiWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId
    };
  }
};

export {
  ConvoAiWorkspaceAri
};
