// @forge/util/packages/ari/chunk-YTKEQOWN.mjs
import {
  LucidLucidchartDesignAriResourceOwner,
  LucidLucidchartDesignAriResourceType
} from "./chunk-JCHNYGGY.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/lucid.lucidchart/design/manifest.ts
var lucidLucidchartDesignAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: LucidLucidchartDesignAriResourceOwner,
  resourceType: LucidLucidchartDesignAriResourceType,
  resourceIdSlug: "document/{documentId}",
  resourceIdSegmentFormats: {
    documentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/lucid.lucidchart/design/index.ts
var LucidLucidchartDesignAri = class _LucidLucidchartDesignAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._documentId = opts.resourceIdSegmentValues.documentId;
  }
  get documentId() {
    return this._documentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: lucidLucidchartDesignAriStaticOpts.qualifier,
      platformQualifier: lucidLucidchartDesignAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: lucidLucidchartDesignAriStaticOpts.resourceOwner,
      resourceType: lucidLucidchartDesignAriStaticOpts.resourceType,
      resourceId: `document/${opts.documentId}`,
      resourceIdSegmentValues: {
        documentId: opts.documentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, lucidLucidchartDesignAriStaticOpts);
    return new _LucidLucidchartDesignAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, lucidLucidchartDesignAriStaticOpts);
    return new _LucidLucidchartDesignAri(opts);
  }
  getVariables() {
    return {
      documentId: this.documentId
    };
  }
};

export {
  LucidLucidchartDesignAri
};
