// @forge/util/packages/ari/chunk-SKW53Q5X.mjs
import {
  RadarPositionAriResourceOwner,
  RadarPositionAriResourceType
} from "./chunk-L3QKGWQW.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/radar/position/manifest.ts
var radarPositionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: RadarPositionAriResourceOwner,
  resourceType: RadarPositionAriResourceType,
  resourceIdSlug: "{workspaceId}/{positionId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    positionId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/radar/position/index.ts
var RadarPositionAri = class _RadarPositionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._positionId = opts.resourceIdSegmentValues.positionId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get positionId() {
    return this._positionId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: radarPositionAriStaticOpts.qualifier,
      platformQualifier: radarPositionAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: radarPositionAriStaticOpts.resourceOwner,
      resourceType: radarPositionAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.positionId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        positionId: opts.positionId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, radarPositionAriStaticOpts);
    return new _RadarPositionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, radarPositionAriStaticOpts);
    return new _RadarPositionAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      positionId: this.positionId
    };
  }
};

export {
  RadarPositionAri
};
