// @forge/util/packages/ari/chunk-3RZZ3UVL.mjs
import {
  JiraPlaybookStepRunAriResourceOwner,
  JiraPlaybookStepRunAriResourceType
} from "./chunk-CPRWGSIC.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira/playbook-step-run/manifest.ts
var jiraPlaybookStepRunAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraPlaybookStepRunAriResourceOwner,
  resourceType: JiraPlaybookStepRunAriResourceType,
  resourceIdSlug: "activation/{activationId}/{playbookStepRunId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    playbookStepRunId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/playbook-step-run/index.ts
var JiraPlaybookStepRunAri = class _JiraPlaybookStepRunAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._playbookStepRunId = opts.resourceIdSegmentValues.playbookStepRunId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get playbookStepRunId() {
    return this._playbookStepRunId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraPlaybookStepRunAriStaticOpts.qualifier,
      platformQualifier: jiraPlaybookStepRunAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraPlaybookStepRunAriStaticOpts.resourceOwner,
      resourceType: jiraPlaybookStepRunAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.playbookStepRunId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        playbookStepRunId: opts.playbookStepRunId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraPlaybookStepRunAriStaticOpts);
    return new _JiraPlaybookStepRunAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraPlaybookStepRunAriStaticOpts);
    return new _JiraPlaybookStepRunAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      playbookStepRunId: this.playbookStepRunId
    };
  }
};

export {
  JiraPlaybookStepRunAri
};
