// @forge/util/packages/ari/chunk-Y4N4DAWM.mjs
import {
  GraphRoleAriResourceOwner,
  GraphRoleAriResourceType
} from "./chunk-RPUSNHXA.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/role/manifest.ts
var graphRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphRoleAriResourceOwner,
  resourceType: GraphRoleAriResourceType,
  resourceIdSlug: "product/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:member|admin)/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/role/index.ts
var GraphRoleAri = class _GraphRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphRoleAriStaticOpts.qualifier,
      platformQualifier: graphRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphRoleAriStaticOpts.resourceOwner,
      resourceType: graphRoleAriStaticOpts.resourceType,
      resourceId: `product/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphRoleAriStaticOpts);
    return new _GraphRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphRoleAriStaticOpts);
    return new _GraphRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

export {
  GraphRoleAri
};
