// @forge/util/packages/ari/chunk-CWTC3RY3.mjs
import {
  TownsquareUpdateNoteAriResourceOwner,
  TownsquareUpdateNoteAriResourceType
} from "./chunk-RLUADO33.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/update-note/manifest.ts
var townsquareUpdateNoteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareUpdateNoteAriResourceOwner,
  resourceType: TownsquareUpdateNoteAriResourceType,
  resourceIdSlug: "{updateNoteUuid}",
  resourceIdSegmentFormats: {
    updateNoteUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/update-note/index.ts
var TownsquareUpdateNoteAri = class _TownsquareUpdateNoteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._updateNoteUuid = opts.resourceIdSegmentValues.updateNoteUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get updateNoteUuid() {
    return this._updateNoteUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareUpdateNoteAriStaticOpts.qualifier,
      platformQualifier: townsquareUpdateNoteAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareUpdateNoteAriStaticOpts.resourceOwner,
      resourceType: townsquareUpdateNoteAriStaticOpts.resourceType,
      resourceId: `${opts.updateNoteUuid}`,
      resourceIdSegmentValues: {
        updateNoteUuid: opts.updateNoteUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareUpdateNoteAriStaticOpts);
    return new _TownsquareUpdateNoteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareUpdateNoteAriStaticOpts);
    return new _TownsquareUpdateNoteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      updateNoteUuid: this.updateNoteUuid
    };
  }
};

export {
  TownsquareUpdateNoteAri
};
