// @forge/util/packages/ari/chunk-3LNAA5VF.mjs
import {
  JiraCustomerServiceRoleAriResourceOwner,
  JiraCustomerServiceRoleAriResourceType
} from "./chunk-HDG7BPFA.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-customer-service/role/manifest.ts
var jiraCustomerServiceRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraCustomerServiceRoleAriResourceOwner,
  resourceType: JiraCustomerServiceRoleAriResourceType,
  resourceIdSlug: "product/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:member|admin|user-admin|customer)/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-customer-service/role/index.ts
var JiraCustomerServiceRoleAri = class _JiraCustomerServiceRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraCustomerServiceRoleAriStaticOpts.qualifier,
      platformQualifier: jiraCustomerServiceRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraCustomerServiceRoleAriStaticOpts.resourceOwner,
      resourceType: jiraCustomerServiceRoleAriStaticOpts.resourceType,
      resourceId: `product/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraCustomerServiceRoleAriStaticOpts);
    return new _JiraCustomerServiceRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraCustomerServiceRoleAriStaticOpts);
    return new _JiraCustomerServiceRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

export {
  JiraCustomerServiceRoleAri
};
