// @forge/util/packages/ari/chunk-AXRF624T.mjs
import {
  JiraDcJiraDcProjectAriResourceOwner,
  JiraDcJiraDcProjectAriResourceType
} from "./chunk-VTX7JY4Y.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc.jira-dc/project/manifest.ts
var jiraDcJiraDcProjectAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcJiraDcProjectAriResourceOwner,
  resourceType: JiraDcJiraDcProjectAriResourceType,
  resourceIdSlug: "{siteId}/{projectId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc.jira-dc/project/index.ts
var JiraDcJiraDcProjectAri = class _JiraDcJiraDcProjectAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._projectId = opts.resourceIdSegmentValues.projectId;
  }
  get siteId() {
    return this._siteId;
  }
  get projectId() {
    return this._projectId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcJiraDcProjectAriStaticOpts.qualifier,
      platformQualifier: jiraDcJiraDcProjectAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcJiraDcProjectAriStaticOpts.resourceOwner,
      resourceType: jiraDcJiraDcProjectAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.projectId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        projectId: opts.projectId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcJiraDcProjectAriStaticOpts);
    return new _JiraDcJiraDcProjectAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcJiraDcProjectAriStaticOpts);
    return new _JiraDcJiraDcProjectAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      projectId: this.projectId
    };
  }
};

export {
  JiraDcJiraDcProjectAri
};
