// @forge/util/packages/ari/chunk-YWH2Q5D3.mjs
import {
  TownsquareProjectUpdateAriResourceOwner,
  TownsquareProjectUpdateAriResourceType
} from "./chunk-MZER64KW.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/project-update/manifest.ts
var townsquareProjectUpdateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareProjectUpdateAriResourceOwner,
  resourceType: TownsquareProjectUpdateAriResourceType,
  resourceIdSlug: "{projectUpdateUuid}",
  resourceIdSegmentFormats: {
    projectUpdateUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/project-update/index.ts
var TownsquareProjectUpdateAri = class _TownsquareProjectUpdateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._projectUpdateUuid = opts.resourceIdSegmentValues.projectUpdateUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get projectUpdateUuid() {
    return this._projectUpdateUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareProjectUpdateAriStaticOpts.qualifier,
      platformQualifier: townsquareProjectUpdateAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareProjectUpdateAriStaticOpts.resourceOwner,
      resourceType: townsquareProjectUpdateAriStaticOpts.resourceType,
      resourceId: `${opts.projectUpdateUuid}`,
      resourceIdSegmentValues: {
        projectUpdateUuid: opts.projectUpdateUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareProjectUpdateAriStaticOpts);
    return new _TownsquareProjectUpdateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareProjectUpdateAriStaticOpts);
    return new _TownsquareProjectUpdateAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      projectUpdateUuid: this.projectUpdateUuid
    };
  }
};

export {
  TownsquareProjectUpdateAri
};
