// @forge/util/packages/ari/chunk-XNAISYEC.mjs
import {
  MercuryChangeProposalAriResourceOwner,
  MercuryChangeProposalAriResourceType
} from "./chunk-P7WJCT2Q.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/mercury/change-proposal/manifest.ts
var mercuryChangeProposalAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MercuryChangeProposalAriResourceOwner,
  resourceType: MercuryChangeProposalAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{changeProposalId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    changeProposalId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/mercury/change-proposal/index.ts
var MercuryChangeProposalAri = class _MercuryChangeProposalAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._changeProposalId = opts.resourceIdSegmentValues.changeProposalId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get changeProposalId() {
    return this._changeProposalId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: mercuryChangeProposalAriStaticOpts.qualifier,
      platformQualifier: mercuryChangeProposalAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: mercuryChangeProposalAriStaticOpts.resourceOwner,
      resourceType: mercuryChangeProposalAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.changeProposalId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        changeProposalId: opts.changeProposalId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, mercuryChangeProposalAriStaticOpts);
    return new _MercuryChangeProposalAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, mercuryChangeProposalAriStaticOpts);
    return new _MercuryChangeProposalAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      changeProposalId: this.changeProposalId
    };
  }
};

export {
  MercuryChangeProposalAri
};
