// @forge/util/packages/ari/chunk-PDWLSMK5.mjs
import {
  PassionfruitAskLinkAriResourceOwner,
  PassionfruitAskLinkAriResourceType
} from "./chunk-3ZCZHB4U.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/ask-link/manifest.ts
var passionfruitAskLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitAskLinkAriResourceOwner,
  resourceType: PassionfruitAskLinkAriResourceType,
  resourceIdSlug: "workspace/{activationId}/{askId}/{linkId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    askId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    linkId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/ask-link/index.ts
var PassionfruitAskLinkAri = class _PassionfruitAskLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._askId = opts.resourceIdSegmentValues.askId, this._linkId = opts.resourceIdSegmentValues.linkId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get askId() {
    return this._askId;
  }
  get linkId() {
    return this._linkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitAskLinkAriStaticOpts.qualifier,
      platformQualifier: passionfruitAskLinkAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: passionfruitAskLinkAriStaticOpts.resourceOwner,
      resourceType: passionfruitAskLinkAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.activationId}/${opts.askId}/${opts.linkId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        askId: opts.askId,
        linkId: opts.linkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitAskLinkAriStaticOpts);
    return new _PassionfruitAskLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitAskLinkAriStaticOpts);
    return new _PassionfruitAskLinkAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      askId: this.askId,
      linkId: this.linkId
    };
  }
};

export {
  PassionfruitAskLinkAri
};
