// @forge/util/packages/ari/chunk-3F6QAPE6.mjs
import {
  GoogleGoogleDriveDocumentAriResourceOwner,
  GoogleGoogleDriveDocumentAriResourceType
} from "./chunk-7MCC3MWT.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/google.google-drive/document/manifest.ts
var googleGoogleDriveDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GoogleGoogleDriveDocumentAriResourceOwner,
  resourceType: GoogleGoogleDriveDocumentAriResourceType,
  resourceIdSlug: "{googleDriveDocumentIdType}/{docId}",
  resourceIdSegmentFormats: {
    googleDriveDocumentIdType: /(documentId|spreadsheetId|presentationId|formId|folderId|imageId|audioId|videoId|pdfId|shortcutId|codeId|archiveId|web-pageId|pageId|blogpostId|otherId)/,
    // eslint-disable-line no-useless-escape
    docId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/google.google-drive/document/index.ts
var GoogleGoogleDriveDocumentAri = class _GoogleGoogleDriveDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._googleDriveDocumentIdType = opts.resourceIdSegmentValues.googleDriveDocumentIdType, this._docId = opts.resourceIdSegmentValues.docId;
  }
  get googleDriveDocumentIdType() {
    return this._googleDriveDocumentIdType;
  }
  get docId() {
    return this._docId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googleGoogleDriveDocumentAriStaticOpts.qualifier,
      platformQualifier: googleGoogleDriveDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googleGoogleDriveDocumentAriStaticOpts.resourceOwner,
      resourceType: googleGoogleDriveDocumentAriStaticOpts.resourceType,
      resourceId: `${opts.googleDriveDocumentIdType}/${opts.docId}`,
      resourceIdSegmentValues: {
        googleDriveDocumentIdType: opts.googleDriveDocumentIdType,
        docId: opts.docId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googleGoogleDriveDocumentAriStaticOpts);
    return new _GoogleGoogleDriveDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googleGoogleDriveDocumentAriStaticOpts);
    return new _GoogleGoogleDriveDocumentAri(opts);
  }
  getVariables() {
    return {
      googleDriveDocumentIdType: this.googleDriveDocumentIdType,
      docId: this.docId
    };
  }
};

export {
  GoogleGoogleDriveDocumentAri
};
