// @forge/util/packages/ari/chunk-BAEDA6TY.mjs
import {
  GraphOrganisationHistoryAriResourceOwner,
  GraphOrganisationHistoryAriResourceType
} from "./chunk-VSZQAEDL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/organisation-history/manifest.ts
var graphOrganisationHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphOrganisationHistoryAriResourceOwner,
  resourceType: GraphOrganisationHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{organisationUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    organisationUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/organisation-history/index.ts
var GraphOrganisationHistoryAri = class _GraphOrganisationHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._organisationUUID = opts.resourceIdSegmentValues.organisationUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get organisationUUID() {
    return this._organisationUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphOrganisationHistoryAriStaticOpts.qualifier,
      platformQualifier: graphOrganisationHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphOrganisationHistoryAriStaticOpts.resourceOwner,
      resourceType: graphOrganisationHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.organisationUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        organisationUUID: opts.organisationUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphOrganisationHistoryAriStaticOpts);
    return new _GraphOrganisationHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphOrganisationHistoryAriStaticOpts);
    return new _GraphOrganisationHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      organisationUUID: this.organisationUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphOrganisationHistoryAri
};
