// @forge/util/packages/ari/chunk-5AOSP2JD.mjs
import {
  MicrosoftPowerBiRemoteLinkAriResourceOwner,
  MicrosoftPowerBiRemoteLinkAriResourceType
} from "./chunk-WCVY5O3S.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft.power-bi/remote-link/manifest.ts
var microsoftPowerBiRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftPowerBiRemoteLinkAriResourceOwner,
  resourceType: MicrosoftPowerBiRemoteLinkAriResourceType,
  resourceIdSlug: "reportId/{reportId}",
  resourceIdSegmentFormats: {
    reportId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft.power-bi/remote-link/index.ts
var MicrosoftPowerBiRemoteLinkAri = class _MicrosoftPowerBiRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._reportId = opts.resourceIdSegmentValues.reportId;
  }
  get reportId() {
    return this._reportId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftPowerBiRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: microsoftPowerBiRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftPowerBiRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: microsoftPowerBiRemoteLinkAriStaticOpts.resourceType,
      resourceId: `reportId/${opts.reportId}`,
      resourceIdSegmentValues: {
        reportId: opts.reportId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftPowerBiRemoteLinkAriStaticOpts);
    return new _MicrosoftPowerBiRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftPowerBiRemoteLinkAriStaticOpts);
    return new _MicrosoftPowerBiRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      reportId: this.reportId
    };
  }
};

export {
  MicrosoftPowerBiRemoteLinkAri
};
