// @forge/util/packages/ari/chunk-EM3WVW33.mjs
import {
  JiraPlaybookInstanceStepAriResourceOwner,
  JiraPlaybookInstanceStepAriResourceType
} from "./chunk-JM2V4RSK.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira/playbook-instance-step/manifest.ts
var jiraPlaybookInstanceStepAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraPlaybookInstanceStepAriResourceOwner,
  resourceType: JiraPlaybookInstanceStepAriResourceType,
  resourceIdSlug: "activation/{activationId}/{playbookId}/{stepId}/{context}/{contextId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    playbookId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    stepId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    context: /(issue|alert)/,
    // eslint-disable-line no-useless-escape
    contextId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/playbook-instance-step/index.ts
var JiraPlaybookInstanceStepAri = class _JiraPlaybookInstanceStepAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._playbookId = opts.resourceIdSegmentValues.playbookId, this._stepId = opts.resourceIdSegmentValues.stepId, this._context = opts.resourceIdSegmentValues.context, this._contextId = opts.resourceIdSegmentValues.contextId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get playbookId() {
    return this._playbookId;
  }
  get stepId() {
    return this._stepId;
  }
  get context() {
    return this._context;
  }
  get contextId() {
    return this._contextId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraPlaybookInstanceStepAriStaticOpts.qualifier,
      platformQualifier: jiraPlaybookInstanceStepAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraPlaybookInstanceStepAriStaticOpts.resourceOwner,
      resourceType: jiraPlaybookInstanceStepAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.playbookId}/${opts.stepId}/${opts.context}/${opts.contextId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        playbookId: opts.playbookId,
        stepId: opts.stepId,
        context: opts.context,
        contextId: opts.contextId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraPlaybookInstanceStepAriStaticOpts);
    return new _JiraPlaybookInstanceStepAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraPlaybookInstanceStepAriStaticOpts);
    return new _JiraPlaybookInstanceStepAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      playbookId: this.playbookId,
      stepId: this.stepId,
      context: this.context,
      contextId: this.contextId
    };
  }
};

export {
  JiraPlaybookInstanceStepAri
};
