// @forge/util/packages/ari/chunk-X26O56BN.mjs
import {
  ConfluenceDcCommentAriResourceOwner,
  ConfluenceDcCommentAriResourceType
} from "./chunk-X3B3PSWP.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/confluence-dc/comment/manifest.ts
var confluenceDcCommentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ConfluenceDcCommentAriResourceOwner,
  resourceType: ConfluenceDcCommentAriResourceType,
  resourceIdSlug: "{siteId}/{commentId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    commentId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/confluence-dc/comment/index.ts
var ConfluenceDcCommentAri = class _ConfluenceDcCommentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._commentId = opts.resourceIdSegmentValues.commentId;
  }
  get siteId() {
    return this._siteId;
  }
  get commentId() {
    return this._commentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: confluenceDcCommentAriStaticOpts.qualifier,
      platformQualifier: confluenceDcCommentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: confluenceDcCommentAriStaticOpts.resourceOwner,
      resourceType: confluenceDcCommentAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.commentId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        commentId: opts.commentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, confluenceDcCommentAriStaticOpts);
    return new _ConfluenceDcCommentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, confluenceDcCommentAriStaticOpts);
    return new _ConfluenceDcCommentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      commentId: this.commentId
    };
  }
};

export {
  ConfluenceDcCommentAri
};
