// @forge/util/packages/ari/chunk-L3CODQ2W.mjs
import {
  JiraPlaybookLabelAriResourceOwner,
  JiraPlaybookLabelAriResourceType
} from "./chunk-TCZPEQFQ.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira/playbook-label/manifest.ts
var jiraPlaybookLabelAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraPlaybookLabelAriResourceOwner,
  resourceType: JiraPlaybookLabelAriResourceType,
  resourceIdSlug: "activation/{activationId}/{labelId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    labelId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/playbook-label/index.ts
var JiraPlaybookLabelAri = class _JiraPlaybookLabelAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._labelId = opts.resourceIdSegmentValues.labelId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get labelId() {
    return this._labelId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraPlaybookLabelAriStaticOpts.qualifier,
      platformQualifier: jiraPlaybookLabelAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraPlaybookLabelAriStaticOpts.resourceOwner,
      resourceType: jiraPlaybookLabelAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.labelId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        labelId: opts.labelId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraPlaybookLabelAriStaticOpts);
    return new _JiraPlaybookLabelAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraPlaybookLabelAriStaticOpts);
    return new _JiraPlaybookLabelAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      labelId: this.labelId
    };
  }
};

export {
  JiraPlaybookLabelAri
};
