// @forge/util/packages/ari/chunk-KXWVJBXF.mjs
import {
  GoogleGmailMessageAriResourceOwner,
  GoogleGmailMessageAriResourceType
} from "./chunk-MPZXQ2VZ.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/google.gmail/message/manifest.ts
var googleGmailMessageAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GoogleGmailMessageAriResourceOwner,
  resourceType: GoogleGmailMessageAriResourceType,
  resourceIdSlug: "{messageId}",
  resourceIdSegmentFormats: {
    messageId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/google.gmail/message/index.ts
var GoogleGmailMessageAri = class _GoogleGmailMessageAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._messageId = opts.resourceIdSegmentValues.messageId;
  }
  get messageId() {
    return this._messageId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googleGmailMessageAriStaticOpts.qualifier,
      platformQualifier: googleGmailMessageAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googleGmailMessageAriStaticOpts.resourceOwner,
      resourceType: googleGmailMessageAriStaticOpts.resourceType,
      resourceId: `${opts.messageId}`,
      resourceIdSegmentValues: {
        messageId: opts.messageId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googleGmailMessageAriStaticOpts);
    return new _GoogleGmailMessageAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googleGmailMessageAriStaticOpts);
    return new _GoogleGmailMessageAri(opts);
  }
  getVariables() {
    return {
      messageId: this.messageId
    };
  }
};

export {
  GoogleGmailMessageAri
};
