// @forge/util/packages/ari/chunk-4QXULHSS.mjs
import {
  ElevateSelfAssessmentAriResourceOwner,
  ElevateSelfAssessmentAriResourceType
} from "./chunk-SOMF35WW.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/elevate/self-assessment/manifest.ts
var elevateSelfAssessmentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ElevateSelfAssessmentAriResourceOwner,
  resourceType: ElevateSelfAssessmentAriResourceType,
  resourceIdSlug: "{selfAssessmentId}",
  resourceIdSegmentFormats: {
    selfAssessmentId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/elevate/self-assessment/index.ts
var ElevateSelfAssessmentAri = class _ElevateSelfAssessmentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._selfAssessmentId = opts.resourceIdSegmentValues.selfAssessmentId;
  }
  get siteId() {
    return this._siteId;
  }
  get selfAssessmentId() {
    return this._selfAssessmentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: elevateSelfAssessmentAriStaticOpts.qualifier,
      platformQualifier: elevateSelfAssessmentAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: elevateSelfAssessmentAriStaticOpts.resourceOwner,
      resourceType: elevateSelfAssessmentAriStaticOpts.resourceType,
      resourceId: `${opts.selfAssessmentId}`,
      resourceIdSegmentValues: {
        selfAssessmentId: opts.selfAssessmentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, elevateSelfAssessmentAriStaticOpts);
    return new _ElevateSelfAssessmentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, elevateSelfAssessmentAriStaticOpts);
    return new _ElevateSelfAssessmentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      selfAssessmentId: this.selfAssessmentId
    };
  }
};

export {
  ElevateSelfAssessmentAri
};
