// @forge/util/packages/ari/chunk-B2BDPA6O.mjs
import {
  TownsquareGoalUpdateAriResourceOwner,
  TownsquareGoalUpdateAriResourceType
} from "./chunk-7RQBTEXE.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/goal-update/manifest.ts
var townsquareGoalUpdateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareGoalUpdateAriResourceOwner,
  resourceType: TownsquareGoalUpdateAriResourceType,
  resourceIdSlug: "{goalUpdateUuid}",
  resourceIdSegmentFormats: {
    goalUpdateUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/goal-update/index.ts
var TownsquareGoalUpdateAri = class _TownsquareGoalUpdateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._goalUpdateUuid = opts.resourceIdSegmentValues.goalUpdateUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get goalUpdateUuid() {
    return this._goalUpdateUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareGoalUpdateAriStaticOpts.qualifier,
      platformQualifier: townsquareGoalUpdateAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareGoalUpdateAriStaticOpts.resourceOwner,
      resourceType: townsquareGoalUpdateAriStaticOpts.resourceType,
      resourceId: `${opts.goalUpdateUuid}`,
      resourceIdSegmentValues: {
        goalUpdateUuid: opts.goalUpdateUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareGoalUpdateAriStaticOpts);
    return new _TownsquareGoalUpdateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareGoalUpdateAriStaticOpts);
    return new _TownsquareGoalUpdateAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      goalUpdateUuid: this.goalUpdateUuid
    };
  }
};

export {
  TownsquareGoalUpdateAri
};
