// @forge/util/packages/ari/chunk-FXWFUBVD.mjs
import {
  AuditLogContextAriResourceOwner,
  AuditLogContextAriResourceType
} from "./chunk-LDMX3ABO.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/audit-log/context/manifest.ts
var auditLogContextAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AuditLogContextAriResourceOwner,
  resourceType: AuditLogContextAriResourceType,
  resourceIdSlug: "event/{eventId}/{contextId}",
  resourceIdSegmentFormats: {
    eventId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    contextId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/audit-log/context/index.ts
var AuditLogContextAri = class _AuditLogContextAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._eventId = opts.resourceIdSegmentValues.eventId, this._contextId = opts.resourceIdSegmentValues.contextId;
  }
  get eventId() {
    return this._eventId;
  }
  get contextId() {
    return this._contextId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: auditLogContextAriStaticOpts.qualifier,
      platformQualifier: auditLogContextAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: auditLogContextAriStaticOpts.resourceOwner,
      resourceType: auditLogContextAriStaticOpts.resourceType,
      resourceId: `event/${opts.eventId}/${opts.contextId}`,
      resourceIdSegmentValues: {
        eventId: opts.eventId,
        contextId: opts.contextId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, auditLogContextAriStaticOpts);
    return new _AuditLogContextAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, auditLogContextAriStaticOpts);
    return new _AuditLogContextAri(opts);
  }
  getVariables() {
    return {
      eventId: this.eventId,
      contextId: this.contextId
    };
  }
};

export {
  AuditLogContextAri
};
