// @forge/util/packages/ari/chunk-MD6YCAO4.mjs
import {
  RovoFileAriResourceOwner,
  RovoFileAriResourceType
} from "./chunk-ICUEPT2G.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/rovo/file/manifest.ts
var rovoFileAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: RovoFileAriResourceOwner,
  resourceType: RovoFileAriResourceType,
  resourceIdSlug: "activation/{activationId}/{fileUploadId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    fileUploadId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/rovo/file/index.ts
var RovoFileAri = class _RovoFileAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._fileUploadId = opts.resourceIdSegmentValues.fileUploadId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get fileUploadId() {
    return this._fileUploadId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: rovoFileAriStaticOpts.qualifier,
      platformQualifier: rovoFileAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: rovoFileAriStaticOpts.resourceOwner,
      resourceType: rovoFileAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.fileUploadId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        fileUploadId: opts.fileUploadId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, rovoFileAriStaticOpts);
    return new _RovoFileAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, rovoFileAriStaticOpts);
    return new _RovoFileAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      fileUploadId: this.fileUploadId
    };
  }
};

export {
  RovoFileAri
};
