// @forge/util/packages/ari/chunk-ZVPXHVXN.mjs
import {
  CanvasSiteAriResourceOwner,
  CanvasSiteAriResourceType
} from "./chunk-VCVOT2RL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/canvas/site/manifest.ts
var canvasSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: CanvasSiteAriResourceOwner,
  resourceType: CanvasSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/canvas/site/index.ts
var CanvasSiteAri = class _CanvasSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: canvasSiteAriStaticOpts.qualifier,
      platformQualifier: canvasSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: canvasSiteAriStaticOpts.resourceOwner,
      resourceType: canvasSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, canvasSiteAriStaticOpts);
    return new _CanvasSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, canvasSiteAriStaticOpts);
    return new _CanvasSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  CanvasSiteAri
};
