// @forge/util/packages/ari/chunk-6U6QESXC.mjs
import {
  MondayMondayDocumentAriResourceOwner,
  MondayMondayDocumentAriResourceType
} from "./chunk-VKXJXWA6.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/monday.monday/document/manifest.ts
var mondayMondayDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MondayMondayDocumentAriResourceOwner,
  resourceType: MondayMondayDocumentAriResourceType,
  resourceIdSlug: "board/{boardId}",
  resourceIdSegmentFormats: {
    boardId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/monday.monday/document/index.ts
var MondayMondayDocumentAri = class _MondayMondayDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._boardId = opts.resourceIdSegmentValues.boardId;
  }
  get boardId() {
    return this._boardId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: mondayMondayDocumentAriStaticOpts.qualifier,
      platformQualifier: mondayMondayDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: mondayMondayDocumentAriStaticOpts.resourceOwner,
      resourceType: mondayMondayDocumentAriStaticOpts.resourceType,
      resourceId: `board/${opts.boardId}`,
      resourceIdSegmentValues: {
        boardId: opts.boardId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, mondayMondayDocumentAriStaticOpts);
    return new _MondayMondayDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, mondayMondayDocumentAriStaticOpts);
    return new _MondayMondayDocumentAri(opts);
  }
  getVariables() {
    return {
      boardId: this.boardId
    };
  }
};

export {
  MondayMondayDocumentAri
};
