// @forge/util/packages/ari/chunk-YQ5KCD7H.mjs
import {
  AirtableAirtableDocumentAriResourceOwner,
  AirtableAirtableDocumentAriResourceType
} from "./chunk-PGCTJ4SL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/airtable.airtable/document/manifest.ts
var airtableAirtableDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AirtableAirtableDocumentAriResourceOwner,
  resourceType: AirtableAirtableDocumentAriResourceType,
  resourceIdSlug: "{compositeAirtableDocument}",
  resourceIdSegmentFormats: {
    compositeAirtableDocument: /(base\/[a-zA-Z0-9_-]+\/table\/[a-zA-Z0-9_-]+|base\/[a-zA-Z0-9_-]+)/
    // eslint-disable-line no-useless-escape
  }
};

// src/airtable.airtable/document/index.ts
var AirtableAirtableDocumentAri = class _AirtableAirtableDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._compositeAirtableDocument = opts.resourceIdSegmentValues.compositeAirtableDocument;
  }
  get compositeAirtableDocument() {
    return this._compositeAirtableDocument;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: airtableAirtableDocumentAriStaticOpts.qualifier,
      platformQualifier: airtableAirtableDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: airtableAirtableDocumentAriStaticOpts.resourceOwner,
      resourceType: airtableAirtableDocumentAriStaticOpts.resourceType,
      resourceId: `${opts.compositeAirtableDocument}`,
      resourceIdSegmentValues: {
        compositeAirtableDocument: opts.compositeAirtableDocument
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, airtableAirtableDocumentAriStaticOpts);
    return new _AirtableAirtableDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, airtableAirtableDocumentAriStaticOpts);
    return new _AirtableAirtableDocumentAri(opts);
  }
  getVariables() {
    return {
      compositeAirtableDocument: this.compositeAirtableDocument
    };
  }
};

export {
  AirtableAirtableDocumentAri
};
