// @forge/util/packages/ari/chunk-FG23VKPJ.mjs
import {
  AmplitudeAmplitudeRemoteLinkAriResourceOwner,
  AmplitudeAmplitudeRemoteLinkAriResourceType
} from "./chunk-63OICGDG.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/amplitude.amplitude/remote-link/manifest.ts
var amplitudeAmplitudeRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AmplitudeAmplitudeRemoteLinkAriResourceOwner,
  resourceType: AmplitudeAmplitudeRemoteLinkAriResourceType,
  resourceIdSlug: "org/{orgId}/chart/{chartId}",
  resourceIdSegmentFormats: {
    orgId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    chartId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/amplitude.amplitude/remote-link/index.ts
var AmplitudeAmplitudeRemoteLinkAri = class _AmplitudeAmplitudeRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._orgId = opts.resourceIdSegmentValues.orgId, this._chartId = opts.resourceIdSegmentValues.chartId;
  }
  get orgId() {
    return this._orgId;
  }
  get chartId() {
    return this._chartId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: amplitudeAmplitudeRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: amplitudeAmplitudeRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: amplitudeAmplitudeRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: amplitudeAmplitudeRemoteLinkAriStaticOpts.resourceType,
      resourceId: `org/${opts.orgId}/chart/${opts.chartId}`,
      resourceIdSegmentValues: {
        orgId: opts.orgId,
        chartId: opts.chartId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, amplitudeAmplitudeRemoteLinkAriStaticOpts);
    return new _AmplitudeAmplitudeRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, amplitudeAmplitudeRemoteLinkAriStaticOpts);
    return new _AmplitudeAmplitudeRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      orgId: this.orgId,
      chartId: this.chartId
    };
  }
};

export {
  AmplitudeAmplitudeRemoteLinkAri
};
