// @forge/util/packages/ari/chunk-4VNQPOCS.mjs
import {
  TownsquareMilestoneAriResourceOwner,
  TownsquareMilestoneAriResourceType
} from "./chunk-42MIAPNU.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/milestone/manifest.ts
var townsquareMilestoneAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareMilestoneAriResourceOwner,
  resourceType: TownsquareMilestoneAriResourceType,
  resourceIdSlug: "{milestoneUuid}",
  resourceIdSegmentFormats: {
    milestoneUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/milestone/index.ts
var TownsquareMilestoneAri = class _TownsquareMilestoneAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._milestoneUuid = opts.resourceIdSegmentValues.milestoneUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get milestoneUuid() {
    return this._milestoneUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareMilestoneAriStaticOpts.qualifier,
      platformQualifier: townsquareMilestoneAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareMilestoneAriStaticOpts.resourceOwner,
      resourceType: townsquareMilestoneAriStaticOpts.resourceType,
      resourceId: `${opts.milestoneUuid}`,
      resourceIdSegmentValues: {
        milestoneUuid: opts.milestoneUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareMilestoneAriStaticOpts);
    return new _TownsquareMilestoneAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareMilestoneAriStaticOpts);
    return new _TownsquareMilestoneAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      milestoneUuid: this.milestoneUuid
    };
  }
};

export {
  TownsquareMilestoneAri
};
