// @forge/util/packages/ari/chunk-T24NS6FQ.mjs
import {
  PassionfruitWorkspaceAriResourceOwner,
  PassionfruitWorkspaceAriResourceType
} from "./chunk-NALH4DTW.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/workspace/manifest.ts
var passionfruitWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitWorkspaceAriResourceOwner,
  resourceType: PassionfruitWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/workspace/index.ts
var PassionfruitWorkspaceAri = class _PassionfruitWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitWorkspaceAriStaticOpts.qualifier,
      platformQualifier: passionfruitWorkspaceAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: passionfruitWorkspaceAriStaticOpts.resourceOwner,
      resourceType: passionfruitWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitWorkspaceAriStaticOpts);
    return new _PassionfruitWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitWorkspaceAriStaticOpts);
    return new _PassionfruitWorkspaceAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId
    };
  }
};

export {
  PassionfruitWorkspaceAri
};
