// @forge/util/packages/ari/chunk-IM5LEZU5.mjs
import {
  GoalGoalTypeAriResourceOwner,
  GoalGoalTypeAriResourceType
} from "./chunk-RE4MHVIY.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/goal/goal-type/manifest.ts
var goalGoalTypeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GoalGoalTypeAriResourceOwner,
  resourceType: GoalGoalTypeAriResourceType,
  resourceIdSlug: "{activationId}/{goalTypeId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    goalTypeId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/goal/goal-type/index.ts
var GoalGoalTypeAri = class _GoalGoalTypeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._goalTypeId = opts.resourceIdSegmentValues.goalTypeId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get goalTypeId() {
    return this._goalTypeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: goalGoalTypeAriStaticOpts.qualifier,
      platformQualifier: goalGoalTypeAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: goalGoalTypeAriStaticOpts.resourceOwner,
      resourceType: goalGoalTypeAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}/${opts.goalTypeId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        goalTypeId: opts.goalTypeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, goalGoalTypeAriStaticOpts);
    return new _GoalGoalTypeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, goalGoalTypeAriStaticOpts);
    return new _GoalGoalTypeAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      goalTypeId: this.goalTypeId
    };
  }
};

export {
  GoalGoalTypeAri
};
