// @forge/util/packages/ari/chunk-2TIIO2DP.mjs
import {
  CompassApiEndpointAriResourceOwner,
  CompassApiEndpointAriResourceType
} from "./chunk-2HMBOSQ3.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/compass/api-endpoint/manifest.ts
var compassApiEndpointAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: CompassApiEndpointAriResourceOwner,
  resourceType: CompassApiEndpointAriResourceType,
  resourceIdSlug: "{workspaceId}/{endpointId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    endpointId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/compass/api-endpoint/index.ts
var CompassApiEndpointAri = class _CompassApiEndpointAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._endpointId = opts.resourceIdSegmentValues.endpointId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get endpointId() {
    return this._endpointId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: compassApiEndpointAriStaticOpts.qualifier,
      platformQualifier: compassApiEndpointAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: compassApiEndpointAriStaticOpts.resourceOwner,
      resourceType: compassApiEndpointAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.endpointId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        endpointId: opts.endpointId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, compassApiEndpointAriStaticOpts);
    return new _CompassApiEndpointAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, compassApiEndpointAriStaticOpts);
    return new _CompassApiEndpointAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      endpointId: this.endpointId
    };
  }
};

export {
  CompassApiEndpointAri
};
