// @forge/util/packages/ari/chunk-MJ3A7RHF.mjs
import {
  SmartsheetSmartsheetRemoteLinkAriResourceOwner,
  SmartsheetSmartsheetRemoteLinkAriResourceType
} from "./chunk-HG7EHHS2.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/smartsheet.smartsheet/remote-link/manifest.ts
var smartsheetSmartsheetRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: SmartsheetSmartsheetRemoteLinkAriResourceOwner,
  resourceType: SmartsheetSmartsheetRemoteLinkAriResourceType,
  resourceIdSlug: "sheet/{sheetId}",
  resourceIdSegmentFormats: {
    sheetId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/smartsheet.smartsheet/remote-link/index.ts
var SmartsheetSmartsheetRemoteLinkAri = class _SmartsheetSmartsheetRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._sheetId = opts.resourceIdSegmentValues.sheetId;
  }
  get sheetId() {
    return this._sheetId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: smartsheetSmartsheetRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: smartsheetSmartsheetRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: smartsheetSmartsheetRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: smartsheetSmartsheetRemoteLinkAriStaticOpts.resourceType,
      resourceId: `sheet/${opts.sheetId}`,
      resourceIdSegmentValues: {
        sheetId: opts.sheetId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, smartsheetSmartsheetRemoteLinkAriStaticOpts);
    return new _SmartsheetSmartsheetRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, smartsheetSmartsheetRemoteLinkAriStaticOpts);
    return new _SmartsheetSmartsheetRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      sheetId: this.sheetId
    };
  }
};

export {
  SmartsheetSmartsheetRemoteLinkAri
};
