// @forge/util/packages/ari/chunk-UM4DZGWX.mjs
import {
  AvpChartAriResourceOwner,
  AvpChartAriResourceType
} from "./chunk-5WYUFPST.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/avp/chart/manifest.ts
var avpChartAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AvpChartAriResourceOwner,
  resourceType: AvpChartAriResourceType,
  resourceIdSlug: "activation/{activationId}/{chartId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    chartId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/chart/index.ts
var AvpChartAri = class _AvpChartAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._chartId = opts.resourceIdSegmentValues.chartId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get chartId() {
    return this._chartId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpChartAriStaticOpts.qualifier,
      platformQualifier: avpChartAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: avpChartAriStaticOpts.resourceOwner,
      resourceType: avpChartAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.chartId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        chartId: opts.chartId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpChartAriStaticOpts);
    return new _AvpChartAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpChartAriStaticOpts);
    return new _AvpChartAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      chartId: this.chartId
    };
  }
};

export {
  AvpChartAri
};
