// @forge/util/packages/ari/chunk-EZAKBOKA.mjs
import {
  ProjectWorkspaceAriResourceOwner,
  ProjectWorkspaceAriResourceType
} from "./chunk-RC6RQXMI.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/project/workspace/manifest.ts
var projectWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ProjectWorkspaceAriResourceOwner,
  resourceType: ProjectWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/project/workspace/index.ts
var ProjectWorkspaceAri = class _ProjectWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: projectWorkspaceAriStaticOpts.qualifier,
      platformQualifier: projectWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: projectWorkspaceAriStaticOpts.resourceOwner,
      resourceType: projectWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, projectWorkspaceAriStaticOpts);
    return new _ProjectWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, projectWorkspaceAriStaticOpts);
    return new _ProjectWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId
    };
  }
};

export {
  ProjectWorkspaceAri
};
