// @forge/util/packages/ari/chunk-NCEASUL6.mjs
import {
  FigmaFigmaRemoteLinkAriResourceOwner,
  FigmaFigmaRemoteLinkAriResourceType
} from "./chunk-2K3H62WY.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/figma.figma/remote-link/manifest.ts
var figmaFigmaRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: FigmaFigmaRemoteLinkAriResourceOwner,
  resourceType: FigmaFigmaRemoteLinkAriResourceType,
  resourceIdSlug: "{remoteLinkId}",
  resourceIdSegmentFormats: {
    remoteLinkId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/figma.figma/remote-link/index.ts
var FigmaFigmaRemoteLinkAri = class _FigmaFigmaRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._remoteLinkId = opts.resourceIdSegmentValues.remoteLinkId;
  }
  get remoteLinkId() {
    return this._remoteLinkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: figmaFigmaRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: figmaFigmaRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: figmaFigmaRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: figmaFigmaRemoteLinkAriStaticOpts.resourceType,
      resourceId: `${opts.remoteLinkId}`,
      resourceIdSegmentValues: {
        remoteLinkId: opts.remoteLinkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, figmaFigmaRemoteLinkAriStaticOpts);
    return new _FigmaFigmaRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, figmaFigmaRemoteLinkAriStaticOpts);
    return new _FigmaFigmaRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      remoteLinkId: this.remoteLinkId
    };
  }
};

export {
  FigmaFigmaRemoteLinkAri
};
