// @forge/util/packages/ari/chunk-TP7IYRNY.mjs
import {
  MicrosoftTeamsRemoteLinkAriResourceOwner,
  MicrosoftTeamsRemoteLinkAriResourceType
} from "./chunk-XYSLL7ZT.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft.teams/remote-link/manifest.ts
var microsoftTeamsRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftTeamsRemoteLinkAriResourceOwner,
  resourceType: MicrosoftTeamsRemoteLinkAriResourceType,
  resourceIdSlug: "teamId/{teamId}/channelId/{channelId}",
  resourceIdSegmentFormats: {
    teamId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    channelId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft.teams/remote-link/index.ts
var MicrosoftTeamsRemoteLinkAri = class _MicrosoftTeamsRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._teamId = opts.resourceIdSegmentValues.teamId, this._channelId = opts.resourceIdSegmentValues.channelId;
  }
  get teamId() {
    return this._teamId;
  }
  get channelId() {
    return this._channelId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftTeamsRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: microsoftTeamsRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftTeamsRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: microsoftTeamsRemoteLinkAriStaticOpts.resourceType,
      resourceId: `teamId/${opts.teamId}/channelId/${opts.channelId}`,
      resourceIdSegmentValues: {
        teamId: opts.teamId,
        channelId: opts.channelId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftTeamsRemoteLinkAriStaticOpts);
    return new _MicrosoftTeamsRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftTeamsRemoteLinkAriStaticOpts);
    return new _MicrosoftTeamsRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      teamId: this.teamId,
      channelId: this.channelId
    };
  }
};

export {
  MicrosoftTeamsRemoteLinkAri
};
