// @forge/util/packages/ari/chunk-5YMW2YB6.mjs
import {
  JiraFieldConfigurationContextAriResourceOwner,
  JiraFieldConfigurationContextAriResourceType
} from "./chunk-OOQFCUQM.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira/field-configuration-context/manifest.ts
var jiraFieldConfigurationContextAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraFieldConfigurationContextAriResourceOwner,
  resourceType: JiraFieldConfigurationContextAriResourceType,
  resourceIdSlug: "activation/{activationId}/{contextId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    contextId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/field-configuration-context/index.ts
var JiraFieldConfigurationContextAri = class _JiraFieldConfigurationContextAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._contextId = opts.resourceIdSegmentValues.contextId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get contextId() {
    return this._contextId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraFieldConfigurationContextAriStaticOpts.qualifier,
      platformQualifier: jiraFieldConfigurationContextAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraFieldConfigurationContextAriStaticOpts.resourceOwner,
      resourceType: jiraFieldConfigurationContextAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.contextId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        contextId: opts.contextId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraFieldConfigurationContextAriStaticOpts);
    return new _JiraFieldConfigurationContextAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraFieldConfigurationContextAriStaticOpts);
    return new _JiraFieldConfigurationContextAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      contextId: this.contextId
    };
  }
};

export {
  JiraFieldConfigurationContextAri
};
