// @forge/util/packages/ari/chunk-5XKGVM5M.mjs
import {
  MercuryChangeAriResourceOwner,
  MercuryChangeAriResourceType
} from "./chunk-ZVFPKZ5M.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/mercury/change/manifest.ts
var mercuryChangeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MercuryChangeAriResourceOwner,
  resourceType: MercuryChangeAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{changeId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    changeId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/mercury/change/index.ts
var MercuryChangeAri = class _MercuryChangeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._changeId = opts.resourceIdSegmentValues.changeId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get changeId() {
    return this._changeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: mercuryChangeAriStaticOpts.qualifier,
      platformQualifier: mercuryChangeAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: mercuryChangeAriStaticOpts.resourceOwner,
      resourceType: mercuryChangeAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.changeId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        changeId: opts.changeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, mercuryChangeAriStaticOpts);
    return new _MercuryChangeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, mercuryChangeAriStaticOpts);
    return new _MercuryChangeAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      changeId: this.changeId
    };
  }
};

export {
  MercuryChangeAri
};
