// @forge/util/packages/ari/chunk-3NNN5ISZ.mjs
import {
  DropboxRemoteLinkAriResourceOwner,
  DropboxRemoteLinkAriResourceType
} from "./chunk-6YI45FOX.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/dropbox/remote-link/manifest.ts
var dropboxRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: DropboxRemoteLinkAriResourceOwner,
  resourceType: DropboxRemoteLinkAriResourceType,
  resourceIdSlug: "file/{fileId}",
  resourceIdSegmentFormats: {
    fileId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/dropbox/remote-link/index.ts
var DropboxRemoteLinkAri = class _DropboxRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._fileId = opts.resourceIdSegmentValues.fileId;
  }
  get fileId() {
    return this._fileId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: dropboxRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: dropboxRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: dropboxRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: dropboxRemoteLinkAriStaticOpts.resourceType,
      resourceId: `file/${opts.fileId}`,
      resourceIdSegmentValues: {
        fileId: opts.fileId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, dropboxRemoteLinkAriStaticOpts);
    return new _DropboxRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, dropboxRemoteLinkAriStaticOpts);
    return new _DropboxRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      fileId: this.fileId
    };
  }
};

export {
  DropboxRemoteLinkAri
};
