// @forge/util/packages/ari/chunk-DYRNZK5R.mjs
import {
  CompassPackageAriResourceOwner,
  CompassPackageAriResourceType
} from "./chunk-LBNPGADB.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/compass/package/manifest.ts
var compassPackageAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: CompassPackageAriResourceOwner,
  resourceType: CompassPackageAriResourceType,
  resourceIdSlug: "{workspaceId}/{packageId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    packageId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/compass/package/index.ts
var CompassPackageAri = class _CompassPackageAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._packageId = opts.resourceIdSegmentValues.packageId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get packageId() {
    return this._packageId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: compassPackageAriStaticOpts.qualifier,
      platformQualifier: compassPackageAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: compassPackageAriStaticOpts.resourceOwner,
      resourceType: compassPackageAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.packageId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        packageId: opts.packageId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, compassPackageAriStaticOpts);
    return new _CompassPackageAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, compassPackageAriStaticOpts);
    return new _CompassPackageAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      packageId: this.packageId
    };
  }
};

export {
  CompassPackageAri
};
