// @forge/util/packages/ari/chunk-6M3OKJRJ.mjs
import {
  AsanaAsanaWorkItemAriResourceOwner,
  AsanaAsanaWorkItemAriResourceType
} from "./chunk-UXEYXOYH.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/asana.asana/work-item/manifest.ts
var asanaAsanaWorkItemAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AsanaAsanaWorkItemAriResourceOwner,
  resourceType: AsanaAsanaWorkItemAriResourceType,
  resourceIdSlug: "task/{taskId}",
  resourceIdSegmentFormats: {
    taskId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/asana.asana/work-item/index.ts
var AsanaAsanaWorkItemAri = class _AsanaAsanaWorkItemAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._taskId = opts.resourceIdSegmentValues.taskId;
  }
  get taskId() {
    return this._taskId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: asanaAsanaWorkItemAriStaticOpts.qualifier,
      platformQualifier: asanaAsanaWorkItemAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: asanaAsanaWorkItemAriStaticOpts.resourceOwner,
      resourceType: asanaAsanaWorkItemAriStaticOpts.resourceType,
      resourceId: `task/${opts.taskId}`,
      resourceIdSegmentValues: {
        taskId: opts.taskId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, asanaAsanaWorkItemAriStaticOpts);
    return new _AsanaAsanaWorkItemAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, asanaAsanaWorkItemAriStaticOpts);
    return new _AsanaAsanaWorkItemAri(opts);
  }
  getVariables() {
    return {
      taskId: this.taskId
    };
  }
};

export {
  AsanaAsanaWorkItemAri
};
