// @forge/util/packages/ari/chunk-63YFV4PZ.mjs
import {
  TrelloPlannerAriResourceOwner,
  TrelloPlannerAriResourceType
} from "./chunk-BFIKUJAR.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/trello/planner/manifest.ts
var trelloPlannerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TrelloPlannerAriResourceOwner,
  resourceType: TrelloPlannerAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{plannerId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-_.]*/,
    // eslint-disable-line no-useless-escape
    plannerId: /[a-zA-Z0-9\-_.]*/
    // eslint-disable-line no-useless-escape
  }
};

// src/trello/planner/index.ts
var TrelloPlannerAri = class _TrelloPlannerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._plannerId = opts.resourceIdSegmentValues.plannerId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get plannerId() {
    return this._plannerId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: trelloPlannerAriStaticOpts.qualifier,
      platformQualifier: trelloPlannerAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: trelloPlannerAriStaticOpts.resourceOwner,
      resourceType: trelloPlannerAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.plannerId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId || "",
        plannerId: opts.plannerId || ""
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, trelloPlannerAriStaticOpts);
    return new _TrelloPlannerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, trelloPlannerAriStaticOpts);
    return new _TrelloPlannerAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      plannerId: this.plannerId
    };
  }
};

export {
  TrelloPlannerAri
};
