// @forge/util/packages/ari/chunk-OP3WKLAR.mjs
import {
  TownsquareMetricValueAriResourceOwner,
  TownsquareMetricValueAriResourceType
} from "./chunk-PJMNKWWL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/townsquare/metric-value/manifest.ts
var townsquareMetricValueAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TownsquareMetricValueAriResourceOwner,
  resourceType: TownsquareMetricValueAriResourceType,
  resourceIdSlug: "{metricValueUuid}",
  resourceIdSegmentFormats: {
    metricValueUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/townsquare/metric-value/index.ts
var TownsquareMetricValueAri = class _TownsquareMetricValueAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._metricValueUuid = opts.resourceIdSegmentValues.metricValueUuid;
  }
  get siteId() {
    return this._siteId;
  }
  get metricValueUuid() {
    return this._metricValueUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: townsquareMetricValueAriStaticOpts.qualifier,
      platformQualifier: townsquareMetricValueAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: townsquareMetricValueAriStaticOpts.resourceOwner,
      resourceType: townsquareMetricValueAriStaticOpts.resourceType,
      resourceId: `${opts.metricValueUuid}`,
      resourceIdSegmentValues: {
        metricValueUuid: opts.metricValueUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, townsquareMetricValueAriStaticOpts);
    return new _TownsquareMetricValueAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, townsquareMetricValueAriStaticOpts);
    return new _TownsquareMetricValueAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      metricValueUuid: this.metricValueUuid
    };
  }
};

export {
  TownsquareMetricValueAri
};
