// @forge/util/packages/ari/chunk-HSYQ5OEW.mjs
import {
  JiraCustomerServiceSiteAriResourceOwner,
  JiraCustomerServiceSiteAriResourceType
} from "./chunk-BVLK4MVL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-customer-service/site/manifest.ts
var jiraCustomerServiceSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraCustomerServiceSiteAriResourceOwner,
  resourceType: JiraCustomerServiceSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-customer-service/site/index.ts
var JiraCustomerServiceSiteAri = class _JiraCustomerServiceSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraCustomerServiceSiteAriStaticOpts.qualifier,
      platformQualifier: jiraCustomerServiceSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraCustomerServiceSiteAriStaticOpts.resourceOwner,
      resourceType: jiraCustomerServiceSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraCustomerServiceSiteAriStaticOpts);
    return new _JiraCustomerServiceSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraCustomerServiceSiteAriStaticOpts);
    return new _JiraCustomerServiceSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  JiraCustomerServiceSiteAri
};
