// @forge/util/packages/ari/chunk-27O375EE.mjs
import {
  PagerdutyPagerdutyRemoteLinkAriResourceOwner,
  PagerdutyPagerdutyRemoteLinkAriResourceType
} from "./chunk-KTT4OW2H.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/pagerduty.pagerduty/remote-link/manifest.ts
var pagerdutyPagerdutyRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PagerdutyPagerdutyRemoteLinkAriResourceOwner,
  resourceType: PagerdutyPagerdutyRemoteLinkAriResourceType,
  resourceIdSlug: "incident/{incidentId}",
  resourceIdSegmentFormats: {
    incidentId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/pagerduty.pagerduty/remote-link/index.ts
var PagerdutyPagerdutyRemoteLinkAri = class _PagerdutyPagerdutyRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._incidentId = opts.resourceIdSegmentValues.incidentId;
  }
  get incidentId() {
    return this._incidentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: pagerdutyPagerdutyRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: pagerdutyPagerdutyRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: pagerdutyPagerdutyRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: pagerdutyPagerdutyRemoteLinkAriStaticOpts.resourceType,
      resourceId: `incident/${opts.incidentId}`,
      resourceIdSegmentValues: {
        incidentId: opts.incidentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, pagerdutyPagerdutyRemoteLinkAriStaticOpts);
    return new _PagerdutyPagerdutyRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, pagerdutyPagerdutyRemoteLinkAriStaticOpts);
    return new _PagerdutyPagerdutyRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      incidentId: this.incidentId
    };
  }
};

export {
  PagerdutyPagerdutyRemoteLinkAri
};
