// @forge/util/packages/ari/chunk-4LHSTBSK.mjs
import {
  GraphDealAriResourceOwner,
  GraphDealAriResourceType
} from "./chunk-LN7ARE4M.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/deal/manifest.ts
var graphDealAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDealAriResourceOwner,
  resourceType: GraphDealAriResourceType,
  resourceIdSlug: "activation/{activationId}/{dealId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    dealId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/deal/index.ts
var GraphDealAri = class _GraphDealAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._dealId = opts.resourceIdSegmentValues.dealId;
  }
  get activationId() {
    return this._activationId;
  }
  get dealId() {
    return this._dealId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDealAriStaticOpts.qualifier,
      platformQualifier: graphDealAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDealAriStaticOpts.resourceOwner,
      resourceType: graphDealAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.dealId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        dealId: opts.dealId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDealAriStaticOpts);
    return new _GraphDealAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDealAriStaticOpts);
    return new _GraphDealAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      dealId: this.dealId
    };
  }
};

export {
  GraphDealAri
};
