// @forge/util/packages/ari/chunk-ZXJEKNUX.mjs
import {
  MercuryViewAriResourceOwner,
  MercuryViewAriResourceType
} from "./chunk-SW47DL6N.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/mercury/view/manifest.ts
var mercuryViewAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MercuryViewAriResourceOwner,
  resourceType: MercuryViewAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{viewId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    viewId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/mercury/view/index.ts
var MercuryViewAri = class _MercuryViewAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._viewId = opts.resourceIdSegmentValues.viewId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get viewId() {
    return this._viewId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: mercuryViewAriStaticOpts.qualifier,
      platformQualifier: mercuryViewAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: mercuryViewAriStaticOpts.resourceOwner,
      resourceType: mercuryViewAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.viewId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        viewId: opts.viewId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, mercuryViewAriStaticOpts);
    return new _MercuryViewAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, mercuryViewAriStaticOpts);
    return new _MercuryViewAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      viewId: this.viewId
    };
  }
};

export {
  MercuryViewAri
};
