// @forge/util/packages/ari/chunk-3HCUELCU.mjs
import {
  GraphPositionHistoryAriResourceOwner,
  GraphPositionHistoryAriResourceType
} from "./chunk-SHV5KRSV.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/position-history/manifest.ts
var graphPositionHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphPositionHistoryAriResourceOwner,
  resourceType: GraphPositionHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{positionUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    positionUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/position-history/index.ts
var GraphPositionHistoryAri = class _GraphPositionHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._positionUUID = opts.resourceIdSegmentValues.positionUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get positionUUID() {
    return this._positionUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphPositionHistoryAriStaticOpts.qualifier,
      platformQualifier: graphPositionHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphPositionHistoryAriStaticOpts.resourceOwner,
      resourceType: graphPositionHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.positionUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        positionUUID: opts.positionUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphPositionHistoryAriStaticOpts);
    return new _GraphPositionHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphPositionHistoryAriStaticOpts);
    return new _GraphPositionHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      positionUUID: this.positionUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphPositionHistoryAri
};
