// @forge/util/packages/ari/chunk-GP2M3LG5.mjs
import {
  MicrosoftAccountAriResourceOwner,
  MicrosoftAccountAriResourceType
} from "./chunk-YIHJDIS5.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft/account/manifest.ts
var microsoftAccountAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftAccountAriResourceOwner,
  resourceType: MicrosoftAccountAriResourceType,
  resourceIdSlug: "{accountId}",
  resourceIdSegmentFormats: {
    accountId: /[a-zA-Z0-9\-_.~@;{}=]{0,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft/account/index.ts
var MicrosoftAccountAri = class _MicrosoftAccountAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._accountId = opts.resourceIdSegmentValues.accountId;
  }
  get accountId() {
    return this._accountId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftAccountAriStaticOpts.qualifier,
      platformQualifier: microsoftAccountAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftAccountAriStaticOpts.resourceOwner,
      resourceType: microsoftAccountAriStaticOpts.resourceType,
      resourceId: `${opts.accountId}`,
      resourceIdSegmentValues: {
        accountId: opts.accountId || ""
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftAccountAriStaticOpts);
    return new _MicrosoftAccountAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftAccountAriStaticOpts);
    return new _MicrosoftAccountAri(opts);
  }
  getVariables() {
    return {
      accountId: this.accountId
    };
  }
};

export {
  MicrosoftAccountAri
};
