// @forge/util/packages/ari/chunk-5W46EKFA.mjs
import {
  WorkdayWorkdayPositionAriResourceOwner,
  WorkdayWorkdayPositionAriResourceType
} from "./chunk-MJXKCGLB.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/workday.workday/position/manifest.ts
var workdayWorkdayPositionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: WorkdayWorkdayPositionAriResourceOwner,
  resourceType: WorkdayWorkdayPositionAriResourceType,
  resourceIdSlug: "{workdayId}",
  resourceIdSegmentFormats: {
    workdayId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday.workday/position/index.ts
var WorkdayWorkdayPositionAri = class _WorkdayWorkdayPositionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayId = opts.resourceIdSegmentValues.workdayId;
  }
  get workdayId() {
    return this._workdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayWorkdayPositionAriStaticOpts.qualifier,
      platformQualifier: workdayWorkdayPositionAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayWorkdayPositionAriStaticOpts.resourceOwner,
      resourceType: workdayWorkdayPositionAriStaticOpts.resourceType,
      resourceId: `${opts.workdayId}`,
      resourceIdSegmentValues: {
        workdayId: opts.workdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayWorkdayPositionAriStaticOpts);
    return new _WorkdayWorkdayPositionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayWorkdayPositionAriStaticOpts);
    return new _WorkdayWorkdayPositionAri(opts);
  }
  getVariables() {
    return {
      workdayId: this.workdayId
    };
  }
};

export {
  WorkdayWorkdayPositionAri
};
