// @forge/util/packages/ari/chunk-QLHWTU4L.mjs
import {
  GithubGithubDeploymentAriResourceOwner,
  GithubGithubDeploymentAriResourceType
} from "./chunk-5OXTM6ZB.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/github.github/deployment/manifest.ts
var githubGithubDeploymentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GithubGithubDeploymentAriResourceOwner,
  resourceType: GithubGithubDeploymentAriResourceType,
  resourceIdSlug: "pipelineId/{pipelineId}/environmentId/{environmentId}/deploymentSequenceNumber/{deploymentSequenceNumber}",
  resourceIdSegmentFormats: {
    pipelineId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    environmentId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    deploymentSequenceNumber: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/github.github/deployment/index.ts
var GithubGithubDeploymentAri = class _GithubGithubDeploymentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._pipelineId = opts.resourceIdSegmentValues.pipelineId, this._environmentId = opts.resourceIdSegmentValues.environmentId, this._deploymentSequenceNumber = opts.resourceIdSegmentValues.deploymentSequenceNumber;
  }
  get pipelineId() {
    return this._pipelineId;
  }
  get environmentId() {
    return this._environmentId;
  }
  get deploymentSequenceNumber() {
    return this._deploymentSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: githubGithubDeploymentAriStaticOpts.qualifier,
      platformQualifier: githubGithubDeploymentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: githubGithubDeploymentAriStaticOpts.resourceOwner,
      resourceType: githubGithubDeploymentAriStaticOpts.resourceType,
      resourceId: `pipelineId/${opts.pipelineId}/environmentId/${opts.environmentId}/deploymentSequenceNumber/${opts.deploymentSequenceNumber}`,
      resourceIdSegmentValues: {
        pipelineId: opts.pipelineId,
        environmentId: opts.environmentId,
        deploymentSequenceNumber: opts.deploymentSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, githubGithubDeploymentAriStaticOpts);
    return new _GithubGithubDeploymentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, githubGithubDeploymentAriStaticOpts);
    return new _GithubGithubDeploymentAri(opts);
  }
  getVariables() {
    return {
      pipelineId: this.pipelineId,
      environmentId: this.environmentId,
      deploymentSequenceNumber: this.deploymentSequenceNumber
    };
  }
};

export {
  GithubGithubDeploymentAri
};
