// @forge/util/packages/ari/chunk-E4EW4JMI.mjs
import {
  AvocadoSiteAriResourceOwner,
  AvocadoSiteAriResourceType
} from "./chunk-3G3BCF2A.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/avocado/site/manifest.ts
var avocadoSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AvocadoSiteAriResourceOwner,
  resourceType: AvocadoSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avocado/site/index.ts
var AvocadoSiteAri = class _AvocadoSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avocadoSiteAriStaticOpts.qualifier,
      platformQualifier: avocadoSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: avocadoSiteAriStaticOpts.resourceOwner,
      resourceType: avocadoSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avocadoSiteAriStaticOpts);
    return new _AvocadoSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avocadoSiteAriStaticOpts);
    return new _AvocadoSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  AvocadoSiteAri
};
