// @forge/util/packages/ari/chunk-FJDJ7H65.mjs
import {
  HubspotHubspotRemoteLinkAriResourceOwner,
  HubspotHubspotRemoteLinkAriResourceType
} from "./chunk-KLQOHOSF.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/hubspot.hubspot/remote-link/manifest.ts
var hubspotHubspotRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: HubspotHubspotRemoteLinkAriResourceOwner,
  resourceType: HubspotHubspotRemoteLinkAriResourceType,
  resourceIdSlug: "deal/{dealId}",
  resourceIdSegmentFormats: {
    dealId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/hubspot.hubspot/remote-link/index.ts
var HubspotHubspotRemoteLinkAri = class _HubspotHubspotRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._dealId = opts.resourceIdSegmentValues.dealId;
  }
  get dealId() {
    return this._dealId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: hubspotHubspotRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: hubspotHubspotRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: hubspotHubspotRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: hubspotHubspotRemoteLinkAriStaticOpts.resourceType,
      resourceId: `deal/${opts.dealId}`,
      resourceIdSegmentValues: {
        dealId: opts.dealId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, hubspotHubspotRemoteLinkAriStaticOpts);
    return new _HubspotHubspotRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, hubspotHubspotRemoteLinkAriStaticOpts);
    return new _HubspotHubspotRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      dealId: this.dealId
    };
  }
};

export {
  HubspotHubspotRemoteLinkAri
};
