// @forge/util/packages/ari/chunk-2KRPWS2R.mjs
import {
  PassionfruitAskActivityAriResourceOwner,
  PassionfruitAskActivityAriResourceType
} from "./chunk-J7F3OJRT.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/ask-activity/manifest.ts
var passionfruitAskActivityAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitAskActivityAriResourceOwner,
  resourceType: PassionfruitAskActivityAriResourceType,
  resourceIdSlug: "workspace/{activationId}/{askId}/{askActivityId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    askId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    askActivityId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/ask-activity/index.ts
var PassionfruitAskActivityAri = class _PassionfruitAskActivityAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._askId = opts.resourceIdSegmentValues.askId, this._askActivityId = opts.resourceIdSegmentValues.askActivityId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get askId() {
    return this._askId;
  }
  get askActivityId() {
    return this._askActivityId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitAskActivityAriStaticOpts.qualifier,
      platformQualifier: passionfruitAskActivityAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: passionfruitAskActivityAriStaticOpts.resourceOwner,
      resourceType: passionfruitAskActivityAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.activationId}/${opts.askId}/${opts.askActivityId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        askId: opts.askId,
        askActivityId: opts.askActivityId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitAskActivityAriStaticOpts);
    return new _PassionfruitAskActivityAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitAskActivityAriStaticOpts);
    return new _PassionfruitAskActivityAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      askId: this.askId,
      askActivityId: this.askActivityId
    };
  }
};

export {
  PassionfruitAskActivityAri
};
