// @forge/util/packages/ari/chunk-A35RPAYV.mjs
import {
  JiraDcTertiaryJiraDcCommentAriResourceOwner,
  JiraDcTertiaryJiraDcCommentAriResourceType
} from "./chunk-TTDPBFH7.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc-tertiary.jira-dc/comment/manifest.ts
var jiraDcTertiaryJiraDcCommentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcTertiaryJiraDcCommentAriResourceOwner,
  resourceType: JiraDcTertiaryJiraDcCommentAriResourceType,
  resourceIdSlug: "{siteId}/{commentId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    commentId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc-tertiary.jira-dc/comment/index.ts
var JiraDcTertiaryJiraDcCommentAri = class _JiraDcTertiaryJiraDcCommentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._commentId = opts.resourceIdSegmentValues.commentId;
  }
  get siteId() {
    return this._siteId;
  }
  get commentId() {
    return this._commentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcTertiaryJiraDcCommentAriStaticOpts.qualifier,
      platformQualifier: jiraDcTertiaryJiraDcCommentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcTertiaryJiraDcCommentAriStaticOpts.resourceOwner,
      resourceType: jiraDcTertiaryJiraDcCommentAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.commentId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        commentId: opts.commentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcTertiaryJiraDcCommentAriStaticOpts);
    return new _JiraDcTertiaryJiraDcCommentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcTertiaryJiraDcCommentAriStaticOpts);
    return new _JiraDcTertiaryJiraDcCommentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      commentId: this.commentId
    };
  }
};

export {
  JiraDcTertiaryJiraDcCommentAri
};
