// @forge/util/packages/ari/chunk-TSCPX54U.mjs
import {
  NotionNotionRemoteLinkAriResourceOwner,
  NotionNotionRemoteLinkAriResourceType
} from "./chunk-FUF75X6C.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/notion.notion/remote-link/manifest.ts
var notionNotionRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: NotionNotionRemoteLinkAriResourceOwner,
  resourceType: NotionNotionRemoteLinkAriResourceType,
  resourceIdSlug: "page/{pageId}",
  resourceIdSegmentFormats: {
    pageId: /[a-z0-9-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/notion.notion/remote-link/index.ts
var NotionNotionRemoteLinkAri = class _NotionNotionRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._pageId = opts.resourceIdSegmentValues.pageId;
  }
  get pageId() {
    return this._pageId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: notionNotionRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: notionNotionRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: notionNotionRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: notionNotionRemoteLinkAriStaticOpts.resourceType,
      resourceId: `page/${opts.pageId}`,
      resourceIdSegmentValues: {
        pageId: opts.pageId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, notionNotionRemoteLinkAriStaticOpts);
    return new _NotionNotionRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, notionNotionRemoteLinkAriStaticOpts);
    return new _NotionNotionRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      pageId: this.pageId
    };
  }
};

export {
  NotionNotionRemoteLinkAri
};
