// @forge/util/packages/ari/chunk-4YWTOI4S.mjs
import {
  GraphWorkItemAriResourceOwner,
  GraphWorkItemAriResourceType
} from "./chunk-WDMLMNL4.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/work-item/manifest.ts
var graphWorkItemAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphWorkItemAriResourceOwner,
  resourceType: GraphWorkItemAriResourceType,
  resourceIdSlug: "activation/{activationId}/{workItemId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    workItemId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/work-item/index.ts
var GraphWorkItemAri = class _GraphWorkItemAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._workItemId = opts.resourceIdSegmentValues.workItemId;
  }
  get activationId() {
    return this._activationId;
  }
  get workItemId() {
    return this._workItemId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphWorkItemAriStaticOpts.qualifier,
      platformQualifier: graphWorkItemAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphWorkItemAriStaticOpts.resourceOwner,
      resourceType: graphWorkItemAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.workItemId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        workItemId: opts.workItemId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphWorkItemAriStaticOpts);
    return new _GraphWorkItemAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphWorkItemAriStaticOpts);
    return new _GraphWorkItemAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      workItemId: this.workItemId
    };
  }
};

export {
  GraphWorkItemAri
};
