// @forge/util/packages/ari/chunk-3S444J2H.mjs
import {
  CompassCampaignAriResourceOwner,
  CompassCampaignAriResourceType
} from "./chunk-FVJCFF7U.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/compass/campaign/manifest.ts
var compassCampaignAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: CompassCampaignAriResourceOwner,
  resourceType: CompassCampaignAriResourceType,
  resourceIdSlug: "{workspaceId}/{campaignId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    campaignId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/compass/campaign/index.ts
var CompassCampaignAri = class _CompassCampaignAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._campaignId = opts.resourceIdSegmentValues.campaignId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get campaignId() {
    return this._campaignId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: compassCampaignAriStaticOpts.qualifier,
      platformQualifier: compassCampaignAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: compassCampaignAriStaticOpts.resourceOwner,
      resourceType: compassCampaignAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.campaignId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        campaignId: opts.campaignId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, compassCampaignAriStaticOpts);
    return new _CompassCampaignAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, compassCampaignAriStaticOpts);
    return new _CompassCampaignAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      campaignId: this.campaignId
    };
  }
};

export {
  CompassCampaignAri
};
