// @forge/util/packages/ari/chunk-DFGH2ASC.mjs
import {
  ProformaHelpcenterFormAriResourceOwner,
  ProformaHelpcenterFormAriResourceType
} from "./chunk-OBKMHUPG.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/proforma/helpcenter-form/manifest.ts
var proformaHelpcenterFormAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ProformaHelpcenterFormAriResourceOwner,
  resourceType: ProformaHelpcenterFormAriResourceType,
  resourceIdSlug: "activation/{activationId}/{formId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    formId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/proforma/helpcenter-form/index.ts
var ProformaHelpcenterFormAri = class _ProformaHelpcenterFormAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._formId = opts.resourceIdSegmentValues.formId;
  }
  get activationId() {
    return this._activationId;
  }
  get formId() {
    return this._formId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: proformaHelpcenterFormAriStaticOpts.qualifier,
      platformQualifier: proformaHelpcenterFormAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: proformaHelpcenterFormAriStaticOpts.resourceOwner,
      resourceType: proformaHelpcenterFormAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.formId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        formId: opts.formId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, proformaHelpcenterFormAriStaticOpts);
    return new _ProformaHelpcenterFormAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, proformaHelpcenterFormAriStaticOpts);
    return new _ProformaHelpcenterFormAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      formId: this.formId
    };
  }
};

export {
  ProformaHelpcenterFormAri
};
