// @forge/util/packages/ari/chunk-IQO5JJSA.mjs
import {
  HubspotHubspotDealAriResourceOwner,
  HubspotHubspotDealAriResourceType
} from "./chunk-EGG5RIMA.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/hubspot.hubspot/deal/manifest.ts
var hubspotHubspotDealAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: HubspotHubspotDealAriResourceOwner,
  resourceType: HubspotHubspotDealAriResourceType,
  resourceIdSlug: "portal/{portalId}/deal/{dealId}",
  resourceIdSegmentFormats: {
    portalId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    dealId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/hubspot.hubspot/deal/index.ts
var HubspotHubspotDealAri = class _HubspotHubspotDealAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._portalId = opts.resourceIdSegmentValues.portalId, this._dealId = opts.resourceIdSegmentValues.dealId;
  }
  get portalId() {
    return this._portalId;
  }
  get dealId() {
    return this._dealId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: hubspotHubspotDealAriStaticOpts.qualifier,
      platformQualifier: hubspotHubspotDealAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: hubspotHubspotDealAriStaticOpts.resourceOwner,
      resourceType: hubspotHubspotDealAriStaticOpts.resourceType,
      resourceId: `portal/${opts.portalId}/deal/${opts.dealId}`,
      resourceIdSegmentValues: {
        portalId: opts.portalId,
        dealId: opts.dealId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, hubspotHubspotDealAriStaticOpts);
    return new _HubspotHubspotDealAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, hubspotHubspotDealAriStaticOpts);
    return new _HubspotHubspotDealAri(opts);
  }
  getVariables() {
    return {
      portalId: this.portalId,
      dealId: this.dealId
    };
  }
};

export {
  HubspotHubspotDealAri
};
