// @forge/util/packages/ari/chunk-GPC7GBLC.mjs
import {
  WorkdayWorkdayOrganisationAriResourceOwner,
  WorkdayWorkdayOrganisationAriResourceType
} from "./chunk-3KFI5HKL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/workday.workday/organisation/manifest.ts
var workdayWorkdayOrganisationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: WorkdayWorkdayOrganisationAriResourceOwner,
  resourceType: WorkdayWorkdayOrganisationAriResourceType,
  resourceIdSlug: "{workdayId}",
  resourceIdSegmentFormats: {
    workdayId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday.workday/organisation/index.ts
var WorkdayWorkdayOrganisationAri = class _WorkdayWorkdayOrganisationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayId = opts.resourceIdSegmentValues.workdayId;
  }
  get workdayId() {
    return this._workdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayWorkdayOrganisationAriStaticOpts.qualifier,
      platformQualifier: workdayWorkdayOrganisationAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayWorkdayOrganisationAriStaticOpts.resourceOwner,
      resourceType: workdayWorkdayOrganisationAriStaticOpts.resourceType,
      resourceId: `${opts.workdayId}`,
      resourceIdSegmentValues: {
        workdayId: opts.workdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayWorkdayOrganisationAriStaticOpts);
    return new _WorkdayWorkdayOrganisationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayWorkdayOrganisationAriStaticOpts);
    return new _WorkdayWorkdayOrganisationAri(opts);
  }
  getVariables() {
    return {
      workdayId: this.workdayId
    };
  }
};

export {
  WorkdayWorkdayOrganisationAri
};
