// @forge/util/packages/ari/chunk-TQVGV52H.mjs
import {
  PlatformTestProvisioningEntityAriResourceOwner,
  PlatformTestProvisioningEntityAriResourceType
} from "./chunk-FF4XBWPF.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/platform/test-provisioning-entity/manifest.ts
var platformTestProvisioningEntityAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PlatformTestProvisioningEntityAriResourceOwner,
  resourceType: PlatformTestProvisioningEntityAriResourceType,
  resourceIdSlug: "{testEntityId}",
  resourceIdSegmentFormats: {
    testEntityId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/platform/test-provisioning-entity/index.ts
var PlatformTestProvisioningEntityAri = class _PlatformTestProvisioningEntityAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._testEntityId = opts.resourceIdSegmentValues.testEntityId;
  }
  get testEntityId() {
    return this._testEntityId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: platformTestProvisioningEntityAriStaticOpts.qualifier,
      platformQualifier: platformTestProvisioningEntityAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: platformTestProvisioningEntityAriStaticOpts.resourceOwner,
      resourceType: platformTestProvisioningEntityAriStaticOpts.resourceType,
      resourceId: `${opts.testEntityId}`,
      resourceIdSegmentValues: {
        testEntityId: opts.testEntityId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, platformTestProvisioningEntityAriStaticOpts);
    return new _PlatformTestProvisioningEntityAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, platformTestProvisioningEntityAriStaticOpts);
    return new _PlatformTestProvisioningEntityAri(opts);
  }
  getVariables() {
    return {
      testEntityId: this.testEntityId
    };
  }
};

export {
  PlatformTestProvisioningEntityAri
};
