// @forge/util/packages/ari/chunk-7UMQCQVN.mjs
import {
  GoogleAccountAriResourceOwner,
  GoogleAccountAriResourceType
} from "./chunk-PYVPLXET.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/google/account/manifest.ts
var googleAccountAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GoogleAccountAriResourceOwner,
  resourceType: GoogleAccountAriResourceType,
  resourceIdSlug: "{accountId}",
  resourceIdSegmentFormats: {
    accountId: /[a-zA-Z0-9\-_.~@;{}=]{0,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/google/account/index.ts
var GoogleAccountAri = class _GoogleAccountAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._accountId = opts.resourceIdSegmentValues.accountId;
  }
  get accountId() {
    return this._accountId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googleAccountAriStaticOpts.qualifier,
      platformQualifier: googleAccountAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googleAccountAriStaticOpts.resourceOwner,
      resourceType: googleAccountAriStaticOpts.resourceType,
      resourceId: `${opts.accountId}`,
      resourceIdSegmentValues: {
        accountId: opts.accountId || ""
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googleAccountAriStaticOpts);
    return new _GoogleAccountAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googleAccountAriStaticOpts);
    return new _GoogleAccountAri(opts);
  }
  getVariables() {
    return {
      accountId: this.accountId
    };
  }
};

export {
  GoogleAccountAri
};
