// @forge/util/packages/ari/chunk-XSEINBJG.mjs
import {
  HomeWorkspaceAriResourceOwner,
  HomeWorkspaceAriResourceType
} from "./chunk-OEOCSY6S.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/home/workspace/manifest.ts
var homeWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: HomeWorkspaceAriResourceOwner,
  resourceType: HomeWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/home/workspace/index.ts
var HomeWorkspaceAri = class _HomeWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: homeWorkspaceAriStaticOpts.qualifier,
      platformQualifier: homeWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: homeWorkspaceAriStaticOpts.resourceOwner,
      resourceType: homeWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, homeWorkspaceAriStaticOpts);
    return new _HomeWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, homeWorkspaceAriStaticOpts);
    return new _HomeWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId
    };
  }
};

export {
  HomeWorkspaceAri
};
