// @forge/util/packages/ari/chunk-NWP2T34O.mjs
import {
  JiraDcJiraDcCommentAriResourceOwner,
  JiraDcJiraDcCommentAriResourceType
} from "./chunk-QECR4HD2.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc.jira-dc/comment/manifest.ts
var jiraDcJiraDcCommentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcJiraDcCommentAriResourceOwner,
  resourceType: JiraDcJiraDcCommentAriResourceType,
  resourceIdSlug: "{siteId}/{commentId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    commentId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc.jira-dc/comment/index.ts
var JiraDcJiraDcCommentAri = class _JiraDcJiraDcCommentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._commentId = opts.resourceIdSegmentValues.commentId;
  }
  get siteId() {
    return this._siteId;
  }
  get commentId() {
    return this._commentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcJiraDcCommentAriStaticOpts.qualifier,
      platformQualifier: jiraDcJiraDcCommentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcJiraDcCommentAriStaticOpts.resourceOwner,
      resourceType: jiraDcJiraDcCommentAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.commentId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        commentId: opts.commentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcJiraDcCommentAriStaticOpts);
    return new _JiraDcJiraDcCommentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcJiraDcCommentAriStaticOpts);
    return new _JiraDcJiraDcCommentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      commentId: this.commentId
    };
  }
};

export {
  JiraDcJiraDcCommentAri
};
