// @forge/util/packages/ari/chunk-AE66OUK7.mjs
import {
  JiraDcJiraDcWorkItemAriResourceOwner,
  JiraDcJiraDcWorkItemAriResourceType
} from "./chunk-F3U77W3D.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc.jira-dc/work-item/manifest.ts
var jiraDcJiraDcWorkItemAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcJiraDcWorkItemAriResourceOwner,
  resourceType: JiraDcJiraDcWorkItemAriResourceType,
  resourceIdSlug: "{siteId}/{workItemId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    workItemId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc.jira-dc/work-item/index.ts
var JiraDcJiraDcWorkItemAri = class _JiraDcJiraDcWorkItemAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._workItemId = opts.resourceIdSegmentValues.workItemId;
  }
  get siteId() {
    return this._siteId;
  }
  get workItemId() {
    return this._workItemId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcJiraDcWorkItemAriStaticOpts.qualifier,
      platformQualifier: jiraDcJiraDcWorkItemAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcJiraDcWorkItemAriStaticOpts.resourceOwner,
      resourceType: jiraDcJiraDcWorkItemAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.workItemId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        workItemId: opts.workItemId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcJiraDcWorkItemAriStaticOpts);
    return new _JiraDcJiraDcWorkItemAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcJiraDcWorkItemAriStaticOpts);
    return new _JiraDcJiraDcWorkItemAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workItemId: this.workItemId
    };
  }
};

export {
  JiraDcJiraDcWorkItemAri
};
