// @forge/util/packages/ari/chunk-WLFIAEY5.mjs
import {
  TrustEngEmailAriResourceOwner,
  TrustEngEmailAriResourceType
} from "./chunk-JRA23ZFE.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/trust-eng/email/manifest.ts
var trustEngEmailAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: TrustEngEmailAriResourceOwner,
  resourceType: TrustEngEmailAriResourceType,
  resourceIdSlug: "{emailID}",
  resourceIdSegmentFormats: {
    emailID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/trust-eng/email/index.ts
var TrustEngEmailAri = class _TrustEngEmailAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._emailID = opts.resourceIdSegmentValues.emailID;
  }
  get emailID() {
    return this._emailID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: trustEngEmailAriStaticOpts.qualifier,
      platformQualifier: trustEngEmailAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: trustEngEmailAriStaticOpts.resourceOwner,
      resourceType: trustEngEmailAriStaticOpts.resourceType,
      resourceId: `${opts.emailID}`,
      resourceIdSegmentValues: {
        emailID: opts.emailID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, trustEngEmailAriStaticOpts);
    return new _TrustEngEmailAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, trustEngEmailAriStaticOpts);
    return new _TrustEngEmailAri(opts);
  }
  getVariables() {
    return {
      emailID: this.emailID
    };
  }
};

export {
  TrustEngEmailAri
};
