// @forge/util/packages/ari/chunk-5WWD7PYR.mjs
import {
  SalesforceSalesforceDealAriResourceOwner,
  SalesforceSalesforceDealAriResourceType
} from "./chunk-IQJKOB2J.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/salesforce.salesforce/deal/manifest.ts
var salesforceSalesforceDealAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: SalesforceSalesforceDealAriResourceOwner,
  resourceType: SalesforceSalesforceDealAriResourceType,
  resourceIdSlug: "org/{instanceDomain}/opportunity/{id}",
  resourceIdSegmentFormats: {
    instanceDomain: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    id: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/salesforce.salesforce/deal/index.ts
var SalesforceSalesforceDealAri = class _SalesforceSalesforceDealAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._instanceDomain = opts.resourceIdSegmentValues.instanceDomain, this._id = opts.resourceIdSegmentValues.id;
  }
  get instanceDomain() {
    return this._instanceDomain;
  }
  get id() {
    return this._id;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: salesforceSalesforceDealAriStaticOpts.qualifier,
      platformQualifier: salesforceSalesforceDealAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: salesforceSalesforceDealAriStaticOpts.resourceOwner,
      resourceType: salesforceSalesforceDealAriStaticOpts.resourceType,
      resourceId: `org/${opts.instanceDomain}/opportunity/${opts.id}`,
      resourceIdSegmentValues: {
        instanceDomain: opts.instanceDomain,
        id: opts.id
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, salesforceSalesforceDealAriStaticOpts);
    return new _SalesforceSalesforceDealAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, salesforceSalesforceDealAriStaticOpts);
    return new _SalesforceSalesforceDealAri(opts);
  }
  getVariables() {
    return {
      instanceDomain: this.instanceDomain,
      id: this.id
    };
  }
};

export {
  SalesforceSalesforceDealAri
};
