// @forge/util/packages/ari/chunk-YNUAAVYJ.mjs
import {
  GraphConversationChunkAriResourceOwner,
  GraphConversationChunkAriResourceType
} from "./chunk-ZTNOGHIC.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/conversation-chunk/manifest.ts
var graphConversationChunkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphConversationChunkAriResourceOwner,
  resourceType: GraphConversationChunkAriResourceType,
  resourceIdSlug: "activation/{activationId}/{chunkId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    chunkId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/conversation-chunk/index.ts
var GraphConversationChunkAri = class _GraphConversationChunkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._chunkId = opts.resourceIdSegmentValues.chunkId;
  }
  get activationId() {
    return this._activationId;
  }
  get chunkId() {
    return this._chunkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphConversationChunkAriStaticOpts.qualifier,
      platformQualifier: graphConversationChunkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphConversationChunkAriStaticOpts.resourceOwner,
      resourceType: graphConversationChunkAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.chunkId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        chunkId: opts.chunkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphConversationChunkAriStaticOpts);
    return new _GraphConversationChunkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphConversationChunkAriStaticOpts);
    return new _GraphConversationChunkAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      chunkId: this.chunkId
    };
  }
};

export {
  GraphConversationChunkAri
};
