// @forge/util/packages/ari/chunk-ILJ5EQHZ.mjs
import {
  GoalWorkspaceAriResourceOwner,
  GoalWorkspaceAriResourceType
} from "./chunk-F4QTF457.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/goal/workspace/manifest.ts
var goalWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GoalWorkspaceAriResourceOwner,
  resourceType: GoalWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/goal/workspace/index.ts
var GoalWorkspaceAri = class _GoalWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: goalWorkspaceAriStaticOpts.qualifier,
      platformQualifier: goalWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: goalWorkspaceAriStaticOpts.resourceOwner,
      resourceType: goalWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, goalWorkspaceAriStaticOpts);
    return new _GoalWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, goalWorkspaceAriStaticOpts);
    return new _GoalWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId
    };
  }
};

export {
  GoalWorkspaceAri
};
