// @forge/util/packages/ari/chunk-O7SO6OWR.mjs
import {
  MicrosoftTeamsMessageAriResourceOwner,
  MicrosoftTeamsMessageAriResourceType
} from "./chunk-K3ZRWGLT.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft.teams/message/manifest.ts
var microsoftTeamsMessageAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftTeamsMessageAriResourceOwner,
  resourceType: MicrosoftTeamsMessageAriResourceType,
  resourceIdSlug: "{compositeMessage}",
  resourceIdSegmentFormats: {
    compositeMessage: /(teamId\/[a-f0-9]{8}-[a-f0-9]{4}-[1-5][a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}\/channelId\/19:[a-zA-Z0-9_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9_-]+\/messageId\/[0-9_-]+|chatId\/19:[a-zA-Z0-9_]+(?:-[a-zA-Z0-9_]+)*@[a-zA-Z0-9._-]+\.[a-zA-Z0-9_-]+\/messageId\/[0-9_-]+)/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft.teams/message/index.ts
var MicrosoftTeamsMessageAri = class _MicrosoftTeamsMessageAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._compositeMessage = opts.resourceIdSegmentValues.compositeMessage;
  }
  get compositeMessage() {
    return this._compositeMessage;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftTeamsMessageAriStaticOpts.qualifier,
      platformQualifier: microsoftTeamsMessageAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftTeamsMessageAriStaticOpts.resourceOwner,
      resourceType: microsoftTeamsMessageAriStaticOpts.resourceType,
      resourceId: `${opts.compositeMessage}`,
      resourceIdSegmentValues: {
        compositeMessage: opts.compositeMessage
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftTeamsMessageAriStaticOpts);
    return new _MicrosoftTeamsMessageAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftTeamsMessageAriStaticOpts);
    return new _MicrosoftTeamsMessageAri(opts);
  }
  getVariables() {
    return {
      compositeMessage: this.compositeMessage
    };
  }
};

export {
  MicrosoftTeamsMessageAri
};
