// @forge/util/packages/ari/chunk-NTIQPGKA.mjs
import {
  JiraDcSecondaryJiraDcWorkItemAriResourceOwner,
  JiraDcSecondaryJiraDcWorkItemAriResourceType
} from "./chunk-CS6RQC3V.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc-secondary.jira-dc/work-item/manifest.ts
var jiraDcSecondaryJiraDcWorkItemAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcSecondaryJiraDcWorkItemAriResourceOwner,
  resourceType: JiraDcSecondaryJiraDcWorkItemAriResourceType,
  resourceIdSlug: "{siteId}/{workItemId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    workItemId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc-secondary.jira-dc/work-item/index.ts
var JiraDcSecondaryJiraDcWorkItemAri = class _JiraDcSecondaryJiraDcWorkItemAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._workItemId = opts.resourceIdSegmentValues.workItemId;
  }
  get siteId() {
    return this._siteId;
  }
  get workItemId() {
    return this._workItemId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcSecondaryJiraDcWorkItemAriStaticOpts.qualifier,
      platformQualifier: jiraDcSecondaryJiraDcWorkItemAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcSecondaryJiraDcWorkItemAriStaticOpts.resourceOwner,
      resourceType: jiraDcSecondaryJiraDcWorkItemAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.workItemId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        workItemId: opts.workItemId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcSecondaryJiraDcWorkItemAriStaticOpts);
    return new _JiraDcSecondaryJiraDcWorkItemAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcSecondaryJiraDcWorkItemAriStaticOpts);
    return new _JiraDcSecondaryJiraDcWorkItemAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workItemId: this.workItemId
    };
  }
};

export {
  JiraDcSecondaryJiraDcWorkItemAri
};
