// @forge/util/packages/ari/chunk-Q623MFIX.mjs
import {
  JiraDcSecondaryJiraDcProjectAriResourceOwner,
  JiraDcSecondaryJiraDcProjectAriResourceType
} from "./chunk-T2BO3UPO.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc-secondary.jira-dc/project/manifest.ts
var jiraDcSecondaryJiraDcProjectAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcSecondaryJiraDcProjectAriResourceOwner,
  resourceType: JiraDcSecondaryJiraDcProjectAriResourceType,
  resourceIdSlug: "{siteId}/{projectId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc-secondary.jira-dc/project/index.ts
var JiraDcSecondaryJiraDcProjectAri = class _JiraDcSecondaryJiraDcProjectAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._projectId = opts.resourceIdSegmentValues.projectId;
  }
  get siteId() {
    return this._siteId;
  }
  get projectId() {
    return this._projectId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcSecondaryJiraDcProjectAriStaticOpts.qualifier,
      platformQualifier: jiraDcSecondaryJiraDcProjectAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcSecondaryJiraDcProjectAriStaticOpts.resourceOwner,
      resourceType: jiraDcSecondaryJiraDcProjectAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.projectId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        projectId: opts.projectId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcSecondaryJiraDcProjectAriStaticOpts);
    return new _JiraDcSecondaryJiraDcProjectAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcSecondaryJiraDcProjectAriStaticOpts);
    return new _JiraDcSecondaryJiraDcProjectAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      projectId: this.projectId
    };
  }
};

export {
  JiraDcSecondaryJiraDcProjectAri
};
