// @forge/util/packages/ari/chunk-YNH2YEXW.mjs
import {
  ZendeskZendeskDocumentAriResourceOwner,
  ZendeskZendeskDocumentAriResourceType
} from "./chunk-SAJNWHHF.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/zendesk.zendesk/document/manifest.ts
var zendeskZendeskDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ZendeskZendeskDocumentAriResourceOwner,
  resourceType: ZendeskZendeskDocumentAriResourceType,
  resourceIdSlug: "domain/{domain}/article/{id}",
  resourceIdSegmentFormats: {
    domain: /[a-zA-Z0-9\-_.]*/,
    // eslint-disable-line no-useless-escape
    id: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/zendesk.zendesk/document/index.ts
var ZendeskZendeskDocumentAri = class _ZendeskZendeskDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._domain = opts.resourceIdSegmentValues.domain, this._id = opts.resourceIdSegmentValues.id;
  }
  get domain() {
    return this._domain;
  }
  get id() {
    return this._id;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: zendeskZendeskDocumentAriStaticOpts.qualifier,
      platformQualifier: zendeskZendeskDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: zendeskZendeskDocumentAriStaticOpts.resourceOwner,
      resourceType: zendeskZendeskDocumentAriStaticOpts.resourceType,
      resourceId: `domain/${opts.domain}/article/${opts.id}`,
      resourceIdSegmentValues: {
        domain: opts.domain || "",
        id: opts.id
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, zendeskZendeskDocumentAriStaticOpts);
    return new _ZendeskZendeskDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, zendeskZendeskDocumentAriStaticOpts);
    return new _ZendeskZendeskDocumentAri(opts);
  }
  getVariables() {
    return {
      domain: this.domain,
      id: this.id
    };
  }
};

export {
  ZendeskZendeskDocumentAri
};
