// @forge/util/packages/ari/chunk-7M5DAIZO.mjs
import {
  GraphDealHistoryAriResourceOwner,
  GraphDealHistoryAriResourceType
} from "./chunk-HDZBCN3T.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/deal-history/manifest.ts
var graphDealHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDealHistoryAriResourceOwner,
  resourceType: GraphDealHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{dealId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    dealId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/deal-history/index.ts
var GraphDealHistoryAri = class _GraphDealHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._dealId = opts.resourceIdSegmentValues.dealId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get dealId() {
    return this._dealId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDealHistoryAriStaticOpts.qualifier,
      platformQualifier: graphDealHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDealHistoryAriStaticOpts.resourceOwner,
      resourceType: graphDealHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.dealId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        dealId: opts.dealId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDealHistoryAriStaticOpts);
    return new _GraphDealHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDealHistoryAriStaticOpts);
    return new _GraphDealHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      dealId: this.dealId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphDealHistoryAri
};
