// @forge/util/packages/ari/chunk-UERM7XJO.mjs
import {
  GraphCommentHistoryAriResourceOwner,
  GraphCommentHistoryAriResourceType
} from "./chunk-QMWB3HAP.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/comment-history/manifest.ts
var graphCommentHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCommentHistoryAriResourceOwner,
  resourceType: GraphCommentHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/comment-history/index.ts
var GraphCommentHistoryAri = class _GraphCommentHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCommentHistoryAriStaticOpts.qualifier,
      platformQualifier: graphCommentHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCommentHistoryAriStaticOpts.resourceOwner,
      resourceType: graphCommentHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCommentHistoryAriStaticOpts);
    return new _GraphCommentHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCommentHistoryAriStaticOpts);
    return new _GraphCommentHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphCommentHistoryAri
};
