// @forge/util/packages/ari/chunk-6MS4NWPD.mjs
import {
  DropboxDocumentAriResourceOwner,
  DropboxDocumentAriResourceType
} from "./chunk-ZDHZYOUG.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/dropbox/document/manifest.ts
var dropboxDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: DropboxDocumentAriResourceOwner,
  resourceType: DropboxDocumentAriResourceType,
  resourceIdSlug: "{documentId}",
  resourceIdSegmentFormats: {
    documentId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/dropbox/document/index.ts
var DropboxDocumentAri = class _DropboxDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._documentId = opts.resourceIdSegmentValues.documentId;
  }
  get documentId() {
    return this._documentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: dropboxDocumentAriStaticOpts.qualifier,
      platformQualifier: dropboxDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: dropboxDocumentAriStaticOpts.resourceOwner,
      resourceType: dropboxDocumentAriStaticOpts.resourceType,
      resourceId: `${opts.documentId}`,
      resourceIdSegmentValues: {
        documentId: opts.documentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, dropboxDocumentAriStaticOpts);
    return new _DropboxDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, dropboxDocumentAriStaticOpts);
    return new _DropboxDocumentAri(opts);
  }
  getVariables() {
    return {
      documentId: this.documentId
    };
  }
};

export {
  DropboxDocumentAri
};
