// @forge/util/packages/ari/chunk-U67CLJIY.mjs
import {
  AvpPipelineQueryAriResourceOwner,
  AvpPipelineQueryAriResourceType
} from "./chunk-5JS6DFBL.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/avp/pipeline-query/manifest.ts
var avpPipelineQueryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: AvpPipelineQueryAriResourceOwner,
  resourceType: AvpPipelineQueryAriResourceType,
  resourceIdSlug: "activation/{activationId}/pipeline/{pipelineId}/{pipelineQueryId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    pipelineId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    pipelineQueryId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/pipeline-query/index.ts
var AvpPipelineQueryAri = class _AvpPipelineQueryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._pipelineId = opts.resourceIdSegmentValues.pipelineId, this._pipelineQueryId = opts.resourceIdSegmentValues.pipelineQueryId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get pipelineId() {
    return this._pipelineId;
  }
  get pipelineQueryId() {
    return this._pipelineQueryId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpPipelineQueryAriStaticOpts.qualifier,
      platformQualifier: avpPipelineQueryAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: avpPipelineQueryAriStaticOpts.resourceOwner,
      resourceType: avpPipelineQueryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/pipeline/${opts.pipelineId}/${opts.pipelineQueryId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        pipelineId: opts.pipelineId,
        pipelineQueryId: opts.pipelineQueryId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpPipelineQueryAriStaticOpts);
    return new _AvpPipelineQueryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpPipelineQueryAriStaticOpts);
    return new _AvpPipelineQueryAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      pipelineId: this.pipelineId,
      pipelineQueryId: this.pipelineQueryId
    };
  }
};

export {
  AvpPipelineQueryAri
};
