// @forge/util/packages/ari/chunk-MWMOPTQE.mjs
import {
  LoomVideoAriResourceOwner,
  LoomVideoAriResourceType
} from "./chunk-HNYYKJLO.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/loom/video/manifest.ts
var loomVideoAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: LoomVideoAriResourceOwner,
  resourceType: LoomVideoAriResourceType,
  resourceIdSlug: "activation/{activationId}/{videoId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    videoId: /[a-fA-F0-9]{32}/
    // eslint-disable-line no-useless-escape
  }
};

// src/loom/video/index.ts
var LoomVideoAri = class _LoomVideoAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._videoId = opts.resourceIdSegmentValues.videoId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get videoId() {
    return this._videoId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: loomVideoAriStaticOpts.qualifier,
      platformQualifier: loomVideoAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: loomVideoAriStaticOpts.resourceOwner,
      resourceType: loomVideoAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.videoId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        videoId: opts.videoId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, loomVideoAriStaticOpts);
    return new _LoomVideoAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, loomVideoAriStaticOpts);
    return new _LoomVideoAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      videoId: this.videoId
    };
  }
};

export {
  LoomVideoAri
};
