// @forge/util/packages/ari/chunk-DEDX4M3Y.mjs
import {
  GoalSiteAriResourceOwner,
  GoalSiteAriResourceType
} from "./chunk-QJAWPXE7.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/goal/site/manifest.ts
var goalSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GoalSiteAriResourceOwner,
  resourceType: GoalSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/goal/site/index.ts
var GoalSiteAri = class _GoalSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: goalSiteAriStaticOpts.qualifier,
      platformQualifier: goalSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: goalSiteAriStaticOpts.resourceOwner,
      resourceType: goalSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, goalSiteAriStaticOpts);
    return new _GoalSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, goalSiteAriStaticOpts);
    return new _GoalSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  GoalSiteAri
};
