// @forge/util/packages/ari/chunk-N622SMMT.mjs
import {
  ZendeskZendeskWorkItemAriResourceOwner,
  ZendeskZendeskWorkItemAriResourceType
} from "./chunk-HQ7UACPC.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/zendesk.zendesk/work-item/manifest.ts
var zendeskZendeskWorkItemAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ZendeskZendeskWorkItemAriResourceOwner,
  resourceType: ZendeskZendeskWorkItemAriResourceType,
  resourceIdSlug: "domain/{domain}/ticket/{id}",
  resourceIdSegmentFormats: {
    domain: /[a-zA-Z0-9\-_.]*/,
    // eslint-disable-line no-useless-escape
    id: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/zendesk.zendesk/work-item/index.ts
var ZendeskZendeskWorkItemAri = class _ZendeskZendeskWorkItemAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._domain = opts.resourceIdSegmentValues.domain, this._id = opts.resourceIdSegmentValues.id;
  }
  get domain() {
    return this._domain;
  }
  get id() {
    return this._id;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: zendeskZendeskWorkItemAriStaticOpts.qualifier,
      platformQualifier: zendeskZendeskWorkItemAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: zendeskZendeskWorkItemAriStaticOpts.resourceOwner,
      resourceType: zendeskZendeskWorkItemAriStaticOpts.resourceType,
      resourceId: `domain/${opts.domain}/ticket/${opts.id}`,
      resourceIdSegmentValues: {
        domain: opts.domain || "",
        id: opts.id
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, zendeskZendeskWorkItemAriStaticOpts);
    return new _ZendeskZendeskWorkItemAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, zendeskZendeskWorkItemAriStaticOpts);
    return new _ZendeskZendeskWorkItemAri(opts);
  }
  getVariables() {
    return {
      domain: this.domain,
      id: this.id
    };
  }
};

export {
  ZendeskZendeskWorkItemAri
};
