// @forge/util/packages/ari/chunk-HC6B6MEQ.mjs
import {
  PassionfruitDependencyRelatedContentAriResourceOwner,
  PassionfruitDependencyRelatedContentAriResourceType
} from "./chunk-CYFQPPII.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/passionfruit/dependency-related-content/manifest.ts
var passionfruitDependencyRelatedContentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: PassionfruitDependencyRelatedContentAriResourceOwner,
  resourceType: PassionfruitDependencyRelatedContentAriResourceType,
  resourceIdSlug: "workspace/{activationId}/{dependencyId}/{relatedContentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    dependencyId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relatedContentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/dependency-related-content/index.ts
var PassionfruitDependencyRelatedContentAri = class _PassionfruitDependencyRelatedContentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._dependencyId = opts.resourceIdSegmentValues.dependencyId, this._relatedContentId = opts.resourceIdSegmentValues.relatedContentId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get dependencyId() {
    return this._dependencyId;
  }
  get relatedContentId() {
    return this._relatedContentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitDependencyRelatedContentAriStaticOpts.qualifier,
      platformQualifier: passionfruitDependencyRelatedContentAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: passionfruitDependencyRelatedContentAriStaticOpts.resourceOwner,
      resourceType: passionfruitDependencyRelatedContentAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.activationId}/${opts.dependencyId}/${opts.relatedContentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        dependencyId: opts.dependencyId,
        relatedContentId: opts.relatedContentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitDependencyRelatedContentAriStaticOpts);
    return new _PassionfruitDependencyRelatedContentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitDependencyRelatedContentAriStaticOpts);
    return new _PassionfruitDependencyRelatedContentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      dependencyId: this.dependencyId,
      relatedContentId: this.relatedContentId
    };
  }
};

export {
  PassionfruitDependencyRelatedContentAri
};
