// @forge/util/packages/ari/chunk-3YUQQX6F.mjs
import {
  ProformaIssueFormAriResourceOwner,
  ProformaIssueFormAriResourceType
} from "./chunk-PZ7IE2Z4.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/proforma/issue-form/manifest.ts
var proformaIssueFormAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ProformaIssueFormAriResourceOwner,
  resourceType: ProformaIssueFormAriResourceType,
  resourceIdSlug: "activation/{activationId}/{formId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    formId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/proforma/issue-form/index.ts
var ProformaIssueFormAri = class _ProformaIssueFormAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._formId = opts.resourceIdSegmentValues.formId;
  }
  get activationId() {
    return this._activationId;
  }
  get formId() {
    return this._formId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: proformaIssueFormAriStaticOpts.qualifier,
      platformQualifier: proformaIssueFormAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: proformaIssueFormAriStaticOpts.resourceOwner,
      resourceType: proformaIssueFormAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.formId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        formId: opts.formId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, proformaIssueFormAriStaticOpts);
    return new _ProformaIssueFormAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, proformaIssueFormAriStaticOpts);
    return new _ProformaIssueFormAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      formId: this.formId
    };
  }
};

export {
  ProformaIssueFormAri
};
