// @forge/util/packages/ari/chunk-ENF6UAUY.mjs
import {
  BoxBoxRemoteLinkAriResourceOwner,
  BoxBoxRemoteLinkAriResourceType
} from "./chunk-5YTZA3BU.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/box.box/remote-link/manifest.ts
var boxBoxRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: BoxBoxRemoteLinkAriResourceOwner,
  resourceType: BoxBoxRemoteLinkAriResourceType,
  resourceIdSlug: "file/{fileId}",
  resourceIdSegmentFormats: {
    fileId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/box.box/remote-link/index.ts
var BoxBoxRemoteLinkAri = class _BoxBoxRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._fileId = opts.resourceIdSegmentValues.fileId;
  }
  get fileId() {
    return this._fileId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: boxBoxRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: boxBoxRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: boxBoxRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: boxBoxRemoteLinkAriStaticOpts.resourceType,
      resourceId: `file/${opts.fileId}`,
      resourceIdSegmentValues: {
        fileId: opts.fileId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, boxBoxRemoteLinkAriStaticOpts);
    return new _BoxBoxRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, boxBoxRemoteLinkAriStaticOpts);
    return new _BoxBoxRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      fileId: this.fileId
    };
  }
};

export {
  BoxBoxRemoteLinkAri
};
