// @forge/util/packages/ari/chunk-YL7V7CED.mjs
import {
  ConfluenceLabelAriResourceOwner,
  ConfluenceLabelAriResourceType
} from "./chunk-5ZGPRH5S.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/confluence/label/manifest.ts
var confluenceLabelAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ConfluenceLabelAriResourceOwner,
  resourceType: ConfluenceLabelAriResourceType,
  resourceIdSlug: "activation/{activationId}/{labelId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    labelId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/confluence/label/index.ts
var ConfluenceLabelAri = class _ConfluenceLabelAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._labelId = opts.resourceIdSegmentValues.labelId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get labelId() {
    return this._labelId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: confluenceLabelAriStaticOpts.qualifier,
      platformQualifier: confluenceLabelAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: confluenceLabelAriStaticOpts.resourceOwner,
      resourceType: confluenceLabelAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.labelId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        labelId: opts.labelId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, confluenceLabelAriStaticOpts);
    return new _ConfluenceLabelAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, confluenceLabelAriStaticOpts);
    return new _ConfluenceLabelAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      labelId: this.labelId
    };
  }
};

export {
  ConfluenceLabelAri
};
