// @forge/util/packages/ari/chunk-OLS7DNW2.mjs
import {
  RadarSiteAriResourceOwner,
  RadarSiteAriResourceType
} from "./chunk-4LGPWYWF.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/radar/site/manifest.ts
var radarSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: RadarSiteAriResourceOwner,
  resourceType: RadarSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/
    // eslint-disable-line no-useless-escape
  }
};

// src/radar/site/index.ts
var RadarSiteAri = class _RadarSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: radarSiteAriStaticOpts.qualifier,
      platformQualifier: radarSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: radarSiteAriStaticOpts.resourceOwner,
      resourceType: radarSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, radarSiteAriStaticOpts);
    return new _RadarSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, radarSiteAriStaticOpts);
    return new _RadarSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  RadarSiteAri
};
