// @forge/util/packages/ari/chunk-LH2T5SM6.mjs
import {
  GraphCalendarEventHistoryAriResourceOwner,
  GraphCalendarEventHistoryAriResourceType
} from "./chunk-62SYOXYN.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/calendar-event-history/manifest.ts
var graphCalendarEventHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCalendarEventHistoryAriResourceOwner,
  resourceType: GraphCalendarEventHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{calendarEventId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    calendarEventId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/calendar-event-history/index.ts
var GraphCalendarEventHistoryAri = class _GraphCalendarEventHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._calendarEventId = opts.resourceIdSegmentValues.calendarEventId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get calendarEventId() {
    return this._calendarEventId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCalendarEventHistoryAriStaticOpts.qualifier,
      platformQualifier: graphCalendarEventHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCalendarEventHistoryAriStaticOpts.resourceOwner,
      resourceType: graphCalendarEventHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.calendarEventId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        calendarEventId: opts.calendarEventId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCalendarEventHistoryAriStaticOpts);
    return new _GraphCalendarEventHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCalendarEventHistoryAriStaticOpts);
    return new _GraphCalendarEventHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      calendarEventId: this.calendarEventId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphCalendarEventHistoryAri
};
