// @forge/util/packages/ari/chunk-NXJK634D.mjs
import {
  ElevatePrecalibrationPillarAriResourceOwner,
  ElevatePrecalibrationPillarAriResourceType
} from "./chunk-7ORFYDUC.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/elevate/precalibration-pillar/manifest.ts
var elevatePrecalibrationPillarAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: ElevatePrecalibrationPillarAriResourceOwner,
  resourceType: ElevatePrecalibrationPillarAriResourceType,
  resourceIdSlug: "{precalibrationId}/{precalibrationPillarId}",
  resourceIdSegmentFormats: {
    precalibrationId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    precalibrationPillarId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/elevate/precalibration-pillar/index.ts
var ElevatePrecalibrationPillarAri = class _ElevatePrecalibrationPillarAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._precalibrationId = opts.resourceIdSegmentValues.precalibrationId, this._precalibrationPillarId = opts.resourceIdSegmentValues.precalibrationPillarId;
  }
  get siteId() {
    return this._siteId;
  }
  get precalibrationId() {
    return this._precalibrationId;
  }
  get precalibrationPillarId() {
    return this._precalibrationPillarId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: elevatePrecalibrationPillarAriStaticOpts.qualifier,
      platformQualifier: elevatePrecalibrationPillarAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: elevatePrecalibrationPillarAriStaticOpts.resourceOwner,
      resourceType: elevatePrecalibrationPillarAriStaticOpts.resourceType,
      resourceId: `${opts.precalibrationId}/${opts.precalibrationPillarId}`,
      resourceIdSegmentValues: {
        precalibrationId: opts.precalibrationId,
        precalibrationPillarId: opts.precalibrationPillarId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, elevatePrecalibrationPillarAriStaticOpts);
    return new _ElevatePrecalibrationPillarAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, elevatePrecalibrationPillarAriStaticOpts);
    return new _ElevatePrecalibrationPillarAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      precalibrationId: this.precalibrationId,
      precalibrationPillarId: this.precalibrationPillarId
    };
  }
};

export {
  ElevatePrecalibrationPillarAri
};
