// @forge/util/packages/ari/chunk-OKLHDWC7.mjs
import {
  LoginAlerterSiteAriResourceOwner,
  LoginAlerterSiteAriResourceType
} from "./chunk-L6MUR4AX.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/login-alerter/site/manifest.ts
var loginAlerterSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: LoginAlerterSiteAriResourceOwner,
  resourceType: LoginAlerterSiteAriResourceType,
  resourceIdSlug: "login-alerter",
  resourceIdSegmentFormats: {}
};

// src/login-alerter/site/index.ts
var LoginAlerterSiteAri = class _LoginAlerterSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: loginAlerterSiteAriStaticOpts.qualifier,
      platformQualifier: loginAlerterSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: loginAlerterSiteAriStaticOpts.resourceOwner,
      resourceType: loginAlerterSiteAriStaticOpts.resourceType,
      resourceId: "login-alerter",
      resourceIdSegmentValues: {}
    }, ariOpts = AriParser.fromOpts(derivedOpts, loginAlerterSiteAriStaticOpts);
    return new _LoginAlerterSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, loginAlerterSiteAriStaticOpts);
    return new _LoginAlerterSiteAri(opts);
  }
  getVariables() {
    return {};
  }
};

export {
  LoginAlerterSiteAri
};
