// @forge/util/packages/ari/chunk-6SGR4JWH.mjs
import {
  JiraDcTertiaryJiraDcSpaceAriResourceOwner,
  JiraDcTertiaryJiraDcSpaceAriResourceType
} from "./chunk-FDSHXPYJ.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc-tertiary.jira-dc/space/manifest.ts
var jiraDcTertiaryJiraDcSpaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcTertiaryJiraDcSpaceAriResourceOwner,
  resourceType: JiraDcTertiaryJiraDcSpaceAriResourceType,
  resourceIdSlug: "{siteId}/{spaceId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    spaceId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc-tertiary.jira-dc/space/index.ts
var JiraDcTertiaryJiraDcSpaceAri = class _JiraDcTertiaryJiraDcSpaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._spaceId = opts.resourceIdSegmentValues.spaceId;
  }
  get siteId() {
    return this._siteId;
  }
  get spaceId() {
    return this._spaceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcTertiaryJiraDcSpaceAriStaticOpts.qualifier,
      platformQualifier: jiraDcTertiaryJiraDcSpaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcTertiaryJiraDcSpaceAriStaticOpts.resourceOwner,
      resourceType: jiraDcTertiaryJiraDcSpaceAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.spaceId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        spaceId: opts.spaceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcTertiaryJiraDcSpaceAriStaticOpts);
    return new _JiraDcTertiaryJiraDcSpaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcTertiaryJiraDcSpaceAriStaticOpts);
    return new _JiraDcTertiaryJiraDcSpaceAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      spaceId: this.spaceId
    };
  }
};

export {
  JiraDcTertiaryJiraDcSpaceAri
};
