// @forge/util/packages/ari/chunk-CWBWH26T.mjs
import {
  KnowledgeServingAndAccessTopicAriResourceOwner,
  KnowledgeServingAndAccessTopicAriResourceType
} from "./chunk-XRVVNMAO.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/knowledge-serving-and-access/topic/manifest.ts
var knowledgeServingAndAccessTopicAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: KnowledgeServingAndAccessTopicAriResourceOwner,
  resourceType: KnowledgeServingAndAccessTopicAriResourceType,
  resourceIdSlug: "{topicId}",
  resourceIdSegmentFormats: {
    topicId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/knowledge-serving-and-access/topic/index.ts
var KnowledgeServingAndAccessTopicAri = class _KnowledgeServingAndAccessTopicAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._topicId = opts.resourceIdSegmentValues.topicId;
  }
  get siteId() {
    return this._siteId;
  }
  get topicId() {
    return this._topicId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: knowledgeServingAndAccessTopicAriStaticOpts.qualifier,
      platformQualifier: knowledgeServingAndAccessTopicAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: knowledgeServingAndAccessTopicAriStaticOpts.resourceOwner,
      resourceType: knowledgeServingAndAccessTopicAriStaticOpts.resourceType,
      resourceId: `${opts.topicId}`,
      resourceIdSegmentValues: {
        topicId: opts.topicId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, knowledgeServingAndAccessTopicAriStaticOpts);
    return new _KnowledgeServingAndAccessTopicAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, knowledgeServingAndAccessTopicAriStaticOpts);
    return new _KnowledgeServingAndAccessTopicAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      topicId: this.topicId
    };
  }
};

export {
  KnowledgeServingAndAccessTopicAri
};
