// @forge/util/packages/ari/chunk-2YAMYMXC.mjs
import {
  FigmaFigmaDesignAriResourceOwner,
  FigmaFigmaDesignAriResourceType
} from "./chunk-T22TGH5G.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/figma.figma/design/manifest.ts
var figmaFigmaDesignAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: FigmaFigmaDesignAriResourceOwner,
  resourceType: FigmaFigmaDesignAriResourceType,
  resourceIdSlug: "{designId}",
  resourceIdSegmentFormats: {
    designId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/figma.figma/design/index.ts
var FigmaFigmaDesignAri = class _FigmaFigmaDesignAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._designId = opts.resourceIdSegmentValues.designId;
  }
  get designId() {
    return this._designId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: figmaFigmaDesignAriStaticOpts.qualifier,
      platformQualifier: figmaFigmaDesignAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: figmaFigmaDesignAriStaticOpts.resourceOwner,
      resourceType: figmaFigmaDesignAriStaticOpts.resourceType,
      resourceId: `${opts.designId}`,
      resourceIdSegmentValues: {
        designId: opts.designId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, figmaFigmaDesignAriStaticOpts);
    return new _FigmaFigmaDesignAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, figmaFigmaDesignAriStaticOpts);
    return new _FigmaFigmaDesignAri(opts);
  }
  getVariables() {
    return {
      designId: this.designId
    };
  }
};

export {
  FigmaFigmaDesignAri
};
