// @forge/util/packages/ari/chunk-FDNQJ433.mjs
import {
  JiraDcSecondaryJiraDcCommentAriResourceOwner,
  JiraDcSecondaryJiraDcCommentAriResourceType
} from "./chunk-UDET3NPV.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/jira-dc-secondary.jira-dc/comment/manifest.ts
var jiraDcSecondaryJiraDcCommentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: JiraDcSecondaryJiraDcCommentAriResourceOwner,
  resourceType: JiraDcSecondaryJiraDcCommentAriResourceType,
  resourceIdSlug: "{siteId}/{commentId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    commentId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-dc-secondary.jira-dc/comment/index.ts
var JiraDcSecondaryJiraDcCommentAri = class _JiraDcSecondaryJiraDcCommentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._commentId = opts.resourceIdSegmentValues.commentId;
  }
  get siteId() {
    return this._siteId;
  }
  get commentId() {
    return this._commentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraDcSecondaryJiraDcCommentAriStaticOpts.qualifier,
      platformQualifier: jiraDcSecondaryJiraDcCommentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraDcSecondaryJiraDcCommentAriStaticOpts.resourceOwner,
      resourceType: jiraDcSecondaryJiraDcCommentAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.commentId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        commentId: opts.commentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraDcSecondaryJiraDcCommentAriStaticOpts);
    return new _JiraDcSecondaryJiraDcCommentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraDcSecondaryJiraDcCommentAriStaticOpts);
    return new _JiraDcSecondaryJiraDcCommentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      commentId: this.commentId
    };
  }
};

export {
  JiraDcSecondaryJiraDcCommentAri
};
