// @forge/util/packages/ari/chunk-XCGYZTQK.mjs
import {
  MicrosoftCalendarEventAriResourceOwner,
  MicrosoftCalendarEventAriResourceType
} from "./chunk-KRNGEAQB.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/microsoft/calendar-event/manifest.ts
var microsoftCalendarEventAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: MicrosoftCalendarEventAriResourceOwner,
  resourceType: MicrosoftCalendarEventAriResourceType,
  resourceIdSlug: "outlook/userId/{userId}/calendarEventId/{calendarEventId}",
  resourceIdSegmentFormats: {
    userId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/,
    // eslint-disable-line no-useless-escape
    calendarEventId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/microsoft/calendar-event/index.ts
var MicrosoftCalendarEventAri = class _MicrosoftCalendarEventAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._userId = opts.resourceIdSegmentValues.userId, this._calendarEventId = opts.resourceIdSegmentValues.calendarEventId;
  }
  get userId() {
    return this._userId;
  }
  get calendarEventId() {
    return this._calendarEventId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: microsoftCalendarEventAriStaticOpts.qualifier,
      platformQualifier: microsoftCalendarEventAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: microsoftCalendarEventAriStaticOpts.resourceOwner,
      resourceType: microsoftCalendarEventAriStaticOpts.resourceType,
      resourceId: `outlook/userId/${opts.userId}/calendarEventId/${opts.calendarEventId}`,
      resourceIdSegmentValues: {
        userId: opts.userId,
        calendarEventId: opts.calendarEventId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, microsoftCalendarEventAriStaticOpts);
    return new _MicrosoftCalendarEventAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, microsoftCalendarEventAriStaticOpts);
    return new _MicrosoftCalendarEventAri(opts);
  }
  getVariables() {
    return {
      userId: this.userId,
      calendarEventId: this.calendarEventId
    };
  }
};

export {
  MicrosoftCalendarEventAri
};
