// @forge/util/packages/ari/chunk-EK3PSSRF.mjs
import {
  LoomMeetingAriResourceOwner,
  LoomMeetingAriResourceType
} from "./chunk-M5DHKYFA.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/loom/meeting/manifest.ts
var loomMeetingAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9_\\-.]{1,255}$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: LoomMeetingAriResourceOwner,
  resourceType: LoomMeetingAriResourceType,
  resourceIdSlug: "activation/{activationId}/{meetingId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    meetingId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/loom/meeting/index.ts
var LoomMeetingAri = class _LoomMeetingAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._meetingId = opts.resourceIdSegmentValues.meetingId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get meetingId() {
    return this._meetingId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: loomMeetingAriStaticOpts.qualifier,
      platformQualifier: loomMeetingAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: loomMeetingAriStaticOpts.resourceOwner,
      resourceType: loomMeetingAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.meetingId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        meetingId: opts.meetingId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, loomMeetingAriStaticOpts);
    return new _LoomMeetingAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, loomMeetingAriStaticOpts);
    return new _LoomMeetingAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      meetingId: this.meetingId
    };
  }
};

export {
  LoomMeetingAri
};
