// @forge/util/packages/ari/chunk-X4TIVCKP.mjs
import {
  VirtualAgentIntentAriResourceOwner,
  VirtualAgentIntentAriResourceType
} from "./chunk-X3TF4A6Y.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/virtual-agent/intent/manifest.ts
var virtualAgentIntentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: VirtualAgentIntentAriResourceOwner,
  resourceType: VirtualAgentIntentAriResourceType,
  resourceIdSlug: "{activationId}/{intentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    intentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/virtual-agent/intent/index.ts
var VirtualAgentIntentAri = class _VirtualAgentIntentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._intentId = opts.resourceIdSegmentValues.intentId;
  }
  get activationId() {
    return this._activationId;
  }
  get intentId() {
    return this._intentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: virtualAgentIntentAriStaticOpts.qualifier,
      platformQualifier: virtualAgentIntentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: virtualAgentIntentAriStaticOpts.resourceOwner,
      resourceType: virtualAgentIntentAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}/${opts.intentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        intentId: opts.intentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, virtualAgentIntentAriStaticOpts);
    return new _VirtualAgentIntentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, virtualAgentIntentAriStaticOpts);
    return new _VirtualAgentIntentAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      intentId: this.intentId
    };
  }
};

export {
  VirtualAgentIntentAri
};
