// @forge/util/packages/ari/chunk-2K3BMXST.mjs
import {
  GraphCustomerOrgHistoryAriResourceOwner,
  GraphCustomerOrgHistoryAriResourceType
} from "./chunk-JOMPZANB.mjs";
import {
  RegisteredAri
} from "./chunk-AB4PPISI.mjs";
import {
  AriParser
} from "./chunk-S7RIZVFI.mjs";

// src/graph/customer-org-history/manifest.ts
var graphCustomerOrgHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCustomerOrgHistoryAriResourceOwner,
  resourceType: GraphCustomerOrgHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{customerOrgId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    customerOrgId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/customer-org-history/index.ts
var GraphCustomerOrgHistoryAri = class _GraphCustomerOrgHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._customerOrgId = opts.resourceIdSegmentValues.customerOrgId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get customerOrgId() {
    return this._customerOrgId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCustomerOrgHistoryAriStaticOpts.qualifier,
      platformQualifier: graphCustomerOrgHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCustomerOrgHistoryAriStaticOpts.resourceOwner,
      resourceType: graphCustomerOrgHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.customerOrgId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        customerOrgId: opts.customerOrgId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCustomerOrgHistoryAriStaticOpts);
    return new _GraphCustomerOrgHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCustomerOrgHistoryAriStaticOpts);
    return new _GraphCustomerOrgHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      customerOrgId: this.customerOrgId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphCustomerOrgHistoryAri
};
