// @forge/util/packages/ari/index.mjs
import "./chunk-OCVP3I7F.mjs";
import {
  ZendeskZendeskDocumentAri
} from "./chunk-YNH2YEXW.mjs";
import "./chunk-SAJNWHHF.mjs";
import {
  ZendeskZendeskWorkItemAri
} from "./chunk-N622SMMT.mjs";
import "./chunk-HQ7UACPC.mjs";
import "./chunk-RKBTPBAW.mjs";
import {
  ZeplinZeplinRemoteLinkAri
} from "./chunk-ATB4PB3W.mjs";
import "./chunk-ONT6HQKB.mjs";
import "./chunk-SIQPZJXB.mjs";
import {
  WorkdayPositionAri
} from "./chunk-EITIKWOL.mjs";
import "./chunk-4PEBAWWW.mjs";
import {
  WorkdayStaffAri
} from "./chunk-ASHD2EKZ.mjs";
import "./chunk-BXN33GTZ.mjs";
import {
  WorkdaySupervisoryOrgAri
} from "./chunk-QGHMIDW3.mjs";
import "./chunk-D7PCKJQ7.mjs";
import {
  WorkdayWorkerAri
} from "./chunk-MFPTPQCO.mjs";
import "./chunk-A4ROCE3Q.mjs";
import {
  WorkdayCostCenterAri
} from "./chunk-ORE23AUB.mjs";
import "./chunk-4UYKCLC2.mjs";
import {
  WorkdayJobProfileAri
} from "./chunk-J6J2O6PQ.mjs";
import "./chunk-UO33O7I4.mjs";
import "./chunk-D4JENJQG.mjs";
import {
  WorkatoJobAri
} from "./chunk-W7ZX4Y2O.mjs";
import "./chunk-BPPBHTKA.mjs";
import {
  WorkatoRecipeAri
} from "./chunk-5FBSSOHX.mjs";
import "./chunk-ZH5P2SLW.mjs";
import "./chunk-TOYA3UTX.mjs";
import {
  WorkdayWorkdayPositionAri
} from "./chunk-5W46EKFA.mjs";
import "./chunk-MJXKCGLB.mjs";
import {
  WorkdayWorkdayWorkerAri
} from "./chunk-E5ZA5FLJ.mjs";
import "./chunk-AT3AV22J.mjs";
import {
  WorkdayWorkdayOrganisationAri
} from "./chunk-GPC7GBLC.mjs";
import "./chunk-3KFI5HKL.mjs";
import "./chunk-6G52RX3M.mjs";
import {
  VirtualAgentIntentQuestionAri
} from "./chunk-5DFVUJJB.mjs";
import "./chunk-5BAWUP2G.mjs";
import "./chunk-SBA32GLM.mjs";
import "./chunk-MJ72T4ZB.mjs";
import {
  VirtualAgentIntentTemplateAri
} from "./chunk-NFJZCBED.mjs";
import "./chunk-66RZPSB4.mjs";
import {
  VirtualAgentIntentAri
} from "./chunk-X4TIVCKP.mjs";
import "./chunk-X3TF4A6Y.mjs";
import {
  VirtualAgentFlowEditorAri
} from "./chunk-5H5KUQSR.mjs";
import "./chunk-ETXO2665.mjs";
import "./chunk-CH4L3UOZ.mjs";
import "./chunk-MPJ7GQCW.mjs";
import "./chunk-YJBKDP2P.mjs";
import "./chunk-FRZCZ4T6.mjs";
import "./chunk-H42OOLXG.mjs";
import {
  UsageTrackingUsageAri
} from "./chunk-SWE627ZW.mjs";
import "./chunk-63SGYTKK.mjs";
import {
  VirtualAgentConfigurationAri
} from "./chunk-PHXNUMZ3.mjs";
import "./chunk-TPOGUUFE.mjs";
import {
  VirtualAgentConversationAri
} from "./chunk-ZUWEDKAO.mjs";
import "./chunk-PF3FHNQU.mjs";
import "./chunk-NJFPSQZT.mjs";
import {
  UnifiedPortableProfileUnifiedProfileAri
} from "./chunk-OIHO6MAS.mjs";
import "./chunk-LRJWQT2R.mjs";
import {
  UnifiedPortableProfileLinkedAccountAri
} from "./chunk-POTRQEZU.mjs";
import "./chunk-GX4LHOQR.mjs";
import {
  UnifiedPortableProfileParentAccountAri
} from "./chunk-VYGYHCMO.mjs";
import "./chunk-PFRWENB5.mjs";
import {
  UnifiedPortableProfileUnifiedAccountAri
} from "./chunk-LW477RLQ.mjs";
import "./chunk-P5I3SEJO.mjs";
import "./chunk-JCXSKI7I.mjs";
import "./chunk-ATN7NKOC.mjs";
import {
  UnifiedHelpRoleAri
} from "./chunk-OEOWPF7P.mjs";
import "./chunk-BQV74DFA.mjs";
import {
  UnifiedHelpSiteAri
} from "./chunk-6ZB6ECVZ.mjs";
import "./chunk-OKKAA5RV.mjs";
import {
  UnifiedHelpWorkspaceAri
} from "./chunk-RMXRMKQ4.mjs";
import "./chunk-GXNUL5AQ.mjs";
import {
  TrustEngEmailAri
} from "./chunk-WLFIAEY5.mjs";
import "./chunk-JRA23ZFE.mjs";
import "./chunk-ILX3IAYS.mjs";
import {
  TrelloStickerAri
} from "./chunk-ZFNZG6GB.mjs";
import "./chunk-YKQLTY3Z.mjs";
import {
  TrelloUserAri
} from "./chunk-5R4C6BRK.mjs";
import "./chunk-OZFY5Y6K.mjs";
import {
  TrelloWorkspaceAri
} from "./chunk-AXG4AOGE.mjs";
import "./chunk-KVD2HPFJ.mjs";
import {
  TrelloPlannerEventCardAri
} from "./chunk-RZCM2CZO.mjs";
import "./chunk-I4CIEUHI.mjs";
import {
  TrelloPlannerAri
} from "./chunk-63YFV4PZ.mjs";
import "./chunk-BFIKUJAR.mjs";
import {
  TrelloRoleAri
} from "./chunk-KIJTU6X5.mjs";
import "./chunk-I5GECIAC.mjs";
import {
  TrelloSiteAri
} from "./chunk-FGMLSXOO.mjs";
import "./chunk-P3FZABZE.mjs";
import {
  TrelloLabelAri
} from "./chunk-MNJMUR4K.mjs";
import "./chunk-JOW7TB5R.mjs";
import {
  TrelloListAri
} from "./chunk-G6J7PDLO.mjs";
import "./chunk-TVSKQH7F.mjs";
import {
  TrelloLogoAri
} from "./chunk-5OVOKP63.mjs";
import "./chunk-SI47IE46.mjs";
import {
  TrelloPlannerCalendarAri
} from "./chunk-CZECT76G.mjs";
import "./chunk-WPEGY4E4.mjs";
import {
  TrelloChecklistAri
} from "./chunk-TL4DYLQ7.mjs";
import "./chunk-2FD4HSCA.mjs";
import {
  TrelloCustomFieldAri
} from "./chunk-YE4P4KFX.mjs";
import "./chunk-KGWMZBU5.mjs";
import {
  TrelloEmojiAri
} from "./chunk-NA2CUYGR.mjs";
import "./chunk-UD76ND34.mjs";
import {
  TrelloEnterpriseAri
} from "./chunk-6QLQNZFX.mjs";
import "./chunk-A3QGWDCS.mjs";
import {
  TrelloBackgroundAri
} from "./chunk-WH7FAXYP.mjs";
import "./chunk-5WRAUVE4.mjs";
import {
  TrelloBoardAri
} from "./chunk-HP62527J.mjs";
import "./chunk-HI4UEG2F.mjs";
import {
  TrelloCardAri
} from "./chunk-GUQ2FRFI.mjs";
import "./chunk-L5R7QT5M.mjs";
import {
  TrelloCheckItemAri
} from "./chunk-635QZO4I.mjs";
import "./chunk-SPJON63N.mjs";
import {
  TrelloAttachmentAri
} from "./chunk-FC35M4RB.mjs";
import "./chunk-BAWQZ3RT.mjs";
import {
  TrelloAvatarAri
} from "./chunk-OJIONXLS.mjs";
import "./chunk-S5DHPCLJ.mjs";
import "./chunk-2DYG6VY2.mjs";
import {
  TownsquareUpdateNoteAri
} from "./chunk-CWTC3RY3.mjs";
import "./chunk-RLUADO33.mjs";
import {
  TownsquareWorkspaceAri
} from "./chunk-X2QBDRA5.mjs";
import "./chunk-UC4GLVA3.mjs";
import {
  TownsquareQuestionAri
} from "./chunk-HIDEUQTZ.mjs";
import "./chunk-IKIQRIWE.mjs";
import {
  TownsquareRoleAri
} from "./chunk-W74BBEC7.mjs";
import "./chunk-RRNPSY4J.mjs";
import {
  TownsquareSiteAri
} from "./chunk-M36TUXDW.mjs";
import "./chunk-MC7VKTYC.mjs";
import {
  TownsquareTagAri
} from "./chunk-SDIRXMWH.mjs";
import "./chunk-7QVDZS4D.mjs";
import {
  TownsquareMetricAri
} from "./chunk-RNK5J6WA.mjs";
import "./chunk-XKTWZDLI.mjs";
import {
  TownsquareMilestoneAri
} from "./chunk-4VNQPOCS.mjs";
import "./chunk-42MIAPNU.mjs";
import {
  TownsquareProjectUpdateAri
} from "./chunk-YWH2Q5D3.mjs";
import "./chunk-MZER64KW.mjs";
import {
  TownsquareProjectAri
} from "./chunk-NCKUKPRU.mjs";
import "./chunk-ICBJW6N7.mjs";
import {
  TownsquareLearningAri
} from "./chunk-BUS4F7GN.mjs";
import "./chunk-4J5O222V.mjs";
import {
  TownsquareMetricTargetAri
} from "./chunk-TKKIRZDB.mjs";
import "./chunk-TQEX6RU4.mjs";
import {
  TownsquareMetricValueAri
} from "./chunk-OP3WKLAR.mjs";
import "./chunk-PJMNKWWL.mjs";
import "./chunk-SLSKSO7Q.mjs";
import "./chunk-7GGI72TL.mjs";
import {
  TownsquareGoalUpdateAri
} from "./chunk-B2BDPA6O.mjs";
import "./chunk-7RQBTEXE.mjs";
import {
  TownsquareGoalAri
} from "./chunk-A2CGJDTV.mjs";
import "./chunk-FPGL2PN7.mjs";
import {
  TownsquareHelpPointerAri
} from "./chunk-GCHBDGAD.mjs";
import "./chunk-GROHW3KB.mjs";
import "./chunk-SSWXVNRV.mjs";
import "./chunk-6XHDPQGW.mjs";
import {
  TodoistTodoistRemoteLinkAri
} from "./chunk-GNIYE5T5.mjs";
import "./chunk-JKOOQ576.mjs";
import {
  TownsquareCommentAri
} from "./chunk-3FZQ733R.mjs";
import "./chunk-GVAPRDNZ.mjs";
import "./chunk-COH6QOWF.mjs";
import {
  StripeStripeRemoteLinkAri
} from "./chunk-XCNIKIGY.mjs";
import "./chunk-NK2XSAGB.mjs";
import {
  SupportCustomerSupportAri
} from "./chunk-L2O7NMJX.mjs";
import "./chunk-55U3VS6Z.mjs";
import {
  SupportCustomerAri
} from "./chunk-OEMVJPIV.mjs";
import "./chunk-PHBXUJAW.mjs";
import "./chunk-IPHUDBLQ.mjs";
import {
  SpinnakerSpinnakerDeploymentAri
} from "./chunk-74URSN7M.mjs";
import "./chunk-EZ6KUXWG.mjs";
import "./chunk-TYQPBEKA.mjs";
import {
  StatuspageRoleAri
} from "./chunk-RY5ZUN3F.mjs";
import "./chunk-BT2GYN5W.mjs";
import {
  StatuspageSiteAri
} from "./chunk-QOZBRGMM.mjs";
import "./chunk-6IR7P2SS.mjs";
import "./chunk-WMXNNYB6.mjs";
import {
  SmartsheetSmartsheetDocumentAri
} from "./chunk-3IOWV2Z2.mjs";
import "./chunk-EOWM7GT2.mjs";
import {
  SmartsheetSmartsheetRemoteLinkAri
} from "./chunk-MJ3A7RHF.mjs";
import "./chunk-HG7EHHS2.mjs";
import "./chunk-6BRB66P2.mjs";
import {
  SlackSlackConversationAri
} from "./chunk-OGJBPPPD.mjs";
import "./chunk-SIDV5ZLZ.mjs";
import {
  SlackSlackMessageAri
} from "./chunk-LTUPEBNJ.mjs";
import "./chunk-RMAZFETB.mjs";
import "./chunk-CDVCXP3O.mjs";
import {
  SentrySentryRemoteLinkAri
} from "./chunk-XI7AS255.mjs";
import "./chunk-R33LKGOS.mjs";
import "./chunk-XS6E6VZA.mjs";
import {
  ServicenowServicenowWorkItemAri
} from "./chunk-T75JGZAD.mjs";
import "./chunk-KJ5FM7F5.mjs";
import "./chunk-TORQOWU2.mjs";
import {
  SettingsServiceSettingsAri
} from "./chunk-GYTLFUGJ.mjs";
import "./chunk-WX2ZBYNS.mjs";
import "./chunk-Q6V4TK26.mjs";
import {
  SearchEventAri
} from "./chunk-PR2UWXFD.mjs";
import "./chunk-MLTMULJ4.mjs";
import {
  SearchForwardDocumentSubscriptionAri
} from "./chunk-TGNWAU7N.mjs";
import "./chunk-LLH27RU3.mjs";
import {
  SearchForwardDocumentAri
} from "./chunk-3RQEZ4PX.mjs";
import "./chunk-I4OEQ7AG.mjs";
import {
  SearchPermissionContainerAri
} from "./chunk-Z2FY7NLX.mjs";
import "./chunk-RELMUIGW.mjs";
import "./chunk-I4MXTKSJ.mjs";
import {
  SalesforceSalesforceDealAri
} from "./chunk-5WWD7PYR.mjs";
import "./chunk-IQJKOB2J.mjs";
import {
  SalesforceSalesforceRemoteLinkAri
} from "./chunk-DP5RQ3DH.mjs";
import "./chunk-UM6AQHHT.mjs";
import "./chunk-HP62X4XV.mjs";
import {
  SandboxServiceSiteAri
} from "./chunk-PXXGT23O.mjs";
import "./chunk-4IDZPUVH.mjs";
import "./chunk-NTJ25HLA.mjs";
import {
  RuntimeAuthClientRoleAri
} from "./chunk-EPLQMWI3.mjs";
import "./chunk-N3DYIPHN.mjs";
import {
  SalesforceSalesforceCustomerOrgAri
} from "./chunk-L4LDQPBH.mjs";
import "./chunk-AVGRLOXF.mjs";
import "./chunk-G43IOKS7.mjs";
import {
  RovoWorkspaceAri
} from "./chunk-RLVKCH5J.mjs";
import "./chunk-WPBXANQL.mjs";
import {
  RovoFileAri
} from "./chunk-MD6YCAO4.mjs";
import "./chunk-ICUEPT2G.mjs";
import {
  RovoMessageAri
} from "./chunk-553W4TP2.mjs";
import "./chunk-7DUOYN26.mjs";
import {
  RovoScenarioAri
} from "./chunk-6KII5J3E.mjs";
import "./chunk-XBHB5O4E.mjs";
import {
  RovoSiteAri
} from "./chunk-Y4WGNVN2.mjs";
import "./chunk-JTG37E43.mjs";
import {
  RovoAgentAri
} from "./chunk-T65G4QSS.mjs";
import "./chunk-AM7JQZBA.mjs";
import {
  RovoCustomActionAri
} from "./chunk-2VXWBVQM.mjs";
import "./chunk-WNV5UZLB.mjs";
import "./chunk-CDFR7AZ5.mjs";
import {
  RadarWorkerAri
} from "./chunk-I2IQRKAZ.mjs";
import "./chunk-GH5TR45G.mjs";
import {
  RadarWorkspaceAri
} from "./chunk-TSUX6PV6.mjs";
import "./chunk-LLK4DGJH.mjs";
import {
  RadarPositionAri
} from "./chunk-SKW53Q5X.mjs";
import "./chunk-L3QKGWQW.mjs";
import {
  RadarSiteAri
} from "./chunk-OLS7DNW2.mjs";
import "./chunk-4LGPWYWF.mjs";
import "./chunk-UK3W2EQK.mjs";
import {
  PureRoleAri
} from "./chunk-Y7ZOW42H.mjs";
import "./chunk-4QXPSVRT.mjs";
import {
  PurePermissionAri
} from "./chunk-DTL5CHR3.mjs";
import "./chunk-BR3GY5NN.mjs";
import {
  PureResourceAri
} from "./chunk-KJXUKVJD.mjs";
import "./chunk-RSU34ULO.mjs";
import "./chunk-3V4LJKQB.mjs";
import {
  ProformaIssueFormAri
} from "./chunk-3YUQQX6F.mjs";
import "./chunk-PZ7IE2Z4.mjs";
import {
  ProformaProjectFormAri
} from "./chunk-AGU6IVS5.mjs";
import "./chunk-GBX2XPCH.mjs";
import {
  ProformaRoleAri
} from "./chunk-OKIMO5EO.mjs";
import "./chunk-ZNKCCQXV.mjs";
import "./chunk-WBAZXW64.mjs";
import {
  ProjectSiteAri
} from "./chunk-MLCBFNO5.mjs";
import "./chunk-FAZ4I6YC.mjs";
import {
  ProjectWorkspaceAri
} from "./chunk-EZAKBOKA.mjs";
import "./chunk-RC6RQXMI.mjs";
import {
  ProformaHelpcenterFormAri
} from "./chunk-DFGH2ASC.mjs";
import "./chunk-OBKMHUPG.mjs";
import "./chunk-27WJ432L.mjs";
import {
  PollinatorCheckAri
} from "./chunk-W7RZYFPV.mjs";
import "./chunk-M6X7UPKX.mjs";
import "./chunk-3VKOG5GD.mjs";
import {
  PostOfficeMessageInstanceAri
} from "./chunk-WQ3PYGQD.mjs";
import "./chunk-6N62CYFU.mjs";
import {
  PostOfficeMessageTemplateAri
} from "./chunk-WDLZTUXM.mjs";
import "./chunk-YT2S6YE6.mjs";
import {
  PostOfficeTriggerAri
} from "./chunk-NT7VKYPC.mjs";
import "./chunk-VNEE4Y6C.mjs";
import "./chunk-MGHXHAL5.mjs";
import {
  PlatformServicesStreamhubSchemaAri
} from "./chunk-LJ625XHV.mjs";
import "./chunk-KM3YDOZJ.mjs";
import "./chunk-SB6LDEPD.mjs";
import {
  PlatformSiteAri
} from "./chunk-QD2LYHMH.mjs";
import "./chunk-QGU7KDYS.mjs";
import {
  PlatformTestProvisioningEntityAri
} from "./chunk-TQVGV52H.mjs";
import "./chunk-FF4XBWPF.mjs";
import {
  PlatformOrgUserAri
} from "./chunk-7KY24GV6.mjs";
import "./chunk-REH6WMJF.mjs";
import {
  PlatformOrgAri
} from "./chunk-I32BISGP.mjs";
import "./chunk-ON6PRZUL.mjs";
import {
  PlatformProductAri
} from "./chunk-DFX2IWF3.mjs";
import "./chunk-EAIJCZUC.mjs";
import {
  PlatformSecureTunnelAri
} from "./chunk-JVKOFILI.mjs";
import "./chunk-7AEMSYL6.mjs";
import {
  PlatformIntegrationAri
} from "./chunk-6XQD66UH.mjs";
import "./chunk-VTU2ZIQ5.mjs";
import {
  PlatformLifecycleResourcePackageTypeAri
} from "./chunk-WN5RFAZT.mjs";
import "./chunk-YUROJJWW.mjs";
import {
  PlatformLifecycleResourcePackageAri
} from "./chunk-3QRNPJQD.mjs";
import "./chunk-AUDUMQTZ.mjs";
import {
  PlatformLifecycleResourceAri
} from "./chunk-NFSPPMFB.mjs";
import "./chunk-T36754GS.mjs";
import {
  PlatformClassificationTagAri
} from "./chunk-PPPY4YEP.mjs";
import "./chunk-JOF4WRFT.mjs";
import {
  PlatformCollaborationContextAri
} from "./chunk-7C3P4PYG.mjs";
import "./chunk-AX7CUYP3.mjs";
import "./chunk-WL6TPSYP.mjs";
import {
  PeoplePerftoolFeedbackAri
} from "./chunk-PUO3H6R4.mjs";
import "./chunk-BUB6ILHZ.mjs";
import {
  PeoplePerftoolPerformanceAri
} from "./chunk-TD6TVNAJ.mjs";
import "./chunk-BBPB2HIH.mjs";
import "./chunk-2C5PNNLA.mjs";
import {
  PipedrivePipedriveRemoteLinkAri
} from "./chunk-GWQ3Z4RB.mjs";
import "./chunk-ZD2TOTMH.mjs";
import {
  PeoplePerftoolCalibrationAri
} from "./chunk-QDER5XTV.mjs";
import "./chunk-YFDO5OIQ.mjs";
import "./chunk-CJCAMJMJ.mjs";
import {
  PeopleDataPositionAri
} from "./chunk-HFHEECKF.mjs";
import "./chunk-GZISAL55.mjs";
import {
  PeopleDataSupervisoryOrgAri
} from "./chunk-YS3KPA5A.mjs";
import "./chunk-PA2MCDK3.mjs";
import {
  PeopleDataWorkerAri
} from "./chunk-R36DM4XQ.mjs";
import "./chunk-2UITM4M3.mjs";
import {
  PeopleDataCostCenterAri
} from "./chunk-TH4QDPAT.mjs";
import "./chunk-KBWC62ZT.mjs";
import {
  PeopleDataJobProfileAri
} from "./chunk-4S3K2FML.mjs";
import "./chunk-MYD6A7PA.mjs";
import "./chunk-M7PX3O7B.mjs";
import {
  PassionfruitWorkspaceAri
} from "./chunk-T24NS6FQ.mjs";
import "./chunk-NALH4DTW.mjs";
import {
  PassionfruitDependencyAri
} from "./chunk-7CHDOIRM.mjs";
import "./chunk-UFPZX76B.mjs";
import {
  PassionfruitRoleAri
} from "./chunk-VTHMKZL4.mjs";
import "./chunk-YYESCI2M.mjs";
import {
  PassionfruitSiteAri
} from "./chunk-HGTZQJ3H.mjs";
import "./chunk-QLHTJETH.mjs";
import "./chunk-OG4WH6OP.mjs";
import "./chunk-Q2W2OS63.mjs";
import {
  PassionfruitAskAri
} from "./chunk-5SHUUCFN.mjs";
import "./chunk-ZODWYKCM.mjs";
import "./chunk-7WK2J226.mjs";
import "./chunk-UK4TPXGH.mjs";
import "./chunk-HC6B6MEQ.mjs";
import "./chunk-CYFQPPII.mjs";
import "./chunk-EXUTBA3J.mjs";
import "./chunk-HI6PTWWS.mjs";
import {
  PassionfruitAskActivityAri
} from "./chunk-2KRPWS2R.mjs";
import "./chunk-J7F3OJRT.mjs";
import {
  PassionfruitAskCommentAri
} from "./chunk-QXNWY4O5.mjs";
import "./chunk-6XWAUVHZ.mjs";
import {
  PassionfruitAskLinkAri
} from "./chunk-PDWLSMK5.mjs";
import "./chunk-3ZCZHB4U.mjs";
import {
  PassionfruitAskUpdateAri
} from "./chunk-AB5I33K3.mjs";
import "./chunk-WRANQECF.mjs";
import "./chunk-GURVNONH.mjs";
import {
  PagerdutyPagerdutyRemoteLinkAri
} from "./chunk-27O375EE.mjs";
import "./chunk-KTT4OW2H.mjs";
import "./chunk-P6ODAFPN.mjs";
import {
  PapiApiAri
} from "./chunk-FL7MNI2Q.mjs";
import "./chunk-5IDHPJZF.mjs";
import {
  PapiRoleAri
} from "./chunk-NYAVAKLN.mjs";
import "./chunk-NLJJ6GJ4.mjs";
import "./chunk-4LMKVFP7.mjs";
import {
  OrgPolicyPolicyAri
} from "./chunk-B3DD6BOI.mjs";
import "./chunk-3N5RUFAN.mjs";
import "./chunk-5YWM6DNB.mjs";
import {
  OpsgenieTimelineAri
} from "./chunk-VLM4JYJY.mjs";
import "./chunk-T4JHTCBL.mjs";
import {
  OpsgenieWorkspaceAri
} from "./chunk-SQCUVRM6.mjs";
import "./chunk-Y2SQLTAK.mjs";
import {
  OpsgenieScheduleAri
} from "./chunk-N7FD5ZRI.mjs";
import "./chunk-OFBJZXGI.mjs";
import {
  OpsgenieSiteAri
} from "./chunk-VJFH7H6D.mjs";
import "./chunk-3A7NUHDU.mjs";
import {
  OpsgenieTeamMemberAri
} from "./chunk-5Q6HR2L4.mjs";
import "./chunk-X2LQHTML.mjs";
import {
  OpsgenieTeamAri
} from "./chunk-ZFFU5EAH.mjs";
import "./chunk-5GDLDNUH.mjs";
import {
  OpsgenieNotificationAri
} from "./chunk-UK4SY7E5.mjs";
import "./chunk-WBXUDE5Y.mjs";
import {
  OpsgenieRoleAri
} from "./chunk-5MH6S5AL.mjs";
import "./chunk-7ZPWMXWE.mjs";
import {
  OpsgenieSavedSearchAri
} from "./chunk-DXTSCW3P.mjs";
import "./chunk-BNSU4GVQ.mjs";
import {
  OpsgenieScheduleRotationAri
} from "./chunk-HWJYDSA4.mjs";
import "./chunk-ILPXSBCV.mjs";
import {
  OpsgenieIncidentStatusUpdateAri
} from "./chunk-6MVDCCKH.mjs";
import "./chunk-RO4YS42G.mjs";
import {
  OpsgenieIncidentAri
} from "./chunk-T6BSHAZR.mjs";
import "./chunk-TLNXXINJ.mjs";
import {
  OpsgenieIncomingCallHistoryAri
} from "./chunk-AKYOIEAA.mjs";
import "./chunk-W4G5P22J.mjs";
import {
  OpsgenieIntegrationAri
} from "./chunk-6THG7YBU.mjs";
import "./chunk-KNRSSKUB.mjs";
import {
  OpsgenieDeploymentAri
} from "./chunk-Y2WYYEOR.mjs";
import "./chunk-BVX743QJ.mjs";
import {
  OpsgenieEscalationAri
} from "./chunk-M6EQREKY.mjs";
import "./chunk-C3Q7ZVPL.mjs";
import {
  OpsgenieEventAri
} from "./chunk-RSNZEEPJ.mjs";
import "./chunk-PW5GVXRG.mjs";
import {
  OpsgenieIncidentAlertLinkAri
} from "./chunk-N2BEN25S.mjs";
import "./chunk-JCLOAQNZ.mjs";
import {
  OpsgenieApiRequestMetricAri
} from "./chunk-2AOKKL53.mjs";
import "./chunk-R4SGT2SX.mjs";
import {
  OpsgenieCallRoutingAri
} from "./chunk-EZ4NBAAG.mjs";
import "./chunk-P2OBBGSQ.mjs";
import {
  OpsgenieCustomRoleAri
} from "./chunk-ZNQSUFLP.mjs";
import "./chunk-RW54AUFT.mjs";
import {
  OpsgenieDeploymentServiceLinkAri
} from "./chunk-GHFBKUGN.mjs";
import "./chunk-VYEUWCEP.mjs";
import {
  OpsgenieAccountLoginAri
} from "./chunk-YGOKNNIM.mjs";
import "./chunk-B2UP53JZ.mjs";
import {
  OpsgenieAccountSettingsAri
} from "./chunk-4DZQDLLN.mjs";
import "./chunk-OZDNDKVC.mjs";
import {
  OpsgenieAlertRecipientLinkAri
} from "./chunk-YFTWHSFQ.mjs";
import "./chunk-JXI5TFG4.mjs";
import {
  OpsgenieAlertAri
} from "./chunk-UGUOBER6.mjs";
import "./chunk-PZKFI7JM.mjs";
import "./chunk-BBW2QZXI.mjs";
import {
  NotionNotionDocumentAri
} from "./chunk-6EWVQ24R.mjs";
import "./chunk-HTQ3SVU2.mjs";
import {
  NotionNotionRemoteLinkAri
} from "./chunk-TSCPX54U.mjs";
import "./chunk-FUF75X6C.mjs";
import "./chunk-IXJTRB7L.mjs";
import {
  OauthClientAri
} from "./chunk-K3B2HAP3.mjs";
import "./chunk-SLRY2ZFP.mjs";
import "./chunk-VF5BTBHC.mjs";
import "./chunk-4ESSGKR2.mjs";
import {
  MondayMondayDocumentAri
} from "./chunk-6U6QESXC.mjs";
import "./chunk-VKXJXWA6.mjs";
import "./chunk-ENXJJMV5.mjs";
import {
  MuralMuralRemoteLinkAri
} from "./chunk-GQU6V3UK.mjs";
import "./chunk-JPXNYGEW.mjs";
import "./chunk-XKE76CNH.mjs";
import {
  MiroMiroDesignAri
} from "./chunk-HXNKYMT5.mjs";
import "./chunk-7BX3X4ZX.mjs";
import {
  MiroMiroRemoteLinkAri
} from "./chunk-EPLRVJSL.mjs";
import "./chunk-O3FMW3NZ.mjs";
import {
  MlRegistryArtifactAri
} from "./chunk-LFYH6LL5.mjs";
import "./chunk-JSBB4LNN.mjs";
import "./chunk-X7MVEKLR.mjs";
import "./chunk-6XRBE35V.mjs";
import "./chunk-2VMP2TVB.mjs";
import "./chunk-QKXY4FYK.mjs";
import "./chunk-TEIQ5OS2.mjs";
import "./chunk-GYBIMR5G.mjs";
import "./chunk-N2H36TJ6.mjs";
import "./chunk-V4IW6ZFW.mjs";
import {
  MigrationReportCenterReportRequestProxyAri
} from "./chunk-7SHM5EBM.mjs";
import "./chunk-G5VJCNUQ.mjs";
import {
  MicrosoftAccountAri
} from "./chunk-GP2M3LG5.mjs";
import "./chunk-YIHJDIS5.mjs";
import "./chunk-XCGYZTQK.mjs";
import "./chunk-KRNGEAQB.mjs";
import "./chunk-SPBJ7WFW.mjs";
import {
  MicrosoftSharepointDocumentAri
} from "./chunk-LMKWBLGX.mjs";
import "./chunk-HF5C63FX.mjs";
import "./chunk-4TBZMI6W.mjs";
import {
  MicrosoftTeamsMessageAri
} from "./chunk-O7SO6OWR.mjs";
import "./chunk-K3ZRWGLT.mjs";
import {
  MicrosoftTeamsRemoteLinkAri
} from "./chunk-TP7IYRNY.mjs";
import "./chunk-XYSLL7ZT.mjs";
import {
  MicrosoftTeamsConversationAri
} from "./chunk-F5CSQIKL.mjs";
import "./chunk-WU5T7JXE.mjs";
import "./chunk-TE36AGBY.mjs";
import "./chunk-2VVBTMPF.mjs";
import {
  MicrosoftOutlookCalendarCalendarEventAri
} from "./chunk-NZFQQBRM.mjs";
import "./chunk-LBWJ6T7X.mjs";
import "./chunk-KDPCRZLH.mjs";
import {
  MicrosoftOutlookMessageAri
} from "./chunk-V5HUGOZP.mjs";
import "./chunk-KOCC56FS.mjs";
import "./chunk-FH5JZU33.mjs";
import {
  MicrosoftPowerBiRemoteLinkAri
} from "./chunk-5AOSP2JD.mjs";
import "./chunk-WCVY5O3S.mjs";
import {
  MicrosoftOnedriveDocumentAri
} from "./chunk-K72IUBFH.mjs";
import "./chunk-P6AW22DS.mjs";
import "./chunk-UXR4UICM.mjs";
import {
  MercuryStrategicEventStatusAri
} from "./chunk-HUQCQ4JQ.mjs";
import "./chunk-UKGLFGAS.mjs";
import {
  MercuryStrategicEventAri
} from "./chunk-5TD7LKMY.mjs";
import "./chunk-AQRHDNVH.mjs";
import {
  MercuryViewAri
} from "./chunk-ZXJEKNUX.mjs";
import "./chunk-SW47DL6N.mjs";
import {
  MercuryWorkspaceAri
} from "./chunk-5PEI2OCP.mjs";
import "./chunk-IRPKQZGM.mjs";
import "./chunk-RTOTM3FZ.mjs";
import "./chunk-DXGTR4X7.mjs";
import "./chunk-ZVDHNU5H.mjs";
import "./chunk-YNUAOINR.mjs";
import {
  MercuryRoleAri
} from "./chunk-MPSZKE7H.mjs";
import "./chunk-2CIDR5KT.mjs";
import {
  MercurySiteAri
} from "./chunk-LA3MIML6.mjs";
import "./chunk-SEACGWCF.mjs";
import {
  MercuryCommentAri
} from "./chunk-VMDV35HO.mjs";
import "./chunk-IP7NNLNW.mjs";
import {
  MercuryFocusAreaStatusUpdateAri
} from "./chunk-H2YV5X3R.mjs";
import "./chunk-QACKR6VL.mjs";
import {
  MercuryFocusAreaTypeAri
} from "./chunk-6KZLQFI3.mjs";
import "./chunk-QPQE5MQK.mjs";
import {
  MercuryFocusAreaAri
} from "./chunk-PRFYCS27.mjs";
import "./chunk-3SEGWH74.mjs";
import "./chunk-DU3KZ7XN.mjs";
import {
  MercuryChangeProposalStatusAri
} from "./chunk-ZRSB7XHK.mjs";
import "./chunk-KLH3PA7K.mjs";
import {
  MercuryChangeProposalAri
} from "./chunk-XNAISYEC.mjs";
import "./chunk-P7WJCT2Q.mjs";
import {
  MercuryChangeAri
} from "./chunk-5XKGVM5M.mjs";
import "./chunk-ZVFPKZ5M.mjs";
import "./chunk-ZNQVUOOQ.mjs";
import {
  MeasurementEmailUuidAri
} from "./chunk-FKAT6PLF.mjs";
import "./chunk-TZWB442U.mjs";
import {
  MeasurementSiteUserAri
} from "./chunk-JTLJWWFY.mjs";
import "./chunk-CHIUNEWO.mjs";
import {
  MeasurementUserAri
} from "./chunk-WGZVKVAW.mjs";
import "./chunk-G4CB4RQJ.mjs";
import {
  MediaFileAri
} from "./chunk-BWNSQOIQ.mjs";
import "./chunk-D3IDOZFP.mjs";
import "./chunk-ITJPM4K6.mjs";
import {
  MarketingCustomerDomainAri
} from "./chunk-VTATTKS4.mjs";
import "./chunk-VAIM2B7I.mjs";
import "./chunk-JEDPRILO.mjs";
import {
  MarketplaceTeamAri
} from "./chunk-UOPQJEYN.mjs";
import "./chunk-IUUUDGF6.mjs";
import "./chunk-E4QOAXOS.mjs";
import {
  LucidLucidchartRemoteLinkAri
} from "./chunk-KMVTHTHM.mjs";
import "./chunk-SCXWFMLG.mjs";
import "./chunk-ZM3JBOHH.mjs";
import {
  MakerSpaceGroupAri
} from "./chunk-RXSAPJKL.mjs";
import "./chunk-TI7RQAMT.mjs";
import "./chunk-ZZK5E3J4.mjs";
import {
  MapiRoleAri
} from "./chunk-5RINPI6I.mjs";
import "./chunk-KZ5P3G67.mjs";
import {
  LucidLucidchartDesignAri
} from "./chunk-YTKEQOWN.mjs";
import "./chunk-JCHNYGGY.mjs";
import "./chunk-7DNBZ34Y.mjs";
import {
  LoomStandaloneWorkspaceAri
} from "./chunk-SPQY2JGE.mjs";
import "./chunk-3ZOKMD7Z.mjs";
import {
  LoomVideoAri
} from "./chunk-MWMOPTQE.mjs";
import "./chunk-HNYYKJLO.mjs";
import {
  LoomWorkspaceAri
} from "./chunk-RPIEJOVB.mjs";
import "./chunk-B775R3QQ.mjs";
import {
  LoomMeetingRecurrenceAri
} from "./chunk-PIBY7ZE3.mjs";
import "./chunk-WQ4ME76A.mjs";
import {
  LoomMeetingAri
} from "./chunk-EK3PSSRF.mjs";
import "./chunk-M5DHKYFA.mjs";
import {
  LoomSiteAri
} from "./chunk-IRNE2HPF.mjs";
import "./chunk-T5DWCQIN.mjs";
import {
  LoomSpaceAri
} from "./chunk-X2CU2FQF.mjs";
import "./chunk-PZ4WNDYC.mjs";
import "./chunk-7UUV4H76.mjs";
import {
  LoomLoomVideoAri
} from "./chunk-LWNWC24L.mjs";
import "./chunk-DEVNHM4I.mjs";
import {
  LoomCommentAri
} from "./chunk-WUYTABO6.mjs";
import "./chunk-KBPT7WSP.mjs";
import {
  LoomFolderAri
} from "./chunk-ZSXBO2NU.mjs";
import "./chunk-TEUWIQME.mjs";
import "./chunk-WFEHQJCC.mjs";
import {
  LearningPlatformLearnerAri
} from "./chunk-J2N55WRD.mjs";
import "./chunk-73GNYBI5.mjs";
import "./chunk-Y2UB4MCB.mjs";
import {
  LinkingPlatformDatasourceAri
} from "./chunk-F74XBNRS.mjs";
import "./chunk-ARLPVCER.mjs";
import "./chunk-A736ZCZO.mjs";
import {
  LoginAlerterSiteAri
} from "./chunk-OKLHDWC7.mjs";
import "./chunk-L6MUR4AX.mjs";
import "./chunk-CZTKEESF.mjs";
import "./chunk-JUH76QAU.mjs";
import {
  KnowledgeServingAndAccessTopicAri
} from "./chunk-CWBWH26T.mjs";
import "./chunk-XRVVNMAO.mjs";
import "./chunk-TZYBB37O.mjs";
import {
  LaunchdarklyLaunchdarklyRemoteLinkAri
} from "./chunk-OHI6FAZH.mjs";
import "./chunk-BVD3WDHP.mjs";
import "./chunk-Q7WEFQHK.mjs";
import {
  JsmChannelOrchestratorConversationAri
} from "./chunk-6ODUCM2G.mjs";
import "./chunk-YXNFQK7U.mjs";
import {
  KnowledgeDiscoveryBookmarkAri
} from "./chunk-OMSDF73F.mjs";
import "./chunk-CSESCAWL.mjs";
import "./chunk-VQOWFW2C.mjs";
import "./chunk-XIM2CLSJ.mjs";
import "./chunk-V5CUEKO2.mjs";
import {
  JiraWorkspaceAri
} from "./chunk-L5SA4ZVO.mjs";
import "./chunk-KOJWXOCJ.mjs";
import {
  JiraVulnerabilityHistoryAri
} from "./chunk-L7VSU6G5.mjs";
import "./chunk-DQSHJLUH.mjs";
import {
  JiraVulnerabilityAri
} from "./chunk-QZPSGUPE.mjs";
import "./chunk-2NDER42Y.mjs";
import {
  JiraWorkflowSchemeAri
} from "./chunk-DRUCFV5E.mjs";
import "./chunk-GF5WTUWI.mjs";
import {
  JiraWorkflowAri
} from "./chunk-73OSZPI2.mjs";
import "./chunk-EYJJDCXK.mjs";
import {
  JiraUserBroadcastMessageAri
} from "./chunk-MQG7NLJF.mjs";
import "./chunk-MHNZKONH.mjs";
import {
  JiraVersionApproverAri
} from "./chunk-L7EEZJ47.mjs";
import "./chunk-CLH2DEO3.mjs";
import {
  JiraVersionAri
} from "./chunk-XQ4YCJ6L.mjs";
import "./chunk-LN36NVHR.mjs";
import {
  JiraViewAri
} from "./chunk-7WZNACN2.mjs";
import "./chunk-RZI34DMQ.mjs";
import {
  JiraSecurityLevelAri
} from "./chunk-6FGPVCNO.mjs";
import "./chunk-PO2YLXFH.mjs";
import {
  JiraSecurityWorkspaceAri
} from "./chunk-TO4JTAL6.mjs";
import "./chunk-7PTBO36C.mjs";
import {
  JiraSiteAri
} from "./chunk-J7GQPOON.mjs";
import "./chunk-XIFA6EHM.mjs";
import {
  JiraSprintAri
} from "./chunk-4SC4DMKZ.mjs";
import "./chunk-FGZEBJS7.mjs";
import {
  JiraScreenSchemeAri
} from "./chunk-SY6XLJUV.mjs";
import "./chunk-BFB2NZ7A.mjs";
import {
  JiraScreenTabAri
} from "./chunk-7G2DKHXF.mjs";
import "./chunk-ES32ZAYS.mjs";
import {
  JiraScreenAri
} from "./chunk-K424B5PK.mjs";
import "./chunk-WX6GALMK.mjs";
import {
  JiraSecurityContainerAri
} from "./chunk-M7E4UE5G.mjs";
import "./chunk-4RYB4RHN.mjs";
import {
  JiraResolutionAri
} from "./chunk-TERO6HHD.mjs";
import "./chunk-BOQOC7ST.mjs";
import {
  JiraResourceUsageMetricAri
} from "./chunk-YUS43HVQ.mjs";
import "./chunk-LLLNN75K.mjs";
import {
  JiraResourceUsageRecommendationAri
} from "./chunk-3WZWQJ2D.mjs";
import "./chunk-D7TMYPCS.mjs";
import {
  JiraRoleAri
} from "./chunk-NIOHLRO4.mjs";
import "./chunk-WAQ2TE25.mjs";
import {
  JiraPullRequestAri
} from "./chunk-3PEI3SZI.mjs";
import "./chunk-5FKWU5BE.mjs";
import {
  JiraRemoteLinkHistoryAri
} from "./chunk-MG2TXJRG.mjs";
import "./chunk-QY36TOVX.mjs";
import {
  JiraRemoteLinkAri
} from "./chunk-KK7GOJV5.mjs";
import "./chunk-KHCVG4TJ.mjs";
import {
  JiraRepositoryAri
} from "./chunk-LFYOSVCM.mjs";
import "./chunk-PYOC3DJR.mjs";
import {
  JiraProjectShortcutAri
} from "./chunk-EBB7STCE.mjs";
import "./chunk-MAZNZE5K.mjs";
import {
  JiraProjectTypeAri
} from "./chunk-6RNO7EQY.mjs";
import "./chunk-OF3K3RDZ.mjs";
import {
  JiraProjectAri
} from "./chunk-2WCAGLMZ.mjs";
import "./chunk-CQAE6WHE.mjs";
import {
  JiraPullRequestHistoryAri
} from "./chunk-VR4SIBEH.mjs";
import "./chunk-IBNDOKW7.mjs";
import {
  JiraProjectFeatureAri
} from "./chunk-JKZZI6H3.mjs";
import "./chunk-QKR33HHC.mjs";
import {
  JiraProjectOverviewAri
} from "./chunk-HBZN3XU3.mjs";
import "./chunk-ORHZXN2K.mjs";
import {
  JiraProjectRoleActorAri
} from "./chunk-ZAPPI4W4.mjs";
import "./chunk-6XVY5ZJP.mjs";
import {
  JiraProjectRoleByProjectAri
} from "./chunk-J7IR7PXB.mjs";
import "./chunk-PTWREV4M.mjs";
import {
  JiraPostIncidentReviewAri
} from "./chunk-U7KPIH7A.mjs";
import "./chunk-D5S55TXO.mjs";
import {
  JiraPriorityAri
} from "./chunk-CVEMA7RM.mjs";
import "./chunk-C4M7DMM2.mjs";
import {
  JiraProductAri
} from "./chunk-26QTOOOQ.mjs";
import "./chunk-DZZ6Y4HB.mjs";
import {
  JiraProjectCategoryAri
} from "./chunk-7PVFN5RH.mjs";
import "./chunk-23L4LUZM.mjs";
import {
  JiraPlaybookStepAri
} from "./chunk-2KCXJPE5.mjs";
import "./chunk-NUL7XK3I.mjs";
import {
  JiraPlaybookAri
} from "./chunk-KI4J6ZXS.mjs";
import "./chunk-IVZ5YFMX.mjs";
import {
  JiraPostIncidentReviewHistoryAri
} from "./chunk-443KB3ZJ.mjs";
import "./chunk-XL2VWTF2.mjs";
import {
  JiraPostIncidentReviewLinkAri
} from "./chunk-HMLL4VQE.mjs";
import "./chunk-WSTI2GHH.mjs";
import {
  JiraPlaybookInstanceStepAri
} from "./chunk-EM3WVW33.mjs";
import "./chunk-JM2V4RSK.mjs";
import {
  JiraPlaybookInstanceAri
} from "./chunk-YFBJUW62.mjs";
import "./chunk-EP5LORX6.mjs";
import {
  JiraPlaybookLabelAri
} from "./chunk-L3CODQ2W.mjs";
import "./chunk-TCZPEQFQ.mjs";
import {
  JiraPlaybookStepRunAri
} from "./chunk-3RZZ3UVL.mjs";
import "./chunk-CPRWGSIC.mjs";
import {
  JiraPlanScenarioIssueAri
} from "./chunk-FHW6UJKS.mjs";
import "./chunk-6345PPDO.mjs";
import {
  JiraPlanScenarioVersionAri
} from "./chunk-DPRCNJ6H.mjs";
import "./chunk-PHVQMJL3.mjs";
import {
  JiraPlanScenarioAri
} from "./chunk-JW4OJFQY.mjs";
import "./chunk-3RMHOHQK.mjs";
import {
  JiraPlanAri
} from "./chunk-MUWNYHMM.mjs";
import "./chunk-GGTGTA2K.mjs";
import {
  JiraPermissionSchemeAri
} from "./chunk-ZQ4QQUDM.mjs";
import "./chunk-SASLBDN4.mjs";
import {
  JiraPlanCrossProjectScenarioVersionAri
} from "./chunk-MRUSXKA6.mjs";
import "./chunk-JU34D5ZO.mjs";
import {
  JiraPlanCrossProjectVersionAri
} from "./chunk-S5XFSXSI.mjs";
import "./chunk-KZ7YADXR.mjs";
import {
  JiraPlanScenarioIssueFieldAri
} from "./chunk-LKFSFKNI.mjs";
import "./chunk-CXJUGHGI.mjs";
import {
  JiraNotificationTypeSchemeAri
} from "./chunk-O6K4FFRH.mjs";
import "./chunk-45SACCRI.mjs";
import {
  JiraNotificationTypeAri
} from "./chunk-2ZHX6IZP.mjs";
import "./chunk-QXA66ZDW.mjs";
import {
  JiraNotificationUserPreferenceAri
} from "./chunk-G7I4CENY.mjs";
import "./chunk-RYGO66NC.mjs";
import {
  JiraOperationsWorkspaceAri
} from "./chunk-VYMAMFTU.mjs";
import "./chunk-QTWVEF4M.mjs";
import {
  JiraNavigationConfigAri
} from "./chunk-LSTXKPUX.mjs";
import "./chunk-C6ASIFDV.mjs";
import {
  JiraNavigationItemAri
} from "./chunk-MPKKD4FA.mjs";
import "./chunk-GXFJTOBO.mjs";
import {
  JiraNavigationMenuAri
} from "./chunk-R7YDMG56.mjs";
import "./chunk-N67HXPHT.mjs";
import {
  JiraNavigationSettingsAri
} from "./chunk-FZUID6OL.mjs";
import "./chunk-DQIMIQK4.mjs";
import {
  JiraIssueTypeAri
} from "./chunk-FYC22F3A.mjs";
import "./chunk-YBV4ZT7H.mjs";
import {
  JiraIssueWorklogAri
} from "./chunk-I5A6O7Q2.mjs";
import "./chunk-O6FNFDKY.mjs";
import {
  JiraIssueAri
} from "./chunk-GQYIJ2PS.mjs";
import "./chunk-6MDCCCXR.mjs";
import {
  JiraIssuefieldvalueAri
} from "./chunk-F7LYFIIK.mjs";
import "./chunk-ZNA5VEMQ.mjs";
import {
  JiraIssueSecurityLevelPermissionAri
} from "./chunk-YOBRFSGN.mjs";
import "./chunk-RHQOSNHG.mjs";
import {
  JiraIssueStatusAri
} from "./chunk-N5C6KS7S.mjs";
import "./chunk-S3WTNWJL.mjs";
import {
  JiraIssueTypeSchemeMappingAri
} from "./chunk-5VQE26JK.mjs";
import "./chunk-VNWAD3NX.mjs";
import {
  JiraIssueTypeSchemeAri
} from "./chunk-4TBEEFZC.mjs";
import "./chunk-QFTRTL74.mjs";
import {
  JiraIssueLinkAri
} from "./chunk-XKZ3WXIJ.mjs";
import "./chunk-QRQQNGKX.mjs";
import {
  JiraIssueRemoteLinkAri
} from "./chunk-CNLP5GG6.mjs";
import "./chunk-QO2BQQRK.mjs";
import {
  JiraIssueSearchViewTypeAri
} from "./chunk-MFK2774G.mjs";
import "./chunk-XE6YVVYA.mjs";
import {
  JiraIssueSearchViewAri
} from "./chunk-YW7V3XXU.mjs";
import "./chunk-4BHYFYKX.mjs";
import {
  JiraIssueFieldMetadataAri
} from "./chunk-YABZAYBP.mjs";
import "./chunk-Q3PRSOZC.mjs";
import {
  JiraIssueFieldOptionAri
} from "./chunk-CFBHOIAU.mjs";
import "./chunk-TKZ6M5JE.mjs";
import {
  JiraIssueHistoryAri
} from "./chunk-5LJG6DKY.mjs";
import "./chunk-N64LAJN2.mjs";
import {
  JiraIssueLinkTypeAri
} from "./chunk-ATPXGPMT.mjs";
import "./chunk-ZJWWJNVZ.mjs";
import {
  JiraIncidentAri
} from "./chunk-XYAOQYWZ.mjs";
import "./chunk-5F3IR2Z7.mjs";
import {
  JiraIssueAttachmentAri
} from "./chunk-QG6DL4XH.mjs";
import "./chunk-AU74CHJB.mjs";
import {
  JiraIssueCommentAri
} from "./chunk-ETLC3EVD.mjs";
import "./chunk-C7LAHHR5.mjs";
import {
  JiraFilterEmailSubscriptionAri
} from "./chunk-7RXEZZ4F.mjs";
import "./chunk-Z6KRUHKY.mjs";
import {
  JiraFilterAri
} from "./chunk-NRG4XEXW.mjs";
import "./chunk-NQBZEVCS.mjs";
import {
  JiraFormAri
} from "./chunk-635QCDHJ.mjs";
import "./chunk-UWOGO6UN.mjs";
import {
  JiraIncidentHistoryAri
} from "./chunk-ELOXESCP.mjs";
import "./chunk-7AJWNJ45.mjs";
import {
  JiraFeatureFlagAri
} from "./chunk-XN6Z7PMA.mjs";
import "./chunk-KSD4G6QQ.mjs";
import {
  JiraFieldConfigurationContextAri
} from "./chunk-5YMW2YB6.mjs";
import "./chunk-OOQFCUQM.mjs";
import {
  JiraFieldConfigurationAri
} from "./chunk-TJYF3X53.mjs";
import "./chunk-X3X4UHNO.mjs";
import {
  JiraFieldSetViewAri
} from "./chunk-V6DFM3VR.mjs";
import "./chunk-2NHJYARG.mjs";
import {
  JiraEntitlementAri
} from "./chunk-FLNS5GZU.mjs";
import "./chunk-N4WHXNB4.mjs";
import {
  JiraEntityPropertyAri
} from "./chunk-VQUUOKNF.mjs";
import "./chunk-WVAVMODN.mjs";
import {
  JiraFavouriteAri
} from "./chunk-OYR7YTQ5.mjs";
import "./chunk-ML5L5N3V.mjs";
import {
  JiraFeatureFlagHistoryAri
} from "./chunk-5QI5VSGI.mjs";
import "./chunk-BLEHWG53.mjs";
import {
  JiraDesignAri
} from "./chunk-QJ7PH2LB.mjs";
import "./chunk-DSHHZHIG.mjs";
import {
  JiraDevopsComponentAri
} from "./chunk-C4OEOED6.mjs";
import "./chunk-7UM4GJSV.mjs";
import {
  JiraDocumentHistoryAri
} from "./chunk-BXS42PQX.mjs";
import "./chunk-Z2JLOUYO.mjs";
import {
  JiraDocumentAri
} from "./chunk-4Y3HCADB.mjs";
import "./chunk-LUBY6O72.mjs";
import {
  JiraDashboardAri
} from "./chunk-UHIP3ZE3.mjs";
import "./chunk-KZLF4LHF.mjs";
import {
  JiraDeploymentHistoryAri
} from "./chunk-HMCQNPPW.mjs";
import "./chunk-UD7ZCFKR.mjs";
import {
  JiraDeploymentAri
} from "./chunk-K4NR5Y6N.mjs";
import "./chunk-ACU6GJRH.mjs";
import {
  JiraDesignHistoryAri
} from "./chunk-AQWSUK4O.mjs";
import "./chunk-UG2AH2TA.mjs";
import "./chunk-SXQT2XIE.mjs";
import "./chunk-UWONVETB.mjs";
import {
  JiraCommitAri
} from "./chunk-6HSCFWH6.mjs";
import "./chunk-YBNRWWV7.mjs";
import {
  JiraComponentAri
} from "./chunk-5IYKPKI7.mjs";
import "./chunk-DUSRJMGF.mjs";
import {
  JiraConnectAppAri
} from "./chunk-TGHIVOEE.mjs";
import "./chunk-VRZRWJMV.mjs";
import {
  JiraBranchAri
} from "./chunk-AGNCBHAW.mjs";
import "./chunk-65K6V5WX.mjs";
import {
  JiraBuildHistoryAri
} from "./chunk-I3UNEMJW.mjs";
import "./chunk-55Z3USGX.mjs";
import {
  JiraBuildAri
} from "./chunk-YN7BQCPX.mjs";
import "./chunk-I56NWCQF.mjs";
import {
  JiraBulkOperationTaskAri
} from "./chunk-5BBODS3G.mjs";
import "./chunk-Z76VAYJY.mjs";
import {
  JiraAnnouncementBannerAri
} from "./chunk-JZYVIETV.mjs";
import "./chunk-VFZPJ7ZD.mjs";
import {
  JiraAvailableFieldsAri
} from "./chunk-JPXPWRIS.mjs";
import "./chunk-ICE3OR6G.mjs";
import {
  JiraBranchHistoryAri
} from "./chunk-N52GXAHP.mjs";
import "./chunk-LZMQ2B5J.mjs";
import "./chunk-FIRLQRLE.mjs";
import {
  JiraSoftwareViewConfigAri
} from "./chunk-GPNBL27J.mjs";
import "./chunk-ZOFOPGOV.mjs";
import {
  JiraSoftwareSiteAri
} from "./chunk-DERC43KG.mjs";
import "./chunk-CKKWVELS.mjs";
import "./chunk-33UWD6HY.mjs";
import "./chunk-F7CN45E2.mjs";
import {
  JiraSoftwareSwimlaneAri
} from "./chunk-P3UFMLJK.mjs";
import "./chunk-IATVHO4Z.mjs";
import {
  JiraSoftwareUserBoardConfigAri
} from "./chunk-34EMNKXQ.mjs";
import "./chunk-3UUDXEXS.mjs";
import {
  JiraSoftwareCardAri
} from "./chunk-RDK5VMVF.mjs";
import "./chunk-6WK3XU6H.mjs";
import {
  JiraSoftwareColumnAri
} from "./chunk-6UGKJPQR.mjs";
import "./chunk-P7LHCIIY.mjs";
import {
  JiraSoftwareCustomFilterAri
} from "./chunk-GDUCPOJW.mjs";
import "./chunk-O6OVDRBO.mjs";
import {
  JiraSoftwareRoleAri
} from "./chunk-6I4RZYWH.mjs";
import "./chunk-SKSGTSPT.mjs";
import {
  JiraSoftwareBoardAri
} from "./chunk-IFSCFF3Q.mjs";
import "./chunk-YJAJS2GF.mjs";
import {
  JiraSoftwareCardColorAri
} from "./chunk-3VDJW3HW.mjs";
import "./chunk-OWEX3L44.mjs";
import {
  JiraSoftwareCardLayoutAri
} from "./chunk-57D7ZHUI.mjs";
import "./chunk-QSH6F5SY.mjs";
import {
  JiraSoftwareCardParentAri
} from "./chunk-OWLEJ5GC.mjs";
import "./chunk-M7R3FF5D.mjs";
import {
  JiraSoftwareBoardFeatureAri
} from "./chunk-B3KX4B7D.mjs";
import "./chunk-E24RP2LJ.mjs";
import {
  JiraSoftwareBoardIssueListAri
} from "./chunk-FV2LAIKW.mjs";
import "./chunk-3Z6ZDJ2A.mjs";
import "./chunk-DGRCF6FW.mjs";
import {
  JiraServicedeskSiteAri
} from "./chunk-LIJ45RCT.mjs";
import "./chunk-F4LW6XYO.mjs";
import {
  JiraServicedeskSlaAri
} from "./chunk-7DG3AORW.mjs";
import "./chunk-BMLW67PZ.mjs";
import {
  JiraServicedeskOrganizationAri
} from "./chunk-3P3433IA.mjs";
import "./chunk-FYLFP7EF.mjs";
import {
  JiraServicedeskQueueAri
} from "./chunk-2O37KGEX.mjs";
import "./chunk-TC6T4VFX.mjs";
import {
  JiraServicedeskRequestTypeAri
} from "./chunk-JMFBW6UU.mjs";
import "./chunk-5SWOR5KD.mjs";
import {
  JiraServicedeskRoleAri
} from "./chunk-IHIWDNNL.mjs";
import "./chunk-VV22WWWI.mjs";
import {
  JiraServicedeskCalendarEventAri
} from "./chunk-FUU6KJZP.mjs";
import "./chunk-SMZUBZEM.mjs";
import {
  JiraServicedeskCannedResponseAri
} from "./chunk-7QJGAFHM.mjs";
import "./chunk-XGGNYA2D.mjs";
import {
  JiraServicedeskOrganizationMemberAri
} from "./chunk-JM5HPVGZ.mjs";
import "./chunk-UUERIETA.mjs";
import "./chunk-LOGAEKNG.mjs";
import {
  JiraDcJiraDcProjectAri
} from "./chunk-AXRF624T.mjs";
import "./chunk-VTX7JY4Y.mjs";
import {
  JiraDcJiraDcSpaceAri
} from "./chunk-PZKWIUJJ.mjs";
import "./chunk-VNXIDWT5.mjs";
import {
  JiraDcJiraDcWorkItemAri
} from "./chunk-AE66OUK7.mjs";
import "./chunk-F3U77W3D.mjs";
import {
  JiraServicedeskApprovalAri
} from "./chunk-OKIKOAIX.mjs";
import "./chunk-AZJHCKNE.mjs";
import {
  JiraDcJiraDcCommentAri
} from "./chunk-NWP2T34O.mjs";
import "./chunk-QECR4HD2.mjs";
import "./chunk-7DTHPART.mjs";
import {
  JiraDcTertiaryJiraDcProjectAri
} from "./chunk-GGIQW4Z4.mjs";
import "./chunk-EBM6ALFK.mjs";
import {
  JiraDcTertiaryJiraDcSpaceAri
} from "./chunk-6SGR4JWH.mjs";
import "./chunk-FDSHXPYJ.mjs";
import {
  JiraDcTertiaryJiraDcWorkItemAri
} from "./chunk-YHIXSUC3.mjs";
import "./chunk-TV27GVTO.mjs";
import {
  JiraDcTertiaryJiraDcCommentAri
} from "./chunk-A35RPAYV.mjs";
import "./chunk-TTDPBFH7.mjs";
import "./chunk-YZ6V66ML.mjs";
import {
  JiraDcSecondaryJiraDcSpaceAri
} from "./chunk-QVEWE3BL.mjs";
import "./chunk-5Q5UL6WL.mjs";
import {
  JiraDcSecondaryJiraDcWorkItemAri
} from "./chunk-NTIQPGKA.mjs";
import "./chunk-CS6RQC3V.mjs";
import {
  JiraDcSecondaryJiraDcCommentAri
} from "./chunk-FDNQJ433.mjs";
import "./chunk-UDET3NPV.mjs";
import {
  JiraDcSecondaryJiraDcProjectAri
} from "./chunk-Q623MFIX.mjs";
import "./chunk-T2BO3UPO.mjs";
import "./chunk-QAQMSQTV.mjs";
import {
  JiraCoreRoleAri
} from "./chunk-GBUPZSZJ.mjs";
import "./chunk-22MNSNE6.mjs";
import {
  JiraCoreSiteAri
} from "./chunk-WKZEUZYP.mjs";
import "./chunk-BYLK2PHJ.mjs";
import "./chunk-A2IOGQNT.mjs";
import {
  JiraCustomerServiceSiteAri
} from "./chunk-HSYQ5OEW.mjs";
import "./chunk-BVLK4MVL.mjs";
import {
  JiraCustomerServiceTemplateFormAri
} from "./chunk-4HAOTHQ3.mjs";
import "./chunk-WCOH2IPR.mjs";
import {
  JiraCustomerServiceRoleAri
} from "./chunk-3LNAA5VF.mjs";
import "./chunk-HDG7BPFA.mjs";
import "./chunk-OFO4T3NF.mjs";
import {
  JiraAlignInstanceAri
} from "./chunk-NVFYVG3L.mjs";
import "./chunk-KIGWQUKL.mjs";
import {
  JiraAlignProjectAri
} from "./chunk-WVQYQ6SP.mjs";
import "./chunk-FTRG34JG.mjs";
import {
  JiraAlignWorkspaceAri
} from "./chunk-IFNDEODP.mjs";
import "./chunk-KSSMXCNA.mjs";
import "./chunk-H5XFSMZL.mjs";
import {
  JiraAiCrocsSentenceAri
} from "./chunk-ITK3ESFK.mjs";
import "./chunk-ZNGWA2WY.mjs";
import {
  JiraAiCrocsWorkspaceAri
} from "./chunk-GSQYTWNQ.mjs";
import "./chunk-KYL4TWI6.mjs";
import "./chunk-26ELLI32.mjs";
import "./chunk-5DRIX3CB.mjs";
import "./chunk-XCXN4KLN.mjs";
import {
  IdentityThirdPartyUserAri
} from "./chunk-ARNS4CQX.mjs";
import "./chunk-4VINCT53.mjs";
import {
  IdentityUserAri
} from "./chunk-NU64SPLY.mjs";
import "./chunk-N5QB45QP.mjs";
import {
  IdentityUserGrantAri
} from "./chunk-NCC44JBR.mjs";
import "./chunk-7Y76725A.mjs";
import {
  IdentityUserbaseAri
} from "./chunk-7DUTCE3K.mjs";
import "./chunk-FKBB466V.mjs";
import {
  IdentityTeamMemberAri
} from "./chunk-VM3B4TVZ.mjs";
import "./chunk-OULHQOIR.mjs";
import {
  IdentityTeamAri
} from "./chunk-3VHM7YXV.mjs";
import "./chunk-T76PG5FI.mjs";
import {
  IdentityThirdPartyDirectoryAri
} from "./chunk-2JMDW7Z5.mjs";
import "./chunk-AJVT7KP5.mjs";
import {
  IdentityThirdPartyGroupAri
} from "./chunk-E3RM7YR2.mjs";
import "./chunk-IMAEBUUM.mjs";
import "./chunk-RI3MMFQT.mjs";
import "./chunk-UTUDGZLV.mjs";
import {
  IdentityRoleAri
} from "./chunk-ABMX4BON.mjs";
import "./chunk-UFIIDAET.mjs";
import {
  IdentityScopedGroupAri
} from "./chunk-6IDTX4T2.mjs";
import "./chunk-CJOJ5VQB.mjs";
import "./chunk-YH7646PT.mjs";
import "./chunk-CXXH2SWL.mjs";
import {
  IdentityDirectoryAri
} from "./chunk-FG2IOQK5.mjs";
import "./chunk-L36L7XSX.mjs";
import {
  IdentityDomainAri
} from "./chunk-WMAEYXXE.mjs";
import "./chunk-S5UAML7R.mjs";
import {
  IdentityForgeInstallationAri
} from "./chunk-Z6KRJVMQ.mjs";
import "./chunk-IV2D7WYN.mjs";
import {
  IdentityGroupAri
} from "./chunk-HTFB6UYN.mjs";
import "./chunk-IZPJNHZU.mjs";
import {
  IdentityAppAri
} from "./chunk-FLAGDVNA.mjs";
import "./chunk-QULKIQO5.mjs";
import {
  IdentityAuthPolicyAri
} from "./chunk-LEDISXJ6.mjs";
import "./chunk-27IAW7JZ.mjs";
import {
  IdentityCustomerDirectoryAri
} from "./chunk-QTQ3ZGBB.mjs";
import "./chunk-KQGM5YYP.mjs";
import {
  IdentityCustomerOrganizationAri
} from "./chunk-IUGUSCGA.mjs";
import "./chunk-XIBIRHB4.mjs";
import "./chunk-SIFZVBMF.mjs";
import {
  HubspotHubspotDealAri
} from "./chunk-IQO5JJSA.mjs";
import "./chunk-EGG5RIMA.mjs";
import {
  HubspotHubspotRemoteLinkAri
} from "./chunk-FJDJ7H65.mjs";
import "./chunk-KLQOHOSF.mjs";
import {
  IdentityAppGrantAri
} from "./chunk-B2JORGIS.mjs";
import "./chunk-CRQ7RP44.mjs";
import "./chunk-QH7UMOHA.mjs";
import {
  HomeWorkspaceAri
} from "./chunk-XSEINBJG.mjs";
import "./chunk-OEOCSY6S.mjs";
import {
  HubspotHubspotCustomerOrgAri
} from "./chunk-Q5L67A6J.mjs";
import "./chunk-SGZ5NPU7.mjs";
import "./chunk-34B6DB3Z.mjs";
import {
  HelpRequestFormAri
} from "./chunk-5FNAHASY.mjs";
import "./chunk-QF6HVRWR.mjs";
import {
  HelpRoleAri
} from "./chunk-C3J23B4Y.mjs";
import "./chunk-TEJV4KBV.mjs";
import {
  HelpLayoutAri
} from "./chunk-LPUKSIWB.mjs";
import "./chunk-HXZGYRZT.mjs";
import {
  HelpPageAri
} from "./chunk-LBWIJ7ZA.mjs";
import "./chunk-CRBMKFEU.mjs";
import {
  HelpPortalAri
} from "./chunk-DVSJH4ZB.mjs";
import "./chunk-DEFOWKTU.mjs";
import {
  HelpArticleAri
} from "./chunk-X7F7MZD6.mjs";
import "./chunk-RDV7AM67.mjs";
import {
  HelpChannelAri
} from "./chunk-KIUHUSQK.mjs";
import "./chunk-H2XK65KU.mjs";
import {
  HelpHelpCenterAri
} from "./chunk-NH3TUJLE.mjs";
import "./chunk-UIZK5L5W.mjs";
import {
  HelpHelpDeskAri
} from "./chunk-OWAPJUS7.mjs";
import "./chunk-EQROJBZ3.mjs";
import "./chunk-HCYUW54X.mjs";
import {
  GraphWorkItemAri
} from "./chunk-4YWTOI4S.mjs";
import "./chunk-WDMLMNL4.mjs";
import {
  GraphWorkerHistoryAri
} from "./chunk-KNNQP5GS.mjs";
import "./chunk-C6RA7M3X.mjs";
import {
  GraphWorkerAri
} from "./chunk-MSBGFKHU.mjs";
import "./chunk-Z4DIOHRR.mjs";
import {
  GraphWorkspaceAri
} from "./chunk-H5OAOK55.mjs";
import "./chunk-SQM2NR5M.mjs";
import {
  GraphVideoAri
} from "./chunk-5TRMBFVA.mjs";
import "./chunk-PDTRZUQJ.mjs";
import {
  GraphVulnerabilityHistoryAri
} from "./chunk-XTSUK2OE.mjs";
import "./chunk-PHKBIUQV.mjs";
import {
  GraphVulnerabilityAri
} from "./chunk-HTXAOB6X.mjs";
import "./chunk-J7VD66AE.mjs";
import {
  GraphWorkItemHistoryAri
} from "./chunk-DCLK5CIC.mjs";
import "./chunk-BGX5LTB4.mjs";
import {
  GraphSpaceHistoryAri
} from "./chunk-XFOYD3JG.mjs";
import "./chunk-CM66JRIL.mjs";
import {
  GraphSpaceAri
} from "./chunk-XBDSIMRD.mjs";
import "./chunk-LSQP2ZSZ.mjs";
import {
  GraphThreadChunkAri
} from "./chunk-AIV2WMB2.mjs";
import "./chunk-JUDWJ3FD.mjs";
import {
  GraphVideoHistoryAri
} from "./chunk-ZNH36KUJ.mjs";
import "./chunk-2IQL4B6E.mjs";
import {
  GraphServiceAndOperationsTeamRelationshipAri
} from "./chunk-QDHMRYZH.mjs";
import "./chunk-E4XM6JBY.mjs";
import {
  GraphServiceAndVcsRepositoryRelationshipAri
} from "./chunk-DM4VJT35.mjs";
import "./chunk-SYSNNBBE.mjs";
import {
  GraphServiceRelationshipAri
} from "./chunk-V6I3GHKK.mjs";
import "./chunk-JNGK7AYM.mjs";
import {
  GraphServiceAri
} from "./chunk-U4QGEEJP.mjs";
import "./chunk-KKDL7RK7.mjs";
import {
  GraphRoleAri
} from "./chunk-Y4N4DAWM.mjs";
import "./chunk-RPUSNHXA.mjs";
import {
  GraphSecurityContainerAri
} from "./chunk-5F2JHVKY.mjs";
import "./chunk-6HYQV6SW.mjs";
import {
  GraphSecurityWorkspaceAri
} from "./chunk-O53XA6VV.mjs";
import "./chunk-4FPKBG6M.mjs";
import {
  GraphServiceAndJiraProjectRelationshipAri
} from "./chunk-LRDVPAQD.mjs";
import "./chunk-723KYNJS.mjs";
import {
  GraphRelationshipTypeAri
} from "./chunk-56MGWW5V.mjs";
import "./chunk-IQMENLZS.mjs";
import {
  GraphRemoteLinkHistoryAri
} from "./chunk-I6R4KGXY.mjs";
import "./chunk-2YKXM3UN.mjs";
import {
  GraphRemoteLinkAri
} from "./chunk-YEPBXFS5.mjs";
import "./chunk-7CH7D4HJ.mjs";
import {
  GraphRepositoryAri
} from "./chunk-WCXD2JKT.mjs";
import "./chunk-SAJO4IGE.mjs";
import {
  GraphProjectHistoryAri
} from "./chunk-7YLRCETH.mjs";
import "./chunk-FKSVEAW3.mjs";
import {
  GraphProjectAri
} from "./chunk-CEW2H2GY.mjs";
import "./chunk-XTUR6LTI.mjs";
import {
  GraphPullRequestHistoryAri
} from "./chunk-G2DOTXH7.mjs";
import "./chunk-K4SS2RGK.mjs";
import {
  GraphPullRequestAri
} from "./chunk-ESRIECJV.mjs";
import "./chunk-Z3FWECDS.mjs";
import {
  GraphPositionHistoryAri
} from "./chunk-3HCUELCU.mjs";
import "./chunk-SHV5KRSV.mjs";
import {
  GraphPositionAri
} from "./chunk-5X6YP4CG.mjs";
import "./chunk-N4KQKV73.mjs";
import {
  GraphPostIncidentReviewHistoryAri
} from "./chunk-EJLJR556.mjs";
import "./chunk-PVBPUKXI.mjs";
import {
  GraphPostIncidentReviewAri
} from "./chunk-WF5R6PNP.mjs";
import "./chunk-SNAPC4SA.mjs";
import {
  GraphMessageAri
} from "./chunk-3VLUHI4T.mjs";
import "./chunk-PTHUIY6N.mjs";
import {
  GraphOperationsWorkspaceAri
} from "./chunk-K2NMN5VS.mjs";
import "./chunk-5HHNCVXF.mjs";
import {
  GraphOrganisationHistoryAri
} from "./chunk-BAEDA6TY.mjs";
import "./chunk-VSZQAEDL.mjs";
import {
  GraphOrganisationAri
} from "./chunk-6RWJX35D.mjs";
import "./chunk-XNVBEJRP.mjs";
import {
  GraphJiraProjectAndDocumentationSpaceRelationshipAri
} from "./chunk-T5JGCK5X.mjs";
import "./chunk-SM4OYQAG.mjs";
import {
  GraphJiraProjectAndOperationsTeamRelationshipAri
} from "./chunk-B5XEEIKP.mjs";
import "./chunk-35GRSR7B.mjs";
import "./chunk-UXP7JLVP.mjs";
import "./chunk-IUR7MTIZ.mjs";
import {
  GraphJiraProjectAndVcsRepositoryRelationshipAri
} from "./chunk-LH6TYC2G.mjs";
import "./chunk-XYDSJQO4.mjs";
import {
  GraphIncidentHistoryAri
} from "./chunk-NA7TPO6F.mjs";
import "./chunk-NTBEFMSE.mjs";
import {
  GraphIncidentAri
} from "./chunk-RF2AN3KB.mjs";
import "./chunk-FKF6Z5DH.mjs";
import {
  GraphIntegrationContextAri
} from "./chunk-P5BYJVW7.mjs";
import "./chunk-CAJWUMVE.mjs";
import {
  GraphJiraProjectAndDevopsToolRelationshipAri
} from "./chunk-BJ73R3FD.mjs";
import "./chunk-2A255S76.mjs";
import {
  GraphEmployeeHistoryAri
} from "./chunk-76YCDCPO.mjs";
import "./chunk-EDEP7UFY.mjs";
import {
  GraphEmployeeAri
} from "./chunk-USDQMZXZ.mjs";
import "./chunk-3SMMF4WR.mjs";
import {
  GraphFeatureFlagHistoryAri
} from "./chunk-EVEXSBJ5.mjs";
import "./chunk-DFRD6YBM.mjs";
import {
  GraphFeatureFlagAri
} from "./chunk-FVHR4QXI.mjs";
import "./chunk-LDUD7O2G.mjs";
import {
  GraphDesignAri
} from "./chunk-CNGEAQG4.mjs";
import "./chunk-RBS5RYKM.mjs";
import {
  GraphDevopsComponentAri
} from "./chunk-HMFX475U.mjs";
import "./chunk-HLKEKPMY.mjs";
import {
  GraphDocumentHistoryAri
} from "./chunk-ENIDATBX.mjs";
import "./chunk-SP4WD2VN.mjs";
import {
  GraphDocumentAri
} from "./chunk-F45RAG4V.mjs";
import "./chunk-J4TEUSCJ.mjs";
import {
  GraphDealAri
} from "./chunk-4LHSTBSK.mjs";
import "./chunk-LN7ARE4M.mjs";
import {
  GraphDeploymentHistoryAri
} from "./chunk-DZCJUTMV.mjs";
import "./chunk-LCOEZ74N.mjs";
import {
  GraphDeploymentAri
} from "./chunk-EKLRRP2Z.mjs";
import "./chunk-OON2ZRPC.mjs";
import {
  GraphDesignHistoryAri
} from "./chunk-4TXCHB67.mjs";
import "./chunk-MTSTC2DF.mjs";
import {
  GraphConversationAri
} from "./chunk-HRIEIUXC.mjs";
import "./chunk-CW3TWCW6.mjs";
import {
  GraphCustomerOrgHistoryAri
} from "./chunk-2K3BMXST.mjs";
import "./chunk-JOMPZANB.mjs";
import {
  GraphCustomerOrgAri
} from "./chunk-EUYWAJZH.mjs";
import "./chunk-UMWZ4TML.mjs";
import {
  GraphDealHistoryAri
} from "./chunk-7M5DAIZO.mjs";
import "./chunk-HDZBCN3T.mjs";
import {
  GraphCommitAri
} from "./chunk-2MWCWDDM.mjs";
import "./chunk-LJ7XURCE.mjs";
import {
  GraphContentAri
} from "./chunk-JRBUNJCS.mjs";
import "./chunk-PBJGJF43.mjs";
import {
  GraphConversationChunkAri
} from "./chunk-YNUAAVYJ.mjs";
import "./chunk-ZTNOGHIC.mjs";
import {
  GraphConversationHistoryAri
} from "./chunk-6AZJYOK3.mjs";
import "./chunk-XCRGDS3T.mjs";
import {
  GraphCalendarEventHistoryAri
} from "./chunk-LH2T5SM6.mjs";
import "./chunk-62SYOXYN.mjs";
import {
  GraphCalendarEventAri
} from "./chunk-CEZUFXIC.mjs";
import "./chunk-GLGAYLO3.mjs";
import {
  GraphCommentHistoryAri
} from "./chunk-UERM7XJO.mjs";
import "./chunk-QMWB3HAP.mjs";
import {
  GraphCommentAri
} from "./chunk-DGZ5AHRM.mjs";
import "./chunk-44PPANH6.mjs";
import {
  GraphBranchHistoryAri
} from "./chunk-TZDFTH22.mjs";
import "./chunk-RL6RLD25.mjs";
import {
  GraphBranchAri
} from "./chunk-L2Q4I2ZD.mjs";
import "./chunk-JJHDDJHM.mjs";
import {
  GraphBuildHistoryAri
} from "./chunk-6PUSL3KY.mjs";
import "./chunk-Q4A3TDFH.mjs";
import {
  GraphBuildAri
} from "./chunk-WKCHFDID.mjs";
import "./chunk-5YTWJ4WK.mjs";
import "./chunk-CBPWBOVC.mjs";
import "./chunk-VZ6DLXWI.mjs";
import "./chunk-X63IP3OI.mjs";
import "./chunk-ZO5HGX2X.mjs";
import "./chunk-5B6S6SWP.mjs";
import "./chunk-TY7CMWCM.mjs";
import "./chunk-466Z3YAU.mjs";
import "./chunk-K2E3OM6F.mjs";
import {
  GoogleGoogleDriveRemoteLinkAri
} from "./chunk-CGUFN27R.mjs";
import "./chunk-ZI2F7BAC.mjs";
import {
  GoogleAccountAri
} from "./chunk-7UMQCQVN.mjs";
import "./chunk-PYVPLXET.mjs";
import "./chunk-QOHF4ZZX.mjs";
import "./chunk-QEA64F52.mjs";
import "./chunk-2AYB4P5U.mjs";
import "./chunk-5VDZQSQU.mjs";
import "./chunk-MHJ7QAFD.mjs";
import {
  GoogleGoogleCalendarCalendarEventAri
} from "./chunk-VBOFQAP6.mjs";
import "./chunk-DUGKATMK.mjs";
import "./chunk-AFKE7ZYW.mjs";
import {
  GoogleGoogleDriveRslDocumentAri
} from "./chunk-2XKVNDU7.mjs";
import "./chunk-JXXXZX76.mjs";
import {
  GoogleGoogleDriveDocumentAri
} from "./chunk-3F6QAPE6.mjs";
import "./chunk-7MCC3MWT.mjs";
import "./chunk-VW7DXTPT.mjs";
import {
  GoogleGmailMessageAri
} from "./chunk-KXWVJBXF.mjs";
import "./chunk-MPZXQ2VZ.mjs";
import "./chunk-3RAAEQVH.mjs";
import {
  GoalSiteAri
} from "./chunk-DEDX4M3Y.mjs";
import "./chunk-QJAWPXE7.mjs";
import {
  GoalWorkspaceAri
} from "./chunk-ILJ5EQHZ.mjs";
import "./chunk-F4QTF457.mjs";
import {
  GoalGoalTypeAri
} from "./chunk-IM5LEZU5.mjs";
import "./chunk-RE4MHVIY.mjs";
import {
  GoalMetricUpdateAri
} from "./chunk-BYO6NU4L.mjs";
import "./chunk-CIDEQG7W.mjs";
import "./chunk-FEGYJAKO.mjs";
import {
  GitlabGitlabRemoteLinkAri
} from "./chunk-ICX2OBCK.mjs";
import "./chunk-7F6GATTB.mjs";
import {
  GitlabGitlabRepositoryAri
} from "./chunk-H3TJDIZ6.mjs";
import "./chunk-IA4I5ARC.mjs";
import "./chunk-H5EU3TS2.mjs";
import {
  GithubGithubRemoteLinkAri
} from "./chunk-N73TY7RZ.mjs";
import "./chunk-3HZTIMQY.mjs";
import {
  GithubGithubRepositoryAri
} from "./chunk-H55VOUEL.mjs";
import "./chunk-PLNY4YWF.mjs";
import {
  GithubGithubVulnerabilityAri
} from "./chunk-E7UK24KR.mjs";
import "./chunk-VMFL6DVG.mjs";
import {
  GithubGithubCommentAri
} from "./chunk-SXFYLCCM.mjs";
import "./chunk-LOCQJBS5.mjs";
import {
  GithubGithubCommitAri
} from "./chunk-MRFTVGA2.mjs";
import "./chunk-HONW3GHX.mjs";
import {
  GithubGithubDeploymentAri
} from "./chunk-QLHWTU4L.mjs";
import "./chunk-5OXTM6ZB.mjs";
import {
  GithubGithubPullRequestAri
} from "./chunk-WWU7R335.mjs";
import "./chunk-5L7TTH2V.mjs";
import {
  GithubGithubBranchAri
} from "./chunk-VK2WUHQD.mjs";
import "./chunk-4XWMIRSM.mjs";
import {
  GithubGithubBuildAri
} from "./chunk-3RDUW2UI.mjs";
import "./chunk-SW7EO4HF.mjs";
import "./chunk-ZMVXG3Z6.mjs";
import {
  ExusExternalUserAri
} from "./chunk-2RSYTJ3M.mjs";
import "./chunk-IWWNXWBX.mjs";
import "./chunk-ZRO65EBB.mjs";
import {
  FigmaFigmaDesignAri
} from "./chunk-2YAMYMXC.mjs";
import "./chunk-T22TGH5G.mjs";
import {
  FigmaFigmaRemoteLinkAri
} from "./chunk-NCEASUL6.mjs";
import "./chunk-2K3H62WY.mjs";
import "./chunk-J6TNPU7E.mjs";
import {
  ElementsCustomEmojiMetadataAri
} from "./chunk-BLLVLRXI.mjs";
import "./chunk-INZ6GYXP.mjs";
import {
  ElementsWorkspaceAri
} from "./chunk-X5KXOLP4.mjs";
import "./chunk-5TZRCHUN.mjs";
import "./chunk-GA6SOHIM.mjs";
import {
  ElevatePrecalibrationPillarAri
} from "./chunk-NXJK634D.mjs";
import "./chunk-7ORFYDUC.mjs";
import {
  ElevatePrecalibrationAri
} from "./chunk-7RCPRLUL.mjs";
import "./chunk-M55RGK4K.mjs";
import {
  ElevateSelfAssessmentPillarAri
} from "./chunk-D66VVGWA.mjs";
import "./chunk-NU3GZ3PV.mjs";
import {
  ElevateSelfAssessmentAri
} from "./chunk-4QXULHSS.mjs";
import "./chunk-SOMF35WW.mjs";
import "./chunk-PQL4RGB3.mjs";
import {
  EcosystemInstallationAri
} from "./chunk-6UBEX26E.mjs";
import "./chunk-NRRB37M2.mjs";
import {
  EcosystemExtensionGroupAri
} from "./chunk-A2H5RCZT.mjs";
import "./chunk-NYRDQJIQ.mjs";
import {
  EcosystemExtensionAri
} from "./chunk-EMCBJB4A.mjs";
import "./chunk-Q6OH5L4G.mjs";
import {
  EcosystemInstallationLinkAri
} from "./chunk-PM3VZQ26.mjs";
import "./chunk-C4DHTYLM.mjs";
import {
  EcosystemAppAri
} from "./chunk-V3SIM7JF.mjs";
import "./chunk-ZSCWAZK4.mjs";
import {
  EcosystemConnectAppAri
} from "./chunk-7BKURJGK.mjs";
import "./chunk-GEYJT4UH.mjs";
import {
  EcosystemConnectModuleAri
} from "./chunk-ESXRJUNW.mjs";
import "./chunk-4UCHZIA4.mjs";
import {
  EcosystemEnvironmentAri
} from "./chunk-DDY4M37M.mjs";
import "./chunk-UJZYZKMO.mjs";
import "./chunk-EGOBPWQ6.mjs";
import {
  DropboxDocumentAri
} from "./chunk-6MS4NWPD.mjs";
import "./chunk-ZDHZYOUG.mjs";
import {
  DropboxRemoteLinkAri
} from "./chunk-3NNN5ISZ.mjs";
import "./chunk-6YI45FOX.mjs";
import {
  EcosystemAppEnvironmentVersionAri
} from "./chunk-IQ5RMXNI.mjs";
import "./chunk-CD56DLNS.mjs";
import "./chunk-WW4VXJAD.mjs";
import {
  DovetailDovetailRemoteLinkAri
} from "./chunk-RBBGLGZT.mjs";
import "./chunk-IQXXUNUN.mjs";
import "./chunk-J6NSTN6D.mjs";
import {
  DropboxDropboxRemoteLinkAri
} from "./chunk-2JFYCBY2.mjs";
import "./chunk-MMP7HB35.mjs";
import {
  DropboxDropboxDocumentAri
} from "./chunk-QY7LJYM7.mjs";
import "./chunk-BXFVNICZ.mjs";
import "./chunk-5ZNUDGAW.mjs";
import {
  DocusignDocusignRemoteLinkAri
} from "./chunk-N73WVBAG.mjs";
import "./chunk-ELGAGBXE.mjs";
import {
  DocusignDocusignDocumentAri
} from "./chunk-YDMQDUYV.mjs";
import "./chunk-T7GLL24I.mjs";
import "./chunk-PTM25UW5.mjs";
import {
  DevopsRelationshipAri
} from "./chunk-6P7KY3SF.mjs";
import "./chunk-PPXOGRS2.mjs";
import {
  DevopsToolAri
} from "./chunk-ZKYFCTB2.mjs";
import "./chunk-WECRO6DG.mjs";
import {
  DevopsContainerAri
} from "./chunk-HPWWKSLI.mjs";
import "./chunk-FGEFELP3.mjs";
import {
  DevopsNamespaceAri
} from "./chunk-UH525T6X.mjs";
import "./chunk-H53KN2GD.mjs";
import {
  DevopsProviderAri
} from "./chunk-OY4ZXZ45.mjs";
import "./chunk-OILGX3NZ.mjs";
import "./chunk-Q676EFQB.mjs";
import {
  DataLakeLakeConnectionAri
} from "./chunk-PSGQFPZF.mjs";
import "./chunk-2Y4LARQO.mjs";
import "./chunk-HZUXQ7LV.mjs";
import {
  DevaiWorkspaceAri
} from "./chunk-2TBYB7LG.mjs";
import "./chunk-SRKXL6M5.mjs";
import {
  DevaiAutodevJobAri
} from "./chunk-U7RQ3KPV.mjs";
import "./chunk-3DEYGGN2.mjs";
import "./chunk-PQ7YGB4H.mjs";
import {
  ConversationalHelpConversationAri
} from "./chunk-V6LBU7SZ.mjs";
import "./chunk-YWVV6JNH.mjs";
import "./chunk-L33DGCYG.mjs";
import {
  ConvoAiWorkspaceAri
} from "./chunk-EMSLGZKT.mjs";
import "./chunk-E3D6AV43.mjs";
import "./chunk-NCM3MHWA.mjs";
import {
  CssXpRoleAri
} from "./chunk-AJLS2RDB.mjs";
import "./chunk-IO5OI7XO.mjs";
import "./chunk-HALR7Y4A.mjs";
import {
  ConnectionRegistryLinkAri
} from "./chunk-FFZFFQ52.mjs";
import "./chunk-WZAFZVQ2.mjs";
import {
  ConversationalHelpConversationMessageAri
} from "./chunk-GRKGIQPN.mjs";
import "./chunk-W6PEKGFC.mjs";
import "./chunk-XUP7ZCCT.mjs";
import {
  ConfluenceWorkflowAri
} from "./chunk-NLGABQTE.mjs";
import "./chunk-ODPYL32A.mjs";
import {
  ConfluenceWorkspaceAri
} from "./chunk-V2WHQ4AT.mjs";
import "./chunk-2BXLBFIW.mjs";
import {
  ConfluenceTaskAri
} from "./chunk-KP4EFNRK.mjs";
import "./chunk-SHZM4TTX.mjs";
import {
  ConfluenceTeamCalendarAri
} from "./chunk-YNYG4Q7E.mjs";
import "./chunk-KKW3GIW7.mjs";
import {
  ConfluenceUserPropertyAri
} from "./chunk-BHJYK2QT.mjs";
import "./chunk-NW7N5UEI.mjs";
import {
  ConfluenceWhiteboardAri
} from "./chunk-G6JJMST7.mjs";
import "./chunk-SRDRWGU4.mjs";
import {
  ConfluencePublicLinkAri
} from "./chunk-7JYV3NUA.mjs";
import "./chunk-HXAMB5JS.mjs";
import {
  ConfluenceRoleAri
} from "./chunk-6BSTGI25.mjs";
import "./chunk-VTWZOXAB.mjs";
import {
  ConfluenceSiteAri
} from "./chunk-HNG2VXJJ.mjs";
import "./chunk-PVOLGFB2.mjs";
import {
  ConfluenceSpaceAri
} from "./chunk-QRGOFIMK.mjs";
import "./chunk-UOPZEJBG.mjs";
import {
  ConfluenceLabelAri
} from "./chunk-YL7V7CED.mjs";
import "./chunk-5ZGPRH5S.mjs";
import {
  ConfluenceLongRunningTaskAri
} from "./chunk-RO27D2PY.mjs";
import "./chunk-E6K4JCID.mjs";
import {
  ConfluenceNoteAri
} from "./chunk-R6UPLDEL.mjs";
import "./chunk-ZWIRILL2.mjs";
import {
  ConfluencePageAri
} from "./chunk-RGLT4MI5.mjs";
import "./chunk-IMF5GHHQ.mjs";
import {
  ConfluenceEmbedAri
} from "./chunk-Z2SJWYLA.mjs";
import "./chunk-N5UKNFK5.mjs";
import {
  ConfluenceEventAri
} from "./chunk-S6P2CY4X.mjs";
import "./chunk-GWVYXALY.mjs";
import {
  ConfluenceFolderAri
} from "./chunk-NORVIKN3.mjs";
import "./chunk-NKU7GI3A.mjs";
import {
  ConfluenceCommentAri
} from "./chunk-QVIWCSY4.mjs";
import "./chunk-ARFRFJQD.mjs";
import {
  ConfluenceContentAri
} from "./chunk-CFNYNOPV.mjs";
import "./chunk-LVQBHR63.mjs";
import {
  ConfluenceCustomContentAri
} from "./chunk-N2FZGV7E.mjs";
import "./chunk-FXD6ATKK.mjs";
import {
  ConfluenceDatabaseAri
} from "./chunk-WOQPCWJ2.mjs";
import "./chunk-PGMODI5S.mjs";
import {
  ConfluenceAdminAnnouncementBannerAri
} from "./chunk-BBBPHAJF.mjs";
import "./chunk-IH6QYLXI.mjs";
import {
  ConfluenceAttachmentAri
} from "./chunk-VTMNLVZW.mjs";
import "./chunk-QCPR2D6J.mjs";
import {
  ConfluenceBlogpostAri
} from "./chunk-OJ3VR7QS.mjs";
import "./chunk-J5SGYOAJ.mjs";
import "./chunk-DNBHNWWK.mjs";
import {
  ConfluenceDcConfluenceDcSpaceAri
} from "./chunk-RBPTEEQE.mjs";
import "./chunk-F4ZZPTKG.mjs";
import {
  ConfluenceDcConfluenceDcCommentAri
} from "./chunk-IRX4UDFX.mjs";
import "./chunk-WWSSF667.mjs";
import {
  ConfluenceDcConfluenceDcDocumentAri
} from "./chunk-JHER7EYD.mjs";
import "./chunk-4G32XNOY.mjs";
import "./chunk-QNHIC5LB.mjs";
import {
  CompassWebhookAri
} from "./chunk-EA5J4EAW.mjs";
import "./chunk-RX4T5I6M.mjs";
import {
  CompassWorkspaceAri
} from "./chunk-QAGG77TG.mjs";
import "./chunk-QQTPW4TZ.mjs";
import {
  CompassRoleAri
} from "./chunk-7WLCZBKA.mjs";
import "./chunk-FJ7KFVKZ.mjs";
import {
  CompassScorecardAri
} from "./chunk-KAHAP27S.mjs";
import "./chunk-2TSZWIWR.mjs";
import {
  CompassSiteAri
} from "./chunk-ZMZO6XRO.mjs";
import "./chunk-SOJW4UXZ.mjs";
import {
  CompassUserDefinedParameterAri
} from "./chunk-ACYGU27X.mjs";
import "./chunk-U7N5T7JC.mjs";
import {
  CompassLibraryScorecardAri
} from "./chunk-D4HFHJPS.mjs";
import "./chunk-TOMPOKNQ.mjs";
import {
  CompassMetricDefinitionAri
} from "./chunk-DCSKBUL7.mjs";
import "./chunk-BS6YPB3T.mjs";
import {
  CompassMetricSourceAri
} from "./chunk-P7D7B3SF.mjs";
import "./chunk-H63VXJES.mjs";
import {
  CompassPackageAri
} from "./chunk-DYRNZK5R.mjs";
import "./chunk-LBNPGADB.mjs";
import {
  CompassDocumentAri
} from "./chunk-QO7SSRBZ.mjs";
import "./chunk-TZHADK5A.mjs";
import {
  CompassDocumentationCategoryAri
} from "./chunk-SOOBWLA6.mjs";
import "./chunk-BCHZ2JY7.mjs";
import {
  CompassEventSourceAri
} from "./chunk-RSE73NDI.mjs";
import "./chunk-G6Y66A5E.mjs";
import {
  CompassIncomingWebhookAri
} from "./chunk-6SQUD2WZ.mjs";
import "./chunk-6S4G5T3V.mjs";
import {
  CompassComponentLinkAri
} from "./chunk-S3YEEHHF.mjs";
import "./chunk-RUUZK374.mjs";
import {
  CompassComponentTypeAri
} from "./chunk-MI5MJZZZ.mjs";
import "./chunk-A55S5SEK.mjs";
import {
  CompassComponentAri
} from "./chunk-W3ZQNKG2.mjs";
import "./chunk-XW2UZKUO.mjs";
import {
  CompassCustomFieldDefinitionAri
} from "./chunk-7ZKBXMHO.mjs";
import "./chunk-K3BVQB6G.mjs";
import {
  CompassApiEndpointAri
} from "./chunk-2TIIO2DP.mjs";
import "./chunk-2HMBOSQ3.mjs";
import {
  CompassAssistantAnswerAri
} from "./chunk-L7WKH6U4.mjs";
import "./chunk-VAWIKJGT.mjs";
import {
  CompassCampaignAri
} from "./chunk-3S444J2H.mjs";
import "./chunk-FVJCFF7U.mjs";
import {
  CompassComponentLabelAri
} from "./chunk-H3PHKV6W.mjs";
import "./chunk-QOMYCLGO.mjs";
import "./chunk-NWYLOPBR.mjs";
import {
  CommerceRoleAri
} from "./chunk-XYS56VYX.mjs";
import "./chunk-T3FMJP7C.mjs";
import {
  CommerceShipToPartyAri
} from "./chunk-MDYKMBBS.mjs";
import "./chunk-MR7K6E3K.mjs";
import {
  CommerceTransactionAccountAri
} from "./chunk-77XUTCTA.mjs";
import "./chunk-IG6GXX55.mjs";
import {
  CommerceWorkbenchResourceAri
} from "./chunk-5KLS37O7.mjs";
import "./chunk-CVI37ASZ.mjs";
import {
  CommercePricingPlanAri
} from "./chunk-4MRXW7L4.mjs";
import "./chunk-66TMLCVM.mjs";
import {
  CommerceProductAri
} from "./chunk-JGTBH6IP.mjs";
import "./chunk-7KRTMBHT.mjs";
import {
  CommercePromotionAri
} from "./chunk-4PN2DXS4.mjs";
import "./chunk-NDIFFXAM.mjs";
import {
  CommerceQuoteAri
} from "./chunk-E3XHWBNO.mjs";
import "./chunk-2PBMRYJM.mjs";
import {
  CommerceOfferingRelationshipTemplateAri
} from "./chunk-NTQZPCEO.mjs";
import "./chunk-2P7D7BWD.mjs";
import {
  CommerceOfferingAri
} from "./chunk-NON2JS2I.mjs";
import "./chunk-ANF2XXE7.mjs";
import {
  CommerceOrderAri
} from "./chunk-4TE24BLX.mjs";
import "./chunk-3AXUCZXV.mjs";
import {
  CommercePaymentMethodAri
} from "./chunk-NLGPZOCA.mjs";
import "./chunk-N3JGQE6H.mjs";
import {
  CommerceEntitlementAri
} from "./chunk-WLYQCRJJ.mjs";
import "./chunk-NINDB5F3.mjs";
import {
  CommerceInitiativeAri
} from "./chunk-NMUDZXWP.mjs";
import "./chunk-NEKFBJUG.mjs";
import {
  CommerceInvoiceGroupAri
} from "./chunk-FAG744FK.mjs";
import "./chunk-MX6GMSIL.mjs";
import {
  CommerceCatalogAccountAri
} from "./chunk-BXWZLPDX.mjs";
import "./chunk-3FQB6Q73.mjs";
import {
  CommerceChangeAri
} from "./chunk-OHSJRD5E.mjs";
import "./chunk-R3MD4VCL.mjs";
import {
  CommerceEntitlementTemplateAri
} from "./chunk-FI7MRGB7.mjs";
import "./chunk-32JDQQPU.mjs";
import "./chunk-7USM4R6S.mjs";
import {
  CmdbWorkspaceAri
} from "./chunk-6S65DHJX.mjs";
import "./chunk-UQ5PWXBJ.mjs";
import {
  CmdbObjectAri
} from "./chunk-BVNEIGLW.mjs";
import "./chunk-BZ6D72WR.mjs";
import {
  CmdbRoleAri
} from "./chunk-ARCUZT3F.mjs";
import "./chunk-WF26OXZU.mjs";
import {
  CmdbSchemaAri
} from "./chunk-HDCBIG77.mjs";
import "./chunk-7NF55AHC.mjs";
import {
  CmdbTypeAri
} from "./chunk-B74OI5SL.mjs";
import "./chunk-Q5AHPTNW.mjs";
import "./chunk-RIJRPWQW.mjs";
import {
  ClickupClickupRemoteLinkAri
} from "./chunk-7FIZBAHX.mjs";
import "./chunk-TZ6DZWZ5.mjs";
import {
  CmdbAttributeAri
} from "./chunk-LYMTZOWM.mjs";
import "./chunk-XAMITGNQ.mjs";
import {
  CmdbImportConfigurationAri
} from "./chunk-HQW7G3EI.mjs";
import "./chunk-W3AHDHPL.mjs";
import "./chunk-ABYQA54S.mjs";
import {
  CiaCprsEmailAri
} from "./chunk-YPBTFLYZ.mjs";
import "./chunk-GSO543SK.mjs";
import "./chunk-BEFLC7IY.mjs";
import {
  CiscoWebexRemoteLinkAri
} from "./chunk-3D5YCFFE.mjs";
import "./chunk-TKNCDH7M.mjs";
import "./chunk-KLYBIY73.mjs";
import {
  CanvasWorkspaceAri
} from "./chunk-YVXE2PZX.mjs";
import "./chunk-RSJD3TU6.mjs";
import {
  CanvasBoardAri
} from "./chunk-XDORMIFK.mjs";
import "./chunk-R3VA6JMD.mjs";
import {
  CanvasDatabaseAri
} from "./chunk-FLNQNRS3.mjs";
import "./chunk-LRE2PRU7.mjs";
import {
  CanvasSiteAri
} from "./chunk-ZVPXHVXN.mjs";
import "./chunk-VCVOT2RL.mjs";
import "./chunk-QKWQ73L2.mjs";
import {
  BrieRoleAri
} from "./chunk-D6ZPSJ7W.mjs";
import "./chunk-S5IRDM4L.mjs";
import {
  BrieSiteAri
} from "./chunk-54HUMSXM.mjs";
import "./chunk-2CDIPTTF.mjs";
import "./chunk-Y6JK6U3Y.mjs";
import {
  CanvaCanvaRemoteLinkAri
} from "./chunk-HDJ6C7QB.mjs";
import "./chunk-4JERSUI5.mjs";
import "./chunk-5RVX54OI.mjs";
import {
  BoxBoxRemoteLinkAri
} from "./chunk-ENF6UAUY.mjs";
import "./chunk-5YTZA3BU.mjs";
import {
  BoxBoxDocumentAri
} from "./chunk-BSKQUBQK.mjs";
import "./chunk-4QS2SFIR.mjs";
import "./chunk-CI2TIA3X.mjs";
import {
  BitbucketTeamAri
} from "./chunk-BM7OAYUI.mjs";
import "./chunk-O3ROTQWU.mjs";
import {
  BitbucketWorkspaceAri
} from "./chunk-VRCK4EFE.mjs";
import "./chunk-A35PGQYK.mjs";
import {
  BitbucketPullrequestAri
} from "./chunk-QVLM6Y5N.mjs";
import "./chunk-7V62G36F.mjs";
import {
  BitbucketRepositoryTransferAri
} from "./chunk-ZXU3C7PI.mjs";
import "./chunk-QL4XDR5W.mjs";
import {
  BitbucketRepositoryAri
} from "./chunk-SZDHRGNZ.mjs";
import "./chunk-4KEXHL4E.mjs";
import {
  BitbucketSiteAri
} from "./chunk-VPNVE6MJ.mjs";
import "./chunk-NKJWWYMZ.mjs";
import {
  BitbucketAppAri
} from "./chunk-4NKKA36Z.mjs";
import "./chunk-HN5XXUJU.mjs";
import {
  BitbucketOauthConsumerAri
} from "./chunk-BZRGJTNI.mjs";
import "./chunk-B6DXOH6V.mjs";
import {
  BitbucketProjectAri
} from "./chunk-ZBJVKFTR.mjs";
import "./chunk-KZCUOZCO.mjs";
import "./chunk-AIEBCD77.mjs";
import {
  BeaconWorkspaceAri
} from "./chunk-JWOHTGXF.mjs";
import "./chunk-POXHAMNN.mjs";
import {
  BeaconRoleAri
} from "./chunk-6XTYVS42.mjs";
import "./chunk-YRF7LNG2.mjs";
import {
  BeaconSiteAri
} from "./chunk-PCOEYRAB.mjs";
import "./chunk-JQQF6HSU.mjs";
import {
  BeaconSubscriptionAri
} from "./chunk-3JPMB2M3.mjs";
import "./chunk-CWWCGBU4.mjs";
import {
  BeaconAlertAri
} from "./chunk-PIMJN7RW.mjs";
import "./chunk-TJOHF5RH.mjs";
import {
  BeaconCustomDetectionAri
} from "./chunk-3T565ECQ.mjs";
import "./chunk-6YGOGAWM.mjs";
import {
  BeaconDetectionSettingAri
} from "./chunk-3BVY6DKX.mjs";
import "./chunk-XIUYJLCH.mjs";
import {
  BeaconDetectionAri
} from "./chunk-PM2OSIKL.mjs";
import "./chunk-L27OOBSM.mjs";
import "./chunk-BJVOSTUY.mjs";
import {
  AzureDevopsAzureDevopsRemoteLinkAri
} from "./chunk-YOITEZ3A.mjs";
import "./chunk-TE65IYB3.mjs";
import {
  AzureDevopsAzureDevopsRepositoryAri
} from "./chunk-ASNLOAY4.mjs";
import "./chunk-ZMPCNVOL.mjs";
import "./chunk-OTT2DMFI.mjs";
import {
  AvpSubscriptionAri
} from "./chunk-P3YAJMFO.mjs";
import "./chunk-GEK45VES.mjs";
import {
  AvpWorkspaceAri
} from "./chunk-G23RQ4C7.mjs";
import "./chunk-ELWZ7UQT.mjs";
import {
  AvpPipelineQueryAri
} from "./chunk-U67CLJIY.mjs";
import "./chunk-5JS6DFBL.mjs";
import {
  AvpRoleAri
} from "./chunk-QGC377PD.mjs";
import "./chunk-SZNOZQ4T.mjs";
import {
  AvpSiteAri
} from "./chunk-BK43G4BF.mjs";
import "./chunk-GFZFXLJN.mjs";
import {
  AvpDashboardChartAri
} from "./chunk-GGQFQYRK.mjs";
import "./chunk-HFTHZG5P.mjs";
import {
  AvpDashboardAri
} from "./chunk-KAGHM5SZ.mjs";
import "./chunk-L3TDTHN7.mjs";
import {
  AvpDataShareAri
} from "./chunk-IXQQVAFS.mjs";
import "./chunk-FFZGZTIN.mjs";
import {
  AvpDatasourceAri
} from "./chunk-G4GXN6P3.mjs";
import "./chunk-MIBCZ475.mjs";
import "./chunk-UM4DZGWX.mjs";
import "./chunk-5WYUFPST.mjs";
import {
  AvpControlAri
} from "./chunk-INEXAAI2.mjs";
import "./chunk-SQJOVVPI.mjs";
import "./chunk-XQOSNG2Y.mjs";
import {
  AuditLogWorkspaceAri
} from "./chunk-UOEAYMCU.mjs";
import "./chunk-7VQ7RXAT.mjs";
import "./chunk-INZF5DMG.mjs";
import "./chunk-XCSGUIVP.mjs";
import "./chunk-U7AZ64SA.mjs";
import {
  AutomationAutomationRuleAri
} from "./chunk-PMHQUGMI.mjs";
import "./chunk-4V2Y2AKP.mjs";
import {
  AutomationRoleAri
} from "./chunk-IETZ4AHS.mjs";
import "./chunk-QUWGD5ND.mjs";
import {
  AuditLogAttributeAri
} from "./chunk-OH2YQ2J4.mjs";
import "./chunk-HMD655OE.mjs";
import {
  AuditLogContainerAri
} from "./chunk-XHJ6YZV4.mjs";
import "./chunk-GRCW5QVO.mjs";
import {
  AuditLogContextAri
} from "./chunk-FXWFUBVD.mjs";
import "./chunk-LDMX3ABO.mjs";
import {
  AuditLogEventAri
} from "./chunk-ATHDKJSA.mjs";
import "./chunk-S2JTDYO4.mjs";
import "./chunk-567EEHI7.mjs";
import {
  AtlashubIdAri
} from "./chunk-AHEN3X4R.mjs";
import "./chunk-C4MJWUCX.mjs";
import {
  AtlashubSiteAri
} from "./chunk-K34XET4M.mjs";
import "./chunk-SJUWQEXL.mjs";
import "./chunk-2XDSMZ2P.mjs";
import {
  AtlassianWebCrawlerAtlassianWebCrawlerDocumentAri
} from "./chunk-OKM6TBQL.mjs";
import "./chunk-PMG3TISW.mjs";
import {
  Arm
} from "./chunk-RZZGPFKT.mjs";
import "./chunk-KUIAUPIG.mjs";
import {
  AsanaAsanaRemoteLinkAri
} from "./chunk-VHDVJ4YH.mjs";
import "./chunk-TOTM77AA.mjs";
import {
  AsanaAsanaWorkItemAri
} from "./chunk-6M3OKJRJ.mjs";
import "./chunk-UXEYXOYH.mjs";
import "./chunk-ZGGM6QS4.mjs";
import {
  ApprovalDeskDecisionTableAri
} from "./chunk-YIA33EBZ.mjs";
import "./chunk-GTNXGX3Z.mjs";
import {
  ApprovalDeskRoleAri
} from "./chunk-GXMW3ADE.mjs";
import "./chunk-OQIMDPGX.mjs";
import {
  ApprovalDeskWorkflowAri
} from "./chunk-334ORLK4.mjs";
import "./chunk-LIEBAJFZ.mjs";
import {
  ApprovalDeskWorkspaceAri
} from "./chunk-OXGGMHZF.mjs";
import "./chunk-YTLJFQ5C.mjs";
import "./chunk-P5EQPJFA.mjs";
import {
  AnalyticsUserAri
} from "./chunk-AKPOIOF4.mjs";
import "./chunk-77DTJ44V.mjs";
import "./chunk-3JT7OIO3.mjs";
import {
  AppMigrationAppDataAri
} from "./chunk-L5ZJFCE7.mjs";
import "./chunk-V2LSCA3O.mjs";
import "./chunk-EYQIVMHY.mjs";
import {
  AirtableAirtableDocumentAri
} from "./chunk-YQ5KCD7H.mjs";
import "./chunk-PGCTJ4SL.mjs";
import {
  AirtableAirtableRemoteLinkAri
} from "./chunk-YJFEKUZZ.mjs";
import "./chunk-5WNGAVVZ.mjs";
import "./chunk-KPNULLMY.mjs";
import {
  AmplitudeAmplitudeRemoteLinkAri
} from "./chunk-FG23VKPJ.mjs";
import "./chunk-63OICGDG.mjs";
import "./chunk-ULDEWI7M.mjs";
import {
  AdobeAdobexdRemoteLinkAri
} from "./chunk-MHTDAQEL.mjs";
import "./chunk-H2R5HU73.mjs";
import "./chunk-W6HXA4MI.mjs";
import {
  AdobeSignDocumentAri
} from "./chunk-ZI7XZNS6.mjs";
import "./chunk-WP7ILJ3M.mjs";
import "./chunk-RGYW4H4I.mjs";
import {
  AhaAhaWorkItemAri
} from "./chunk-MTF7V6HO.mjs";
import "./chunk-AESEW5OU.mjs";
import "./chunk-AB4PPISI.mjs";
import {
  AnyAri,
  Ari
} from "./chunk-YFR5EBRK.mjs";
import {
  Ati
} from "./chunk-S7RIZVFI.mjs";
import "./chunk-GAA5ADPB.mjs";
import {
  ValidationError
} from "./chunk-FAZYALZ2.mjs";

// src/core/types/deprecated/resource-owner-enum.ts
var ResourceOwnerEnum = /* @__PURE__ */ ((ResourceOwnerEnum2) => (ResourceOwnerEnum2.AdobeAdobexd = "adobe.adobexd", ResourceOwnerEnum2.AdobeSign = "adobe.sign", ResourceOwnerEnum2.AhaAha = "aha.aha", ResourceOwnerEnum2.AirtableAirtable = "airtable.airtable", ResourceOwnerEnum2.AmplitudeAmplitude = "amplitude.amplitude", ResourceOwnerEnum2.Analytics = "analytics", ResourceOwnerEnum2.AppMigration = "app-migration", ResourceOwnerEnum2.ApprovalDesk = "approval-desk", ResourceOwnerEnum2.AsanaAsana = "asana.asana", ResourceOwnerEnum2.Atlashub = "atlashub", ResourceOwnerEnum2.AtlassianWebCrawlerAtlassianWebCrawler = "atlassian-web-crawler.atlassian-web-crawler", ResourceOwnerEnum2.AuditLog = "audit-log", ResourceOwnerEnum2.Automation = "automation", ResourceOwnerEnum2.Avocado = "avocado", ResourceOwnerEnum2.Avp = "avp", ResourceOwnerEnum2.AzureDevopsAzureDevops = "azure-devops.azure-devops", ResourceOwnerEnum2.Beacon = "beacon", ResourceOwnerEnum2.Bitbucket = "bitbucket", ResourceOwnerEnum2.BoxBox = "box.box", ResourceOwnerEnum2.Brie = "brie", ResourceOwnerEnum2.CanvaCanva = "canva.canva", ResourceOwnerEnum2.Canvas = "canvas", ResourceOwnerEnum2.Cia = "cia", ResourceOwnerEnum2.CiscoWebex = "cisco.webex", ResourceOwnerEnum2.ClickupClickup = "clickup.clickup", ResourceOwnerEnum2.Cmdb = "cmdb", ResourceOwnerEnum2.Commerce = "commerce", ResourceOwnerEnum2.Compass = "compass", ResourceOwnerEnum2.Confluence = "confluence", ResourceOwnerEnum2.ConfluenceDc = "confluence-dc", ResourceOwnerEnum2.ConfluenceDcConfluenceDc = "confluence-dc.confluence-dc", ResourceOwnerEnum2.ConnectionRegistry = "connection-registry", ResourceOwnerEnum2.ConversationalHelp = "conversational-help", ResourceOwnerEnum2.ConvoAi = "convo-ai", ResourceOwnerEnum2.CssXp = "css-xp", ResourceOwnerEnum2.DataLake = "data-lake", ResourceOwnerEnum2.Devai = "devai", ResourceOwnerEnum2.Devops = "devops", ResourceOwnerEnum2.DocusignDocusign = "docusign.docusign", ResourceOwnerEnum2.DovetailDovetail = "dovetail.dovetail", ResourceOwnerEnum2.Dropbox = "dropbox", ResourceOwnerEnum2.DropboxDropbox = "dropbox.dropbox", ResourceOwnerEnum2.Ecosystem = "ecosystem", ResourceOwnerEnum2.Elements = "elements", ResourceOwnerEnum2.Elevate = "elevate", ResourceOwnerEnum2.ExusExternal = "exus-external", ResourceOwnerEnum2.Figma = "figma", ResourceOwnerEnum2.FigmaFigma = "figma.figma", ResourceOwnerEnum2.GithubGithub = "github.github", ResourceOwnerEnum2.GitlabGitlab = "gitlab.gitlab", ResourceOwnerEnum2.Goal = "goal", ResourceOwnerEnum2.Google = "google", ResourceOwnerEnum2.GoogleGmail = "google.gmail", ResourceOwnerEnum2.GoogleGoogleCalendar = "google.google-calendar", ResourceOwnerEnum2.GoogleGoogleDrive = "google.google-drive", ResourceOwnerEnum2.GoogleGoogleDriveRsl = "google.google-drive-rsl", ResourceOwnerEnum2.Graph = "graph", ResourceOwnerEnum2.Help = "help", ResourceOwnerEnum2.Home = "home", ResourceOwnerEnum2.HubspotHubspot = "hubspot.hubspot", ResourceOwnerEnum2.Identity = "identity", ResourceOwnerEnum2.Jira = "jira", ResourceOwnerEnum2.JiraAiCrocs = "jira-ai-crocs", ResourceOwnerEnum2.JiraAlign = "jira-align", ResourceOwnerEnum2.JiraCore = "jira-core", ResourceOwnerEnum2.JiraCustomerService = "jira-customer-service", ResourceOwnerEnum2.JiraDcJiraDc = "jira-dc.jira-dc", ResourceOwnerEnum2.JiraDcSecondaryJiraDc = "jira-dc-secondary.jira-dc", ResourceOwnerEnum2.JiraDcTertiaryJiraDc = "jira-dc-tertiary.jira-dc", ResourceOwnerEnum2.JiraServicedesk = "jira-servicedesk", ResourceOwnerEnum2.JiraSoftware = "jira-software", ResourceOwnerEnum2.JsmChannelOrchestrator = "jsm-channel-orchestrator", ResourceOwnerEnum2.KnowledgeDiscovery = "knowledge-discovery", ResourceOwnerEnum2.KnowledgeServingAndAccess = "knowledge-serving-and-access", ResourceOwnerEnum2.LaunchdarklyLaunchdarkly = "launchdarkly.launchdarkly", ResourceOwnerEnum2.LearningPlatform = "learning-platform", ResourceOwnerEnum2.LinkingPlatform = "linking-platform", ResourceOwnerEnum2.LoginAlerter = "login-alerter", ResourceOwnerEnum2.Loom = "loom", ResourceOwnerEnum2.LoomLoom = "loom.loom", ResourceOwnerEnum2.LucidLucidchart = "lucid.lucidchart", ResourceOwnerEnum2.MakerSpace = "maker-space", ResourceOwnerEnum2.Mapi = "mapi", ResourceOwnerEnum2.Marketing = "marketing", ResourceOwnerEnum2.Marketplace = "marketplace", ResourceOwnerEnum2.Measurement = "measurement", ResourceOwnerEnum2.Media = "media", ResourceOwnerEnum2.Mercury = "mercury", ResourceOwnerEnum2.Microsoft = "microsoft", ResourceOwnerEnum2.MicrosoftAzureDevops = "microsoft.azure-devops", ResourceOwnerEnum2.MicrosoftOnedrive = "microsoft.onedrive", ResourceOwnerEnum2.MicrosoftOutlook = "microsoft.outlook", ResourceOwnerEnum2.MicrosoftOutlookCalendar = "microsoft.outlook-calendar", ResourceOwnerEnum2.MicrosoftPowerBi = "microsoft.power-bi", ResourceOwnerEnum2.MicrosoftSharepoint = "microsoft.sharepoint", ResourceOwnerEnum2.MicrosoftTeams = "microsoft.teams", ResourceOwnerEnum2.MigrationReportCenter = "migration-report-center", ResourceOwnerEnum2.MiroMiro = "miro.miro", ResourceOwnerEnum2.MlRegistry = "ml-registry", ResourceOwnerEnum2.MondayMonday = "monday.monday", ResourceOwnerEnum2.MuralMural = "mural.mural", ResourceOwnerEnum2.NotionNotion = "notion.notion", ResourceOwnerEnum2.Oauth = "oauth", ResourceOwnerEnum2.Opsgenie = "opsgenie", ResourceOwnerEnum2.OrgPolicy = "org-policy", ResourceOwnerEnum2.PagerdutyPagerduty = "pagerduty.pagerduty", ResourceOwnerEnum2.Papi = "papi", ResourceOwnerEnum2.Passionfruit = "passionfruit", ResourceOwnerEnum2.PeopleData = "people-data", ResourceOwnerEnum2.PeoplePerftool = "people-perftool", ResourceOwnerEnum2.PipedrivePipedrive = "pipedrive.pipedrive", ResourceOwnerEnum2.Platform = "platform", ResourceOwnerEnum2.PlatformServices = "platform-services", ResourceOwnerEnum2.Pollinator = "pollinator", ResourceOwnerEnum2.PostOffice = "post-office", ResourceOwnerEnum2.Proforma = "proforma", ResourceOwnerEnum2.Project = "project", ResourceOwnerEnum2.Pure = "pure", ResourceOwnerEnum2.Radar = "radar", ResourceOwnerEnum2.Rovo = "rovo", ResourceOwnerEnum2.RuntimeAuthClient = "runtime-auth-client", ResourceOwnerEnum2.SalesforceSalesforce = "salesforce.salesforce", ResourceOwnerEnum2.SandboxService = "sandbox-service", ResourceOwnerEnum2.Search = "search", ResourceOwnerEnum2.SentrySentry = "sentry.sentry", ResourceOwnerEnum2.ServicenowServicenow = "servicenow.servicenow", ResourceOwnerEnum2.SettingsService = "settings-service", ResourceOwnerEnum2.Slack = "slack", ResourceOwnerEnum2.SlackSlack = "slack.slack", ResourceOwnerEnum2.SmartsheetSmartsheet = "smartsheet.smartsheet", ResourceOwnerEnum2.SpinnakerSpinnaker = "spinnaker.spinnaker", ResourceOwnerEnum2.Statuspage = "statuspage", ResourceOwnerEnum2.StripeStripe = "stripe.stripe", ResourceOwnerEnum2.Support = "support", ResourceOwnerEnum2.Teams = "teams", ResourceOwnerEnum2.TodoistTodoist = "todoist.todoist", ResourceOwnerEnum2.Townsquare = "townsquare", ResourceOwnerEnum2.Trello = "trello", ResourceOwnerEnum2.TrustEng = "trust-eng", ResourceOwnerEnum2.UnifiedHelp = "unified-help", ResourceOwnerEnum2.UnifiedPortableProfile = "unified-portable-profile", ResourceOwnerEnum2.UsageTracking = "usage-tracking", ResourceOwnerEnum2.VirtualAgent = "virtual-agent", ResourceOwnerEnum2.Workato = "workato", ResourceOwnerEnum2.Workday = "workday", ResourceOwnerEnum2.WorkdayWorkday = "workday.workday", ResourceOwnerEnum2.ZendeskZendesk = "zendesk.zendesk", ResourceOwnerEnum2.ZeplinZeplin = "zeplin.zeplin", ResourceOwnerEnum2))(ResourceOwnerEnum || {});

// src/core/types/deprecated/resource-type-enum.ts
var ResourceTypeEnum = /* @__PURE__ */ ((ResourceTypeEnum2) => (ResourceTypeEnum2.Account = "account", ResourceTypeEnum2.AccountLogin = "account-login", ResourceTypeEnum2.AccountSettings = "account-settings", ResourceTypeEnum2.AdminAnnouncementBanner = "admin-announcement-banner", ResourceTypeEnum2.Agent = "agent", ResourceTypeEnum2.Alert = "alert", ResourceTypeEnum2.AlertRecipientLink = "alert-recipient-link", ResourceTypeEnum2.AnnouncementBanner = "announcement-banner", ResourceTypeEnum2.Answer = "answer", ResourceTypeEnum2.Api = "api", ResourceTypeEnum2.ApiEndpoint = "api-endpoint", ResourceTypeEnum2.ApiRequestMetric = "api-request-metric", ResourceTypeEnum2.App = "app", ResourceTypeEnum2.AppData = "app-data", ResourceTypeEnum2.AppEnvironmentVersion = "app-environment-version", ResourceTypeEnum2.AppGrant = "app-grant", ResourceTypeEnum2.Approval = "approval", ResourceTypeEnum2.Article = "article", ResourceTypeEnum2.Artifact = "artifact", ResourceTypeEnum2.Ask = "ask", ResourceTypeEnum2.AskActivity = "ask-activity", ResourceTypeEnum2.AskComment = "ask-comment", ResourceTypeEnum2.AskLink = "ask-link", ResourceTypeEnum2.AskUpdate = "ask-update", ResourceTypeEnum2.AssistantAnswer = "assistant-answer", ResourceTypeEnum2.Attachment = "attachment", ResourceTypeEnum2.Attribute = "attribute", ResourceTypeEnum2.AuthPolicy = "auth-policy", ResourceTypeEnum2.AutodevJob = "autodev-job", ResourceTypeEnum2.AutomationRule = "automation-rule", ResourceTypeEnum2.AvailableFields = "available-fields", ResourceTypeEnum2.Avatar = "avatar", ResourceTypeEnum2.Background = "background", ResourceTypeEnum2.Blogpost = "blogpost", ResourceTypeEnum2.Board = "board", ResourceTypeEnum2.BoardFeature = "board-feature", ResourceTypeEnum2.BoardIssueList = "board-issue-list", ResourceTypeEnum2.Bookmark = "bookmark", ResourceTypeEnum2.Branch = "branch", ResourceTypeEnum2.BranchHistory = "branch-history", ResourceTypeEnum2.Build = "build", ResourceTypeEnum2.BuildHistory = "build-history", ResourceTypeEnum2.BulkOperationTask = "bulk-operation-task", ResourceTypeEnum2.CalendarEvent = "calendar-event", ResourceTypeEnum2.CalendarEventHistory = "calendar-event-history", ResourceTypeEnum2.Calibration = "calibration", ResourceTypeEnum2.CallRouting = "call-routing", ResourceTypeEnum2.Campaign = "campaign", ResourceTypeEnum2.CannedResponse = "canned-response", ResourceTypeEnum2.Card = "card", ResourceTypeEnum2.CardColor = "card-color", ResourceTypeEnum2.CardLayout = "card-layout", ResourceTypeEnum2.CardParent = "card-parent", ResourceTypeEnum2.CatalogAccount = "catalog-account", ResourceTypeEnum2.Change = "change", ResourceTypeEnum2.ChangeProposal = "change-proposal", ResourceTypeEnum2.ChangeProposalStatus = "change-proposal-status", ResourceTypeEnum2.Channel = "channel", ResourceTypeEnum2.Chart = "chart", ResourceTypeEnum2.Chat = "chat", ResourceTypeEnum2.Check = "check", ResourceTypeEnum2.CheckItem = "check-item", ResourceTypeEnum2.Checklist = "checklist", ResourceTypeEnum2.ClassificationTag = "classification-tag", ResourceTypeEnum2.Client = "client", ResourceTypeEnum2.CollaborationContext = "collaboration-context", ResourceTypeEnum2.Column = "column", ResourceTypeEnum2.Comment = "comment", ResourceTypeEnum2.CommentHistory = "comment-history", ResourceTypeEnum2.Commit = "commit", ResourceTypeEnum2.Component = "component", ResourceTypeEnum2.ComponentLabel = "component-label", ResourceTypeEnum2.ComponentLink = "component-link", ResourceTypeEnum2.ComponentType = "component-type", ResourceTypeEnum2.Configuration = "configuration", ResourceTypeEnum2.ConnectApp = "connect-app", ResourceTypeEnum2.ConnectModule = "connect-module", ResourceTypeEnum2.Container = "container", ResourceTypeEnum2.Content = "content", ResourceTypeEnum2.Context = "context", ResourceTypeEnum2.Control = "control", ResourceTypeEnum2.Conversation = "conversation", ResourceTypeEnum2.ConversationChunk = "conversation-chunk", ResourceTypeEnum2.ConversationHistory = "conversation-history", ResourceTypeEnum2.ConversationMessage = "conversation-message", ResourceTypeEnum2.CostCenter = "cost-center", ResourceTypeEnum2.CprsEmail = "cprs-email", ResourceTypeEnum2.CustomAction = "custom-action", ResourceTypeEnum2.CustomContent = "custom-content", ResourceTypeEnum2.CustomDetection = "custom-detection", ResourceTypeEnum2.CustomEmojiMetadata = "custom-emoji-metadata", ResourceTypeEnum2.Customer = "customer", ResourceTypeEnum2.CustomerDirectory = "customer-directory", ResourceTypeEnum2.CustomerDomain = "customer-domain", ResourceTypeEnum2.CustomerOrg = "customer-org", ResourceTypeEnum2.CustomerOrganization = "customer-organization", ResourceTypeEnum2.CustomerOrgHistory = "customer-org-history", ResourceTypeEnum2.CustomerSupport = "customer-support", ResourceTypeEnum2.CustomField = "custom-field", ResourceTypeEnum2.CustomFieldDefinition = "custom-field-definition", ResourceTypeEnum2.CustomFilter = "custom-filter", ResourceTypeEnum2.CustomRole = "custom-role", ResourceTypeEnum2.Dashboard = "dashboard", ResourceTypeEnum2.DashboardChart = "dashboard-chart", ResourceTypeEnum2.Database = "database", ResourceTypeEnum2.DataShare = "data-share", ResourceTypeEnum2.Datasource = "datasource", ResourceTypeEnum2.Deal = "deal", ResourceTypeEnum2.DealHistory = "deal-history", ResourceTypeEnum2.DecisionTable = "decision-table", ResourceTypeEnum2.Dependency = "dependency", ResourceTypeEnum2.DependencyComment = "dependency-comment", ResourceTypeEnum2.DependencyRelatedContent = "dependency-related-content", ResourceTypeEnum2.DependencyUpdate = "dependency-update", ResourceTypeEnum2.Deployment = "deployment", ResourceTypeEnum2.DeploymentHistory = "deployment-history", ResourceTypeEnum2.DeploymentServiceLink = "deployment-service-link", ResourceTypeEnum2.Design = "design", ResourceTypeEnum2.DesignHistory = "design-history", ResourceTypeEnum2.Detection = "detection", ResourceTypeEnum2.DetectionSetting = "detection-setting", ResourceTypeEnum2.DevopsComponent = "devops-component", ResourceTypeEnum2.Directory = "directory", ResourceTypeEnum2.Document = "document", ResourceTypeEnum2.DocumentationCategory = "documentation-category", ResourceTypeEnum2.DocumentHistory = "document-history", ResourceTypeEnum2.Domain = "domain", ResourceTypeEnum2.Email = "email", ResourceTypeEnum2.EmailUuid = "email-uuid", ResourceTypeEnum2.Embed = "embed", ResourceTypeEnum2.Emoji = "emoji", ResourceTypeEnum2.Employee = "employee", ResourceTypeEnum2.EmployeeHistory = "employee-history", ResourceTypeEnum2.Enterprise = "enterprise", ResourceTypeEnum2.Entitlement = "entitlement", ResourceTypeEnum2.EntitlementTemplate = "entitlement-template", ResourceTypeEnum2.EntityProperty = "entity-property", ResourceTypeEnum2.Environment = "environment", ResourceTypeEnum2.Epic = "epic", ResourceTypeEnum2.Escalation = "escalation", ResourceTypeEnum2.Event = "event", ResourceTypeEnum2.EventSource = "event-source", ResourceTypeEnum2.Extension = "extension", ResourceTypeEnum2.ExtensionGroup = "extension-group", ResourceTypeEnum2.Favourite = "favourite", ResourceTypeEnum2.FeatureFlag = "feature-flag", ResourceTypeEnum2.FeatureFlagHistory = "feature-flag-history", ResourceTypeEnum2.Feedback = "feedback", ResourceTypeEnum2.FieldConfiguration = "field-configuration", ResourceTypeEnum2.FieldConfigurationContext = "field-configuration-context", ResourceTypeEnum2.FieldSetView = "field-set-view", ResourceTypeEnum2.File = "file", ResourceTypeEnum2.Filter = "filter", ResourceTypeEnum2.FilterEmailSubscription = "filter-email-subscription", ResourceTypeEnum2.FlowEditor = "flow-editor", ResourceTypeEnum2.FocusArea = "focus-area", ResourceTypeEnum2.FocusAreaStatusUpdate = "focus-area-status-update", ResourceTypeEnum2.FocusAreaType = "focus-area-type", ResourceTypeEnum2.Folder = "folder", ResourceTypeEnum2.ForgeInstallation = "forge-installation", ResourceTypeEnum2.Form = "form", ResourceTypeEnum2.ForwardDocument = "forward-document", ResourceTypeEnum2.ForwardDocumentSubscription = "forward-document-subscription", ResourceTypeEnum2.Goal = "goal", ResourceTypeEnum2.GoalMetricUpdate = "goal-metric-update", ResourceTypeEnum2.GoalType = "goal-type", ResourceTypeEnum2.GoalUpdate = "goal-update", ResourceTypeEnum2.Group = "group", ResourceTypeEnum2.HelpCenter = "help-center", ResourceTypeEnum2.HelpcenterForm = "helpcenter-form", ResourceTypeEnum2.HelpDesk = "help-desk", ResourceTypeEnum2.HelpPointer = "help-pointer", ResourceTypeEnum2.Id = "id", ResourceTypeEnum2.ImportConfiguration = "import-configuration", ResourceTypeEnum2.Incident = "incident", ResourceTypeEnum2.IncidentAlertLink = "incident-alert-link", ResourceTypeEnum2.IncidentHistory = "incident-history", ResourceTypeEnum2.IncidentStatusUpdate = "incident-status-update", ResourceTypeEnum2.IncomingCallHistory = "incoming-call-history", ResourceTypeEnum2.IncomingWebhook = "incoming-webhook", ResourceTypeEnum2.Initiative = "initiative", ResourceTypeEnum2.Installation = "installation", ResourceTypeEnum2.InstallationLink = "installation-link", ResourceTypeEnum2.Instance = "instance", ResourceTypeEnum2.Integration = "integration", ResourceTypeEnum2.IntegrationContext = "integration-context", ResourceTypeEnum2.Intent = "intent", ResourceTypeEnum2.IntentProjection = "intent-projection", ResourceTypeEnum2.IntentQuestion = "intent-question", ResourceTypeEnum2.IntentQuestionProjection = "intent-question-projection", ResourceTypeEnum2.IntentRuleProjection = "intent-rule-projection", ResourceTypeEnum2.IntentTemplate = "intent-template", ResourceTypeEnum2.InvoiceGroup = "invoice-group", ResourceTypeEnum2.Issue = "issue", ResourceTypeEnum2.IssueAttachment = "issue-attachment", ResourceTypeEnum2.IssueComment = "issue-comment", ResourceTypeEnum2.IssueFieldMetadata = "issue-field-metadata", ResourceTypeEnum2.IssueFieldOption = "issue-field-option", ResourceTypeEnum2.Issuefieldvalue = "issuefieldvalue", ResourceTypeEnum2.IssueForm = "issue-form", ResourceTypeEnum2.IssueHistory = "issue-history", ResourceTypeEnum2.IssueLink = "issue-link", ResourceTypeEnum2.IssueLinkType = "issue-link-type", ResourceTypeEnum2.IssueRemoteLink = "issue-remote-link", ResourceTypeEnum2.IssueSearchView = "issue-search-view", ResourceTypeEnum2.IssueSearchViewType = "issue-search-view-type", ResourceTypeEnum2.IssueSecurityLevelPermission = "issue-security-level-permission", ResourceTypeEnum2.IssueStatus = "issue-status", ResourceTypeEnum2.IssueType = "issue-type", ResourceTypeEnum2.IssueTypeScheme = "issue-type-scheme", ResourceTypeEnum2.IssueTypeSchemeMapping = "issue-type-scheme-mapping", ResourceTypeEnum2.IssueWorklog = "issue-worklog", ResourceTypeEnum2.JiraProjectAndDevopsToolRelationship = "jira-project-and-devops-tool-relationship", ResourceTypeEnum2.JiraProjectAndDocumentationSpaceRelationship = "jira-project-and-documentation-space-relationship", ResourceTypeEnum2.JiraProjectAndOperationsTeamRelationship = "jira-project-and-operations-team-relationship", ResourceTypeEnum2.JiraProjectAndVcsRepositoryRelationship = "jira-project-and-vcs-repository-relationship", ResourceTypeEnum2.JiraProjectAndVcsRepositoryRelationshipTempMigration = "jira-project-and-vcs-repository-relationship-temp-migration", ResourceTypeEnum2.Job = "job", ResourceTypeEnum2.JobProfile = "job-profile", ResourceTypeEnum2.Label = "label", ResourceTypeEnum2.LakeConnection = "lake-connection", ResourceTypeEnum2.Layout = "layout", ResourceTypeEnum2.Learner = "learner", ResourceTypeEnum2.Learning = "learning", ResourceTypeEnum2.LibraryScorecard = "library-scorecard", ResourceTypeEnum2.LifecycleResource = "lifecycle-resource", ResourceTypeEnum2.LifecycleResourcePackage = "lifecycle-resource-package", ResourceTypeEnum2.LifecycleResourcePackageType = "lifecycle-resource-package-type", ResourceTypeEnum2.Link = "link", ResourceTypeEnum2.LinkedAccount = "linked-account", ResourceTypeEnum2.List = "list", ResourceTypeEnum2.Logo = "logo", ResourceTypeEnum2.LongRunningTask = "long-running-task", ResourceTypeEnum2.Meeting = "meeting", ResourceTypeEnum2.MeetingRecurrence = "meeting-recurrence", ResourceTypeEnum2.Message = "message", ResourceTypeEnum2.MessageInstance = "message-instance", ResourceTypeEnum2.MessageTemplate = "message-template", ResourceTypeEnum2.Metric = "metric", ResourceTypeEnum2.MetricDefinition = "metric-definition", ResourceTypeEnum2.MetricSource = "metric-source", ResourceTypeEnum2.MetricTarget = "metric-target", ResourceTypeEnum2.MetricUpdate = "metric-update", ResourceTypeEnum2.MetricValue = "metric-value", ResourceTypeEnum2.Milestone = "milestone", ResourceTypeEnum2.Namespace = "namespace", ResourceTypeEnum2.NavigationConfig = "navigation-config", ResourceTypeEnum2.NavigationItem = "navigation-item", ResourceTypeEnum2.NavigationMenu = "navigation-menu", ResourceTypeEnum2.NavigationSettings = "navigation-settings", ResourceTypeEnum2.Note = "note", ResourceTypeEnum2.Notification = "notification", ResourceTypeEnum2.NotificationType = "notification-type", ResourceTypeEnum2.NotificationTypeScheme = "notification-type-scheme", ResourceTypeEnum2.NotificationUserPreference = "notification-user-preference", ResourceTypeEnum2.OauthClient = "oauthClient", ResourceTypeEnum2.OauthConsumer = "oauth-consumer", ResourceTypeEnum2.Object = "object", ResourceTypeEnum2.Offering = "offering", ResourceTypeEnum2.OfferingRelationshipTemplate = "offering-relationship-template", ResourceTypeEnum2.OperationsWorkspace = "operations-workspace", ResourceTypeEnum2.Order = "order", ResourceTypeEnum2.Org = "org", ResourceTypeEnum2.Organisation = "organisation", ResourceTypeEnum2.OrganisationHistory = "organisation-history", ResourceTypeEnum2.Organization = "organization", ResourceTypeEnum2.OrganizationMember = "organization-member", ResourceTypeEnum2.OrgUser = "org-user", ResourceTypeEnum2.Package = "package", ResourceTypeEnum2.Page = "page", ResourceTypeEnum2.ParentAccount = "parent-account", ResourceTypeEnum2.PaymentMethod = "payment-method", ResourceTypeEnum2.Performance = "performance", ResourceTypeEnum2.Permission = "permission", ResourceTypeEnum2.PermissionContainer = "permission-container", ResourceTypeEnum2.PermissionScheme = "permission-scheme", ResourceTypeEnum2.PipelineQuery = "pipeline-query", ResourceTypeEnum2.Plan = "plan", ResourceTypeEnum2.PlanCrossProjectScenarioVersion = "plan-cross-project-scenario-version", ResourceTypeEnum2.PlanCrossProjectVersion = "plan-cross-project-version", ResourceTypeEnum2.Planner = "planner", ResourceTypeEnum2.PlannerCalendar = "planner-calendar", ResourceTypeEnum2.PlannerEventCard = "planner-event-card", ResourceTypeEnum2.PlanScenario = "plan-scenario", ResourceTypeEnum2.PlanScenarioIssue = "plan-scenario-issue", ResourceTypeEnum2.PlanScenarioIssueField = "plan-scenario-issue-field", ResourceTypeEnum2.PlanScenarioVersion = "plan-scenario-version", ResourceTypeEnum2.Playbook = "playbook", ResourceTypeEnum2.PlaybookInstance = "playbook-instance", ResourceTypeEnum2.PlaybookInstanceStep = "playbook-instance-step", ResourceTypeEnum2.PlaybookLabel = "playbook-label", ResourceTypeEnum2.PlaybookStep = "playbook-step", ResourceTypeEnum2.PlaybookStepRun = "playbook-step-run", ResourceTypeEnum2.Policy = "policy", ResourceTypeEnum2.Portal = "portal", ResourceTypeEnum2.Position = "position", ResourceTypeEnum2.PositionHistory = "position-history", ResourceTypeEnum2.PostIncidentReview = "post-incident-review", ResourceTypeEnum2.PostIncidentReviewHistory = "post-incident-review-history", ResourceTypeEnum2.PostIncidentReviewLink = "post-incident-review-link", ResourceTypeEnum2.Precalibration = "precalibration", ResourceTypeEnum2.PrecalibrationPillar = "precalibration-pillar", ResourceTypeEnum2.Presentation = "presentation", ResourceTypeEnum2.PricingPlan = "pricing-plan", ResourceTypeEnum2.Priority = "priority", ResourceTypeEnum2.Product = "product", ResourceTypeEnum2.Program = "program", ResourceTypeEnum2.ProgramStatusUpdate = "program-status-update", ResourceTypeEnum2.Project = "project", ResourceTypeEnum2.ProjectCategory = "project-category", ResourceTypeEnum2.ProjectFeature = "project-feature", ResourceTypeEnum2.ProjectForm = "project-form", ResourceTypeEnum2.ProjectHistory = "project-history", ResourceTypeEnum2.ProjectOverview = "project-overview", ResourceTypeEnum2.ProjectRoleActor = "project-role-actor", ResourceTypeEnum2.ProjectRoleByProject = "project-role-by-project", ResourceTypeEnum2.ProjectShortcut = "project-shortcut", ResourceTypeEnum2.ProjectType = "project-type", ResourceTypeEnum2.ProjectUpdate = "project-update", ResourceTypeEnum2.Promotion = "promotion", ResourceTypeEnum2.Provider = "provider", ResourceTypeEnum2.PublicLink = "public-link", ResourceTypeEnum2.Pullrequest = "pullrequest", ResourceTypeEnum2.PullRequest = "pull-request", ResourceTypeEnum2.PullRequestHistory = "pull-request-history", ResourceTypeEnum2.Question = "question", ResourceTypeEnum2.Queue = "queue", ResourceTypeEnum2.Quote = "quote", ResourceTypeEnum2.Recipe = "recipe", ResourceTypeEnum2.Relationship = "relationship", ResourceTypeEnum2.RelationshipType = "relationship-type", ResourceTypeEnum2.RemoteLink = "remote-link", ResourceTypeEnum2.RemoteLinkHistory = "remote-link-history", ResourceTypeEnum2.ReportRequestProxy = "report-request-proxy", ResourceTypeEnum2.Repository = "repository", ResourceTypeEnum2.RepositoryTransfer = "repository-transfer", ResourceTypeEnum2.RequestForm = "request-form", ResourceTypeEnum2.RequestType = "request-type", ResourceTypeEnum2.Resolution = "resolution", ResourceTypeEnum2.Resource = "resource", ResourceTypeEnum2.ResourceUsageMetric = "resource-usage-metric", ResourceTypeEnum2.ResourceUsageRecommendation = "resource-usage-recommendation", ResourceTypeEnum2.Role = "role", ResourceTypeEnum2.Rule = "rule", ResourceTypeEnum2.SavedSearch = "saved-search", ResourceTypeEnum2.Scenario = "scenario", ResourceTypeEnum2.Schedule = "schedule", ResourceTypeEnum2.ScheduleRotation = "schedule-rotation", ResourceTypeEnum2.Schema = "schema", ResourceTypeEnum2.ScopedGroup = "scoped-group", ResourceTypeEnum2.Scorecard = "scorecard", ResourceTypeEnum2.Screen = "screen", ResourceTypeEnum2.ScreenScheme = "screen-scheme", ResourceTypeEnum2.ScreenTab = "screen-tab", ResourceTypeEnum2.SecureTunnel = "secure-tunnel", ResourceTypeEnum2.SecurityContainer = "security-container", ResourceTypeEnum2.SecurityLevel = "security-level", ResourceTypeEnum2.SecurityWorkspace = "security-workspace", ResourceTypeEnum2.SelfAssessment = "self-assessment", ResourceTypeEnum2.SelfAssessmentPillar = "self-assessment-pillar", ResourceTypeEnum2.Sentence = "sentence", ResourceTypeEnum2.Service = "service", ResourceTypeEnum2.ServiceAndJiraProjectRelationship = "service-and-jira-project-relationship", ResourceTypeEnum2.ServiceAndOperationsTeamRelationship = "service-and-operations-team-relationship", ResourceTypeEnum2.ServiceAndVcsRepositoryRelationship = "service-and-vcs-repository-relationship", ResourceTypeEnum2.ServiceRelationship = "service-relationship", ResourceTypeEnum2.Settings = "settings", ResourceTypeEnum2.SharepointDocument = "sharepoint-document", ResourceTypeEnum2.ShipToParty = "ship-to-party", ResourceTypeEnum2.Site = "site", ResourceTypeEnum2.SiteUser = "site-user", ResourceTypeEnum2.Sla = "sla", ResourceTypeEnum2.Space = "space", ResourceTypeEnum2.SpaceHistory = "space-history", ResourceTypeEnum2.Spreadsheet = "spreadsheet", ResourceTypeEnum2.Sprint = "sprint", ResourceTypeEnum2.Staff = "staff", ResourceTypeEnum2.StandaloneWorkspace = "standalone-workspace", ResourceTypeEnum2.Sticker = "sticker", ResourceTypeEnum2.StrategicEvent = "strategic-event", ResourceTypeEnum2.StrategicEventStatus = "strategic-event-status", ResourceTypeEnum2.StreamhubSchema = "streamhub-schema", ResourceTypeEnum2.Subscription = "subscription", ResourceTypeEnum2.SupervisoryOrg = "supervisory-org", ResourceTypeEnum2.Swimlane = "swimlane", ResourceTypeEnum2.Tag = "tag", ResourceTypeEnum2.Task = "task", ResourceTypeEnum2.Team = "team", ResourceTypeEnum2.TeamCalendar = "team-calendar", ResourceTypeEnum2.TeamMember = "team-member", ResourceTypeEnum2.TemplateForm = "template-form", ResourceTypeEnum2.TestProvisioningEntity = "test-provisioning-entity", ResourceTypeEnum2.ThirdPartyDirectory = "third-party-directory", ResourceTypeEnum2.ThirdPartyGroup = "third-party-group", ResourceTypeEnum2.ThirdPartyUser = "third-party-user", ResourceTypeEnum2.ThreadChunk = "thread-chunk", ResourceTypeEnum2.Timeline = "timeline", ResourceTypeEnum2.Tool = "tool", ResourceTypeEnum2.Topic = "topic", ResourceTypeEnum2.TransactionAccount = "transaction-account", ResourceTypeEnum2.Trigger = "trigger", ResourceTypeEnum2.Type = "type", ResourceTypeEnum2.UnifiedAccount = "unified-account", ResourceTypeEnum2.UnifiedProfile = "unified-profile", ResourceTypeEnum2.UpdateNote = "update-note", ResourceTypeEnum2.Usage = "usage", ResourceTypeEnum2.User = "user", ResourceTypeEnum2.Userbase = "userbase", ResourceTypeEnum2.UserBoardConfig = "user-board-config", ResourceTypeEnum2.UserBroadcastMessage = "user-broadcast-message", ResourceTypeEnum2.UserDefinedParameter = "user-defined-parameter", ResourceTypeEnum2.UserGrant = "userGrant", ResourceTypeEnum2.UserProperty = "user-property", ResourceTypeEnum2.Version = "version", ResourceTypeEnum2.VersionApprover = "version-approver", ResourceTypeEnum2.Video = "video", ResourceTypeEnum2.VideoHistory = "video-history", ResourceTypeEnum2.View = "view", ResourceTypeEnum2.ViewConfig = "view-config", ResourceTypeEnum2.Vulnerability = "vulnerability", ResourceTypeEnum2.VulnerabilityHistory = "vulnerability-history", ResourceTypeEnum2.Webhook = "webhook", ResourceTypeEnum2.Whiteboard = "whiteboard", ResourceTypeEnum2.WorkbenchResource = "workbench-resource", ResourceTypeEnum2.Worker = "worker", ResourceTypeEnum2.WorkerHistory = "worker-history", ResourceTypeEnum2.Workflow = "workflow", ResourceTypeEnum2.WorkflowScheme = "workflow-scheme", ResourceTypeEnum2.WorkItem = "work-item", ResourceTypeEnum2.WorkItemHistory = "work-item-history", ResourceTypeEnum2.Worklog = "worklog", ResourceTypeEnum2.Workspace = "workspace", ResourceTypeEnum2))(ResourceTypeEnum || {});
export {
  AdobeAdobexdRemoteLinkAri,
  AdobeSignDocumentAri,
  AhaAhaWorkItemAri,
  AirtableAirtableDocumentAri,
  AirtableAirtableRemoteLinkAri,
  AmplitudeAmplitudeRemoteLinkAri,
  AnalyticsUserAri,
  AnyAri,
  AppMigrationAppDataAri,
  ApprovalDeskDecisionTableAri,
  ApprovalDeskRoleAri,
  ApprovalDeskWorkflowAri,
  ApprovalDeskWorkspaceAri,
  Ari,
  Arm,
  AsanaAsanaRemoteLinkAri,
  AsanaAsanaWorkItemAri,
  Ati,
  AtlashubIdAri,
  AtlashubSiteAri,
  AtlassianWebCrawlerAtlassianWebCrawlerDocumentAri,
  AuditLogAttributeAri,
  AuditLogContainerAri,
  AuditLogContextAri,
  AuditLogEventAri,
  AuditLogWorkspaceAri,
  AutomationAutomationRuleAri,
  AutomationRoleAri,
  AvpControlAri,
  AvpDashboardAri,
  AvpDashboardChartAri,
  AvpDataShareAri,
  AvpDatasourceAri,
  AvpPipelineQueryAri,
  AvpRoleAri,
  AvpSiteAri,
  AvpSubscriptionAri,
  AvpWorkspaceAri,
  AzureDevopsAzureDevopsRemoteLinkAri,
  AzureDevopsAzureDevopsRepositoryAri,
  BeaconAlertAri,
  BeaconCustomDetectionAri,
  BeaconDetectionAri,
  BeaconDetectionSettingAri,
  BeaconRoleAri,
  BeaconSiteAri,
  BeaconSubscriptionAri,
  BeaconWorkspaceAri,
  BitbucketAppAri,
  BitbucketOauthConsumerAri,
  BitbucketProjectAri,
  BitbucketPullrequestAri,
  BitbucketRepositoryAri,
  BitbucketRepositoryTransferAri,
  BitbucketSiteAri,
  BitbucketTeamAri,
  BitbucketWorkspaceAri,
  BoxBoxDocumentAri,
  BoxBoxRemoteLinkAri,
  BrieRoleAri,
  BrieSiteAri,
  CanvaCanvaRemoteLinkAri,
  CanvasBoardAri,
  CanvasDatabaseAri,
  CanvasSiteAri,
  CanvasWorkspaceAri,
  CiaCprsEmailAri,
  CiscoWebexRemoteLinkAri,
  ClickupClickupRemoteLinkAri,
  CmdbAttributeAri,
  CmdbImportConfigurationAri,
  CmdbObjectAri,
  CmdbRoleAri,
  CmdbSchemaAri,
  CmdbTypeAri,
  CmdbWorkspaceAri,
  CommerceCatalogAccountAri,
  CommerceChangeAri,
  CommerceEntitlementAri,
  CommerceEntitlementTemplateAri,
  CommerceInitiativeAri,
  CommerceInvoiceGroupAri,
  CommerceOfferingAri,
  CommerceOfferingRelationshipTemplateAri,
  CommerceOrderAri,
  CommercePaymentMethodAri,
  CommercePricingPlanAri,
  CommerceProductAri,
  CommercePromotionAri,
  CommerceQuoteAri,
  CommerceRoleAri,
  CommerceShipToPartyAri,
  CommerceTransactionAccountAri,
  CommerceWorkbenchResourceAri,
  CompassApiEndpointAri,
  CompassAssistantAnswerAri,
  CompassCampaignAri,
  CompassComponentAri,
  CompassComponentLabelAri,
  CompassComponentLinkAri,
  CompassComponentTypeAri,
  CompassCustomFieldDefinitionAri,
  CompassDocumentAri,
  CompassDocumentationCategoryAri,
  CompassEventSourceAri,
  CompassIncomingWebhookAri,
  CompassLibraryScorecardAri,
  CompassMetricDefinitionAri,
  CompassMetricSourceAri,
  CompassPackageAri,
  CompassRoleAri,
  CompassScorecardAri,
  CompassSiteAri,
  CompassUserDefinedParameterAri,
  CompassWebhookAri,
  CompassWorkspaceAri,
  ConfluenceAdminAnnouncementBannerAri,
  ConfluenceAttachmentAri,
  ConfluenceBlogpostAri,
  ConfluenceCommentAri,
  ConfluenceContentAri,
  ConfluenceCustomContentAri,
  ConfluenceDatabaseAri,
  ConfluenceDcConfluenceDcCommentAri,
  ConfluenceDcConfluenceDcDocumentAri,
  ConfluenceDcConfluenceDcSpaceAri,
  ConfluenceEmbedAri,
  ConfluenceEventAri,
  ConfluenceFolderAri,
  ConfluenceLabelAri,
  ConfluenceLongRunningTaskAri,
  ConfluenceNoteAri,
  ConfluencePageAri,
  ConfluencePublicLinkAri,
  ConfluenceRoleAri,
  ConfluenceSiteAri,
  ConfluenceSpaceAri,
  ConfluenceTaskAri,
  ConfluenceTeamCalendarAri,
  ConfluenceUserPropertyAri,
  ConfluenceWhiteboardAri,
  ConfluenceWorkflowAri,
  ConfluenceWorkspaceAri,
  ConnectionRegistryLinkAri,
  ConversationalHelpConversationAri,
  ConversationalHelpConversationMessageAri,
  ConvoAiWorkspaceAri,
  CssXpRoleAri,
  DataLakeLakeConnectionAri,
  DevaiAutodevJobAri,
  DevaiWorkspaceAri,
  DevopsContainerAri,
  DevopsNamespaceAri,
  DevopsProviderAri,
  DevopsRelationshipAri,
  DevopsToolAri,
  DocusignDocusignDocumentAri,
  DocusignDocusignRemoteLinkAri,
  DovetailDovetailRemoteLinkAri,
  DropboxDocumentAri,
  DropboxDropboxDocumentAri,
  DropboxDropboxRemoteLinkAri,
  DropboxRemoteLinkAri,
  EcosystemAppAri,
  EcosystemAppEnvironmentVersionAri,
  EcosystemConnectAppAri,
  EcosystemConnectModuleAri,
  EcosystemEnvironmentAri,
  EcosystemExtensionAri,
  EcosystemExtensionGroupAri,
  EcosystemInstallationAri,
  EcosystemInstallationLinkAri,
  ElementsCustomEmojiMetadataAri,
  ElementsWorkspaceAri,
  ElevatePrecalibrationAri,
  ElevatePrecalibrationPillarAri,
  ElevateSelfAssessmentAri,
  ElevateSelfAssessmentPillarAri,
  ExusExternalUserAri,
  FigmaFigmaDesignAri,
  FigmaFigmaRemoteLinkAri,
  GithubGithubBranchAri,
  GithubGithubBuildAri,
  GithubGithubCommentAri,
  GithubGithubCommitAri,
  GithubGithubDeploymentAri,
  GithubGithubPullRequestAri,
  GithubGithubRemoteLinkAri,
  GithubGithubRepositoryAri,
  GithubGithubVulnerabilityAri,
  GitlabGitlabRemoteLinkAri,
  GitlabGitlabRepositoryAri,
  GoalGoalTypeAri,
  GoalMetricUpdateAri,
  GoalSiteAri,
  GoalWorkspaceAri,
  GoogleAccountAri,
  GoogleGmailMessageAri,
  GoogleGoogleCalendarCalendarEventAri,
  GoogleGoogleDriveDocumentAri,
  GoogleGoogleDriveRemoteLinkAri,
  GoogleGoogleDriveRslDocumentAri,
  GraphBranchAri,
  GraphBranchHistoryAri,
  GraphBuildAri,
  GraphBuildHistoryAri,
  GraphCalendarEventAri,
  GraphCalendarEventHistoryAri,
  GraphCommentAri,
  GraphCommentHistoryAri,
  GraphCommitAri,
  GraphContentAri,
  GraphConversationAri,
  GraphConversationChunkAri,
  GraphConversationHistoryAri,
  GraphCustomerOrgAri,
  GraphCustomerOrgHistoryAri,
  GraphDealAri,
  GraphDealHistoryAri,
  GraphDeploymentAri,
  GraphDeploymentHistoryAri,
  GraphDesignAri,
  GraphDesignHistoryAri,
  GraphDevopsComponentAri,
  GraphDocumentAri,
  GraphDocumentHistoryAri,
  GraphEmployeeAri,
  GraphEmployeeHistoryAri,
  GraphFeatureFlagAri,
  GraphFeatureFlagHistoryAri,
  GraphIncidentAri,
  GraphIncidentHistoryAri,
  GraphIntegrationContextAri,
  GraphJiraProjectAndDevopsToolRelationshipAri,
  GraphJiraProjectAndDocumentationSpaceRelationshipAri,
  GraphJiraProjectAndOperationsTeamRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipAri,
  GraphMessageAri,
  GraphOperationsWorkspaceAri,
  GraphOrganisationAri,
  GraphOrganisationHistoryAri,
  GraphPositionAri,
  GraphPositionHistoryAri,
  GraphPostIncidentReviewAri,
  GraphPostIncidentReviewHistoryAri,
  GraphProjectAri,
  GraphProjectHistoryAri,
  GraphPullRequestAri,
  GraphPullRequestHistoryAri,
  GraphRelationshipTypeAri,
  GraphRemoteLinkAri,
  GraphRemoteLinkHistoryAri,
  GraphRepositoryAri,
  GraphRoleAri,
  GraphSecurityContainerAri,
  GraphSecurityWorkspaceAri,
  GraphServiceAndJiraProjectRelationshipAri,
  GraphServiceAndOperationsTeamRelationshipAri,
  GraphServiceAndVcsRepositoryRelationshipAri,
  GraphServiceAri,
  GraphServiceRelationshipAri,
  GraphSpaceAri,
  GraphSpaceHistoryAri,
  GraphThreadChunkAri,
  GraphVideoAri,
  GraphVideoHistoryAri,
  GraphVulnerabilityAri,
  GraphVulnerabilityHistoryAri,
  GraphWorkItemAri,
  GraphWorkItemHistoryAri,
  GraphWorkerAri,
  GraphWorkerHistoryAri,
  GraphWorkspaceAri,
  HelpArticleAri,
  HelpChannelAri,
  HelpHelpCenterAri,
  HelpHelpDeskAri,
  HelpLayoutAri,
  HelpPageAri,
  HelpPortalAri,
  HelpRequestFormAri,
  HelpRoleAri,
  HomeWorkspaceAri,
  HubspotHubspotCustomerOrgAri,
  HubspotHubspotDealAri,
  HubspotHubspotRemoteLinkAri,
  IdentityAppAri,
  IdentityAppGrantAri,
  IdentityAuthPolicyAri,
  IdentityCustomerDirectoryAri,
  IdentityCustomerOrganizationAri,
  IdentityDirectoryAri,
  IdentityDomainAri,
  IdentityForgeInstallationAri,
  IdentityGroupAri,
  IdentityRoleAri,
  IdentityScopedGroupAri,
  IdentityTeamAri,
  IdentityTeamMemberAri,
  IdentityThirdPartyDirectoryAri,
  IdentityThirdPartyGroupAri,
  IdentityThirdPartyUserAri,
  IdentityUserAri,
  IdentityUserGrantAri,
  IdentityUserbaseAri,
  JiraAiCrocsSentenceAri,
  JiraAiCrocsWorkspaceAri,
  JiraAlignInstanceAri,
  JiraAlignProjectAri,
  JiraAlignWorkspaceAri,
  JiraAnnouncementBannerAri,
  JiraAvailableFieldsAri,
  JiraBranchAri,
  JiraBranchHistoryAri,
  JiraBuildAri,
  JiraBuildHistoryAri,
  JiraBulkOperationTaskAri,
  JiraCommitAri,
  JiraComponentAri,
  JiraConnectAppAri,
  JiraCoreRoleAri,
  JiraCoreSiteAri,
  JiraCustomerServiceRoleAri,
  JiraCustomerServiceSiteAri,
  JiraCustomerServiceTemplateFormAri,
  JiraDashboardAri,
  JiraDcJiraDcCommentAri,
  JiraDcJiraDcProjectAri,
  JiraDcJiraDcSpaceAri,
  JiraDcJiraDcWorkItemAri,
  JiraDcSecondaryJiraDcCommentAri,
  JiraDcSecondaryJiraDcProjectAri,
  JiraDcSecondaryJiraDcSpaceAri,
  JiraDcSecondaryJiraDcWorkItemAri,
  JiraDcTertiaryJiraDcCommentAri,
  JiraDcTertiaryJiraDcProjectAri,
  JiraDcTertiaryJiraDcSpaceAri,
  JiraDcTertiaryJiraDcWorkItemAri,
  JiraDeploymentAri,
  JiraDeploymentHistoryAri,
  JiraDesignAri,
  JiraDesignHistoryAri,
  JiraDevopsComponentAri,
  JiraDocumentAri,
  JiraDocumentHistoryAri,
  JiraEntitlementAri,
  JiraEntityPropertyAri,
  JiraFavouriteAri,
  JiraFeatureFlagAri,
  JiraFeatureFlagHistoryAri,
  JiraFieldConfigurationAri,
  JiraFieldConfigurationContextAri,
  JiraFieldSetViewAri,
  JiraFilterAri,
  JiraFilterEmailSubscriptionAri,
  JiraFormAri,
  JiraIncidentAri,
  JiraIncidentHistoryAri,
  JiraIssueAri,
  JiraIssueAttachmentAri,
  JiraIssueCommentAri,
  JiraIssueFieldMetadataAri,
  JiraIssueFieldOptionAri,
  JiraIssueHistoryAri,
  JiraIssueLinkAri,
  JiraIssueLinkTypeAri,
  JiraIssueRemoteLinkAri,
  JiraIssueSearchViewAri,
  JiraIssueSearchViewTypeAri,
  JiraIssueSecurityLevelPermissionAri,
  JiraIssueStatusAri,
  JiraIssueTypeAri,
  JiraIssueTypeSchemeAri,
  JiraIssueTypeSchemeMappingAri,
  JiraIssueWorklogAri,
  JiraIssuefieldvalueAri,
  JiraNavigationConfigAri,
  JiraNavigationItemAri,
  JiraNavigationMenuAri,
  JiraNavigationSettingsAri,
  JiraNotificationTypeAri,
  JiraNotificationTypeSchemeAri,
  JiraNotificationUserPreferenceAri,
  JiraOperationsWorkspaceAri,
  JiraPermissionSchemeAri,
  JiraPlanAri,
  JiraPlanCrossProjectScenarioVersionAri,
  JiraPlanCrossProjectVersionAri,
  JiraPlanScenarioAri,
  JiraPlanScenarioIssueAri,
  JiraPlanScenarioIssueFieldAri,
  JiraPlanScenarioVersionAri,
  JiraPlaybookAri,
  JiraPlaybookInstanceAri,
  JiraPlaybookInstanceStepAri,
  JiraPlaybookLabelAri,
  JiraPlaybookStepAri,
  JiraPlaybookStepRunAri,
  JiraPostIncidentReviewAri,
  JiraPostIncidentReviewHistoryAri,
  JiraPostIncidentReviewLinkAri,
  JiraPriorityAri,
  JiraProductAri,
  JiraProjectAri,
  JiraProjectCategoryAri,
  JiraProjectFeatureAri,
  JiraProjectOverviewAri,
  JiraProjectRoleActorAri,
  JiraProjectRoleByProjectAri,
  JiraProjectShortcutAri,
  JiraProjectTypeAri,
  JiraPullRequestAri,
  JiraPullRequestHistoryAri,
  JiraRemoteLinkAri,
  JiraRemoteLinkHistoryAri,
  JiraRepositoryAri,
  JiraResolutionAri,
  JiraResourceUsageMetricAri,
  JiraResourceUsageRecommendationAri,
  JiraRoleAri,
  JiraScreenAri,
  JiraScreenSchemeAri,
  JiraScreenTabAri,
  JiraSecurityContainerAri,
  JiraSecurityLevelAri,
  JiraSecurityWorkspaceAri,
  JiraServicedeskApprovalAri,
  JiraServicedeskCalendarEventAri,
  JiraServicedeskCannedResponseAri,
  JiraServicedeskOrganizationAri,
  JiraServicedeskOrganizationMemberAri,
  JiraServicedeskQueueAri,
  JiraServicedeskRequestTypeAri,
  JiraServicedeskRoleAri,
  JiraServicedeskSiteAri,
  JiraServicedeskSlaAri,
  JiraSiteAri,
  JiraSoftwareBoardAri,
  JiraSoftwareBoardFeatureAri,
  JiraSoftwareBoardIssueListAri,
  JiraSoftwareCardAri,
  JiraSoftwareCardColorAri,
  JiraSoftwareCardLayoutAri,
  JiraSoftwareCardParentAri,
  JiraSoftwareColumnAri,
  JiraSoftwareCustomFilterAri,
  JiraSoftwareRoleAri,
  JiraSoftwareSiteAri,
  JiraSoftwareSwimlaneAri,
  JiraSoftwareUserBoardConfigAri,
  JiraSoftwareViewConfigAri,
  JiraSprintAri,
  JiraUserBroadcastMessageAri,
  JiraVersionApproverAri,
  JiraVersionAri,
  JiraViewAri,
  JiraVulnerabilityAri,
  JiraVulnerabilityHistoryAri,
  JiraWorkflowAri,
  JiraWorkflowSchemeAri,
  JiraWorkspaceAri,
  JsmChannelOrchestratorConversationAri,
  KnowledgeDiscoveryBookmarkAri,
  KnowledgeServingAndAccessTopicAri,
  LaunchdarklyLaunchdarklyRemoteLinkAri,
  LearningPlatformLearnerAri,
  LinkingPlatformDatasourceAri,
  LoginAlerterSiteAri,
  LoomCommentAri,
  LoomFolderAri,
  LoomLoomVideoAri,
  LoomMeetingAri,
  LoomMeetingRecurrenceAri,
  LoomSiteAri,
  LoomSpaceAri,
  LoomStandaloneWorkspaceAri,
  LoomVideoAri,
  LoomWorkspaceAri,
  LucidLucidchartDesignAri,
  LucidLucidchartRemoteLinkAri,
  MakerSpaceGroupAri,
  MapiRoleAri,
  MarketingCustomerDomainAri,
  MarketplaceTeamAri,
  MeasurementEmailUuidAri,
  MeasurementSiteUserAri,
  MeasurementUserAri,
  MediaFileAri,
  MercuryChangeAri,
  MercuryChangeProposalAri,
  MercuryChangeProposalStatusAri,
  MercuryCommentAri,
  MercuryFocusAreaAri,
  MercuryFocusAreaStatusUpdateAri,
  MercuryFocusAreaTypeAri,
  MercuryRoleAri,
  MercurySiteAri,
  MercuryStrategicEventAri,
  MercuryStrategicEventStatusAri,
  MercuryViewAri,
  MercuryWorkspaceAri,
  MicrosoftAccountAri,
  MicrosoftOnedriveDocumentAri,
  MicrosoftOutlookCalendarCalendarEventAri,
  MicrosoftOutlookMessageAri,
  MicrosoftPowerBiRemoteLinkAri,
  MicrosoftSharepointDocumentAri,
  MicrosoftTeamsConversationAri,
  MicrosoftTeamsMessageAri,
  MicrosoftTeamsRemoteLinkAri,
  MigrationReportCenterReportRequestProxyAri,
  MiroMiroDesignAri,
  MiroMiroRemoteLinkAri,
  MlRegistryArtifactAri,
  MondayMondayDocumentAri,
  MuralMuralRemoteLinkAri,
  NotionNotionDocumentAri,
  NotionNotionRemoteLinkAri,
  OauthClientAri,
  OpsgenieAccountLoginAri,
  OpsgenieAccountSettingsAri,
  OpsgenieAlertAri,
  OpsgenieAlertRecipientLinkAri,
  OpsgenieApiRequestMetricAri,
  OpsgenieCallRoutingAri,
  OpsgenieCustomRoleAri,
  OpsgenieDeploymentAri,
  OpsgenieDeploymentServiceLinkAri,
  OpsgenieEscalationAri,
  OpsgenieEventAri,
  OpsgenieIncidentAlertLinkAri,
  OpsgenieIncidentAri,
  OpsgenieIncidentStatusUpdateAri,
  OpsgenieIncomingCallHistoryAri,
  OpsgenieIntegrationAri,
  OpsgenieNotificationAri,
  OpsgenieRoleAri,
  OpsgenieSavedSearchAri,
  OpsgenieScheduleAri,
  OpsgenieScheduleRotationAri,
  OpsgenieSiteAri,
  OpsgenieTeamAri,
  OpsgenieTeamMemberAri,
  OpsgenieTimelineAri,
  OpsgenieWorkspaceAri,
  OrgPolicyPolicyAri,
  PagerdutyPagerdutyRemoteLinkAri,
  PapiApiAri,
  PapiRoleAri,
  PassionfruitAskActivityAri,
  PassionfruitAskAri,
  PassionfruitAskCommentAri,
  PassionfruitAskLinkAri,
  PassionfruitAskUpdateAri,
  PassionfruitDependencyAri,
  PassionfruitRoleAri,
  PassionfruitSiteAri,
  PassionfruitWorkspaceAri,
  PeopleDataCostCenterAri,
  PeopleDataJobProfileAri,
  PeopleDataPositionAri,
  PeopleDataSupervisoryOrgAri,
  PeopleDataWorkerAri,
  PeoplePerftoolCalibrationAri,
  PeoplePerftoolFeedbackAri,
  PeoplePerftoolPerformanceAri,
  PipedrivePipedriveRemoteLinkAri,
  PlatformClassificationTagAri,
  PlatformCollaborationContextAri,
  PlatformIntegrationAri,
  PlatformLifecycleResourceAri,
  PlatformLifecycleResourcePackageAri,
  PlatformLifecycleResourcePackageTypeAri,
  PlatformOrgAri,
  PlatformOrgUserAri,
  PlatformProductAri,
  PlatformSecureTunnelAri,
  PlatformServicesStreamhubSchemaAri,
  PlatformSiteAri,
  PlatformTestProvisioningEntityAri,
  PollinatorCheckAri,
  PostOfficeMessageInstanceAri,
  PostOfficeMessageTemplateAri,
  PostOfficeTriggerAri,
  ProformaHelpcenterFormAri,
  ProformaIssueFormAri,
  ProformaProjectFormAri,
  ProformaRoleAri,
  ProjectSiteAri,
  ProjectWorkspaceAri,
  PurePermissionAri,
  PureResourceAri,
  PureRoleAri,
  RadarPositionAri,
  RadarSiteAri,
  RadarWorkerAri,
  RadarWorkspaceAri,
  ResourceOwnerEnum as ResourceOwner,
  ResourceTypeEnum as ResourceType,
  RovoAgentAri,
  RovoCustomActionAri,
  RovoFileAri,
  RovoMessageAri,
  RovoScenarioAri,
  RovoSiteAri,
  RovoWorkspaceAri,
  RuntimeAuthClientRoleAri,
  SalesforceSalesforceCustomerOrgAri,
  SalesforceSalesforceDealAri,
  SalesforceSalesforceRemoteLinkAri,
  SandboxServiceSiteAri,
  SearchEventAri,
  SearchForwardDocumentAri,
  SearchForwardDocumentSubscriptionAri,
  SearchPermissionContainerAri,
  SentrySentryRemoteLinkAri,
  ServicenowServicenowWorkItemAri,
  SettingsServiceSettingsAri,
  SlackSlackConversationAri,
  SlackSlackMessageAri,
  SmartsheetSmartsheetDocumentAri,
  SmartsheetSmartsheetRemoteLinkAri,
  SpinnakerSpinnakerDeploymentAri,
  StatuspageRoleAri,
  StatuspageSiteAri,
  StripeStripeRemoteLinkAri,
  SupportCustomerAri,
  SupportCustomerSupportAri,
  TodoistTodoistRemoteLinkAri,
  TownsquareCommentAri,
  TownsquareGoalAri,
  TownsquareGoalUpdateAri,
  TownsquareHelpPointerAri,
  TownsquareLearningAri,
  TownsquareMetricAri,
  TownsquareMetricTargetAri,
  TownsquareMetricValueAri,
  TownsquareMilestoneAri,
  TownsquareProjectAri,
  TownsquareProjectUpdateAri,
  TownsquareQuestionAri,
  TownsquareRoleAri,
  TownsquareSiteAri,
  TownsquareTagAri,
  TownsquareUpdateNoteAri,
  TownsquareWorkspaceAri,
  TrelloAttachmentAri,
  TrelloAvatarAri,
  TrelloBackgroundAri,
  TrelloBoardAri,
  TrelloCardAri,
  TrelloCheckItemAri,
  TrelloChecklistAri,
  TrelloCustomFieldAri,
  TrelloEmojiAri,
  TrelloEnterpriseAri,
  TrelloLabelAri,
  TrelloListAri,
  TrelloLogoAri,
  TrelloPlannerAri,
  TrelloPlannerCalendarAri,
  TrelloPlannerEventCardAri,
  TrelloRoleAri,
  TrelloSiteAri,
  TrelloStickerAri,
  TrelloUserAri,
  TrelloWorkspaceAri,
  TrustEngEmailAri,
  UnifiedHelpRoleAri,
  UnifiedHelpSiteAri,
  UnifiedHelpWorkspaceAri,
  UnifiedPortableProfileLinkedAccountAri,
  UnifiedPortableProfileParentAccountAri,
  UnifiedPortableProfileUnifiedAccountAri,
  UnifiedPortableProfileUnifiedProfileAri,
  UsageTrackingUsageAri,
  ValidationError,
  VirtualAgentConfigurationAri,
  VirtualAgentConversationAri,
  VirtualAgentFlowEditorAri,
  VirtualAgentIntentAri,
  VirtualAgentIntentQuestionAri,
  VirtualAgentIntentTemplateAri,
  WorkatoJobAri,
  WorkatoRecipeAri,
  WorkdayCostCenterAri,
  WorkdayJobProfileAri,
  WorkdayPositionAri,
  WorkdayStaffAri,
  WorkdaySupervisoryOrgAri,
  WorkdayWorkdayOrganisationAri,
  WorkdayWorkdayPositionAri,
  WorkdayWorkdayWorkerAri,
  WorkdayWorkerAri,
  ZendeskZendeskDocumentAri,
  ZendeskZendeskWorkItemAri,
  ZeplinZeplinRemoteLinkAri
};
