// @forge/lint/out/lint/linters/handler-linter/verifiers/handlers-verifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HandlersVerifier = void 0;
const linter_interface_1 = require("../../../linter-interface");
const verifier_interface_1 = require("../../verifier-interface");
const text_1 = require("../../../text");
class HandlersVerifier extends verifier_interface_1.BaseLintIssueVerifier {
    getLintClass() {
        return linter_interface_1.LintClass.Warning;
    }
    async process(input) {
        if (!this.manifest?.modules?.function) {
            return [];
        }
        const missingMethods = input.expectedMethods.filter((func) => !input.exportedMethods.find((exportedMethod) => exportedMethod.method === func.method));
        return missingMethods.map((func) => ({
            class: this.getLintClass(),
            message: text_1.messages.verifiers.handler.message(func.method, func.key),
            reference: text_1.messages.verifiers.handler.reference,
            column: 0,
            line: 0
        }));
    }
}
exports.HandlersVerifier = HandlersVerifier;
