// @forge/cli/out/analytics-client/analytics-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalyticsClientReporter = void 0;
const tslib_1 = require("tslib");
const os = tslib_1.__importStar(require("os"));
const url_1 = require("url");
const uuid_1 = require("uuid");
const analytics_node_client_1 = require("@forge/util/packages/analytics-node-client");
const ari_1 = require("@forge/util/packages/ari");
const cli_shared_1 = require("@forge/cli-shared");
const anon_user_id_1 = require("../command-line/anon-user-id");
const errors_1 = require("../command-line/errors");
const unique_machine_id_1 = require("../command-line/unique-machine-id");
const analytics_message_handler_1 = require("./analytics-message-handler");
const Identity = (value) => value;
function appAidOrUndefined(appAri) {
    try {
        return ari_1.EcosystemAppAri.parse(appAri).appId;
    }
    catch (e) {
        return undefined;
    }
}
class AnalyticsClientReporter {
    storage;
    configService;
    credentialStore;
    constructor(storage, configService, credentialStore) {
        this.storage = storage;
        this.configService = configService;
        this.credentialStore = credentialStore;
    }
    source = 'forge/cli';
    async reportSuccess(cmdName, attributes) {
        this.processAnalyticsEvent({
            id: (0, uuid_1.v4)(),
            eventType: analytics_message_handler_1.EventType.TRACK,
            event: {
                timestamp: new Date(),
                os: {
                    name: os.platform(),
                    version: os.release()
                },
                ...(await this._getUserId()),
                trackEvent: {
                    source: this.source,
                    action: 'invoked',
                    actionSubject: cmdName,
                    attributes: this._getAttributes(attributes),
                    containers: this._getContainer(attributes)
                }
            }
        });
    }
    async reportFailure(cmdName, attributes, e) {
        attributes = {
            ...attributes,
            ...this._getErrorDetails(e)
        };
        await this.reportOperationalEvent('invoked', cmdName, attributes);
    }
    async reportCommandInvoke(cmdName, attributes) {
        await this.reportOperationalEvent('cmdInvoked', cmdName, attributes);
    }
    async reportInvokeFailure(cmdName, attributes, e) {
        if (e) {
            attributes = { ...attributes, ...this._getErrorListDetails(e) };
        }
        await this.reportOperationalEvent('cmdInvokeFailed', cmdName, attributes);
    }
    async reportAppPackaged(attributes) {
        await this.reportOperationalEvent('packaged', 'app', attributes);
    }
    async reportDevSpaceCreated(attributes) {
        await this.reportTrackEvent('created', 'developerSpace', {
            ...attributes
        });
    }
    async reportDevSpaceRetryAttempt(attributes) {
        await this.reportTrackEvent('createRetryAttempted', 'developerSpace', {
            ...attributes
        });
    }
    async reportTrackEvent(action, actionSubject, attributes) {
        this.processAnalyticsEvent({
            id: (0, uuid_1.v4)(),
            eventType: analytics_message_handler_1.EventType.TRACK,
            event: {
                timestamp: new Date(),
                os: {
                    name: os.platform(),
                    version: os.release()
                },
                ...(await this._getUserId()),
                trackEvent: {
                    source: this.source,
                    action: action,
                    actionSubject: actionSubject,
                    attributes: this._getAttributes(attributes),
                    containers: this._getContainer(attributes)
                }
            }
        });
    }
    async reportOperationalEvent(action, actionSubject, attributes) {
        this.processAnalyticsEvent({
            id: (0, uuid_1.v4)(),
            eventType: analytics_message_handler_1.EventType.OPERATIONAL,
            event: {
                timestamp: new Date(),
                os: {
                    name: os.platform(),
                    version: os.release()
                },
                ...(await this._getUserId()),
                operationalEvent: {
                    source: this.source,
                    action,
                    actionSubject,
                    attributes: {
                        ...this._getAttributes(attributes)
                    },
                    containers: this._getContainer(attributes)
                }
            }
        });
    }
    processAnalyticsEvent(analyticsEvent) {
        if (this.configService.getAnalyticsPreferences() === false) {
            return;
        }
        this.storage.addAnalyticsEvent(analyticsEvent);
    }
    async _getUserId() {
        try {
            const { accountId: userId } = await this.credentialStore.getCredentials();
            return {
                userIdType: analytics_node_client_1.userTypes.ATLASSIAN_ACCOUNT,
                userId
            };
        }
        catch {
            return {
                anonymousId: (0, anon_user_id_1.getAnonId)()
            };
        }
    }
    _getErrorDetails(e) {
        if (e instanceof cli_shared_1.GraphQlMutationError) {
            return { error: e.getCode() };
        }
        return { error: e.constructor.name };
    }
    _getErrorListDetails(e) {
        const errorList = [];
        const errors = e instanceof errors_1.DeferredErrors ? e.getErrors() : [e];
        errors.forEach((error) => {
            if (error instanceof cli_shared_1.GraphQlMutationError) {
                errorList.push(error.getCode());
            }
            else {
                errorList.push(error.constructor.name);
            }
        });
        return { error: errorList.join(',') };
    }
    _asContainer(attrValue, containerType, type, mapper = Identity) {
        const mappedValue = mapper(attrValue);
        return mappedValue
            ? {
                [containerType]: {
                    id: mappedValue,
                    type
                }
            }
            : undefined;
    }
    _nodeVersion() {
        return { node: process.version };
    }
    _machineId() {
        return { machineId: (0, unique_machine_id_1.getMachineId)() };
    }
    _asAttribute(attrValue, name, mapper = Identity) {
        const mappedValue = mapper(attrValue);
        return mappedValue ? { [name]: mappedValue } : {};
    }
    _getAttributes(attributes) {
        return {
            ...attributes,
            ...this._asAttribute(attributes.appId, 'appId', appAidOrUndefined),
            ...this._nodeVersion(),
            ...this._machineId()
        };
    }
    _getContainer(attributes) {
        return {
            ...this._asContainer(attributes.appId, 'app', 'appId', appAidOrUndefined),
            ...this._asContainer(attributes.appEnv, 'appEnv', 'environment'),
            ...this._asContainer(attributes.product, 'product', 'product')
        };
    }
    static formatValue(value) {
        return value instanceof url_1.URL
            ? value.href.toString()
            : typeof value === 'string'
                ? value
                : value === undefined
                    ? 'undefined'
                    : JSON.stringify(value);
    }
}
exports.AnalyticsClientReporter = AnalyticsClientReporter;
