// @forge/util/packages/ari/graph/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);

// src/graph/index.ts
var graph_exports = {};
__export(graph_exports, {
  GraphBranchAri: () => GraphBranchAri,
  GraphBranchHistoryAri: () => GraphBranchHistoryAri,
  GraphBuildAri: () => GraphBuildAri,
  GraphBuildHistoryAri: () => GraphBuildHistoryAri,
  GraphCalendarEventAri: () => GraphCalendarEventAri,
  GraphCalendarEventHistoryAri: () => GraphCalendarEventHistoryAri,
  GraphCommentAri: () => GraphCommentAri,
  GraphCommentHistoryAri: () => GraphCommentHistoryAri,
  GraphCommitAri: () => GraphCommitAri,
  GraphContentAri: () => GraphContentAri,
  GraphConversationAri: () => GraphConversationAri,
  GraphConversationChunkAri: () => GraphConversationChunkAri,
  GraphConversationHistoryAri: () => GraphConversationHistoryAri,
  GraphCustomerOrgAri: () => GraphCustomerOrgAri,
  GraphCustomerOrgHistoryAri: () => GraphCustomerOrgHistoryAri,
  GraphDealAri: () => GraphDealAri,
  GraphDealHistoryAri: () => GraphDealHistoryAri,
  GraphDeploymentAri: () => GraphDeploymentAri,
  GraphDeploymentHistoryAri: () => GraphDeploymentHistoryAri,
  GraphDesignAri: () => GraphDesignAri,
  GraphDesignHistoryAri: () => GraphDesignHistoryAri,
  GraphDevopsComponentAri: () => GraphDevopsComponentAri,
  GraphDocumentAri: () => GraphDocumentAri,
  GraphDocumentHistoryAri: () => GraphDocumentHistoryAri,
  GraphEmployeeAri: () => GraphEmployeeAri,
  GraphEmployeeHistoryAri: () => GraphEmployeeHistoryAri,
  GraphFeatureFlagAri: () => GraphFeatureFlagAri,
  GraphFeatureFlagHistoryAri: () => GraphFeatureFlagHistoryAri,
  GraphIncidentAri: () => GraphIncidentAri,
  GraphIncidentHistoryAri: () => GraphIncidentHistoryAri,
  GraphIntegrationContextAri: () => GraphIntegrationContextAri,
  GraphJiraProjectAndDevopsToolRelationshipAri: () => GraphJiraProjectAndDevopsToolRelationshipAri,
  GraphJiraProjectAndDocumentationSpaceRelationshipAri: () => GraphJiraProjectAndDocumentationSpaceRelationshipAri,
  GraphJiraProjectAndOperationsTeamRelationshipAri: () => GraphJiraProjectAndOperationsTeamRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipAri: () => GraphJiraProjectAndVcsRepositoryRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri: () => GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri,
  GraphMessageAri: () => GraphMessageAri,
  GraphOperationsWorkspaceAri: () => GraphOperationsWorkspaceAri,
  GraphOrganisationAri: () => GraphOrganisationAri,
  GraphOrganisationHistoryAri: () => GraphOrganisationHistoryAri,
  GraphPositionAri: () => GraphPositionAri,
  GraphPositionHistoryAri: () => GraphPositionHistoryAri,
  GraphPostIncidentReviewAri: () => GraphPostIncidentReviewAri,
  GraphPostIncidentReviewHistoryAri: () => GraphPostIncidentReviewHistoryAri,
  GraphProjectAri: () => GraphProjectAri,
  GraphProjectHistoryAri: () => GraphProjectHistoryAri,
  GraphPullRequestAri: () => GraphPullRequestAri,
  GraphPullRequestHistoryAri: () => GraphPullRequestHistoryAri,
  GraphRelationshipTypeAri: () => GraphRelationshipTypeAri,
  GraphRemoteLinkAri: () => GraphRemoteLinkAri,
  GraphRemoteLinkHistoryAri: () => GraphRemoteLinkHistoryAri,
  GraphRepositoryAri: () => GraphRepositoryAri,
  GraphRoleAri: () => GraphRoleAri,
  GraphSecurityContainerAri: () => GraphSecurityContainerAri,
  GraphSecurityWorkspaceAri: () => GraphSecurityWorkspaceAri,
  GraphServiceAndJiraProjectRelationshipAri: () => GraphServiceAndJiraProjectRelationshipAri,
  GraphServiceAndOperationsTeamRelationshipAri: () => GraphServiceAndOperationsTeamRelationshipAri,
  GraphServiceAndVcsRepositoryRelationshipAri: () => GraphServiceAndVcsRepositoryRelationshipAri,
  GraphServiceAri: () => GraphServiceAri,
  GraphServiceRelationshipAri: () => GraphServiceRelationshipAri,
  GraphSpaceAri: () => GraphSpaceAri,
  GraphSpaceHistoryAri: () => GraphSpaceHistoryAri,
  GraphThreadChunkAri: () => GraphThreadChunkAri,
  GraphVideoAri: () => GraphVideoAri,
  GraphVideoHistoryAri: () => GraphVideoHistoryAri,
  GraphVulnerabilityAri: () => GraphVulnerabilityAri,
  GraphVulnerabilityHistoryAri: () => GraphVulnerabilityHistoryAri,
  GraphWorkItemAri: () => GraphWorkItemAri,
  GraphWorkItemHistoryAri: () => GraphWorkItemHistoryAri,
  GraphWorkerAri: () => GraphWorkerAri,
  GraphWorkerHistoryAri: () => GraphWorkerHistoryAri,
  GraphWorkspaceAri: () => GraphWorkspaceAri
});
module.exports = __toCommonJS(graph_exports);

// src/errors.ts
var ValidationError = class extends Error {
};

// src/core/rules/cloud-id.ts
function validateCloudId(cloudId, format = new RegExp("^[a-zA-Z0-9_\\-.]{0,255}$")) {
  if (!cloudId.match(format))
    throw new ValidationError(`Invalid cloud ID, expected ID of format ${format}.`);
}

// src/core/rules/platform-qualifier.ts
function validatePlatformQualifier(qualifier) {
  if (qualifier !== "cloud" && qualifier !== "third-party")
    throw new ValidationError(`Identifier must have a qualifier of 'cloud' or 'third-party'. Received: ${qualifier}`);
}

// src/core/rules/qualifier.ts
function validateAtiQualifier(qualifier) {
  if (qualifier !== "ati")
    throw new ValidationError(`ATI must have a qualifier of 'ati'. Received: ${qualifier}`);
}
function validateAriQualifier(qualifier) {
  if (qualifier !== "ari")
    throw new ValidationError(`ARI must have a qualifier of 'ari'. Received: ${qualifier}`);
}

// src/core/rules/resource-id.ts
function validateResourceId(id, format = new RegExp("[!a-zA-Z0-9\\-_.~@:;{}=]+(/[!a-zA-Z0-9\\-_.~@:;{}=]+)*" /* ANY_RESOURCE_ID */), key = "resourceId") {
  let formatWithCarets = new RegExp(`^${format.source}$`);
  if (!id.match(formatWithCarets))
    throw new ValidationError(`Invalid ${key} - ${id}, expected ID of format ${formatWithCarets}.`);
}

// src/core/rules/resource-id-segments.ts
function validateResourceIdSegments(resourceIdSegmentValues, resourceIdSegmentFormats) {
  Object.entries(resourceIdSegmentValues).forEach(([resourceIdKey, resourceIdValue]) => {
    validateResourceId(resourceIdValue, resourceIdSegmentFormats == null ? void 0 : resourceIdSegmentFormats[resourceIdKey], resourceIdKey);
  });
}

// src/core/rules/resource-owner.ts
function validateResourceOwner(owner, expectedResourceOwner) {
  if (expectedResourceOwner && owner !== expectedResourceOwner)
    throw new ValidationError(`Invalid resource owner - ${owner}, expected ${expectedResourceOwner}.`);
}

// src/core/rules/resource-type.ts
function validateResourceType(type, expectedResourceType) {
  if (expectedResourceType && type !== expectedResourceType)
    throw new ValidationError(`Invalid resource type - ${type}, expected ${expectedResourceType}.`);
}

// src/core/parser/base.ts
var NUMBER_OF_BASE_SEGMENTS = 5, SEGMENT_SEPARATOR = ":", BaseParser = class {
  static getIdentifierSegments(inputStr, numberOfSegmentsRequired) {
    let allSegments = inputStr.split(SEGMENT_SEPARATOR);
    if (allSegments.length < numberOfSegmentsRequired)
      throw new ValidationError(`Input string must have ${numberOfSegmentsRequired} segments.`);
    if (allSegments.length > numberOfSegmentsRequired) {
      let segments = allSegments.slice(0, NUMBER_OF_BASE_SEGMENTS - 1), segmentsForResourceId = allSegments.slice(NUMBER_OF_BASE_SEGMENTS - 1, allSegments.length);
      return [...segments, segmentsForResourceId.join(SEGMENT_SEPARATOR)];
    }
    return allSegments;
  }
};

// src/core/parser/ari.ts
var NUMBER_OF_ARI_SEGMENTS = 5, AriParser = class extends BaseParser {
  static fromString(maybeAri, ariProperties) {
    let segments = this.getIdentifierSegments(maybeAri, NUMBER_OF_ARI_SEGMENTS), [qualifier, platformQualifier, resourceOwner, cloudId, resourceTypeAndId] = segments, [resourceType, ...resourceIdSegments] = resourceTypeAndId.split("/"), maybeAriDerivedProperties = {
      qualifier,
      platformQualifier,
      cloudId,
      resourceOwner,
      resourceType,
      resourceId: resourceIdSegments.join("/"),
      resourceIdSegmentValues: ariProperties != null && ariProperties.resourceIdSlug ? this.ariSegmentValuesFromSlug(
        ariProperties.resourceIdSlug,
        resourceIdSegments.join("/"),
        Object.keys(ariProperties.resourceIdSegmentFormats)
      ) : {}
    };
    return this.fromOpts(maybeAriDerivedProperties, ariProperties);
  }
  static fromOpts(maybeAriOpts, ariProperties) {
    let {
      qualifier = "ari",
      platformQualifier = "cloud",
      resourceOwner,
      cloudId,
      resourceId,
      resourceIdSegmentValues,
      resourceType
    } = maybeAriOpts;
    return validateAriQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateCloudId(maybeAriOpts.cloudId || "", ariProperties == null ? void 0 : ariProperties.cloudId), validateResourceOwner(resourceOwner, ariProperties == null ? void 0 : ariProperties.resourceOwner), validateResourceType(resourceType, ariProperties == null ? void 0 : ariProperties.resourceType), validateResourceIdSegments(resourceIdSegmentValues, ariProperties == null ? void 0 : ariProperties.resourceIdSegmentFormats), {
      qualifier,
      resourceOwner,
      resourceType,
      platformQualifier,
      cloudId,
      resourceId,
      resourceIdSegmentValues
    };
  }
  static ariSegmentValuesFromSlug(slug, incomingSegments, expectedKeys) {
    let regexpResult = new RegExp("^" + slug.replace(/\{(.*?)\}/g, "(?<$1>.*?)") + "$").exec(incomingSegments);
    if (!regexpResult)
      throw new ValidationError(`Segment '${incomingSegments}' don't match expected slug: ${slug}`);
    return expectedKeys.forEach((expectedKey) => {
      var _a;
      if (!((_a = regexpResult.groups) != null && _a[expectedKey]))
        throw new ValidationError(`No value supplied for '${expectedKey}' based on slug ${slug}`);
    }), regexpResult.groups || {};
  }
};

// src/core/parser/ati.ts
var NUMBER_OF_ATI_SEGMENTS = 4, AtiParser = class extends BaseParser {
  static fromString(inputStr, atiOpts) {
    let segments = this.getIdentifierSegments(inputStr, NUMBER_OF_ATI_SEGMENTS), [qualifier = "ati", platformQualifier = "cloud", resourceOwner, resourceType] = segments;
    return validateAtiQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateResourceOwner(resourceOwner, atiOpts == null ? void 0 : atiOpts.resourceOwner), validateResourceType(resourceType, atiOpts == null ? void 0 : atiOpts.resourceType), { platformQualifier, resourceOwner, resourceType };
  }
  static fromOpts(maybeAtiOpts, atiOpts) {
    let { qualifier, platformQualifier, resourceOwner, resourceType } = maybeAtiOpts;
    return validateAtiQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateResourceOwner(resourceOwner, atiOpts == null ? void 0 : atiOpts.resourceOwner), validateResourceType(resourceType, atiOpts == null ? void 0 : atiOpts.resourceType), {
      platformQualifier,
      resourceOwner,
      resourceType
    };
  }
};

// src/ati.ts
var Ati = class _Ati {
  constructor(platformQualifier, resourceOwner, resourceType) {
    this._platformQualifier = platformQualifier, this._resourceOwner = resourceOwner, this._resourceType = resourceType;
  }
  get platformQualifier() {
    return this._platformQualifier;
  }
  get resourceOwner() {
    return this._resourceOwner;
  }
  get resourceType() {
    return this._resourceType;
  }
  static create(opts) {
    return new _Ati(opts.platformQualifier || "cloud", opts.resourceOwner, opts.resourceType);
  }
  static parse(maybeAti, atiOpts) {
    let opts = AtiParser.fromString(maybeAti, atiOpts);
    return new _Ati(opts.platformQualifier, opts.resourceOwner, opts.resourceType);
  }
  toString() {
    return `ati:${this.platformQualifier}:${this.resourceOwner}:${this.resourceType}`;
  }
  toJSON() {
    return this.toString();
  }
  toOpts() {
    return {
      platformQualifier: this._platformQualifier,
      resourceOwner: this._resourceOwner,
      resourceType: this._resourceType
    };
  }
};

// src/core/ari.ts
var Ari = class {
  constructor(opts) {
    this._ati = Ati.create({
      platformQualifier: opts.platformQualifier || "cloud",
      resourceOwner: opts.resourceOwner,
      resourceType: opts.resourceType
    }), this._cloudId = opts.cloudId !== "" ? opts.cloudId : void 0, this._resourceId = opts.resourceId;
  }
  get platformQualifier() {
    return this._ati.platformQualifier;
  }
  get cloudId() {
    return this._cloudId;
  }
  get resourceOwner() {
    return this._ati.resourceOwner;
  }
  get resourceType() {
    return this._ati.resourceType;
  }
  get resourceId() {
    return this._resourceId;
  }
  get ati() {
    return this._ati;
  }
  equals(other) {
    return this.toString() === other.toString();
  }
  toString() {
    return `ari:${this.platformQualifier}:${this.resourceOwner}:${this.cloudId || ""}:${this.resourceType}/${this.resourceId}`;
  }
  toJSON() {
    return this.toString();
  }
  toOpts() {
    return {
      platformQualifier: this.platformQualifier,
      resourceOwner: this.resourceOwner,
      cloudId: this.cloudId,
      resourceType: this.resourceType,
      resourceId: this.resourceId
    };
  }
};

// src/any-ari.ts
var AnyAri = class _AnyAri extends Ari {
  constructor(opts) {
    super(opts);
  }
  static create(ariOpts, ariStaticOpts) {
    let ariOptsWithDefaults = { ...ariOpts, resourceIdSegmentValues: {} }, validatedOpts = AriParser.fromOpts(ariOptsWithDefaults, ariStaticOpts);
    return new _AnyAri(validatedOpts);
  }
  static parse(maybeAri, ariStaticOpts) {
    let validatedOpts = AriParser.fromString(maybeAri, ariStaticOpts);
    return new _AnyAri(validatedOpts);
  }
  static check(maybeAri) {
    try {
      return _AnyAri.parse(maybeAri.toString()), !0;
    } catch (err) {
      return !1;
    }
  }
  asAnyAri() {
    return this;
  }
};

// src/core/registered-ari.ts
var RegisteredAri = class extends Ari {
  static check(maybeAri) {
    try {
      return this.parse(maybeAri.toString()), !0;
    } catch (err) {
      return !1;
    }
  }
  asAnyAri() {
    return AnyAri.parse(this.toString());
  }
};

// src/graph/branch/types.ts
var GraphBranchAriResourceOwner = "graph", GraphBranchAriResourceType = "branch";

// src/graph/branch/manifest.ts
var graphBranchAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphBranchAriResourceOwner,
  resourceType: GraphBranchAriResourceType,
  resourceIdSlug: "activation/{activationId}/{branchId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    branchId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/branch/index.ts
var GraphBranchAri = class _GraphBranchAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._branchId = opts.resourceIdSegmentValues.branchId;
  }
  get activationId() {
    return this._activationId;
  }
  get branchId() {
    return this._branchId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphBranchAriStaticOpts.qualifier,
      platformQualifier: graphBranchAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphBranchAriStaticOpts.resourceOwner,
      resourceType: graphBranchAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.branchId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        branchId: opts.branchId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphBranchAriStaticOpts);
    return new _GraphBranchAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphBranchAriStaticOpts);
    return new _GraphBranchAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      branchId: this.branchId
    };
  }
};

// src/graph/branch-history/types.ts
var GraphBranchHistoryAriResourceOwner = "graph", GraphBranchHistoryAriResourceType = "branch-history";

// src/graph/branch-history/manifest.ts
var graphBranchHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphBranchHistoryAriResourceOwner,
  resourceType: GraphBranchHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{branchId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    branchId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/branch-history/index.ts
var GraphBranchHistoryAri = class _GraphBranchHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._branchId = opts.resourceIdSegmentValues.branchId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get branchId() {
    return this._branchId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphBranchHistoryAriStaticOpts.qualifier,
      platformQualifier: graphBranchHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphBranchHistoryAriStaticOpts.resourceOwner,
      resourceType: graphBranchHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.branchId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        branchId: opts.branchId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphBranchHistoryAriStaticOpts);
    return new _GraphBranchHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphBranchHistoryAriStaticOpts);
    return new _GraphBranchHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      branchId: this.branchId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/build/types.ts
var GraphBuildAriResourceOwner = "graph", GraphBuildAriResourceType = "build";

// src/graph/build/manifest.ts
var graphBuildAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphBuildAriResourceOwner,
  resourceType: GraphBuildAriResourceType,
  resourceIdSlug: "activation/{activationId}/{buildId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    buildId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/build/index.ts
var GraphBuildAri = class _GraphBuildAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._buildId = opts.resourceIdSegmentValues.buildId;
  }
  get activationId() {
    return this._activationId;
  }
  get buildId() {
    return this._buildId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphBuildAriStaticOpts.qualifier,
      platformQualifier: graphBuildAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphBuildAriStaticOpts.resourceOwner,
      resourceType: graphBuildAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.buildId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        buildId: opts.buildId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphBuildAriStaticOpts);
    return new _GraphBuildAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphBuildAriStaticOpts);
    return new _GraphBuildAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      buildId: this.buildId
    };
  }
};

// src/graph/build-history/types.ts
var GraphBuildHistoryAriResourceOwner = "graph", GraphBuildHistoryAriResourceType = "build-history";

// src/graph/build-history/manifest.ts
var graphBuildHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphBuildHistoryAriResourceOwner,
  resourceType: GraphBuildHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{buildId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    buildId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/build-history/index.ts
var GraphBuildHistoryAri = class _GraphBuildHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._buildId = opts.resourceIdSegmentValues.buildId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get buildId() {
    return this._buildId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphBuildHistoryAriStaticOpts.qualifier,
      platformQualifier: graphBuildHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphBuildHistoryAriStaticOpts.resourceOwner,
      resourceType: graphBuildHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.buildId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        buildId: opts.buildId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphBuildHistoryAriStaticOpts);
    return new _GraphBuildHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphBuildHistoryAriStaticOpts);
    return new _GraphBuildHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      buildId: this.buildId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/calendar-event/types.ts
var GraphCalendarEventAriResourceOwner = "graph", GraphCalendarEventAriResourceType = "calendar-event";

// src/graph/calendar-event/manifest.ts
var graphCalendarEventAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCalendarEventAriResourceOwner,
  resourceType: GraphCalendarEventAriResourceType,
  resourceIdSlug: "activation/{activationId}/{calendarEventId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    calendarEventId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/calendar-event/index.ts
var GraphCalendarEventAri = class _GraphCalendarEventAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._calendarEventId = opts.resourceIdSegmentValues.calendarEventId;
  }
  get activationId() {
    return this._activationId;
  }
  get calendarEventId() {
    return this._calendarEventId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCalendarEventAriStaticOpts.qualifier,
      platformQualifier: graphCalendarEventAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCalendarEventAriStaticOpts.resourceOwner,
      resourceType: graphCalendarEventAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.calendarEventId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        calendarEventId: opts.calendarEventId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCalendarEventAriStaticOpts);
    return new _GraphCalendarEventAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCalendarEventAriStaticOpts);
    return new _GraphCalendarEventAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      calendarEventId: this.calendarEventId
    };
  }
};

// src/graph/calendar-event-history/types.ts
var GraphCalendarEventHistoryAriResourceOwner = "graph", GraphCalendarEventHistoryAriResourceType = "calendar-event-history";

// src/graph/calendar-event-history/manifest.ts
var graphCalendarEventHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCalendarEventHistoryAriResourceOwner,
  resourceType: GraphCalendarEventHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{calendarEventId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    calendarEventId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/calendar-event-history/index.ts
var GraphCalendarEventHistoryAri = class _GraphCalendarEventHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._calendarEventId = opts.resourceIdSegmentValues.calendarEventId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get calendarEventId() {
    return this._calendarEventId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCalendarEventHistoryAriStaticOpts.qualifier,
      platformQualifier: graphCalendarEventHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCalendarEventHistoryAriStaticOpts.resourceOwner,
      resourceType: graphCalendarEventHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.calendarEventId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        calendarEventId: opts.calendarEventId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCalendarEventHistoryAriStaticOpts);
    return new _GraphCalendarEventHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCalendarEventHistoryAriStaticOpts);
    return new _GraphCalendarEventHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      calendarEventId: this.calendarEventId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/comment/types.ts
var GraphCommentAriResourceOwner = "graph", GraphCommentAriResourceType = "comment";

// src/graph/comment/manifest.ts
var graphCommentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCommentAriResourceOwner,
  resourceType: GraphCommentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/comment/index.ts
var GraphCommentAri = class _GraphCommentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCommentAriStaticOpts.qualifier,
      platformQualifier: graphCommentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCommentAriStaticOpts.resourceOwner,
      resourceType: graphCommentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCommentAriStaticOpts);
    return new _GraphCommentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCommentAriStaticOpts);
    return new _GraphCommentAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID
    };
  }
};

// src/graph/comment-history/types.ts
var GraphCommentHistoryAriResourceOwner = "graph", GraphCommentHistoryAriResourceType = "comment-history";

// src/graph/comment-history/manifest.ts
var graphCommentHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCommentHistoryAriResourceOwner,
  resourceType: GraphCommentHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/comment-history/index.ts
var GraphCommentHistoryAri = class _GraphCommentHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCommentHistoryAriStaticOpts.qualifier,
      platformQualifier: graphCommentHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCommentHistoryAriStaticOpts.resourceOwner,
      resourceType: graphCommentHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCommentHistoryAriStaticOpts);
    return new _GraphCommentHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCommentHistoryAriStaticOpts);
    return new _GraphCommentHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/commit/types.ts
var GraphCommitAriResourceOwner = "graph", GraphCommitAriResourceType = "commit";

// src/graph/commit/manifest.ts
var graphCommitAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCommitAriResourceOwner,
  resourceType: GraphCommitAriResourceType,
  resourceIdSlug: "activation/{activationId}/{commitId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    commitId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/commit/index.ts
var GraphCommitAri = class _GraphCommitAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._commitId = opts.resourceIdSegmentValues.commitId;
  }
  get activationId() {
    return this._activationId;
  }
  get commitId() {
    return this._commitId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCommitAriStaticOpts.qualifier,
      platformQualifier: graphCommitAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCommitAriStaticOpts.resourceOwner,
      resourceType: graphCommitAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.commitId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        commitId: opts.commitId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCommitAriStaticOpts);
    return new _GraphCommitAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCommitAriStaticOpts);
    return new _GraphCommitAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      commitId: this.commitId
    };
  }
};

// src/graph/content/types.ts
var GraphContentAriResourceOwner = "graph", GraphContentAriResourceType = "content";

// src/graph/content/manifest.ts
var graphContentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphContentAriResourceOwner,
  resourceType: GraphContentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{contentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    contentId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/content/index.ts
var GraphContentAri = class _GraphContentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._contentId = opts.resourceIdSegmentValues.contentId;
  }
  get activationId() {
    return this._activationId;
  }
  get contentId() {
    return this._contentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphContentAriStaticOpts.qualifier,
      platformQualifier: graphContentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphContentAriStaticOpts.resourceOwner,
      resourceType: graphContentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.contentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        contentId: opts.contentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphContentAriStaticOpts);
    return new _GraphContentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphContentAriStaticOpts);
    return new _GraphContentAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      contentId: this.contentId
    };
  }
};

// src/graph/conversation/types.ts
var GraphConversationAriResourceOwner = "graph", GraphConversationAriResourceType = "conversation";

// src/graph/conversation/manifest.ts
var graphConversationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphConversationAriResourceOwner,
  resourceType: GraphConversationAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/conversation/index.ts
var GraphConversationAri = class _GraphConversationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphConversationAriStaticOpts.qualifier,
      platformQualifier: graphConversationAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphConversationAriStaticOpts.resourceOwner,
      resourceType: graphConversationAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphConversationAriStaticOpts);
    return new _GraphConversationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphConversationAriStaticOpts);
    return new _GraphConversationAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID
    };
  }
};

// src/graph/conversation-chunk/types.ts
var GraphConversationChunkAriResourceOwner = "graph", GraphConversationChunkAriResourceType = "conversation-chunk";

// src/graph/conversation-chunk/manifest.ts
var graphConversationChunkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphConversationChunkAriResourceOwner,
  resourceType: GraphConversationChunkAriResourceType,
  resourceIdSlug: "activation/{activationId}/{chunkId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    chunkId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/conversation-chunk/index.ts
var GraphConversationChunkAri = class _GraphConversationChunkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._chunkId = opts.resourceIdSegmentValues.chunkId;
  }
  get activationId() {
    return this._activationId;
  }
  get chunkId() {
    return this._chunkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphConversationChunkAriStaticOpts.qualifier,
      platformQualifier: graphConversationChunkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphConversationChunkAriStaticOpts.resourceOwner,
      resourceType: graphConversationChunkAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.chunkId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        chunkId: opts.chunkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphConversationChunkAriStaticOpts);
    return new _GraphConversationChunkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphConversationChunkAriStaticOpts);
    return new _GraphConversationChunkAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      chunkId: this.chunkId
    };
  }
};

// src/graph/conversation-history/types.ts
var GraphConversationHistoryAriResourceOwner = "graph", GraphConversationHistoryAriResourceType = "conversation-history";

// src/graph/conversation-history/manifest.ts
var graphConversationHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphConversationHistoryAriResourceOwner,
  resourceType: GraphConversationHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/conversation-history/index.ts
var GraphConversationHistoryAri = class _GraphConversationHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphConversationHistoryAriStaticOpts.qualifier,
      platformQualifier: graphConversationHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphConversationHistoryAriStaticOpts.resourceOwner,
      resourceType: graphConversationHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphConversationHistoryAriStaticOpts);
    return new _GraphConversationHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphConversationHistoryAriStaticOpts);
    return new _GraphConversationHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/customer-org/types.ts
var GraphCustomerOrgAriResourceOwner = "graph", GraphCustomerOrgAriResourceType = "customer-org";

// src/graph/customer-org/manifest.ts
var graphCustomerOrgAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCustomerOrgAriResourceOwner,
  resourceType: GraphCustomerOrgAriResourceType,
  resourceIdSlug: "activation/{activationId}/{customerOrgId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    customerOrgId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/customer-org/index.ts
var GraphCustomerOrgAri = class _GraphCustomerOrgAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._customerOrgId = opts.resourceIdSegmentValues.customerOrgId;
  }
  get activationId() {
    return this._activationId;
  }
  get customerOrgId() {
    return this._customerOrgId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCustomerOrgAriStaticOpts.qualifier,
      platformQualifier: graphCustomerOrgAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCustomerOrgAriStaticOpts.resourceOwner,
      resourceType: graphCustomerOrgAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.customerOrgId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        customerOrgId: opts.customerOrgId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCustomerOrgAriStaticOpts);
    return new _GraphCustomerOrgAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCustomerOrgAriStaticOpts);
    return new _GraphCustomerOrgAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      customerOrgId: this.customerOrgId
    };
  }
};

// src/graph/customer-org-history/types.ts
var GraphCustomerOrgHistoryAriResourceOwner = "graph", GraphCustomerOrgHistoryAriResourceType = "customer-org-history";

// src/graph/customer-org-history/manifest.ts
var graphCustomerOrgHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphCustomerOrgHistoryAriResourceOwner,
  resourceType: GraphCustomerOrgHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{customerOrgId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    customerOrgId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/customer-org-history/index.ts
var GraphCustomerOrgHistoryAri = class _GraphCustomerOrgHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._customerOrgId = opts.resourceIdSegmentValues.customerOrgId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get customerOrgId() {
    return this._customerOrgId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphCustomerOrgHistoryAriStaticOpts.qualifier,
      platformQualifier: graphCustomerOrgHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphCustomerOrgHistoryAriStaticOpts.resourceOwner,
      resourceType: graphCustomerOrgHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.customerOrgId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        customerOrgId: opts.customerOrgId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphCustomerOrgHistoryAriStaticOpts);
    return new _GraphCustomerOrgHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphCustomerOrgHistoryAriStaticOpts);
    return new _GraphCustomerOrgHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      customerOrgId: this.customerOrgId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/deal/types.ts
var GraphDealAriResourceOwner = "graph", GraphDealAriResourceType = "deal";

// src/graph/deal/manifest.ts
var graphDealAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDealAriResourceOwner,
  resourceType: GraphDealAriResourceType,
  resourceIdSlug: "activation/{activationId}/{dealId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    dealId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/deal/index.ts
var GraphDealAri = class _GraphDealAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._dealId = opts.resourceIdSegmentValues.dealId;
  }
  get activationId() {
    return this._activationId;
  }
  get dealId() {
    return this._dealId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDealAriStaticOpts.qualifier,
      platformQualifier: graphDealAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDealAriStaticOpts.resourceOwner,
      resourceType: graphDealAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.dealId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        dealId: opts.dealId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDealAriStaticOpts);
    return new _GraphDealAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDealAriStaticOpts);
    return new _GraphDealAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      dealId: this.dealId
    };
  }
};

// src/graph/deal-history/types.ts
var GraphDealHistoryAriResourceOwner = "graph", GraphDealHistoryAriResourceType = "deal-history";

// src/graph/deal-history/manifest.ts
var graphDealHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDealHistoryAriResourceOwner,
  resourceType: GraphDealHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{dealId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    dealId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/deal-history/index.ts
var GraphDealHistoryAri = class _GraphDealHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._dealId = opts.resourceIdSegmentValues.dealId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get dealId() {
    return this._dealId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDealHistoryAriStaticOpts.qualifier,
      platformQualifier: graphDealHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDealHistoryAriStaticOpts.resourceOwner,
      resourceType: graphDealHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.dealId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        dealId: opts.dealId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDealHistoryAriStaticOpts);
    return new _GraphDealHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDealHistoryAriStaticOpts);
    return new _GraphDealHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      dealId: this.dealId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/deployment/types.ts
var GraphDeploymentAriResourceOwner = "graph", GraphDeploymentAriResourceType = "deployment";

// src/graph/deployment/manifest.ts
var graphDeploymentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDeploymentAriResourceOwner,
  resourceType: GraphDeploymentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{deploymentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    deploymentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/deployment/index.ts
var GraphDeploymentAri = class _GraphDeploymentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._deploymentId = opts.resourceIdSegmentValues.deploymentId;
  }
  get activationId() {
    return this._activationId;
  }
  get deploymentId() {
    return this._deploymentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDeploymentAriStaticOpts.qualifier,
      platformQualifier: graphDeploymentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDeploymentAriStaticOpts.resourceOwner,
      resourceType: graphDeploymentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.deploymentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        deploymentId: opts.deploymentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDeploymentAriStaticOpts);
    return new _GraphDeploymentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDeploymentAriStaticOpts);
    return new _GraphDeploymentAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      deploymentId: this.deploymentId
    };
  }
};

// src/graph/deployment-history/types.ts
var GraphDeploymentHistoryAriResourceOwner = "graph", GraphDeploymentHistoryAriResourceType = "deployment-history";

// src/graph/deployment-history/manifest.ts
var graphDeploymentHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDeploymentHistoryAriResourceOwner,
  resourceType: GraphDeploymentHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{deploymentId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    deploymentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/deployment-history/index.ts
var GraphDeploymentHistoryAri = class _GraphDeploymentHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._deploymentId = opts.resourceIdSegmentValues.deploymentId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get deploymentId() {
    return this._deploymentId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDeploymentHistoryAriStaticOpts.qualifier,
      platformQualifier: graphDeploymentHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDeploymentHistoryAriStaticOpts.resourceOwner,
      resourceType: graphDeploymentHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.deploymentId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        deploymentId: opts.deploymentId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDeploymentHistoryAriStaticOpts);
    return new _GraphDeploymentHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDeploymentHistoryAriStaticOpts);
    return new _GraphDeploymentHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      deploymentId: this.deploymentId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/design/types.ts
var GraphDesignAriResourceOwner = "graph", GraphDesignAriResourceType = "design";

// src/graph/design/manifest.ts
var graphDesignAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDesignAriResourceOwner,
  resourceType: GraphDesignAriResourceType,
  resourceIdSlug: "activation/{activationId}/{designId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    designId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/design/index.ts
var GraphDesignAri = class _GraphDesignAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._designId = opts.resourceIdSegmentValues.designId;
  }
  get activationId() {
    return this._activationId;
  }
  get designId() {
    return this._designId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDesignAriStaticOpts.qualifier,
      platformQualifier: graphDesignAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDesignAriStaticOpts.resourceOwner,
      resourceType: graphDesignAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.designId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        designId: opts.designId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDesignAriStaticOpts);
    return new _GraphDesignAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDesignAriStaticOpts);
    return new _GraphDesignAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      designId: this.designId
    };
  }
};

// src/graph/design-history/types.ts
var GraphDesignHistoryAriResourceOwner = "graph", GraphDesignHistoryAriResourceType = "design-history";

// src/graph/design-history/manifest.ts
var graphDesignHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDesignHistoryAriResourceOwner,
  resourceType: GraphDesignHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{designId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    designId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/design-history/index.ts
var GraphDesignHistoryAri = class _GraphDesignHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._designId = opts.resourceIdSegmentValues.designId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get designId() {
    return this._designId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDesignHistoryAriStaticOpts.qualifier,
      platformQualifier: graphDesignHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDesignHistoryAriStaticOpts.resourceOwner,
      resourceType: graphDesignHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.designId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        designId: opts.designId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDesignHistoryAriStaticOpts);
    return new _GraphDesignHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDesignHistoryAriStaticOpts);
    return new _GraphDesignHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      designId: this.designId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/devops-component/types.ts
var GraphDevopsComponentAriResourceOwner = "graph", GraphDevopsComponentAriResourceType = "devops-component";

// src/graph/devops-component/manifest.ts
var graphDevopsComponentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDevopsComponentAriResourceOwner,
  resourceType: GraphDevopsComponentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{devopsComponentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    devopsComponentId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/devops-component/index.ts
var GraphDevopsComponentAri = class _GraphDevopsComponentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._devopsComponentId = opts.resourceIdSegmentValues.devopsComponentId;
  }
  get activationId() {
    return this._activationId;
  }
  get devopsComponentId() {
    return this._devopsComponentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDevopsComponentAriStaticOpts.qualifier,
      platformQualifier: graphDevopsComponentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDevopsComponentAriStaticOpts.resourceOwner,
      resourceType: graphDevopsComponentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.devopsComponentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        devopsComponentId: opts.devopsComponentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDevopsComponentAriStaticOpts);
    return new _GraphDevopsComponentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDevopsComponentAriStaticOpts);
    return new _GraphDevopsComponentAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      devopsComponentId: this.devopsComponentId
    };
  }
};

// src/graph/document/types.ts
var GraphDocumentAriResourceOwner = "graph", GraphDocumentAriResourceType = "document";

// src/graph/document/manifest.ts
var graphDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDocumentAriResourceOwner,
  resourceType: GraphDocumentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{documentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    documentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/document/index.ts
var GraphDocumentAri = class _GraphDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._documentId = opts.resourceIdSegmentValues.documentId;
  }
  get activationId() {
    return this._activationId;
  }
  get documentId() {
    return this._documentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDocumentAriStaticOpts.qualifier,
      platformQualifier: graphDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDocumentAriStaticOpts.resourceOwner,
      resourceType: graphDocumentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.documentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        documentId: opts.documentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDocumentAriStaticOpts);
    return new _GraphDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDocumentAriStaticOpts);
    return new _GraphDocumentAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      documentId: this.documentId
    };
  }
};

// src/graph/document-history/types.ts
var GraphDocumentHistoryAriResourceOwner = "graph", GraphDocumentHistoryAriResourceType = "document-history";

// src/graph/document-history/manifest.ts
var graphDocumentHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphDocumentHistoryAriResourceOwner,
  resourceType: GraphDocumentHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{documentId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    documentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/document-history/index.ts
var GraphDocumentHistoryAri = class _GraphDocumentHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._documentId = opts.resourceIdSegmentValues.documentId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get documentId() {
    return this._documentId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphDocumentHistoryAriStaticOpts.qualifier,
      platformQualifier: graphDocumentHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphDocumentHistoryAriStaticOpts.resourceOwner,
      resourceType: graphDocumentHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.documentId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        documentId: opts.documentId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphDocumentHistoryAriStaticOpts);
    return new _GraphDocumentHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphDocumentHistoryAriStaticOpts);
    return new _GraphDocumentHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      documentId: this.documentId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/employee/types.ts
var GraphEmployeeAriResourceOwner = "graph", GraphEmployeeAriResourceType = "employee";

// src/graph/employee/manifest.ts
var graphEmployeeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphEmployeeAriResourceOwner,
  resourceType: GraphEmployeeAriResourceType,
  resourceIdSlug: "activation/{activationId}/{employeeId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    employeeId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/employee/index.ts
var GraphEmployeeAri = class _GraphEmployeeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._employeeId = opts.resourceIdSegmentValues.employeeId;
  }
  get activationId() {
    return this._activationId;
  }
  get employeeId() {
    return this._employeeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphEmployeeAriStaticOpts.qualifier,
      platformQualifier: graphEmployeeAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphEmployeeAriStaticOpts.resourceOwner,
      resourceType: graphEmployeeAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.employeeId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        employeeId: opts.employeeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphEmployeeAriStaticOpts);
    return new _GraphEmployeeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphEmployeeAriStaticOpts);
    return new _GraphEmployeeAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      employeeId: this.employeeId
    };
  }
};

// src/graph/employee-history/types.ts
var GraphEmployeeHistoryAriResourceOwner = "graph", GraphEmployeeHistoryAriResourceType = "employee-history";

// src/graph/employee-history/manifest.ts
var graphEmployeeHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphEmployeeHistoryAriResourceOwner,
  resourceType: GraphEmployeeHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{employeeId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    employeeId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/employee-history/index.ts
var GraphEmployeeHistoryAri = class _GraphEmployeeHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._employeeId = opts.resourceIdSegmentValues.employeeId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get employeeId() {
    return this._employeeId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphEmployeeHistoryAriStaticOpts.qualifier,
      platformQualifier: graphEmployeeHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphEmployeeHistoryAriStaticOpts.resourceOwner,
      resourceType: graphEmployeeHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.employeeId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        employeeId: opts.employeeId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphEmployeeHistoryAriStaticOpts);
    return new _GraphEmployeeHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphEmployeeHistoryAriStaticOpts);
    return new _GraphEmployeeHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      employeeId: this.employeeId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/feature-flag/types.ts
var GraphFeatureFlagAriResourceOwner = "graph", GraphFeatureFlagAriResourceType = "feature-flag";

// src/graph/feature-flag/manifest.ts
var graphFeatureFlagAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphFeatureFlagAriResourceOwner,
  resourceType: GraphFeatureFlagAriResourceType,
  resourceIdSlug: "activation/{activationId}/{featureFlagId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    featureFlagId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/feature-flag/index.ts
var GraphFeatureFlagAri = class _GraphFeatureFlagAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._featureFlagId = opts.resourceIdSegmentValues.featureFlagId;
  }
  get activationId() {
    return this._activationId;
  }
  get featureFlagId() {
    return this._featureFlagId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphFeatureFlagAriStaticOpts.qualifier,
      platformQualifier: graphFeatureFlagAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphFeatureFlagAriStaticOpts.resourceOwner,
      resourceType: graphFeatureFlagAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.featureFlagId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        featureFlagId: opts.featureFlagId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphFeatureFlagAriStaticOpts);
    return new _GraphFeatureFlagAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphFeatureFlagAriStaticOpts);
    return new _GraphFeatureFlagAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      featureFlagId: this.featureFlagId
    };
  }
};

// src/graph/feature-flag-history/types.ts
var GraphFeatureFlagHistoryAriResourceOwner = "graph", GraphFeatureFlagHistoryAriResourceType = "feature-flag-history";

// src/graph/feature-flag-history/manifest.ts
var graphFeatureFlagHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphFeatureFlagHistoryAriResourceOwner,
  resourceType: GraphFeatureFlagHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{featureFlagId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    featureFlagId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/feature-flag-history/index.ts
var GraphFeatureFlagHistoryAri = class _GraphFeatureFlagHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._featureFlagId = opts.resourceIdSegmentValues.featureFlagId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get featureFlagId() {
    return this._featureFlagId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphFeatureFlagHistoryAriStaticOpts.qualifier,
      platformQualifier: graphFeatureFlagHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphFeatureFlagHistoryAriStaticOpts.resourceOwner,
      resourceType: graphFeatureFlagHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.featureFlagId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        featureFlagId: opts.featureFlagId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphFeatureFlagHistoryAriStaticOpts);
    return new _GraphFeatureFlagHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphFeatureFlagHistoryAriStaticOpts);
    return new _GraphFeatureFlagHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      featureFlagId: this.featureFlagId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/incident/types.ts
var GraphIncidentAriResourceOwner = "graph", GraphIncidentAriResourceType = "incident";

// src/graph/incident/manifest.ts
var graphIncidentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphIncidentAriResourceOwner,
  resourceType: GraphIncidentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{incidentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    incidentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/incident/index.ts
var GraphIncidentAri = class _GraphIncidentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._incidentId = opts.resourceIdSegmentValues.incidentId;
  }
  get activationId() {
    return this._activationId;
  }
  get incidentId() {
    return this._incidentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphIncidentAriStaticOpts.qualifier,
      platformQualifier: graphIncidentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphIncidentAriStaticOpts.resourceOwner,
      resourceType: graphIncidentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.incidentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        incidentId: opts.incidentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphIncidentAriStaticOpts);
    return new _GraphIncidentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphIncidentAriStaticOpts);
    return new _GraphIncidentAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      incidentId: this.incidentId
    };
  }
};

// src/graph/incident-history/types.ts
var GraphIncidentHistoryAriResourceOwner = "graph", GraphIncidentHistoryAriResourceType = "incident-history";

// src/graph/incident-history/manifest.ts
var graphIncidentHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphIncidentHistoryAriResourceOwner,
  resourceType: GraphIncidentHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{incidentId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    incidentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/incident-history/index.ts
var GraphIncidentHistoryAri = class _GraphIncidentHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._incidentId = opts.resourceIdSegmentValues.incidentId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get incidentId() {
    return this._incidentId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphIncidentHistoryAriStaticOpts.qualifier,
      platformQualifier: graphIncidentHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphIncidentHistoryAriStaticOpts.resourceOwner,
      resourceType: graphIncidentHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.incidentId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        incidentId: opts.incidentId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphIncidentHistoryAriStaticOpts);
    return new _GraphIncidentHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphIncidentHistoryAriStaticOpts);
    return new _GraphIncidentHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      incidentId: this.incidentId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/integration-context/types.ts
var GraphIntegrationContextAriResourceOwner = "graph", GraphIntegrationContextAriResourceType = "integration-context";

// src/graph/integration-context/manifest.ts
var graphIntegrationContextAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphIntegrationContextAriResourceOwner,
  resourceType: GraphIntegrationContextAriResourceType,
  resourceIdSlug: "{integrationContextId}",
  resourceIdSegmentFormats: {
    integrationContextId: /(ecda99d9-9b42-4bf7-8b4f-ecb5fcf5868c|3d1b1176-a2e3-4233-82bd-29a9cfc80cab|161675af-ebe2-4e6b-b5fd-e3fdf9f5ad84)/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/integration-context/index.ts
var GraphIntegrationContextAri = class _GraphIntegrationContextAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._integrationContextId = opts.resourceIdSegmentValues.integrationContextId;
  }
  get integrationContextId() {
    return this._integrationContextId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphIntegrationContextAriStaticOpts.qualifier,
      platformQualifier: graphIntegrationContextAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphIntegrationContextAriStaticOpts.resourceOwner,
      resourceType: graphIntegrationContextAriStaticOpts.resourceType,
      resourceId: `${opts.integrationContextId}`,
      resourceIdSegmentValues: {
        integrationContextId: opts.integrationContextId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphIntegrationContextAriStaticOpts);
    return new _GraphIntegrationContextAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphIntegrationContextAriStaticOpts);
    return new _GraphIntegrationContextAri(opts);
  }
  getVariables() {
    return {
      integrationContextId: this.integrationContextId
    };
  }
};

// src/graph/jira-project-and-devops-tool-relationship/types.ts
var GraphJiraProjectAndDevopsToolRelationshipAriResourceOwner = "graph", GraphJiraProjectAndDevopsToolRelationshipAriResourceType = "jira-project-and-devops-tool-relationship";

// src/graph/jira-project-and-devops-tool-relationship/manifest.ts
var graphJiraProjectAndDevopsToolRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphJiraProjectAndDevopsToolRelationshipAriResourceOwner,
  resourceType: GraphJiraProjectAndDevopsToolRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/jira-project-and-devops-tool-relationship/index.ts
var GraphJiraProjectAndDevopsToolRelationshipAri = class _GraphJiraProjectAndDevopsToolRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphJiraProjectAndDevopsToolRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphJiraProjectAndDevopsToolRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphJiraProjectAndDevopsToolRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphJiraProjectAndDevopsToolRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphJiraProjectAndDevopsToolRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndDevopsToolRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphJiraProjectAndDevopsToolRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndDevopsToolRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/jira-project-and-documentation-space-relationship/types.ts
var GraphJiraProjectAndDocumentationSpaceRelationshipAriResourceOwner = "graph", GraphJiraProjectAndDocumentationSpaceRelationshipAriResourceType = "jira-project-and-documentation-space-relationship";

// src/graph/jira-project-and-documentation-space-relationship/manifest.ts
var graphJiraProjectAndDocumentationSpaceRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphJiraProjectAndDocumentationSpaceRelationshipAriResourceOwner,
  resourceType: GraphJiraProjectAndDocumentationSpaceRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/jira-project-and-documentation-space-relationship/index.ts
var GraphJiraProjectAndDocumentationSpaceRelationshipAri = class _GraphJiraProjectAndDocumentationSpaceRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphJiraProjectAndDocumentationSpaceRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphJiraProjectAndDocumentationSpaceRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphJiraProjectAndDocumentationSpaceRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphJiraProjectAndDocumentationSpaceRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphJiraProjectAndDocumentationSpaceRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndDocumentationSpaceRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphJiraProjectAndDocumentationSpaceRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndDocumentationSpaceRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/jira-project-and-operations-team-relationship/types.ts
var GraphJiraProjectAndOperationsTeamRelationshipAriResourceOwner = "graph", GraphJiraProjectAndOperationsTeamRelationshipAriResourceType = "jira-project-and-operations-team-relationship";

// src/graph/jira-project-and-operations-team-relationship/manifest.ts
var graphJiraProjectAndOperationsTeamRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphJiraProjectAndOperationsTeamRelationshipAriResourceOwner,
  resourceType: GraphJiraProjectAndOperationsTeamRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/jira-project-and-operations-team-relationship/index.ts
var GraphJiraProjectAndOperationsTeamRelationshipAri = class _GraphJiraProjectAndOperationsTeamRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphJiraProjectAndOperationsTeamRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphJiraProjectAndOperationsTeamRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphJiraProjectAndOperationsTeamRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphJiraProjectAndOperationsTeamRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphJiraProjectAndOperationsTeamRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndOperationsTeamRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphJiraProjectAndOperationsTeamRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndOperationsTeamRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/jira-project-and-vcs-repository-relationship/types.ts
var GraphJiraProjectAndVcsRepositoryRelationshipAriResourceOwner = "graph", GraphJiraProjectAndVcsRepositoryRelationshipAriResourceType = "jira-project-and-vcs-repository-relationship";

// src/graph/jira-project-and-vcs-repository-relationship/manifest.ts
var graphJiraProjectAndVcsRepositoryRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphJiraProjectAndVcsRepositoryRelationshipAriResourceOwner,
  resourceType: GraphJiraProjectAndVcsRepositoryRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/jira-project-and-vcs-repository-relationship/index.ts
var GraphJiraProjectAndVcsRepositoryRelationshipAri = class _GraphJiraProjectAndVcsRepositoryRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphJiraProjectAndVcsRepositoryRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphJiraProjectAndVcsRepositoryRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphJiraProjectAndVcsRepositoryRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphJiraProjectAndVcsRepositoryRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphJiraProjectAndVcsRepositoryRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndVcsRepositoryRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphJiraProjectAndVcsRepositoryRelationshipAriStaticOpts);
    return new _GraphJiraProjectAndVcsRepositoryRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/jira-project-and-vcs-repository-relationship-temp-migration/types.ts
var GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriResourceOwner = "graph", GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriResourceType = "jira-project-and-vcs-repository-relationship-temp-migration";

// src/graph/jira-project-and-vcs-repository-relationship-temp-migration/manifest.ts
var graphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriResourceOwner,
  resourceType: GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriResourceType,
  resourceIdSlug: "{siteId}/{projectId}/{activationId}/{repositoryId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9_\-.]{1,255}/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9]+/,
    // eslint-disable-line no-useless-escape
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    repositoryId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/jira-project-and-vcs-repository-relationship-temp-migration/index.ts
var GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri = class _GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId, this._projectId = opts.resourceIdSegmentValues.projectId, this._activationId = opts.resourceIdSegmentValues.activationId, this._repositoryId = opts.resourceIdSegmentValues.repositoryId;
  }
  get siteId() {
    return this._siteId;
  }
  get projectId() {
    return this._projectId;
  }
  get activationId() {
    return this._activationId;
  }
  get repositoryId() {
    return this._repositoryId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriStaticOpts.qualifier,
      platformQualifier: graphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriStaticOpts.resourceOwner,
      resourceType: graphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}/${opts.projectId}/${opts.activationId}/${opts.repositoryId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId,
        projectId: opts.projectId,
        activationId: opts.activationId,
        repositoryId: opts.repositoryId
      }
    }, ariOpts = AriParser.fromOpts(
      derivedOpts,
      graphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriStaticOpts
    );
    return new _GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphJiraProjectAndVcsRepositoryRelationshipTempMigrationAriStaticOpts);
    return new _GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      projectId: this.projectId,
      activationId: this.activationId,
      repositoryId: this.repositoryId
    };
  }
};

// src/graph/message/types.ts
var GraphMessageAriResourceOwner = "graph", GraphMessageAriResourceType = "message";

// src/graph/message/manifest.ts
var graphMessageAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphMessageAriResourceOwner,
  resourceType: GraphMessageAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/message/index.ts
var GraphMessageAri = class _GraphMessageAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphMessageAriStaticOpts.qualifier,
      platformQualifier: graphMessageAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphMessageAriStaticOpts.resourceOwner,
      resourceType: graphMessageAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphMessageAriStaticOpts);
    return new _GraphMessageAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphMessageAriStaticOpts);
    return new _GraphMessageAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID
    };
  }
};

// src/graph/operations-workspace/types.ts
var GraphOperationsWorkspaceAriResourceOwner = "graph", GraphOperationsWorkspaceAriResourceType = "operations-workspace";

// src/graph/operations-workspace/manifest.ts
var graphOperationsWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphOperationsWorkspaceAriResourceOwner,
  resourceType: GraphOperationsWorkspaceAriResourceType,
  resourceIdSlug: "activation/{activationId}/{providerAppId}/{operationsWorkspaceId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    providerAppId: /[a-zA-Z0-9-._]+/,
    // eslint-disable-line no-useless-escape
    operationsWorkspaceId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/operations-workspace/index.ts
var GraphOperationsWorkspaceAri = class _GraphOperationsWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._providerAppId = opts.resourceIdSegmentValues.providerAppId, this._operationsWorkspaceId = opts.resourceIdSegmentValues.operationsWorkspaceId;
  }
  get activationId() {
    return this._activationId;
  }
  get providerAppId() {
    return this._providerAppId;
  }
  get operationsWorkspaceId() {
    return this._operationsWorkspaceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphOperationsWorkspaceAriStaticOpts.qualifier,
      platformQualifier: graphOperationsWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphOperationsWorkspaceAriStaticOpts.resourceOwner,
      resourceType: graphOperationsWorkspaceAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.providerAppId}/${opts.operationsWorkspaceId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        providerAppId: opts.providerAppId,
        operationsWorkspaceId: opts.operationsWorkspaceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphOperationsWorkspaceAriStaticOpts);
    return new _GraphOperationsWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphOperationsWorkspaceAriStaticOpts);
    return new _GraphOperationsWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      providerAppId: this.providerAppId,
      operationsWorkspaceId: this.operationsWorkspaceId
    };
  }
};

// src/graph/organisation/types.ts
var GraphOrganisationAriResourceOwner = "graph", GraphOrganisationAriResourceType = "organisation";

// src/graph/organisation/manifest.ts
var graphOrganisationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphOrganisationAriResourceOwner,
  resourceType: GraphOrganisationAriResourceType,
  resourceIdSlug: "activation/{activationId}/{organisationUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    organisationUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/organisation/index.ts
var GraphOrganisationAri = class _GraphOrganisationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._organisationUUID = opts.resourceIdSegmentValues.organisationUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get organisationUUID() {
    return this._organisationUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphOrganisationAriStaticOpts.qualifier,
      platformQualifier: graphOrganisationAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphOrganisationAriStaticOpts.resourceOwner,
      resourceType: graphOrganisationAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.organisationUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        organisationUUID: opts.organisationUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphOrganisationAriStaticOpts);
    return new _GraphOrganisationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphOrganisationAriStaticOpts);
    return new _GraphOrganisationAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      organisationUUID: this.organisationUUID
    };
  }
};

// src/graph/organisation-history/types.ts
var GraphOrganisationHistoryAriResourceOwner = "graph", GraphOrganisationHistoryAriResourceType = "organisation-history";

// src/graph/organisation-history/manifest.ts
var graphOrganisationHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphOrganisationHistoryAriResourceOwner,
  resourceType: GraphOrganisationHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{organisationUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    organisationUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/organisation-history/index.ts
var GraphOrganisationHistoryAri = class _GraphOrganisationHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._organisationUUID = opts.resourceIdSegmentValues.organisationUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get organisationUUID() {
    return this._organisationUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphOrganisationHistoryAriStaticOpts.qualifier,
      platformQualifier: graphOrganisationHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphOrganisationHistoryAriStaticOpts.resourceOwner,
      resourceType: graphOrganisationHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.organisationUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        organisationUUID: opts.organisationUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphOrganisationHistoryAriStaticOpts);
    return new _GraphOrganisationHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphOrganisationHistoryAriStaticOpts);
    return new _GraphOrganisationHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      organisationUUID: this.organisationUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/position/types.ts
var GraphPositionAriResourceOwner = "graph", GraphPositionAriResourceType = "position";

// src/graph/position/manifest.ts
var graphPositionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphPositionAriResourceOwner,
  resourceType: GraphPositionAriResourceType,
  resourceIdSlug: "activation/{activationId}/{positionUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    positionUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/position/index.ts
var GraphPositionAri = class _GraphPositionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._positionUUID = opts.resourceIdSegmentValues.positionUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get positionUUID() {
    return this._positionUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphPositionAriStaticOpts.qualifier,
      platformQualifier: graphPositionAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphPositionAriStaticOpts.resourceOwner,
      resourceType: graphPositionAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.positionUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        positionUUID: opts.positionUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphPositionAriStaticOpts);
    return new _GraphPositionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphPositionAriStaticOpts);
    return new _GraphPositionAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      positionUUID: this.positionUUID
    };
  }
};

// src/graph/position-history/types.ts
var GraphPositionHistoryAriResourceOwner = "graph", GraphPositionHistoryAriResourceType = "position-history";

// src/graph/position-history/manifest.ts
var graphPositionHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphPositionHistoryAriResourceOwner,
  resourceType: GraphPositionHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{positionUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    positionUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/position-history/index.ts
var GraphPositionHistoryAri = class _GraphPositionHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._positionUUID = opts.resourceIdSegmentValues.positionUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get positionUUID() {
    return this._positionUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphPositionHistoryAriStaticOpts.qualifier,
      platformQualifier: graphPositionHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphPositionHistoryAriStaticOpts.resourceOwner,
      resourceType: graphPositionHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.positionUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        positionUUID: opts.positionUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphPositionHistoryAriStaticOpts);
    return new _GraphPositionHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphPositionHistoryAriStaticOpts);
    return new _GraphPositionHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      positionUUID: this.positionUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/post-incident-review/types.ts
var GraphPostIncidentReviewAriResourceOwner = "graph", GraphPostIncidentReviewAriResourceType = "post-incident-review";

// src/graph/post-incident-review/manifest.ts
var graphPostIncidentReviewAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphPostIncidentReviewAriResourceOwner,
  resourceType: GraphPostIncidentReviewAriResourceType,
  resourceIdSlug: "activation/{activationId}/{postIncidentReviewId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    postIncidentReviewId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/post-incident-review/index.ts
var GraphPostIncidentReviewAri = class _GraphPostIncidentReviewAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._postIncidentReviewId = opts.resourceIdSegmentValues.postIncidentReviewId;
  }
  get activationId() {
    return this._activationId;
  }
  get postIncidentReviewId() {
    return this._postIncidentReviewId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphPostIncidentReviewAriStaticOpts.qualifier,
      platformQualifier: graphPostIncidentReviewAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphPostIncidentReviewAriStaticOpts.resourceOwner,
      resourceType: graphPostIncidentReviewAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.postIncidentReviewId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        postIncidentReviewId: opts.postIncidentReviewId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphPostIncidentReviewAriStaticOpts);
    return new _GraphPostIncidentReviewAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphPostIncidentReviewAriStaticOpts);
    return new _GraphPostIncidentReviewAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      postIncidentReviewId: this.postIncidentReviewId
    };
  }
};

// src/graph/post-incident-review-history/types.ts
var GraphPostIncidentReviewHistoryAriResourceOwner = "graph", GraphPostIncidentReviewHistoryAriResourceType = "post-incident-review-history";

// src/graph/post-incident-review-history/manifest.ts
var graphPostIncidentReviewHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphPostIncidentReviewHistoryAriResourceOwner,
  resourceType: GraphPostIncidentReviewHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{postIncidentReviewId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    postIncidentReviewId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/post-incident-review-history/index.ts
var GraphPostIncidentReviewHistoryAri = class _GraphPostIncidentReviewHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._postIncidentReviewId = opts.resourceIdSegmentValues.postIncidentReviewId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get postIncidentReviewId() {
    return this._postIncidentReviewId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphPostIncidentReviewHistoryAriStaticOpts.qualifier,
      platformQualifier: graphPostIncidentReviewHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphPostIncidentReviewHistoryAriStaticOpts.resourceOwner,
      resourceType: graphPostIncidentReviewHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.postIncidentReviewId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        postIncidentReviewId: opts.postIncidentReviewId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphPostIncidentReviewHistoryAriStaticOpts);
    return new _GraphPostIncidentReviewHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphPostIncidentReviewHistoryAriStaticOpts);
    return new _GraphPostIncidentReviewHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      postIncidentReviewId: this.postIncidentReviewId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/project/types.ts
var GraphProjectAriResourceOwner = "graph", GraphProjectAriResourceType = "project";

// src/graph/project/manifest.ts
var graphProjectAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphProjectAriResourceOwner,
  resourceType: GraphProjectAriResourceType,
  resourceIdSlug: "activation/{activationId}/{projectId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/project/index.ts
var GraphProjectAri = class _GraphProjectAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._projectId = opts.resourceIdSegmentValues.projectId;
  }
  get activationId() {
    return this._activationId;
  }
  get projectId() {
    return this._projectId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphProjectAriStaticOpts.qualifier,
      platformQualifier: graphProjectAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphProjectAriStaticOpts.resourceOwner,
      resourceType: graphProjectAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.projectId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        projectId: opts.projectId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphProjectAriStaticOpts);
    return new _GraphProjectAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphProjectAriStaticOpts);
    return new _GraphProjectAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      projectId: this.projectId
    };
  }
};

// src/graph/project-history/types.ts
var GraphProjectHistoryAriResourceOwner = "graph", GraphProjectHistoryAriResourceType = "project-history";

// src/graph/project-history/manifest.ts
var graphProjectHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphProjectHistoryAriResourceOwner,
  resourceType: GraphProjectHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{projectId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/project-history/index.ts
var GraphProjectHistoryAri = class _GraphProjectHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._projectId = opts.resourceIdSegmentValues.projectId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get projectId() {
    return this._projectId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphProjectHistoryAriStaticOpts.qualifier,
      platformQualifier: graphProjectHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphProjectHistoryAriStaticOpts.resourceOwner,
      resourceType: graphProjectHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.projectId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        projectId: opts.projectId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphProjectHistoryAriStaticOpts);
    return new _GraphProjectHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphProjectHistoryAriStaticOpts);
    return new _GraphProjectHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      projectId: this.projectId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/pull-request/types.ts
var GraphPullRequestAriResourceOwner = "graph", GraphPullRequestAriResourceType = "pull-request";

// src/graph/pull-request/manifest.ts
var graphPullRequestAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphPullRequestAriResourceOwner,
  resourceType: GraphPullRequestAriResourceType,
  resourceIdSlug: "activation/{activationId}/{pullRequestId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    pullRequestId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/pull-request/index.ts
var GraphPullRequestAri = class _GraphPullRequestAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._pullRequestId = opts.resourceIdSegmentValues.pullRequestId;
  }
  get activationId() {
    return this._activationId;
  }
  get pullRequestId() {
    return this._pullRequestId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphPullRequestAriStaticOpts.qualifier,
      platformQualifier: graphPullRequestAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphPullRequestAriStaticOpts.resourceOwner,
      resourceType: graphPullRequestAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.pullRequestId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        pullRequestId: opts.pullRequestId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphPullRequestAriStaticOpts);
    return new _GraphPullRequestAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphPullRequestAriStaticOpts);
    return new _GraphPullRequestAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      pullRequestId: this.pullRequestId
    };
  }
};

// src/graph/pull-request-history/types.ts
var GraphPullRequestHistoryAriResourceOwner = "graph", GraphPullRequestHistoryAriResourceType = "pull-request-history";

// src/graph/pull-request-history/manifest.ts
var graphPullRequestHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphPullRequestHistoryAriResourceOwner,
  resourceType: GraphPullRequestHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{pullRequestId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    pullRequestId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/pull-request-history/index.ts
var GraphPullRequestHistoryAri = class _GraphPullRequestHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._pullRequestId = opts.resourceIdSegmentValues.pullRequestId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get pullRequestId() {
    return this._pullRequestId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphPullRequestHistoryAriStaticOpts.qualifier,
      platformQualifier: graphPullRequestHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphPullRequestHistoryAriStaticOpts.resourceOwner,
      resourceType: graphPullRequestHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.pullRequestId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        pullRequestId: opts.pullRequestId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphPullRequestHistoryAriStaticOpts);
    return new _GraphPullRequestHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphPullRequestHistoryAriStaticOpts);
    return new _GraphPullRequestHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      pullRequestId: this.pullRequestId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/relationship-type/types.ts
var GraphRelationshipTypeAriResourceOwner = "graph", GraphRelationshipTypeAriResourceType = "relationship-type";

// src/graph/relationship-type/manifest.ts
var graphRelationshipTypeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphRelationshipTypeAriResourceOwner,
  resourceType: GraphRelationshipTypeAriResourceType,
  resourceIdSlug: "{relationshipTypeId}",
  resourceIdSegmentFormats: {
    relationshipTypeId: /[a-zA-Z0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/relationship-type/index.ts
var GraphRelationshipTypeAri = class _GraphRelationshipTypeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._relationshipTypeId = opts.resourceIdSegmentValues.relationshipTypeId;
  }
  get relationshipTypeId() {
    return this._relationshipTypeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphRelationshipTypeAriStaticOpts.qualifier,
      platformQualifier: graphRelationshipTypeAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphRelationshipTypeAriStaticOpts.resourceOwner,
      resourceType: graphRelationshipTypeAriStaticOpts.resourceType,
      resourceId: `${opts.relationshipTypeId}`,
      resourceIdSegmentValues: {
        relationshipTypeId: opts.relationshipTypeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphRelationshipTypeAriStaticOpts);
    return new _GraphRelationshipTypeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphRelationshipTypeAriStaticOpts);
    return new _GraphRelationshipTypeAri(opts);
  }
  getVariables() {
    return {
      relationshipTypeId: this.relationshipTypeId
    };
  }
};

// src/graph/remote-link/types.ts
var GraphRemoteLinkAriResourceOwner = "graph", GraphRemoteLinkAriResourceType = "remote-link";

// src/graph/remote-link/manifest.ts
var graphRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphRemoteLinkAriResourceOwner,
  resourceType: GraphRemoteLinkAriResourceType,
  resourceIdSlug: "activation/{activationId}/{remoteLinkId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    remoteLinkId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/remote-link/index.ts
var GraphRemoteLinkAri = class _GraphRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._remoteLinkId = opts.resourceIdSegmentValues.remoteLinkId;
  }
  get activationId() {
    return this._activationId;
  }
  get remoteLinkId() {
    return this._remoteLinkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: graphRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: graphRemoteLinkAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.remoteLinkId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        remoteLinkId: opts.remoteLinkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphRemoteLinkAriStaticOpts);
    return new _GraphRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphRemoteLinkAriStaticOpts);
    return new _GraphRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      remoteLinkId: this.remoteLinkId
    };
  }
};

// src/graph/remote-link-history/types.ts
var GraphRemoteLinkHistoryAriResourceOwner = "graph", GraphRemoteLinkHistoryAriResourceType = "remote-link-history";

// src/graph/remote-link-history/manifest.ts
var graphRemoteLinkHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphRemoteLinkHistoryAriResourceOwner,
  resourceType: GraphRemoteLinkHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{remoteLinkId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    remoteLinkId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/remote-link-history/index.ts
var GraphRemoteLinkHistoryAri = class _GraphRemoteLinkHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._remoteLinkId = opts.resourceIdSegmentValues.remoteLinkId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get remoteLinkId() {
    return this._remoteLinkId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphRemoteLinkHistoryAriStaticOpts.qualifier,
      platformQualifier: graphRemoteLinkHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphRemoteLinkHistoryAriStaticOpts.resourceOwner,
      resourceType: graphRemoteLinkHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.remoteLinkId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        remoteLinkId: opts.remoteLinkId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphRemoteLinkHistoryAriStaticOpts);
    return new _GraphRemoteLinkHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphRemoteLinkHistoryAriStaticOpts);
    return new _GraphRemoteLinkHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      remoteLinkId: this.remoteLinkId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/repository/types.ts
var GraphRepositoryAriResourceOwner = "graph", GraphRepositoryAriResourceType = "repository";

// src/graph/repository/manifest.ts
var graphRepositoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphRepositoryAriResourceOwner,
  resourceType: GraphRepositoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{repositoryId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    repositoryId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/repository/index.ts
var GraphRepositoryAri = class _GraphRepositoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._repositoryId = opts.resourceIdSegmentValues.repositoryId;
  }
  get activationId() {
    return this._activationId;
  }
  get repositoryId() {
    return this._repositoryId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphRepositoryAriStaticOpts.qualifier,
      platformQualifier: graphRepositoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphRepositoryAriStaticOpts.resourceOwner,
      resourceType: graphRepositoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.repositoryId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        repositoryId: opts.repositoryId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphRepositoryAriStaticOpts);
    return new _GraphRepositoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphRepositoryAriStaticOpts);
    return new _GraphRepositoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      repositoryId: this.repositoryId
    };
  }
};

// src/graph/role/types.ts
var GraphRoleAriResourceOwner = "graph", GraphRoleAriResourceType = "role";

// src/graph/role/manifest.ts
var graphRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphRoleAriResourceOwner,
  resourceType: GraphRoleAriResourceType,
  resourceIdSlug: "product/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:member|admin)/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/role/index.ts
var GraphRoleAri = class _GraphRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphRoleAriStaticOpts.qualifier,
      platformQualifier: graphRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphRoleAriStaticOpts.resourceOwner,
      resourceType: graphRoleAriStaticOpts.resourceType,
      resourceId: `product/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphRoleAriStaticOpts);
    return new _GraphRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphRoleAriStaticOpts);
    return new _GraphRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

// src/graph/security-container/types.ts
var GraphSecurityContainerAriResourceOwner = "graph", GraphSecurityContainerAriResourceType = "security-container";

// src/graph/security-container/manifest.ts
var graphSecurityContainerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphSecurityContainerAriResourceOwner,
  resourceType: GraphSecurityContainerAriResourceType,
  resourceIdSlug: "activation/{activationId}/{providerAppId}/{securityContainerId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    providerAppId: /[a-zA-Z0-9-._]+/,
    // eslint-disable-line no-useless-escape
    securityContainerId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/security-container/index.ts
var GraphSecurityContainerAri = class _GraphSecurityContainerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._providerAppId = opts.resourceIdSegmentValues.providerAppId, this._securityContainerId = opts.resourceIdSegmentValues.securityContainerId;
  }
  get activationId() {
    return this._activationId;
  }
  get providerAppId() {
    return this._providerAppId;
  }
  get securityContainerId() {
    return this._securityContainerId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphSecurityContainerAriStaticOpts.qualifier,
      platformQualifier: graphSecurityContainerAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphSecurityContainerAriStaticOpts.resourceOwner,
      resourceType: graphSecurityContainerAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.providerAppId}/${opts.securityContainerId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        providerAppId: opts.providerAppId,
        securityContainerId: opts.securityContainerId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphSecurityContainerAriStaticOpts);
    return new _GraphSecurityContainerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphSecurityContainerAriStaticOpts);
    return new _GraphSecurityContainerAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      providerAppId: this.providerAppId,
      securityContainerId: this.securityContainerId
    };
  }
};

// src/graph/security-workspace/types.ts
var GraphSecurityWorkspaceAriResourceOwner = "graph", GraphSecurityWorkspaceAriResourceType = "security-workspace";

// src/graph/security-workspace/manifest.ts
var graphSecurityWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphSecurityWorkspaceAriResourceOwner,
  resourceType: GraphSecurityWorkspaceAriResourceType,
  resourceIdSlug: "activation/{activationId}/{providerAppId}/{securityWorkspaceId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    providerAppId: /[a-zA-Z0-9-._]+/,
    // eslint-disable-line no-useless-escape
    securityWorkspaceId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/security-workspace/index.ts
var GraphSecurityWorkspaceAri = class _GraphSecurityWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._providerAppId = opts.resourceIdSegmentValues.providerAppId, this._securityWorkspaceId = opts.resourceIdSegmentValues.securityWorkspaceId;
  }
  get activationId() {
    return this._activationId;
  }
  get providerAppId() {
    return this._providerAppId;
  }
  get securityWorkspaceId() {
    return this._securityWorkspaceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphSecurityWorkspaceAriStaticOpts.qualifier,
      platformQualifier: graphSecurityWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphSecurityWorkspaceAriStaticOpts.resourceOwner,
      resourceType: graphSecurityWorkspaceAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.providerAppId}/${opts.securityWorkspaceId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        providerAppId: opts.providerAppId,
        securityWorkspaceId: opts.securityWorkspaceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphSecurityWorkspaceAriStaticOpts);
    return new _GraphSecurityWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphSecurityWorkspaceAriStaticOpts);
    return new _GraphSecurityWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      providerAppId: this.providerAppId,
      securityWorkspaceId: this.securityWorkspaceId
    };
  }
};

// src/graph/service/types.ts
var GraphServiceAriResourceOwner = "graph", GraphServiceAriResourceType = "service";

// src/graph/service/manifest.ts
var graphServiceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphServiceAriResourceOwner,
  resourceType: GraphServiceAriResourceType,
  resourceIdSlug: "{workspaceId}/{serviceId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    serviceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/service/index.ts
var GraphServiceAri = class _GraphServiceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._serviceId = opts.resourceIdSegmentValues.serviceId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get serviceId() {
    return this._serviceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphServiceAriStaticOpts.qualifier,
      platformQualifier: graphServiceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphServiceAriStaticOpts.resourceOwner,
      resourceType: graphServiceAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.serviceId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        serviceId: opts.serviceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphServiceAriStaticOpts);
    return new _GraphServiceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphServiceAriStaticOpts);
    return new _GraphServiceAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      serviceId: this.serviceId
    };
  }
};

// src/graph/service-and-jira-project-relationship/types.ts
var GraphServiceAndJiraProjectRelationshipAriResourceOwner = "graph", GraphServiceAndJiraProjectRelationshipAriResourceType = "service-and-jira-project-relationship";

// src/graph/service-and-jira-project-relationship/manifest.ts
var graphServiceAndJiraProjectRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphServiceAndJiraProjectRelationshipAriResourceOwner,
  resourceType: GraphServiceAndJiraProjectRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/service-and-jira-project-relationship/index.ts
var GraphServiceAndJiraProjectRelationshipAri = class _GraphServiceAndJiraProjectRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphServiceAndJiraProjectRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphServiceAndJiraProjectRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphServiceAndJiraProjectRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphServiceAndJiraProjectRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphServiceAndJiraProjectRelationshipAriStaticOpts);
    return new _GraphServiceAndJiraProjectRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphServiceAndJiraProjectRelationshipAriStaticOpts);
    return new _GraphServiceAndJiraProjectRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/service-and-operations-team-relationship/types.ts
var GraphServiceAndOperationsTeamRelationshipAriResourceOwner = "graph", GraphServiceAndOperationsTeamRelationshipAriResourceType = "service-and-operations-team-relationship";

// src/graph/service-and-operations-team-relationship/manifest.ts
var graphServiceAndOperationsTeamRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphServiceAndOperationsTeamRelationshipAriResourceOwner,
  resourceType: GraphServiceAndOperationsTeamRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/service-and-operations-team-relationship/index.ts
var GraphServiceAndOperationsTeamRelationshipAri = class _GraphServiceAndOperationsTeamRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphServiceAndOperationsTeamRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphServiceAndOperationsTeamRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphServiceAndOperationsTeamRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphServiceAndOperationsTeamRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphServiceAndOperationsTeamRelationshipAriStaticOpts);
    return new _GraphServiceAndOperationsTeamRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphServiceAndOperationsTeamRelationshipAriStaticOpts);
    return new _GraphServiceAndOperationsTeamRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/service-and-vcs-repository-relationship/types.ts
var GraphServiceAndVcsRepositoryRelationshipAriResourceOwner = "graph", GraphServiceAndVcsRepositoryRelationshipAriResourceType = "service-and-vcs-repository-relationship";

// src/graph/service-and-vcs-repository-relationship/manifest.ts
var graphServiceAndVcsRepositoryRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphServiceAndVcsRepositoryRelationshipAriResourceOwner,
  resourceType: GraphServiceAndVcsRepositoryRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/service-and-vcs-repository-relationship/index.ts
var GraphServiceAndVcsRepositoryRelationshipAri = class _GraphServiceAndVcsRepositoryRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphServiceAndVcsRepositoryRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphServiceAndVcsRepositoryRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphServiceAndVcsRepositoryRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphServiceAndVcsRepositoryRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphServiceAndVcsRepositoryRelationshipAriStaticOpts);
    return new _GraphServiceAndVcsRepositoryRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphServiceAndVcsRepositoryRelationshipAriStaticOpts);
    return new _GraphServiceAndVcsRepositoryRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/service-relationship/types.ts
var GraphServiceRelationshipAriResourceOwner = "graph", GraphServiceRelationshipAriResourceType = "service-relationship";

// src/graph/service-relationship/manifest.ts
var graphServiceRelationshipAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphServiceRelationshipAriResourceOwner,
  resourceType: GraphServiceRelationshipAriResourceType,
  resourceIdSlug: "{workspaceId}/{relationshipId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    relationshipId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/service-relationship/index.ts
var GraphServiceRelationshipAri = class _GraphServiceRelationshipAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._relationshipId = opts.resourceIdSegmentValues.relationshipId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get relationshipId() {
    return this._relationshipId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphServiceRelationshipAriStaticOpts.qualifier,
      platformQualifier: graphServiceRelationshipAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphServiceRelationshipAriStaticOpts.resourceOwner,
      resourceType: graphServiceRelationshipAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.relationshipId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        relationshipId: opts.relationshipId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphServiceRelationshipAriStaticOpts);
    return new _GraphServiceRelationshipAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphServiceRelationshipAriStaticOpts);
    return new _GraphServiceRelationshipAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      relationshipId: this.relationshipId
    };
  }
};

// src/graph/space/types.ts
var GraphSpaceAriResourceOwner = "graph", GraphSpaceAriResourceType = "space";

// src/graph/space/manifest.ts
var graphSpaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphSpaceAriResourceOwner,
  resourceType: GraphSpaceAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/space/index.ts
var GraphSpaceAri = class _GraphSpaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphSpaceAriStaticOpts.qualifier,
      platformQualifier: graphSpaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphSpaceAriStaticOpts.resourceOwner,
      resourceType: graphSpaceAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphSpaceAriStaticOpts);
    return new _GraphSpaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphSpaceAriStaticOpts);
    return new _GraphSpaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID
    };
  }
};

// src/graph/space-history/types.ts
var GraphSpaceHistoryAriResourceOwner = "graph", GraphSpaceHistoryAriResourceType = "space-history";

// src/graph/space-history/manifest.ts
var graphSpaceHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphSpaceHistoryAriResourceOwner,
  resourceType: GraphSpaceHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/space-history/index.ts
var GraphSpaceHistoryAri = class _GraphSpaceHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphSpaceHistoryAriStaticOpts.qualifier,
      platformQualifier: graphSpaceHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphSpaceHistoryAriStaticOpts.resourceOwner,
      resourceType: graphSpaceHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphSpaceHistoryAriStaticOpts);
    return new _GraphSpaceHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphSpaceHistoryAriStaticOpts);
    return new _GraphSpaceHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/thread-chunk/types.ts
var GraphThreadChunkAriResourceOwner = "graph", GraphThreadChunkAriResourceType = "thread-chunk";

// src/graph/thread-chunk/manifest.ts
var graphThreadChunkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphThreadChunkAriResourceOwner,
  resourceType: GraphThreadChunkAriResourceType,
  resourceIdSlug: "activation/{activationId}/{chunkId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    chunkId: /[!a-zA-Z0-9\-_.~@:;{}=]+(\/[!a-zA-Z0-9\-_.~@:;{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/thread-chunk/index.ts
var GraphThreadChunkAri = class _GraphThreadChunkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._chunkId = opts.resourceIdSegmentValues.chunkId;
  }
  get activationId() {
    return this._activationId;
  }
  get chunkId() {
    return this._chunkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphThreadChunkAriStaticOpts.qualifier,
      platformQualifier: graphThreadChunkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphThreadChunkAriStaticOpts.resourceOwner,
      resourceType: graphThreadChunkAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.chunkId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        chunkId: opts.chunkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphThreadChunkAriStaticOpts);
    return new _GraphThreadChunkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphThreadChunkAriStaticOpts);
    return new _GraphThreadChunkAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      chunkId: this.chunkId
    };
  }
};

// src/graph/video/types.ts
var GraphVideoAriResourceOwner = "graph", GraphVideoAriResourceType = "video";

// src/graph/video/manifest.ts
var graphVideoAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphVideoAriResourceOwner,
  resourceType: GraphVideoAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/video/index.ts
var GraphVideoAri = class _GraphVideoAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphVideoAriStaticOpts.qualifier,
      platformQualifier: graphVideoAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphVideoAriStaticOpts.resourceOwner,
      resourceType: graphVideoAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphVideoAriStaticOpts);
    return new _GraphVideoAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphVideoAriStaticOpts);
    return new _GraphVideoAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID
    };
  }
};

// src/graph/video-history/types.ts
var GraphVideoHistoryAriResourceOwner = "graph", GraphVideoHistoryAriResourceType = "video-history";

// src/graph/video-history/manifest.ts
var graphVideoHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphVideoHistoryAriResourceOwner,
  resourceType: GraphVideoHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/video-history/index.ts
var GraphVideoHistoryAri = class _GraphVideoHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphVideoHistoryAriStaticOpts.qualifier,
      platformQualifier: graphVideoHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphVideoHistoryAriStaticOpts.resourceOwner,
      resourceType: graphVideoHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphVideoHistoryAriStaticOpts);
    return new _GraphVideoHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphVideoHistoryAriStaticOpts);
    return new _GraphVideoHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/vulnerability/types.ts
var GraphVulnerabilityAriResourceOwner = "graph", GraphVulnerabilityAriResourceType = "vulnerability";

// src/graph/vulnerability/manifest.ts
var graphVulnerabilityAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphVulnerabilityAriResourceOwner,
  resourceType: GraphVulnerabilityAriResourceType,
  resourceIdSlug: "activation/{activationId}/{vulnerabilityId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    vulnerabilityId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/vulnerability/index.ts
var GraphVulnerabilityAri = class _GraphVulnerabilityAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._vulnerabilityId = opts.resourceIdSegmentValues.vulnerabilityId;
  }
  get activationId() {
    return this._activationId;
  }
  get vulnerabilityId() {
    return this._vulnerabilityId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphVulnerabilityAriStaticOpts.qualifier,
      platformQualifier: graphVulnerabilityAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphVulnerabilityAriStaticOpts.resourceOwner,
      resourceType: graphVulnerabilityAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.vulnerabilityId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        vulnerabilityId: opts.vulnerabilityId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphVulnerabilityAriStaticOpts);
    return new _GraphVulnerabilityAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphVulnerabilityAriStaticOpts);
    return new _GraphVulnerabilityAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      vulnerabilityId: this.vulnerabilityId
    };
  }
};

// src/graph/vulnerability-history/types.ts
var GraphVulnerabilityHistoryAriResourceOwner = "graph", GraphVulnerabilityHistoryAriResourceType = "vulnerability-history";

// src/graph/vulnerability-history/manifest.ts
var graphVulnerabilityHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphVulnerabilityHistoryAriResourceOwner,
  resourceType: GraphVulnerabilityHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{vulnerabilityId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    vulnerabilityId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/vulnerability-history/index.ts
var GraphVulnerabilityHistoryAri = class _GraphVulnerabilityHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._vulnerabilityId = opts.resourceIdSegmentValues.vulnerabilityId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get vulnerabilityId() {
    return this._vulnerabilityId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphVulnerabilityHistoryAriStaticOpts.qualifier,
      platformQualifier: graphVulnerabilityHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphVulnerabilityHistoryAriStaticOpts.resourceOwner,
      resourceType: graphVulnerabilityHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.vulnerabilityId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        vulnerabilityId: opts.vulnerabilityId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphVulnerabilityHistoryAriStaticOpts);
    return new _GraphVulnerabilityHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphVulnerabilityHistoryAriStaticOpts);
    return new _GraphVulnerabilityHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      vulnerabilityId: this.vulnerabilityId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/work-item/types.ts
var GraphWorkItemAriResourceOwner = "graph", GraphWorkItemAriResourceType = "work-item";

// src/graph/work-item/manifest.ts
var graphWorkItemAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphWorkItemAriResourceOwner,
  resourceType: GraphWorkItemAriResourceType,
  resourceIdSlug: "activation/{activationId}/{workItemId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    workItemId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/work-item/index.ts
var GraphWorkItemAri = class _GraphWorkItemAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._workItemId = opts.resourceIdSegmentValues.workItemId;
  }
  get activationId() {
    return this._activationId;
  }
  get workItemId() {
    return this._workItemId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphWorkItemAriStaticOpts.qualifier,
      platformQualifier: graphWorkItemAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphWorkItemAriStaticOpts.resourceOwner,
      resourceType: graphWorkItemAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.workItemId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        workItemId: opts.workItemId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphWorkItemAriStaticOpts);
    return new _GraphWorkItemAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphWorkItemAriStaticOpts);
    return new _GraphWorkItemAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      workItemId: this.workItemId
    };
  }
};

// src/graph/work-item-history/types.ts
var GraphWorkItemHistoryAriResourceOwner = "graph", GraphWorkItemHistoryAriResourceType = "work-item-history";

// src/graph/work-item-history/manifest.ts
var graphWorkItemHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphWorkItemHistoryAriResourceOwner,
  resourceType: GraphWorkItemHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{workItemId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    workItemId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/work-item-history/index.ts
var GraphWorkItemHistoryAri = class _GraphWorkItemHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._workItemId = opts.resourceIdSegmentValues.workItemId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get workItemId() {
    return this._workItemId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphWorkItemHistoryAriStaticOpts.qualifier,
      platformQualifier: graphWorkItemHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphWorkItemHistoryAriStaticOpts.resourceOwner,
      resourceType: graphWorkItemHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.workItemId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        workItemId: opts.workItemId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphWorkItemHistoryAriStaticOpts);
    return new _GraphWorkItemHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphWorkItemHistoryAriStaticOpts);
    return new _GraphWorkItemHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      workItemId: this.workItemId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/worker/types.ts
var GraphWorkerAriResourceOwner = "graph", GraphWorkerAriResourceType = "worker";

// src/graph/worker/manifest.ts
var graphWorkerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphWorkerAriResourceOwner,
  resourceType: GraphWorkerAriResourceType,
  resourceIdSlug: "activation/{activationId}/{workerUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    workerUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/worker/index.ts
var GraphWorkerAri = class _GraphWorkerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._workerUUID = opts.resourceIdSegmentValues.workerUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get workerUUID() {
    return this._workerUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphWorkerAriStaticOpts.qualifier,
      platformQualifier: graphWorkerAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphWorkerAriStaticOpts.resourceOwner,
      resourceType: graphWorkerAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.workerUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        workerUUID: opts.workerUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphWorkerAriStaticOpts);
    return new _GraphWorkerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphWorkerAriStaticOpts);
    return new _GraphWorkerAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      workerUUID: this.workerUUID
    };
  }
};

// src/graph/worker-history/types.ts
var GraphWorkerHistoryAriResourceOwner = "graph", GraphWorkerHistoryAriResourceType = "worker-history";

// src/graph/worker-history/manifest.ts
var graphWorkerHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphWorkerHistoryAriResourceOwner,
  resourceType: GraphWorkerHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{workerUUID}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    workerUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/worker-history/index.ts
var GraphWorkerHistoryAri = class _GraphWorkerHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._workerUUID = opts.resourceIdSegmentValues.workerUUID, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get workerUUID() {
    return this._workerUUID;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphWorkerHistoryAriStaticOpts.qualifier,
      platformQualifier: graphWorkerHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphWorkerHistoryAriStaticOpts.resourceOwner,
      resourceType: graphWorkerHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.workerUUID}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        workerUUID: opts.workerUUID,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphWorkerHistoryAriStaticOpts);
    return new _GraphWorkerHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphWorkerHistoryAriStaticOpts);
    return new _GraphWorkerHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      workerUUID: this.workerUUID,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

// src/graph/workspace/types.ts
var GraphWorkspaceAriResourceOwner = "graph", GraphWorkspaceAriResourceType = "workspace";

// src/graph/workspace/manifest.ts
var graphWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  // eslint-disable-line no-useless-escape
  resourceOwner: GraphWorkspaceAriResourceOwner,
  resourceType: GraphWorkspaceAriResourceType,
  resourceIdSlug: "{workspaceId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/workspace/index.ts
var GraphWorkspaceAri = class _GraphWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphWorkspaceAriStaticOpts.qualifier,
      platformQualifier: graphWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphWorkspaceAriStaticOpts.resourceOwner,
      resourceType: graphWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphWorkspaceAriStaticOpts);
    return new _GraphWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphWorkspaceAriStaticOpts);
    return new _GraphWorkspaceAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId
    };
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  GraphBranchAri,
  GraphBranchHistoryAri,
  GraphBuildAri,
  GraphBuildHistoryAri,
  GraphCalendarEventAri,
  GraphCalendarEventHistoryAri,
  GraphCommentAri,
  GraphCommentHistoryAri,
  GraphCommitAri,
  GraphContentAri,
  GraphConversationAri,
  GraphConversationChunkAri,
  GraphConversationHistoryAri,
  GraphCustomerOrgAri,
  GraphCustomerOrgHistoryAri,
  GraphDealAri,
  GraphDealHistoryAri,
  GraphDeploymentAri,
  GraphDeploymentHistoryAri,
  GraphDesignAri,
  GraphDesignHistoryAri,
  GraphDevopsComponentAri,
  GraphDocumentAri,
  GraphDocumentHistoryAri,
  GraphEmployeeAri,
  GraphEmployeeHistoryAri,
  GraphFeatureFlagAri,
  GraphFeatureFlagHistoryAri,
  GraphIncidentAri,
  GraphIncidentHistoryAri,
  GraphIntegrationContextAri,
  GraphJiraProjectAndDevopsToolRelationshipAri,
  GraphJiraProjectAndDocumentationSpaceRelationshipAri,
  GraphJiraProjectAndOperationsTeamRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri,
  GraphMessageAri,
  GraphOperationsWorkspaceAri,
  GraphOrganisationAri,
  GraphOrganisationHistoryAri,
  GraphPositionAri,
  GraphPositionHistoryAri,
  GraphPostIncidentReviewAri,
  GraphPostIncidentReviewHistoryAri,
  GraphProjectAri,
  GraphProjectHistoryAri,
  GraphPullRequestAri,
  GraphPullRequestHistoryAri,
  GraphRelationshipTypeAri,
  GraphRemoteLinkAri,
  GraphRemoteLinkHistoryAri,
  GraphRepositoryAri,
  GraphRoleAri,
  GraphSecurityContainerAri,
  GraphSecurityWorkspaceAri,
  GraphServiceAndJiraProjectRelationshipAri,
  GraphServiceAndOperationsTeamRelationshipAri,
  GraphServiceAndVcsRepositoryRelationshipAri,
  GraphServiceAri,
  GraphServiceRelationshipAri,
  GraphSpaceAri,
  GraphSpaceHistoryAri,
  GraphThreadChunkAri,
  GraphVideoAri,
  GraphVideoHistoryAri,
  GraphVulnerabilityAri,
  GraphVulnerabilityHistoryAri,
  GraphWorkItemAri,
  GraphWorkItemHistoryAri,
  GraphWorkerAri,
  GraphWorkerHistoryAri,
  GraphWorkspaceAri
});
