// @forge/cli/out/environment/graphql-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GraphqlClient = exports.MissingAppError = exports.APP_HAS_INSTALLATIONS_CODE = void 0;
const cli_shared_1 = require("@forge/cli-shared");
exports.APP_HAS_INSTALLATIONS_CODE = 'APP_HAS_INSTALLATIONS';
class MissingAppError extends cli_shared_1.UserError {
}
exports.MissingAppError = MissingAppError;
class GraphqlClient {
    graphqlClient;
    constructor(graphqlClient) {
        this.graphqlClient = graphqlClient;
    }
    async createEnvironment(details) {
        const mutation = `
      mutation forge_cli_createEnvironment($input: CreateAppEnvironmentInput!) {
        ecosystem {
          createAppEnvironment(input: $input) {
            success
            errors {
              message
              extensions {
                errorType
                statusCode
              }
            }
          }
        }
      }
    `;
        const { response: { ecosystem: { createAppEnvironment } }, requestId } = await this.graphqlClient.mutate(mutation, {
            input: {
                appAri: details.appId,
                environmentKey: details.environmentKey,
                environmentType: cli_shared_1.AppEnvironmentType.Development
            }
        });
        if (!createAppEnvironment) {
            throw new cli_shared_1.GraphQlMutationError(`Unable to get a response (requestId: ${requestId || 'unknown'})`, { requestId });
        }
        const { success, errors } = createAppEnvironment;
        const error = (0, cli_shared_1.getError)(errors);
        if (!success) {
            throw new cli_shared_1.GraphQlMutationError(`${error.message} (requestId: ${requestId || 'unknown'})`, {
                requestId,
                code: error.code,
                statusCode: error.statusCode
            });
        }
    }
    async listEnvironment(details) {
        const query = `
      query forge_cli_listEnvironment($id: ID!) {
        app(id: $id) {
          environments {
            id
            key
            type
            createdAt
            versions(first: 1) {
              nodes {
                updatedAt
                requiredProducts
              }
            }
          }
        }
      }
    `;
        const { app } = await this.graphqlClient.query(query, {
            id: details.appId
        });
        if (!app) {
            throw new MissingAppError();
        }
        return app.environments.map((environment) => {
            return {
                id: environment.id,
                type: environment.type,
                key: environment.key,
                lastDeployedAt: environment.versions?.nodes?.[0]?.updatedAt || environment.createdAt,
                requiredProducts: environment.versions?.nodes?.[0]?.requiredProducts || []
            };
        });
    }
    async deleteEnvironments(details) {
        const results = [];
        for (const environmentKey of details.environmentKeys) {
            try {
                const success = await this.deleteEnvironment({
                    ...details,
                    environmentKey
                });
                results.push({ environmentKey, successful: success });
            }
            catch (err) {
                results.push({ environmentKey, successful: false, error: err });
            }
        }
        return results;
    }
    async deleteEnvironment(details) {
        const mutation = `
      mutation forge_cli_deleteEnvironment($input: DeleteAppEnvironmentInput!) {
        ecosystem {
          deleteAppEnvironment(input: $input) {
            success
            errors {
              message
              extensions {
                errorType
                statusCode
              }
            }
          }
        }
      }
    `;
        const { response: { ecosystem: { deleteAppEnvironment } }, requestId } = await this.graphqlClient.mutate(mutation, {
            input: {
                appAri: details.appId,
                environmentKey: details.environmentKey
            }
        });
        if (!deleteAppEnvironment) {
            throw new cli_shared_1.GraphQlMutationError(`Unable to get a response (requestId: ${requestId || 'unknown'})`, {
                requestId
            });
        }
        const { success, errors } = deleteAppEnvironment;
        const error = (0, cli_shared_1.getError)(errors);
        let errorMessage = error.message;
        if (error.code === exports.APP_HAS_INSTALLATIONS_CODE) {
            errorMessage = cli_shared_1.Text.deleteEnvironment.hasInstallationError(details.environmentKey);
        }
        if (!success) {
            throw new cli_shared_1.GraphQlMutationError(`${errorMessage} (requestId: ${requestId || 'unknown'})`, {
                requestId,
                code: error.code,
                statusCode: error.statusCode
            });
        }
        return success;
    }
}
exports.GraphqlClient = GraphqlClient;
