// @forge/bundler/out/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BundlerError = exports.mergeMetadata = exports.emptyMetadata = void 0;
const cli_shared_1 = require("@forge/cli-shared");
function emptyMetadata() {
    return {
        jsFiles: 0,
        tsFiles: 0,
        esm: false,
        dependencies: new Set(),
        sdkImports: new Set(),
        errors: 0
    };
}
exports.emptyMetadata = emptyMetadata;
function mergeMetadata(metadata1, metadata2) {
    return {
        ...((metadata1.modules || metadata2.modules) && {
            modules: [...(metadata1.modules ?? []), ...(metadata2.modules ?? [])]
        }),
        jsFiles: metadata1.jsFiles + metadata2.jsFiles,
        tsFiles: metadata1.tsFiles + metadata2.tsFiles,
        esm: metadata1.esm || metadata2.esm,
        dependencies: new Set([...metadata1.dependencies, ...metadata2.dependencies]),
        sdkImports: new Set([...metadata1.sdkImports, ...metadata2.sdkImports]),
        errors: metadata1.errors + metadata2.errors,
        bundler: metadata1.bundler || metadata2.bundler
    };
}
exports.mergeMetadata = mergeMetadata;
class BundlerError extends cli_shared_1.UserError {
}
exports.BundlerError = BundlerError;
