// @forge/cli/out/command-line/view/eligibility-view.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EligibilityView = void 0;
const cli_shared_1 = require("@forge/cli-shared");
class EligibilityView {
    ui;
    constructor(ui) {
        this.ui = ui;
    }
    getLogger() {
        return this.ui;
    }
    async displayProgress(progress) {
        return this.ui.displayProgress(progress, cli_shared_1.Text.eligibility.cmd.start, cli_shared_1.Text.eligibility.cmd.end);
    }
    displayRoaEligibilityFailureMessage(appVersion, envKey, envType, reason) {
        this.ui.emptyLine();
        this.ui.info(cli_shared_1.Text.eligibility.cmd.status.notEligible(envKey, envType, appVersion, reason.map(cli_shared_1.Text.eligibility.cmd.rule.conversion)));
    }
    displayRoaEligibilitySuccessMessage(appVersion, envKey, envType) {
        this.ui.emptyLine();
        this.ui.info(cli_shared_1.Text.eligibility.cmd.status.eligible(envKey, envType, appVersion));
    }
}
exports.EligibilityView = EligibilityView;
