// @forge/cli-shared/out/auth/personal/me-graphql-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeGraphqlClient = exports.ScopedTokenError = exports.UserNotFoundError = void 0;
const shared_1 = require("../../shared");
const ui_1 = require("../../ui");
class UserNotFoundError extends shared_1.UserError {
}
exports.UserNotFoundError = UserNotFoundError;
class ScopedTokenError extends shared_1.UserError {
    constructor() {
        super(ui_1.Text.login.scopedToken.error);
    }
}
exports.ScopedTokenError = ScopedTokenError;
class MeGraphqlClient {
    graphqlClient;
    logger;
    constructor(graphqlClient, logger) {
        this.graphqlClient = graphqlClient;
        this.logger = logger;
    }
    async getUser() {
        try {
            const query = `
        query forge_cli_getUserDetails {
          me {
            user {
              name
              accountStatus
              accountId
            }
          }
        }`;
            const result = await this.graphqlClient.query(query, {});
            if (result.me.user && result.me.user.accountStatus === 'active') {
                return result.me.user;
            }
        }
        catch (error) {
            this.logger.debug((0, ui_1.errorMessage)(error));
            if ((0, ui_1.errorMessage)(error).includes('This request does not contain the right authorisation scopes')) {
                throw new ScopedTokenError();
            }
        }
        throw new UserNotFoundError();
    }
}
exports.MeGraphqlClient = MeGraphqlClient;
