// @forge/cli/out/service/cached-config-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CachedConfigService = void 0;
const ANALYTICS_PREFERENCES_KEY = 'analytics-preferences';
const DEFAULT_ENVIRONMENT_KEY = 'default-environment';
const ASSISTANT_NAME_KEY = 'assistant-name';
class CachedConfigService {
    cachedConf;
    constructor(cachedConf) {
        this.cachedConf = cachedConf;
    }
    getAnalyticsPreferences() {
        if (process.env.FORGE_DISABLE_ANALYTICS) {
            return false;
        }
        return this.cachedConf.get(ANALYTICS_PREFERENCES_KEY);
    }
    setAnalyticsPreferences(preference) {
        this.cachedConf.set(ANALYTICS_PREFERENCES_KEY, preference);
    }
    getDefaultEnvironments() {
        return this.cachedConf.get(DEFAULT_ENVIRONMENT_KEY) || {};
    }
    getDefaultEnvironment(appId) {
        const defaultEnvironments = this.getDefaultEnvironments();
        return defaultEnvironments[appId];
    }
    setDefaultEnvironment(appId, preference) {
        const defaultEnvironments = this.getDefaultEnvironments();
        defaultEnvironments[appId] = preference;
        this.cachedConf.set(DEFAULT_ENVIRONMENT_KEY, defaultEnvironments);
    }
    getAssistantName() {
        return this.cachedConf.get(ASSISTANT_NAME_KEY) ?? null;
    }
    setAssistantName(name) {
        if (name === null) {
            this.cachedConf.delete(ASSISTANT_NAME_KEY);
        }
        else {
            this.cachedConf.set(ASSISTANT_NAME_KEY, name);
        }
    }
}
exports.CachedConfigService = CachedConfigService;
