// @forge/cli/out/service/docker-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DockerService = exports.DockerErrorCode = exports.DOCKER_DOWNLOAD_LINK = exports.DockerError = void 0;
const tslib_1 = require("tslib");
const child_process_1 = require("child_process");
const command_exists_1 = tslib_1.__importDefault(require("command-exists"));
const cross_spawn_1 = require("cross-spawn");
const cli_shared_1 = require("@forge/cli-shared");
class DockerError extends cli_shared_1.UserError {
    code;
    constructor(message, code) {
        super(message);
        this.code = code;
    }
}
exports.DockerError = DockerError;
exports.DOCKER_DOWNLOAD_LINK = 'https://docs.docker.com/get-docker/';
var DockerErrorCode;
(function (DockerErrorCode) {
    DockerErrorCode[DockerErrorCode["DAEMON_NOT_RUNNING"] = 0] = "DAEMON_NOT_RUNNING";
    DockerErrorCode[DockerErrorCode["NOT_INSTALLED"] = 1] = "NOT_INSTALLED";
})(DockerErrorCode = exports.DockerErrorCode || (exports.DockerErrorCode = {}));
class DockerService {
    verifyInstalled() {
        if (!command_exists_1.default.sync('docker')) {
            throw new DockerError(cli_shared_1.Text.tunnel.error.dockerNotInstalled(exports.DOCKER_DOWNLOAD_LINK), DockerErrorCode.NOT_INSTALLED);
        }
    }
    authenticateDocker(args) {
        this.verifyInstalled();
        return (0, cross_spawn_1.spawn)('docker', ['login', ...args], {
            stdio: ['pipe', 'inherit', 'inherit']
        });
    }
    async getDockerVersion(debugEnabled) {
        this.verifyInstalled();
        const { err, stdout } = await this.execPromise('docker version --format "{{.Server.Version}}"');
        if (err) {
            throw new DockerError(cli_shared_1.Text.tunnel.error.dockerDaemonNotRunning(err.message, debugEnabled), DockerErrorCode.DAEMON_NOT_RUNNING);
        }
        const dockerVersion = stdout.trim();
        return dockerVersion;
    }
    execPromise(cmd) {
        return new Promise((resolve) => {
            (0, child_process_1.exec)(cmd, (err, stdout, stderr) => {
                resolve({
                    err: err,
                    stdout: stdout,
                    stderr: stderr
                });
            });
        });
    }
}
exports.DockerService = DockerService;
