// @forge/cli/out/service/lint-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LintService = exports.DEFAULT_DIRECTORY = void 0;
const tslib_1 = require("tslib");
const cli_shared_1 = require("@forge/cli-shared");
const lint_1 = require("@forge/lint");
const path_1 = tslib_1.__importDefault(require("path"));
exports.DEFAULT_DIRECTORY = './src';
class LintService {
    configFile;
    fileSystemReader;
    statsigService;
    constructor(configFile, fileSystemReader, statsigService) {
        this.configFile = configFile;
        this.fileSystemReader = fileSystemReader;
        this.statsigService = statsigService;
    }
    async run(environment, { fix }, logger, directory = exports.DEFAULT_DIRECTORY, lintFunction = lint_1.lint) {
        const UIKitResources = await this.configFile.getResources(['nativeUI']);
        const UIKitDirectories = UIKitResources.map((resource) => path_1.default.dirname(resource.path));
        const exclude = [...(await (0, cli_shared_1.listGitIgnoreFiles)(this.fileSystemReader)), '.git', 'node_modules'];
        const [filesToLint, ...UIKitFilesByDirectory] = await Promise.all([
            this.fileSystemReader.recursiveReadDir(directory, exclude),
            ...UIKitDirectories.map((directory) => this.fileSystemReader.recursiveReadDir(directory, exclude))
        ]);
        const UIKitFilesToLint = UIKitFilesByDirectory.reduce((allFiles, directoryFiles) => allFiles.concat(directoryFiles), []);
        const allFilesToLint = filesToLint
            .concat(UIKitFilesToLint)
            .filter((value, index, arr) => arr.indexOf(value) === index);
        const lintResults = await lintFunction(allFilesToLint, await this.configFile.readConfig(), environment, logger, this.statsigService);
        if (fix) {
            return { lintResults, fixResults: await this.fix(lintResults) };
        }
        return { lintResults };
    }
    async fix(lintResults) {
        const initialState = { errorsFixed: 0, warningsFixed: 0, configFile: this.configFile };
        return lintResults.reduce((promiseChain, nextResult) => promiseChain.then((state) => nextResult.runFixer(state)), Promise.resolve(initialState));
    }
    problemCount(lintResults) {
        return (0, lint_1.problemCount)(lintResults);
    }
    failedScopesFromLintResult({ errors, warnings }) {
        return Array.from((0, lint_1.findMissingPermissions)(errors, warnings, 'missingPermission'));
    }
    failedScopes(lintResults) {
        return lintResults.reduce((acc, lintResult) => {
            const failedScopes = this.failedScopesFromLintResult(lintResult);
            acc.push(...failedScopes);
            return acc;
        }, []);
    }
}
exports.LintService = LintService;
