// @forge/lint/out/lint/linters/permission-linter/verifiers/storage-api-verifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAPIVerifier = void 0;
const linter_interface_1 = require("../../../linter-interface");
const verifier_interface_1 = require("../../verifier-interface");
const text_1 = require("../../../text");
const utils_1 = require("../../utils");
const STORAGE_PERMISSION = 'storage:app';
class StorageAPIVerifier extends verifier_interface_1.BaseLintIssueVerifier {
    getLintClass() {
        return linter_interface_1.LintClass.Error;
    }
    async process(apiCalls) {
        if ((0, utils_1.getAllScopeKeys)(this.manifest.permissions?.scopes ?? []).includes(STORAGE_PERMISSION)) {
            return [];
        }
        return apiCalls.map((apiCall) => ({
            class: this.getLintClass(),
            message: text_1.messages.verifiers.storage.message(STORAGE_PERMISSION),
            reference: text_1.messages.verifiers.storage.reference,
            ...apiCall,
            metadata: {
                missingPermission: STORAGE_PERMISSION
            }
        }));
    }
}
exports.StorageAPIVerifier = StorageAPIVerifier;
